// selenidejs/built/collection.js
"use strict";
// Copyright 2018 Knowledge Expert SA
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Collection = void 0;
const configuration_1 = require("./configuration");
const element_1 = require("./element");
const entity_1 = require("./entity");
const byConditionWebElementLocator_1 = require("./locators/byConditionWebElementLocator");
const byIndexWebElementLocator_1 = require("./locators/byIndexWebElementLocator");
const cashedWebElementLocator_1 = require("./locators/cashedWebElementLocator");
const collectedByLocator_1 = require("./locators/collectedByLocator");
const filteredByConditionWebElementsLocator_1 = require("./locators/filteredByConditionWebElementsLocator");
const slicedWebElementsLocator_1 = require("./locators/slicedWebElementsLocator");
const wait_1 = require("./wait");
class Collection extends entity_1.Entity {
    constructor(locator, configuration) {
        super(configuration);
        this.locator = locator;
    }
    with(customConfig) {
        return new Collection(this.locator, new configuration_1.Configuration(Object.assign(Object.assign({}, this.configuration), customConfig)));
    }
    // todo: should not we move it to queries?, or rename to asCashedArray() ?
    async getAsCashedArray() {
        return (await this.getWebElements())
            .map((it, index) => new element_1.Element(new cashedWebElementLocator_1.CashedWebElementLocator(it, `${this}[${index}]`), this.configuration));
    }
    elementAt(index) {
        return new element_1.Element(new byIndexWebElementLocator_1.ByIndexWebElementLocator(index, this), this.configuration);
    }
    get first() {
        return this.elementAt(0);
    }
    get second() {
        return this.elementAt(1);
    }
    /**
     * Represents a new collection sliced from 'start' element index to 'end' element index exclusive.
     * @param start The inclusive "start" index of collection to be sliced.
     * @param end The exclusive "end" index of collection to be sliced
     */
    sliced(start, end) {
        return new Collection(new slicedWebElementsLocator_1.SlicedWebElementsLocator(start, end, this), this.configuration);
    }
    filteredBy(...conditions) {
        return this.by(...conditions);
    }
    by(...conditions) {
        return new Collection(new filteredByConditionWebElementsLocator_1.FilteredByConditionWebElementsLocator(wait_1.Condition.all(...conditions), this), this.configuration);
    }
    elementBy(...conditions) {
        return new element_1.Element(new byConditionWebElementLocator_1.ElementByConditionWebElementLocator(wait_1.Condition.all(...conditions), this), this.configuration);
    }
    collected(searchFunction) {
        return new Collection(new collectedByLocator_1.CollectedByLocator(searchFunction, this), this.configuration);
    }
    async getWebElements() {
        return this.locator.find();
    }
    toString() {
        return this.locator.toString();
    }
}
exports.Collection = Collection;
