// pega-model/src/widgets/controls/dropdown/dropdownCommon.ts
// Copyright 2020 Knowledge Expert SA
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import {Selector, I} from 'test-maker';
import {controlsCommonActions} from '../controlsCommonActions';
import {methodOptions} from '../../../../helpers/options/methodOptions';
import {PegaErrorMessages} from '../pega-error-messages/pegaErrorMessages';

export class DropdownCommon {

    public readonly element: string;
    public selected: boolean;

    constructor(element: string) {
        this.element = element;
    }

    private actualOptions;

    async waitUntilVisibility(options?: { timeout?: number, interval?: number, retries?: number }) {
        await controlsCommonActions.waitUntilVisibility(this.element, options);
    }

    async waitUntilVisibilityByText(text: string, options?: { timeout?: number, interval?: number, retries?: number }) {
        await controlsCommonActions.waitUntilVisibilityByText(this.element, text, options);
    }

    async waitUntilVisibilityByAttribute(attributeName: string, attributeValue?: string,
                                         options?: { timeout?: number, interval?: number, retries?: number }) {
        await controlsCommonActions.waitUntilVisibilityByAttribute(this.element, attributeName, attributeValue, options);
    }

    async waitUntilVisibilityByIndex(index: number, options?: { timeout?: number, interval?: number, retries?: number }) {
        await controlsCommonActions.waitUntilVisibilityByIndex(this.element, index, options);
    }

    async waitUntilCollectionOfElementsSizeIsGreaterOrEqual(expectedSize: number, options?: { timeout?: number, interval?: number, retries?: number }) {
        await controlsCommonActions.waitUntilCollectionOfElementsSizeIsGreaterOrEqual(this.element, expectedSize, options);
    }

    async waitUntilCollectionOfElementsSizeIsLessThan(expectedSize: number, options?: { timeout?: number, interval?: number, retries?: number }) {
        await controlsCommonActions.waitUntilCollectionOfElementsSizeIsLessThan(this.element, expectedSize, options);
    }

    async waitUntilInvisibility(options?: { timeout?: number, interval?: number, retries?: number }) {
        await controlsCommonActions.waitUntilInvisibility(this.element, options);
    }

    async waitUntilInvisibilityByIndex(index: number, options?: { timeout?: number, interval?: number, retries?: number }) {
        await controlsCommonActions.waitUntilInvisibilityByIndex(this.element, index, options);
    }

    async click(options?: { index?: number, timeout?: number, interval?: number, retries?: number }) {
        await controlsCommonActions.click(this.element, options);
    }

    async focus(options?: { timeout?: number, interval?: number, retries?: number }) {
        await controlsCommonActions.focus(this.element, options);
    }

    async pressTab() {
        await controlsCommonActions.pressTab();
    }

    async shouldBeVisible(options?: { index?: number, timeout?: number, assertionTimeout?: number, interval?: number, retries?: number }) {
        await controlsCommonActions.shouldBeVisible(this.element, options);
    }

    async shouldNotBeVisible(options?: { index?: number, timeout?: number, assertionTimeout?: number, interval?: number, retries?: number }) {
        await controlsCommonActions.shouldNotBeVisible(this.element, options);
    }

    async waitUntilElementExists(options?: { timeout?: number, assertionTimeout?: number, interval?: number, retries?: number }) {
        await controlsCommonActions.waitUntilElementExists(this.element, options);
    }

    async waitUntilElementNotExist(options?: { timeout?: number, assertionTimeout?: number, interval?: number, retries?: number }) {
        await controlsCommonActions.waitUntilElementNotExist(this.element, options);
    }

    async waitForSelectActionabilityCheck(checks, isFilterByVisibility = true) {
        await controlsCommonActions.waitForSelectActionabilityCheck(this.element, checks, isFilterByVisibility = true);
    }

    async shouldExist(options?: { index?: number, timeout?: number, interval?: number, retries?: number }) {
        await controlsCommonActions.shouldExist(this.element, options);
    }

    async shouldNotExist(options?: { index?: number, timeout?: number, interval?: number, retries?: number }) {
        await controlsCommonActions.shouldNotExist(this.element, options);
    }

    async shouldHaveText(value: string, options?: { index?: number, timeout?: number, assertionTimeout?: number, interval?: number, retries?: number, isCaseSensitive?: boolean }) {
        await controlsCommonActions.shouldHaveText(this.element, value, options);
    }

    async shouldHaveExactText(value: string, options?: { index?: number, timeout?: number, assertionTimeout?: number, interval?: number, retries?: number }) {
        await controlsCommonActions.shouldHaveExactText(this.element, value, options);
    }

    async shouldNotHaveExactText(value: string, options?: { index?: number, timeout?: number, assertionTimeout?: number, interval?: number, retries?: number }) {
        await controlsCommonActions.shouldNotHaveExactText(this.element, value, options);
    }

    async shouldNotHaveText(value: string, options?: { index?: number, timeout?: number, assertionTimeout?: number, interval?: number, retries?: number, isCaseSensitive?: boolean }) {
        await controlsCommonActions.shouldNotHaveText(this.element, value, options);
    }

    async shouldHaveValue(value: string | number, options?: { index?: number, timeout?: number, assertionTimeout?: number, interval?: number, retries?: number, isCaseSensitive?: boolean }) {
        await controlsCommonActions.shouldHaveValue(this.element, value, options);
    }

    async shouldNotHaveValue(value: string | number, options?: { index?: number, timeout?: number, assertionTimeout?: number, interval?: number, retries?: number, isCaseSensitive?: boolean }) {
        await controlsCommonActions.shouldNotHaveValue(this.element, value, options);
    }

    async shouldBeEnabled(options?: { index?: number, timeout?: number, assertionTimeout?: number, interval?: number, retries?: number }) {
        await controlsCommonActions.shouldBeEnabled(this.element, options);
    }

    async shouldBeDisabled(options?: { index?: number, timeout?: number, assertionTimeout?: number, interval?: number, retries?: number }) {
        await controlsCommonActions.shouldBeDisabled(this.element, options);
    }

    async waitForTheOptionToBeVisible(value: string, options?: { timeout?: number, interval?: number, retries?: number }) {
        this.actualOptions = methodOptions.setOptions(options);
        await I.waitForSelectorToBeVisible(Selector(`${this.element}//option[@value="${value}"]`, {
            timeout: this.actualOptions.timeout,
            interval: this.actualOptions.interval,
            retries: this.actualOptions.retries
        }));
    }

    async selectAndCheckTheValue(value: string, options?: { index?: number, timeout?: number, interval?: number, retries?: number }) {
        await this.select(value, options);
        await this.shouldHaveSelectedItem(value);
    }

    async select(value: string, options?: { index?: number, timeout?: number, interval?: number, retries?: number }) {
        if (value === undefined || value === null || value === '') {
            return;
        }
        await this.shouldBeVisible();
        await this.shouldBeEnabled();
        this.actualOptions = methodOptions.setOptions(options);
        await I.selectOption(Selector(this.element, this.actualOptions).nth(this.actualOptions.index), [value]);
    }

    async shouldHaveSelectedItem(value: string, options?: { index?: number, timeout?: number, assertionTimeout?: number, interval?: number, retries?: number }) {
        await this.shouldHaveValue(value, options);
    }

    async shouldNotHaveSelectedItem(value: string, options?: { index?: number, timeout?: number, assertionTimeout?: number, interval?: number, retries?: number }) {
        await this.shouldNotHaveValue(value, options);
    }

    async isSelected(value: string, options?: { index?: number, timeout?: number, assertionTimeout?: number, interval?: number, retries?: number }) {
        this.actualOptions = methodOptions.setOptions(options);
        let actualValue = await Selector(`${this.element}`, this.actualOptions)
            .nth(this.actualOptions.index).value;
        if (actualValue == value) {
            this.selected = true;
        }
        return this.selected;
    }

    async shouldHaveAttributeSelected(value: string, options?: { index?: number, timeout?: number, assertionTimeout?: number, interval?: number, retries?: number }) {
        this.actualOptions = methodOptions.setOptions(options);
        await I.expect(Selector(`${this.element}//option[@value="${value}"]`, this.actualOptions)
            .nth(this.actualOptions.index).hasAttribute('selected'), {
            timeout: this.actualOptions.assertionTimeout,
            interval: this.actualOptions.interval,
            retries: this.actualOptions.retries
        }).toBeOk();
    }

    async shouldNotHaveAttributeSelected(value: string, options?: { index?: number, timeout?: number, assertionTimeout?: number, interval?: number, retries?: number }) {
        this.actualOptions = methodOptions.setOptions(options);
        await I.expect(Selector(`${this.element}//option[@value="${value}"]`, this.actualOptions)
            .nth(this.actualOptions.index).hasAttribute('selected'), {
            timeout: this.actualOptions.assertionTimeout,
            interval: this.actualOptions.interval,
            retries: this.actualOptions.retries
        }).not.toBeOk();
    }

    async hasSelectedAttribute(value: string, options?: { index?: number, timeout?: number, assertionTimeout?: number, interval?: number, retries?: number }) {
        this.actualOptions = methodOptions.setOptions(options);
        this.selected = await Selector(`${this.element}//option[@value="${value}"]`, this.actualOptions)
            .nth(this.actualOptions.index).hasAttribute('selected'), {
            timeout: this.actualOptions.assertionTimeout,
            interval: this.actualOptions.interval,
            retries: this.actualOptions.retries
        }
        return this.selected;
    }

    async shouldHaveSuggestedItems(values: string[], options?: { timeout?: number, assertionTimeout?: number, interval?: number, retries?: number }) {
        this.actualOptions = methodOptions.setOptions(options);
        for (let i = 0; i < values.length; i++) {
            let value = values[i];
            await I.expectSelector(Selector(`${this.element}//option[@value="${value}"]`, this.actualOptions), {
                timeout: this.actualOptions.assertionTimeout,
                interval: this.actualOptions.interval,
                retries: this.actualOptions.retries
            }).toExist();
        }
    }

    async shouldOnlyHaveSuggestedItems(values: string[], isPlaceholderPresent = true, options?: { timeout?: number, assertionTimeout?: number, interval?: number, retries?: number }) {
        this.actualOptions = methodOptions.setOptions(options);
        let valuesNumber = values.length;
        let dropdownValuesNumber = await Selector(`${this.element}//option[@value]`, this.actualOptions).count;
        if (isPlaceholderPresent) {
            await I.expect(valuesNumber).toEqual(dropdownValuesNumber - 1);
        }
        else {
            await I.expect(valuesNumber).toEqual(dropdownValuesNumber - 1);
        }
        for (let i = 0; i < values.length; i++) {
            let value = values[i];
            await I.expectSelector(Selector(`${this.element}//option[@value="${value}"]`, this.actualOptions), {
                timeout: this.actualOptions.assertionTimeout,
                interval: this.actualOptions.interval,
                retries: this.actualOptions.retries
            }).toExist();
        }
    }

    async shouldHaveSuggestedItemsByInnerText(innerTexts: string[], options?: { timeout?: number, assertionTimeout?: number, interval?: number, retries?: number }) {
        this.actualOptions = methodOptions.setOptions(options);
        for (let i = 0; i < innerTexts.length; i++) {
            let text = innerTexts[i];
            await I.expectSelector(Selector(`${this.element}//option[text()="${text}"]`, this.actualOptions), {
                timeout: this.actualOptions.assertionTimeout,
                interval: this.actualOptions.interval,
                retries: this.actualOptions.retries
            }).toExist();
        }
    }

    async shouldNotHaveSuggestedItemsByInnerText(innerTexts: string[], options?: { timeout?: number, assertionTimeout?: number, interval?: number, retries?: number }) {
        this.actualOptions = methodOptions.setOptions(options);
        for (let i = 0; i < innerTexts.length; i++) {
            let text = innerTexts[i];
            await I.expectSelector(Selector(`${this.element}//option[text()="${text}"]`, this.actualOptions), {
                timeout: this.actualOptions.assertionTimeout,
                interval: this.actualOptions.interval,
                retries: this.actualOptions.retries
            }).not.toExist();
        }
    }

    async shouldHaveInnerTextPerOptionValue(optionValue: string, innerText: string, options?: { timeout?: number, assertionTimeout?: number, interval?: number, retries?: number }) {
        this.actualOptions = methodOptions.setOptions(options);
        await I.expectSelector(Selector(`${this.element}//option[@value="${optionValue}"]`), {
            timeout: this.actualOptions.assertionTimeout,
            interval: this.actualOptions.interval,
            retries: this.actualOptions.retries
        }).toHaveInnerText(innerText);
    }


    async shouldNotHaveInnerTextPerOptionValue(optionValue: string, innerText: string, options?: { timeout?: number, assertionTimeout?: number, interval?: number, retries?: number }) {
        this.actualOptions = methodOptions.setOptions(options);
        await I.expectSelector(Selector(`${this.element}//option[@value="${optionValue}"]`), {
            timeout: this.actualOptions.assertionTimeout,
            interval: this.actualOptions.interval,
            retries: this.actualOptions.retries
        }).not.toHaveInnerText(innerText);
    }

    async shouldHaveInnerTextsPerOptionValues(optionValues: string[], innerTexts: string[], options?: { timeout?: number, assertionTimeout?: number, interval?: number, retries?: number }) {
        for (let i = 0; i < optionValues.length; i++) {
            let value = optionValues[i];
            let text = innerTexts[i];
            await this.shouldHaveInnerTextPerOptionValue(value, text, options);
        }
    }

    async shouldNotHaveSuggestedItems(values: string[], options?: { timeout?: number, assertionTimeout?: number, interval?: number, retries?: number }) {
        this.actualOptions = methodOptions.setOptions(options);
        //await this.click(this.actualOptions);
        //await I.wait(1000);
        for (let i = 0; i < values.length; i++) {
            let value = values[i];
            await I.expectSelector(Selector(`${this.element}//option[@value="${value}"]`, this.actualOptions), {
                timeout: this.actualOptions.assertionTimeout,
                interval: this.actualOptions.interval,
                retries: this.actualOptions.retries
            }).not.toExist();
        }
    }

    async shouldBeRequired(options?: { timeout?: number, assertionTimeout?: number, interval?: number, retries?: number }) {
        await controlsCommonActions.shouldBeRequired(this.element, options);
    }

    async shouldNotBeRequired(options?: { timeout?: number, assertionTimeout?: number, interval?: number, retries?: number }) {
        await controlsCommonActions.shouldNotBeRequired(this.element, options);
    }

    async shouldHaveValidationType(validationType: string, options?: { timeout?: number, assertionTimeout?: number, interval?: number, retries?: number }) {
        await controlsCommonActions.shouldHaveValidationType(this.element, validationType, options);
    }

    async shouldNotHaveValidationType(validationType: string, options?: { timeout?: number, assertionTimeout?: number, interval?: number, retries?: number }) {
        await controlsCommonActions.shouldNotHaveValidationType(this.element, validationType, options);
    }


    async isListOfItemsVisible(values: string[], options?: { timeout?: number, assertionTimeout?: number, interval?: number, retries?: number }): Promise<boolean> {
        this.actualOptions = methodOptions.setOptions(options);
        for (let i = 0; i < values.length; i++) {
            let value = values[i];
            let item = Selector(`${this.element}//option[@value="${value}"]`, this.actualOptions).filterVisible();
            if (!item.visible) {
                return false;
            }
        }
        return true;
    }

    async getValue() {
        return await controlsCommonActions.getValue(this.element);
    }

    async getText() {
        return await controlsCommonActions.getText(this.element);
    }

    error() {
        return new PegaErrorMessages(this.element);
    }

    async isVisible(options?: { filterByVisibility?: boolean, index?: number, timeout?: number, assertionTimeout?: number, interval?: number, retries?: number }) {
        return await controlsCommonActions.isVisible(this.element, options);
    }

    async isExists(options?: { index?: number, timeout?: number, assertionTimeout?: number, interval?: number, retries?: number }) {
        return await controlsCommonActions.isExists(this.element, options);
    }
}
