// pega-model/src/widgets/controls/radiobuttons/radiobuttonCommon.ts
// Copyright 2020 Knowledge Expert SA
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import {methodOptions, Options} from '../../../../helpers/options/methodOptions';
import {I, Selector} from 'test-maker';

export class RadiobuttonCommon {

    private actualOptions;

    async shouldBeChecked(elementToCheck: string, options?: { index?: number, timeout?: number, assertionTimeout?: number, interval?: number, retries?: number }) {
        this.actualOptions = methodOptions.setOptions(options);
        await I.expect(Selector(elementToCheck, this.actualOptions).filterVisible().nth(this.actualOptions.index).checked, {
            timeout: this.actualOptions.assertionTimeout,
            interval: this.actualOptions.interval,
            retries: this.actualOptions.retries
        }).toBeOk();
    }

    async shouldNotBeChecked(elementToCheck: string, options?: { index?: number, timeout?: number, assertionTimeout?: number, interval?: number, retries?: number }) {
        this.actualOptions = methodOptions.setOptions(options);
        await I.expect(Selector(elementToCheck, this.actualOptions).filterVisible().nth(this.actualOptions.index).hasAttribute('checked'), {
            timeout: this.actualOptions.assertionTimeout,
            interval: this.actualOptions.interval,
            retries: this.actualOptions.retries
        }).not.toBeOk();
    }

}

export const radiobuttonCommon = new RadiobuttonCommon();
