// test-maker/dist/102.js
'use strict';
exports['id'] = 0x66, exports['ids'] = [
    0x66,
    0xb4
], exports['modules'] = {
    0x17166: (_0x3bb08e, _0x45a2f8, _0x3f26d3) => {
        _0x3f26d3['r'](_0x45a2f8), _0x3f26d3['d'](_0x45a2f8, {
            'Runner': () => _0x42c5a9,
            'runner': () => _0x2e4273
        });
        var _0x387783 = _0x3f26d3(0x1145c), _0x150351 = _0x3f26d3['n'](_0x387783), _0x156753 = _0x3f26d3(0x5885), _0x5aa407 = _0x3f26d3(0x177ce), _0x5d76f8 = _0x3f26d3['n'](_0x5aa407), _0xa23ca = _0x3f26d3(0x17fbe), _0x527471 = _0x3f26d3(0x9677), _0x521759 = _0x3f26d3(0x5dbf), _0x2d4ad6 = _0x3f26d3['n'](_0x521759), _0x4c1121 = _0x3f26d3(0x3b1), _0xaadb68 = _0x3f26d3['n'](_0x4c1121), _0x28ef3d = _0x3f26d3(0x135fa), _0x49568a = _0x3f26d3['n'](_0x28ef3d), _0x593367 = _0x3f26d3(0xbb05), _0xd6a305 = _0x3f26d3(0x14e4b), _0x127752 = _0x3f26d3(0xe982), _0x1c6815 = _0x3f26d3(0x17067), _0x4ad6ac = _0x3f26d3(0x2fa5), _0xbcb6e4 = _0x3f26d3(0xc491);
        ;
        function _0x805c31(_0x51a1f1) {
            const _0x380bf7 = _0x51a1f1['toLowerCase']();
            switch (_0x380bf7) {
            case 'true':
                return !![];
            case 'false':
                return ![];
            case 'nan':
                return NaN;
            case '':
            case 'undefined':
                return undefined;
            case 'null':
                return null;
            default:
                return isNaN(Number(_0x380bf7)) ? _0x51a1f1 : Number(_0x380bf7);
            }
        }
        var _0xdfcefa = _0x3f26d3(0x17f4f), _0x482b81 = _0x3f26d3(0xa2b4), _0x2381b0 = _0x3f26d3(0x677b), _0x38dc7e = _0x3f26d3(0x16259), _0x3cb4fa = _0x3f26d3(0x18425), _0x3054a9 = _0x3f26d3(0x81d5), _0x49f515 = _0x3f26d3(0x1eb4), _0x1cb4c2 = _0x3f26d3(0x1201f);
        ;
        class _0x42c5a9 {
            async ['start'](_0x5b7dc7) {
                this['options'] = _0x5b7dc7, await this['compileConfig']();
                if (_0xd6a305['configManager']['configuration']['runner']['liveMode'])
                    await this['watchTestMakerConfig']();
                _0x1c6815['stateManager']['state']['cliArgs'] = this['options'], _0x1cb4c2['internalLogger']['level'] = _0xd6a305['configManager']['configuration']['logging']['globalLogLevel'], await _0x127752['instanceManager']['init'](), await _0xbcb6e4['tmEventsManger']['emit'](_0x4ad6ac['TMEvents']['EngineStart']);
            }
            async ['watchTestMakerConfig']() {
                if (this['options']['config'])
                    _0x156753['watch'](_0xaadb68()['joinSafe'](_0x593367['projectPath'], this['options']['config']), {
                        'persistent': !![],
                        'awaitWriteFinish': !![],
                        'ignoreInitial': !![]
                    })['on']('all', async () => {
                        _0x5d76f8()['warn']('Test\x20Maker\x20Config\x20file\x20has\x20been\x20modified,\x20Test\x20Maker\x20needs\x20to\x20be\x20restarted.'), process['exitCode'] = 0x400, (0x0, _0xdfcefa['exitProcess'])(0x0);
                    });
            }
            async ['compileConfig']() {
                if (this['options']['config']) {
                    const _0x3883ba = _0xaadb68()['resolve'](_0xaadb68()['normalizeTrim'](this['options']['config']));
                    _0x5d76f8()['info']('Compiling\x20Test\x20Maker\x20Config\x20File:\x20' + _0x3883ba);
                    const _0x1cb31d = await (0x0, _0x49f515['webpackMemoryCompiler'])({
                            'webpackCompiler': _0x49568a(),
                            'config': (await _0x3f26d3['e'](0x3e3)['then'](_0x3f26d3['bind'](_0x3f26d3, 0x10d83)))['getBundleConfig'](this['options']['config'], _0x3883ba)
                        }), _0x32dd6d = _0x593367['testMakerArtifactsPath'] + '/' + _0xaadb68()['basename'](_0x3883ba);
                    await _0xa23ca['outputFile'](_0x32dd6d, _0x1cb31d);
                    const _0x570b5e = (0x0, _0x38dc7e['nativeRequire'])(_0x32dd6d), _0x154fc7 = this['options']['config'] ? _0x570b5e['default'] : {};
                    this['setupOptions']({
                        'options': this['options'],
                        'fileConfiguration': _0x154fc7
                    });
                } else
                    this['setupOptions']({
                        'options': this['options'],
                        'fileConfiguration': {}
                    });
            }
            ['setupOptions']({
                options: _0x18c01d,
                fileConfiguration: _0xcc23bb
            }) {
                this['mergeConfig']({
                    'options': _0x18c01d,
                    'fileConfiguration': _0xcc23bb
                }), this['validateNodeversionOption'](), this['normalizeFilteringOptions'](), this['normalizeOverrideOptions'](_0x18c01d), this['normalizeSuitesOptions'](_0x18c01d), this['normalizeExtractions'](_0x18c01d), this['normalizeOutputOptions']();
            }
            ['parseExtraOptions'](_0x1bac44, _0x2c6237) {
                try {
                    const _0x311116 = new _0x3cb4fa['PropertyAccessor'](_0x2c6237), _0x4b8e1b = {}, _0x1523c1 = Array['isArray'](_0x1bac44) ? _0x1bac44 : [exports];
                    return _0x1523c1['forEach'](_0x111a10 => {
                        const _0x3ee5b0 = _0x111a10['split']('=');
                        if (_0x3ee5b0['length'] > 0x1) {
                            const _0xd29e4d = _0x805c31(_0x3ee5b0[0x1]);
                            _0x311116['target'] = (0x0, _0x2381b0['mergeAndConcat'])(_0x311116['target'], _0x311116['set'](_0x3ee5b0[0x0], _0xd29e4d)), _0x4b8e1b[_0x3ee5b0[0x0]] = _0xd29e4d;
                        } else
                            _0x4b8e1b[_0x111a10] = _0x311116['get'](_0x111a10);
                    }), _0x4b8e1b;
                } catch (_0x306d92) {
                    _0x306d92['name'] = 'Extra\x20Options\x20Setup', _0x306d92['message'] = '' + _0x306d92['message'];
                    throw _0x306d92;
                }
            }
            ['normalizeFilteringOptions']() {
                const _0x1e3dc1 = _0xd6a305['configManager']['configuration']['filtering'], _0x49a536 = _0x5b5751 => {
                        return _0x5b5751 ? Array['isArray'](_0x5b5751) ? _0x5b5751 : [_0x5b5751] : [];
                    };
                _0x1e3dc1 && (_0x1e3dc1['features'] = _0x49a536(_0x1e3dc1['features']), _0x1e3dc1['featuresTags'] = _0x49a536(_0x1e3dc1['featuresTags']), _0x1e3dc1['scenarios'] = _0x49a536(_0x1e3dc1['scenarios']), _0x1e3dc1['scenariosTags'] = _0x49a536(_0x1e3dc1['scenariosTags']), _0x1e3dc1['steps'] = _0x49a536(_0x1e3dc1['steps']), _0x1e3dc1['stepsTags'] = _0x49a536(_0x1e3dc1['stepsTags']));
            }
            ['validateNodeversionOption']() {
                var _0x50098f, _0x11064b;
                if ((_0x50098f = _0xd6a305['configManager']['configuration']['runner']) === null || _0x50098f === void 0x0 ? void 0x0 : _0x50098f['nodeVersion'])
                    !_0x2d4ad6()['satisfies'](process['version'], (_0x11064b = _0xd6a305['configManager']['configuration']['runner']) === null || _0x11064b === void 0x0 ? void 0x0 : _0x11064b['nodeVersion']) && (_0x5d76f8()['error']('The\x20Specified\x20NodeJS\x20version:' + _0x150351()['yellowBright'](_0xd6a305['configManager']['configuration']['runner']['nodeVersion']) + '\x20in\x20config\x20does\x20not\x20satisfy\x20the\x20one\x20installed:\x20' + _0x150351()['yellowBright'](process['version'])), (0x0, _0xdfcefa['exitProcess'])(0x1));
                else
                    !_0x2d4ad6()['satisfies'](process['version'], '>=14.x') && _0x5d76f8()['warn']('Test\x20Maker\x20may\x20not\x20work\x20correctly\x20with\x20Node\x20Version\x20less\x20than\x20\x2214x\x20-\x20current\x20installed\x20Node\x20version:\x20' + process['version']);
            }
            ['normalizeSuitesOptions'](_0x297f35) {
                const _0x1c7033 = {};
                _0x297f35['suites']['forEach'](_0x2a8012 => {
                    var _0x112435;
                    let _0x44d654 = (_0x112435 = _0xd6a305['configManager']['configuration']['suites']) === null || _0x112435 === void 0x0 ? void 0x0 : _0x112435[_0x2a8012];
                    if (!_0x44d654)
                        throw new Error('Suite:\x20\x22' + _0x2a8012 + '\x22\x20is\x20not\x20defined\x20in\x20Config\x20file.\x20please\x20add\x20one\x20with\x20the\x20same\x20name.');
                    if (Array['isArray'](_0x44d654))
                        _0x44d654 = { 'path': _0x44d654 };
                    _0x1c7033[_0x2a8012] = _0x44d654;
                });
                if (Object['keys'](_0x1c7033)['length'])
                    _0xd6a305['configManager']['configuration']['suites'] = _0x1c7033;
            }
            ['normalizeExtractions'](_0x2569fa) {
                this['parseExtraOptions'](_0x2569fa['extra'], _0xd6a305['configManager']['configuration']['extra'] || {});
            }
            ['normalizeOutputOptions']() {
                var _0x29713b, _0x4b8767, _0x4183f6, _0x29e3d1;
                if ((_0x4b8767 = (_0x29713b = _0xd6a305['configManager']['configuration']['build']) === null || _0x29713b === void 0x0 ? void 0x0 : _0x29713b['output']) === null || _0x4b8767 === void 0x0 ? void 0x0 : _0x4b8767['path'])
                    _0xd6a305['configManager']['configuration']['build']['output']['path'] = _0xaadb68()['resolve'](((_0x29e3d1 = (_0x4183f6 = _0xd6a305['configManager']['configuration']['build']) === null || _0x4183f6 === void 0x0 ? void 0x0 : _0x4183f6['output']) === null || _0x29e3d1 === void 0x0 ? void 0x0 : _0x29e3d1['path']) || '');
            }
            ['normalizeOverrideOptions'](_0x459020) {
                if (_0x459020['override'] && _0x459020['override'] !== 'true' && _0x459020['override']['length'])
                    _0xd6a305['configManager']['configuration'] = (0x0, _0x527471['merge'])(_0xd6a305['configManager']['configuration'], (0x0, _0x3054a9['virtualModule'])('module.exports=\x20' + _0x459020['override'], (0x0, _0x482b81['generateId'])()));
            }
            ['mergeConfig']({
                options: _0x3fddf4,
                fileConfiguration: _0x1d3484
            }) {
                _0xd6a305['configManager']['configuration'] = (0x0, _0x527471['merge'])(_0xd6a305['configManager']['configuration'], _0x1d3484);
                const _0x475714 = {
                    'runner': {
                        'startPage': _0x3fddf4['startPage'],
                        'liveMode': _0x3fddf4['liveMode'],
                        'headless': _0x3fddf4['headless'],
                        'parallel': {
                            'features': _0x3fddf4['parallelFeatures'],
                            'clients': _0x3fddf4['parallelClients'],
                            'adapters': _0x3fddf4['parallelAdapters']
                        }
                    },
                    'reporting': { 'reporters': _0x3fddf4['reporters'] },
                    'source': _0x3fddf4['source'],
                    'filtering': {
                        'glob': _0x3fddf4['filterGlob'],
                        'features': _0x3fddf4['filterFeature'],
                        'featuresTags': _0x3fddf4['filterFeatureTag'],
                        'scenarios': _0x3fddf4['filterScenario'],
                        'scenariosTags': _0x3fddf4['filterScenarioTag'],
                        'steps': _0x3fddf4['filterStep'],
                        'stepsTags': _0x3fddf4['filterStepTag']
                    }
                };
                _0xd6a305['configManager']['configuration'] = (0x0, _0x527471['mergeAndCompare'])((_0x174a59, _0x433e43) => {
                    return _0x433e43 === undefined ? _0x174a59 : _0x433e43;
                }, _0xd6a305['configManager']['configuration'], _0x475714);
            }
        }
        const _0x2e4273 = new _0x42c5a9();
    },
    0x79cc: (_0x106a5a, _0x1ed847, _0x1408ca) => {
        _0x1408ca['r'](_0x1ed847), _0x1408ca['d'](_0x1ed847, { 'logGenericError': () => _0x5d0298 });
        var _0x548f59 = _0x1408ca(0x1145c), _0x477177 = _0x1408ca['n'](_0x548f59);
        async function _0x5d0298({
            error: _0x6f08fa,
            isWarning: _0x5af2c2,
            show: show = !![]
        }) {
            let _0x22a51d;
            try {
                _0x22a51d ? _0x5af2c2 ? console['warn'](_0x22a51d['formatted']['ansi']) : console['error'](_0x22a51d['formatted']['ansi']) : _0x5af2c2 ? console['warn'](_0x6f08fa) : console['error'](_0x477177()['redBright'](_0x6f08fa['stack'] || _0x6f08fa));
            } catch (_0x19309) {
                console['error'](_0x477177()['redBright'](_0x19309['stack'] || _0x19309));
            } finally {
                console['info'](_0x477177()['cyanBright']('See\x20Log\x20file\x20for\x20more\x20Details...'));
            }
        }
    },
    0x1eb4: (_0xea0e58, _0x37fd54, _0x1455f4) => {
        _0x1455f4['r'](_0x37fd54), _0x1455f4['d'](_0x37fd54, { 'webpackMemoryCompiler': () => _0x3bf719 });
        var _0x53d467 = _0x1455f4(0x17fbe), _0x4efeee = _0x1455f4['n'](_0x53d467), _0xdacb13 = _0x1455f4(0x5b3f), _0x56ac24 = _0x1455f4['n'](_0xdacb13), _0x4c7bf1 = _0x1455f4(0x5615), _0x540e13 = _0x1455f4['n'](_0x4c7bf1), _0x364ce2 = _0x1455f4(0x11569), _0x31d9cf = _0x1455f4['n'](_0x364ce2), _0x4227e1 = _0x1455f4(0xa2b4);
        function _0xc35a38(_0x2e0297 = (0x0, _0x4227e1['generateId'])()) {
            const _0x442378 = (0x0, _0x4c7bf1['tmpdir'])(), _0x2fd504 = _0x2e0297 + '.js', _0x3f5102 = (0x0, _0x364ce2['join'])(_0x442378, _0x2fd504), _0x3ed086 = async _0x2a4308 => {
                    await (0x0, _0x53d467['writeFile'])(_0x3f5102, _0x2a4308);
                }, _0x3191b4 = async () => {
                    await (0x0, _0x53d467['unlink'])(_0x3f5102);
                };
            return {
                'path': _0x442378,
                'filename': _0x2fd504,
                'filepath': _0x3f5102,
                'write': _0x3ed086,
                'remove': _0x3191b4
            };
        }
        async function _0x3bf719({
            webpackCompiler: _0x1b54fe,
            config: config = { 'mode': 'production' },
            code: _0x5dfb60
        }) {
            const _0x3ef290 = {
                ...config,
                'output': {
                    ...config['output'],
                    'path': '/',
                    'filename': 'output.js'
                }
            };
            let _0x5dd19a;
            _0x5dfb60 && (_0x5dd19a = _0xc35a38(), await _0x5dd19a['write'](_0x5dfb60), _0x3ef290['entry'] = _0x5dd19a['filepath']);
            const _0x24a067 = _0x1b54fe(_0x3ef290);
            _0x24a067['outputFileSystem'] = new (_0x56ac24())();
            const _0x28889d = await new Promise((_0x8f690c, _0x245c71) => {
                _0x24a067['run']((_0x49c1e1, _0x411fda) => {
                    if (_0x49c1e1)
                        return _0x245c71(_0x49c1e1);
                    if (_0x411fda === null || _0x411fda === void 0x0 ? void 0x0 : _0x411fda['hasErrors']())
                        return _0x245c71(new Error(_0x411fda['toString']({
                            'errorDetails': !![],
                            'warnings': !![]
                        })));
                    _0x8f690c(_0x24a067['outputFileSystem']['data']['output.js']['toString']());
                });
            });
            if (_0x5dd19a)
                await _0x5dd19a['remove']();
            return _0x28889d;
        }
    }
};
;
