// test-maker/dist/123.js
exports['id'] = 0x7b, exports['ids'] = [0x7b], exports['modules'] = {
    0x93bc: (_0x150c0f, _0x214f4b, _0x337343) => {
        _0x214f4b['formatArgs'] = _0x1d254b, _0x214f4b['save'] = _0x3f36be, _0x214f4b['load'] = _0x3c0e63, _0x214f4b['useColors'] = _0x229423, _0x214f4b['storage'] = _0x3c8458(), _0x214f4b['destroy'] = ((() => {
            let _0x1eec47 = ![];
            return () => {
                !_0x1eec47 && (_0x1eec47 = !![], console['warn']('Instance\x20method\x20`debug.destroy()`\x20is\x20deprecated\x20and\x20no\x20longer\x20does\x20anything.\x20It\x20will\x20be\x20removed\x20in\x20the\x20next\x20major\x20version\x20of\x20`debug`.'));
            };
        })()), _0x214f4b['colors'] = [
            '#0000CC',
            '#0000FF',
            '#0033CC',
            '#0033FF',
            '#0066CC',
            '#0066FF',
            '#0099CC',
            '#0099FF',
            '#00CC00',
            '#00CC33',
            '#00CC66',
            '#00CC99',
            '#00CCCC',
            '#00CCFF',
            '#3300CC',
            '#3300FF',
            '#3333CC',
            '#3333FF',
            '#3366CC',
            '#3366FF',
            '#3399CC',
            '#3399FF',
            '#33CC00',
            '#33CC33',
            '#33CC66',
            '#33CC99',
            '#33CCCC',
            '#33CCFF',
            '#6600CC',
            '#6600FF',
            '#6633CC',
            '#6633FF',
            '#66CC00',
            '#66CC33',
            '#9900CC',
            '#9900FF',
            '#9933CC',
            '#9933FF',
            '#99CC00',
            '#99CC33',
            '#CC0000',
            '#CC0033',
            '#CC0066',
            '#CC0099',
            '#CC00CC',
            '#CC00FF',
            '#CC3300',
            '#CC3333',
            '#CC3366',
            '#CC3399',
            '#CC33CC',
            '#CC33FF',
            '#CC6600',
            '#CC6633',
            '#CC9900',
            '#CC9933',
            '#CCCC00',
            '#CCCC33',
            '#FF0000',
            '#FF0033',
            '#FF0066',
            '#FF0099',
            '#FF00CC',
            '#FF00FF',
            '#FF3300',
            '#FF3333',
            '#FF3366',
            '#FF3399',
            '#FF33CC',
            '#FF33FF',
            '#FF6600',
            '#FF6633',
            '#FF9900',
            '#FF9933',
            '#FFCC00',
            '#FFCC33'
        ];
        function _0x229423() {
            if (typeof window !== 'undefined' && window['process'] && (window['process']['type'] === 'renderer' || window['process']['__nwjs']))
                return !![];
            if (typeof navigator !== 'undefined' && navigator['userAgent'] && navigator['userAgent']['toLowerCase']()['match'](/(edge|trident)\/(\d+)/))
                return ![];
            return typeof document !== 'undefined' && document['documentElement'] && document['documentElement']['style'] && document['documentElement']['style']['WebkitAppearance'] || typeof window !== 'undefined' && window['console'] && (window['console']['firebug'] || window['console']['exception'] && window['console']['table']) || typeof navigator !== 'undefined' && navigator['userAgent'] && navigator['userAgent']['toLowerCase']()['match'](/firefox\/(\d+)/) && parseInt(RegExp['$1'], 0xa) >= 0x1f || typeof navigator !== 'undefined' && navigator['userAgent'] && navigator['userAgent']['toLowerCase']()['match'](/applewebkit\/(\d+)/);
        }
        function _0x1d254b(_0x465370) {
            _0x465370[0x0] = (this['useColors'] ? '%c' : '') + this['namespace'] + (this['useColors'] ? '\x20%c' : '\x20') + _0x465370[0x0] + (this['useColors'] ? '%c\x20' : '\x20') + '+' + _0x150c0f['exports']['humanize'](this['diff']);
            if (!this['useColors'])
                return;
            const _0x527fc5 = 'color:\x20' + this['color'];
            _0x465370['splice'](0x1, 0x0, _0x527fc5, 'color:\x20inherit');
            let _0x25f79e = 0x0, _0x387388 = 0x0;
            _0x465370[0x0]['replace'](/%[a-zA-Z%]/g, _0x7701f7 => {
                if (_0x7701f7 === '%%')
                    return;
                _0x25f79e++, _0x7701f7 === '%c' && (_0x387388 = _0x25f79e);
            }), _0x465370['splice'](_0x387388, 0x0, _0x527fc5);
        }
        _0x214f4b['log'] = console['debug'] || console['log'] || (() => {
        });
        function _0x3f36be(_0x247525) {
            try {
                _0x247525 ? _0x214f4b['storage']['setItem']('debug', _0x247525) : _0x214f4b['storage']['removeItem']('debug');
            } catch (_0x401a5d) {
            }
        }
        function _0x3c0e63() {
            let _0x3fc857;
            try {
                _0x3fc857 = _0x214f4b['storage']['getItem']('debug');
            } catch (_0x4e1625) {
            }
            return !_0x3fc857 && typeof process !== 'undefined' && 'env' in process && (_0x3fc857 = process['env']['DEBUG']), _0x3fc857;
        }
        function _0x3c8458() {
            try {
                return localStorage;
            } catch (_0x375541) {
            }
        }
        _0x150c0f['exports'] = _0x337343(0x4817)(_0x214f4b);
        const {formatters: _0xe5fa1c} = _0x150c0f['exports'];
        _0xe5fa1c['j'] = function (_0x5438be) {
            try {
                return JSON['stringify'](_0x5438be);
            } catch (_0x459708) {
                return '[UnexpectedJSONParseError]:\x20' + _0x459708['message'];
            }
        };
    },
    0x4817: (_0x32ed4f, _0x49ac7f, _0x41f961) => {
        function _0x171ace(_0x34353e) {
            _0x5463ed['debug'] = _0x5463ed, _0x5463ed['default'] = _0x5463ed, _0x5463ed['coerce'] = _0x1944e0, _0x5463ed['disable'] = _0x1d0916, _0x5463ed['enable'] = _0x39ba6e, _0x5463ed['enabled'] = _0x283035, _0x5463ed['humanize'] = _0x41f961(0xd552), _0x5463ed['destroy'] = _0x1f5944, Object['keys'](_0x34353e)['forEach'](_0x4029f5 => {
                _0x5463ed[_0x4029f5] = _0x34353e[_0x4029f5];
            }), _0x5463ed['names'] = [], _0x5463ed['skips'] = [], _0x5463ed['formatters'] = {};
            function _0xb8381e(_0x54247c) {
                let _0x3eca97 = 0x0;
                for (let _0x3b729e = 0x0; _0x3b729e < _0x54247c['length']; _0x3b729e++) {
                    _0x3eca97 = (_0x3eca97 << 0x5) - _0x3eca97 + _0x54247c['charCodeAt'](_0x3b729e), _0x3eca97 |= 0x0;
                }
                return _0x5463ed['colors'][Math['abs'](_0x3eca97) % _0x5463ed['colors']['length']];
            }
            _0x5463ed['selectColor'] = _0xb8381e;
            function _0x5463ed(_0x257c8a) {
                let _0x3d5e75, _0x23bee9 = null, _0x77141a, _0x127f99;
                function _0x1f4044(..._0x52072d) {
                    if (!_0x1f4044['enabled'])
                        return;
                    const _0x34ea09 = _0x1f4044, _0x39f190 = Number(new Date()), _0x3037dd = _0x39f190 - (_0x3d5e75 || _0x39f190);
                    _0x34ea09['diff'] = _0x3037dd, _0x34ea09['prev'] = _0x3d5e75, _0x34ea09['curr'] = _0x39f190, _0x3d5e75 = _0x39f190, _0x52072d[0x0] = _0x5463ed['coerce'](_0x52072d[0x0]);
                    typeof _0x52072d[0x0] !== 'string' && _0x52072d['unshift']('%O');
                    let _0x191204 = 0x0;
                    _0x52072d[0x0] = _0x52072d[0x0]['replace'](/%([a-zA-Z%])/g, (_0x7adf13, _0x28b398) => {
                        if (_0x7adf13 === '%%')
                            return '%';
                        _0x191204++;
                        const _0x26e59f = _0x5463ed['formatters'][_0x28b398];
                        if (typeof _0x26e59f === 'function') {
                            const _0x173c16 = _0x52072d[_0x191204];
                            _0x7adf13 = _0x26e59f['call'](_0x34ea09, _0x173c16), _0x52072d['splice'](_0x191204, 0x1), _0x191204--;
                        }
                        return _0x7adf13;
                    }), _0x5463ed['formatArgs']['call'](_0x34ea09, _0x52072d);
                    const _0x4237b7 = _0x34ea09['log'] || _0x5463ed['log'];
                    _0x4237b7['apply'](_0x34ea09, _0x52072d);
                }
                return _0x1f4044['namespace'] = _0x257c8a, _0x1f4044['useColors'] = _0x5463ed['useColors'](), _0x1f4044['color'] = _0x5463ed['selectColor'](_0x257c8a), _0x1f4044['extend'] = _0x252c25, _0x1f4044['destroy'] = _0x5463ed['destroy'], Object['defineProperty'](_0x1f4044, 'enabled', {
                    'enumerable': !![],
                    'configurable': ![],
                    'get': () => {
                        if (_0x23bee9 !== null)
                            return _0x23bee9;
                        return _0x77141a !== _0x5463ed['namespaces'] && (_0x77141a = _0x5463ed['namespaces'], _0x127f99 = _0x5463ed['enabled'](_0x257c8a)), _0x127f99;
                    },
                    'set': _0x414744 => {
                        _0x23bee9 = _0x414744;
                    }
                }), typeof _0x5463ed['init'] === 'function' && _0x5463ed['init'](_0x1f4044), _0x1f4044;
            }
            function _0x252c25(_0x3576c9, _0x5bbc10) {
                const _0x2cdbf1 = _0x5463ed(this['namespace'] + (typeof _0x5bbc10 === 'undefined' ? ':' : _0x5bbc10) + _0x3576c9);
                return _0x2cdbf1['log'] = this['log'], _0x2cdbf1;
            }
            function _0x39ba6e(_0x31d206) {
                _0x5463ed['save'](_0x31d206), _0x5463ed['namespaces'] = _0x31d206, _0x5463ed['names'] = [], _0x5463ed['skips'] = [];
                let _0x1318bd;
                const _0x256def = (typeof _0x31d206 === 'string' ? _0x31d206 : '')['split'](/[\s,]+/), _0x58c9f7 = _0x256def['length'];
                for (_0x1318bd = 0x0; _0x1318bd < _0x58c9f7; _0x1318bd++) {
                    if (!_0x256def[_0x1318bd])
                        continue;
                    _0x31d206 = _0x256def[_0x1318bd]['replace'](/\*/g, '.*?'), _0x31d206[0x0] === '-' ? _0x5463ed['skips']['push'](new RegExp('^' + _0x31d206['slice'](0x1) + '$')) : _0x5463ed['names']['push'](new RegExp('^' + _0x31d206 + '$'));
                }
            }
            function _0x1d0916() {
                const _0x3c33d1 = [
                    ..._0x5463ed['names']['map'](_0x299f6e),
                    ..._0x5463ed['skips']['map'](_0x299f6e)['map'](_0x557681 => '-' + _0x557681)
                ]['join'](',');
                return _0x5463ed['enable'](''), _0x3c33d1;
            }
            function _0x283035(_0x15a199) {
                if (_0x15a199[_0x15a199['length'] - 0x1] === '*')
                    return !![];
                let _0x5c4143, _0x113b02;
                for (_0x5c4143 = 0x0, _0x113b02 = _0x5463ed['skips']['length']; _0x5c4143 < _0x113b02; _0x5c4143++) {
                    if (_0x5463ed['skips'][_0x5c4143]['test'](_0x15a199))
                        return ![];
                }
                for (_0x5c4143 = 0x0, _0x113b02 = _0x5463ed['names']['length']; _0x5c4143 < _0x113b02; _0x5c4143++) {
                    if (_0x5463ed['names'][_0x5c4143]['test'](_0x15a199))
                        return !![];
                }
                return ![];
            }
            function _0x299f6e(_0x8ec189) {
                return _0x8ec189['toString']()['substring'](0x2, _0x8ec189['toString']()['length'] - 0x2)['replace'](/\.\*\?$/, '*');
            }
            function _0x1944e0(_0x272e32) {
                if (_0x272e32 instanceof Error)
                    return _0x272e32['stack'] || _0x272e32['message'];
                return _0x272e32;
            }
            function _0x1f5944() {
                console['warn']('Instance\x20method\x20`debug.destroy()`\x20is\x20deprecated\x20and\x20no\x20longer\x20does\x20anything.\x20It\x20will\x20be\x20removed\x20in\x20the\x20next\x20major\x20version\x20of\x20`debug`.');
            }
            return _0x5463ed['enable'](_0x5463ed['load']()), _0x5463ed;
        }
        _0x32ed4f['exports'] = _0x171ace;
    },
    0x566b: (_0x1722c5, _0x4322d9, _0x517c95) => {
        typeof process === 'undefined' || process['type'] === 'renderer' || process['browser'] === !![] || process['__nwjs'] ? _0x1722c5['exports'] = _0x517c95(0x93bc) : _0x1722c5['exports'] = _0x517c95(0xc933);
    },
    0xc933: (_0x9b2084, _0xec5ee2, _0x2c7057) => {
        const _0x1a66bb = _0x2c7057(0x129c0), _0x2b2e3c = _0x2c7057(0x1206d);
        _0xec5ee2['init'] = _0x464ff0, _0xec5ee2['log'] = _0xadc755, _0xec5ee2['formatArgs'] = _0x2fe862, _0xec5ee2['save'] = _0x3ffcc5, _0xec5ee2['load'] = _0x5c6a5f, _0xec5ee2['useColors'] = _0x3f348b, _0xec5ee2['destroy'] = _0x2b2e3c['deprecate'](() => {
        }, 'Instance\x20method\x20`debug.destroy()`\x20is\x20deprecated\x20and\x20no\x20longer\x20does\x20anything.\x20It\x20will\x20be\x20removed\x20in\x20the\x20next\x20major\x20version\x20of\x20`debug`.'), _0xec5ee2['colors'] = [
            0x6,
            0x2,
            0x3,
            0x4,
            0x5,
            0x1
        ];
        try {
            const _0x5d80d0 = _0x2c7057(0x9583);
            _0x5d80d0 && (_0x5d80d0['stderr'] || _0x5d80d0)['level'] >= 0x2 && (_0xec5ee2['colors'] = [
                0x14,
                0x15,
                0x1a,
                0x1b,
                0x20,
                0x21,
                0x26,
                0x27,
                0x28,
                0x29,
                0x2a,
                0x2b,
                0x2c,
                0x2d,
                0x38,
                0x39,
                0x3e,
                0x3f,
                0x44,
                0x45,
                0x4a,
                0x4b,
                0x4c,
                0x4d,
                0x4e,
                0x4f,
                0x50,
                0x51,
                0x5c,
                0x5d,
                0x62,
                0x63,
                0x70,
                0x71,
                0x80,
                0x81,
                0x86,
                0x87,
                0x94,
                0x95,
                0xa0,
                0xa1,
                0xa2,
                0xa3,
                0xa4,
                0xa5,
                0xa6,
                0xa7,
                0xa8,
                0xa9,
                0xaa,
                0xab,
                0xac,
                0xad,
                0xb2,
                0xb3,
                0xb8,
                0xb9,
                0xc4,
                0xc5,
                0xc6,
                0xc7,
                0xc8,
                0xc9,
                0xca,
                0xcb,
                0xcc,
                0xcd,
                0xce,
                0xcf,
                0xd0,
                0xd1,
                0xd6,
                0xd7,
                0xdc,
                0xdd
            ]);
        } catch (_0x3325c4) {
        }
        _0xec5ee2['inspectOpts'] = Object['keys'](process['env'])['filter'](_0x3162cb => {
            return /^debug_/i['test'](_0x3162cb);
        })['reduce']((_0x2ec0bb, _0x23fed3) => {
            const _0x354c5e = _0x23fed3['substring'](0x6)['toLowerCase']()['replace'](/_([a-z])/g, (_0xc99dfc, _0x53f58b) => {
                return _0x53f58b['toUpperCase']();
            });
            let _0x1adcb3 = process['env'][_0x23fed3];
            if (/^(yes|on|true|enabled)$/i['test'](_0x1adcb3))
                _0x1adcb3 = !![];
            else {
                if (/^(no|off|false|disabled)$/i['test'](_0x1adcb3))
                    _0x1adcb3 = ![];
                else
                    _0x1adcb3 === 'null' ? _0x1adcb3 = null : _0x1adcb3 = Number(_0x1adcb3);
            }
            return _0x2ec0bb[_0x354c5e] = _0x1adcb3, _0x2ec0bb;
        }, {});
        function _0x3f348b() {
            return 'colors' in _0xec5ee2['inspectOpts'] ? Boolean(_0xec5ee2['inspectOpts']['colors']) : _0x1a66bb['isatty'](process['stderr']['fd']);
        }
        function _0x2fe862(_0x13864c) {
            const {
                namespace: _0x2cf972,
                useColors: _0x556021
            } = this;
            if (_0x556021) {
                const _0x1e500c = this['color'], _0xc7b71f = '\x1b[3' + (_0x1e500c < 0x8 ? _0x1e500c : '8;5;' + _0x1e500c), _0x320f7f = '\x20\x20' + _0xc7b71f + ';1m' + _0x2cf972 + '\x20\x1b[0m';
                _0x13864c[0x0] = _0x320f7f + _0x13864c[0x0]['split']('\x0a')['join']('\x0a' + _0x320f7f), _0x13864c['push'](_0xc7b71f + 'm+' + _0x9b2084['exports']['humanize'](this['diff']) + '\x1b[0m');
            } else
                _0x13864c[0x0] = _0x7cfa38() + _0x2cf972 + '\x20' + _0x13864c[0x0];
        }
        function _0x7cfa38() {
            if (_0xec5ee2['inspectOpts']['hideDate'])
                return '';
            return new Date()['toISOString']() + '\x20';
        }
        function _0xadc755(..._0x510475) {
            return process['stderr']['write'](_0x2b2e3c['format'](..._0x510475) + '\x0a');
        }
        function _0x3ffcc5(_0x4f9ea2) {
            _0x4f9ea2 ? process['env']['DEBUG'] = _0x4f9ea2 : delete process['env']['DEBUG'];
        }
        function _0x5c6a5f() {
            return process['env']['DEBUG'];
        }
        function _0x464ff0(_0x49e381) {
            _0x49e381['inspectOpts'] = {};
            const _0x2f786b = Object['keys'](_0xec5ee2['inspectOpts']);
            for (let _0x4b790a = 0x0; _0x4b790a < _0x2f786b['length']; _0x4b790a++) {
                _0x49e381['inspectOpts'][_0x2f786b[_0x4b790a]] = _0xec5ee2['inspectOpts'][_0x2f786b[_0x4b790a]];
            }
        }
        _0x9b2084['exports'] = _0x2c7057(0x4817)(_0xec5ee2);
        const {formatters: _0x592d3e} = _0x9b2084['exports'];
        _0x592d3e['o'] = function (_0xeb8f5a) {
            return this['inspectOpts']['colors'] = this['useColors'], _0x2b2e3c['inspect'](_0xeb8f5a, this['inspectOpts'])['split']('\x0a')['map'](_0x2bc33a => _0x2bc33a['trim']())['join']('\x20');
        }, _0x592d3e['O'] = function (_0x441acd) {
            return this['inspectOpts']['colors'] = this['useColors'], _0x2b2e3c['inspect'](_0x441acd, this['inspectOpts']);
        };
    },
    0xd552: _0x4fc7e7 => {
        var _0x5f5a5a = 0x3e8, _0x4b8d32 = _0x5f5a5a * 0x3c, _0x4a7ea4 = _0x4b8d32 * 0x3c, _0x722dc5 = _0x4a7ea4 * 0x18, _0x49a810 = _0x722dc5 * 0x7, _0x5cf24d = _0x722dc5 * 365.25;
        _0x4fc7e7['exports'] = function (_0x1265db, _0x440616) {
            _0x440616 = _0x440616 || {};
            var _0x520f9d = typeof _0x1265db;
            if (_0x520f9d === 'string' && _0x1265db['length'] > 0x0)
                return _0x2e15fb(_0x1265db);
            else {
                if (_0x520f9d === 'number' && isFinite(_0x1265db))
                    return _0x440616['long'] ? _0x4b2332(_0x1265db) : _0x97d933(_0x1265db);
            }
            throw new Error('val\x20is\x20not\x20a\x20non-empty\x20string\x20or\x20a\x20valid\x20number.\x20val=' + JSON['stringify'](_0x1265db));
        };
        function _0x2e15fb(_0x1120d4) {
            _0x1120d4 = String(_0x1120d4);
            if (_0x1120d4['length'] > 0x64)
                return;
            var _0x548a8f = /^(-?(?:\d+)?\.?\d+) *(milliseconds?|msecs?|ms|seconds?|secs?|s|minutes?|mins?|m|hours?|hrs?|h|days?|d|weeks?|w|years?|yrs?|y)?$/i['exec'](_0x1120d4);
            if (!_0x548a8f)
                return;
            var _0x589200 = parseFloat(_0x548a8f[0x1]), _0x468238 = (_0x548a8f[0x2] || 'ms')['toLowerCase']();
            switch (_0x468238) {
            case 'years':
            case 'year':
            case 'yrs':
            case 'yr':
            case 'y':
                return _0x589200 * _0x5cf24d;
            case 'weeks':
            case 'week':
            case 'w':
                return _0x589200 * _0x49a810;
            case 'days':
            case 'day':
            case 'd':
                return _0x589200 * _0x722dc5;
            case 'hours':
            case 'hour':
            case 'hrs':
            case 'hr':
            case 'h':
                return _0x589200 * _0x4a7ea4;
            case 'minutes':
            case 'minute':
            case 'mins':
            case 'min':
            case 'm':
                return _0x589200 * _0x4b8d32;
            case 'seconds':
            case 'second':
            case 'secs':
            case 'sec':
            case 's':
                return _0x589200 * _0x5f5a5a;
            case 'milliseconds':
            case 'millisecond':
            case 'msecs':
            case 'msec':
            case 'ms':
                return _0x589200;
            default:
                return undefined;
            }
        }
        function _0x97d933(_0x99e493) {
            var _0x4d1f9b = Math['abs'](_0x99e493);
            if (_0x4d1f9b >= _0x722dc5)
                return Math['round'](_0x99e493 / _0x722dc5) + 'd';
            if (_0x4d1f9b >= _0x4a7ea4)
                return Math['round'](_0x99e493 / _0x4a7ea4) + 'h';
            if (_0x4d1f9b >= _0x4b8d32)
                return Math['round'](_0x99e493 / _0x4b8d32) + 'm';
            if (_0x4d1f9b >= _0x5f5a5a)
                return Math['round'](_0x99e493 / _0x5f5a5a) + 's';
            return _0x99e493 + 'ms';
        }
        function _0x4b2332(_0x9b3fbd) {
            var _0x137b28 = Math['abs'](_0x9b3fbd);
            if (_0x137b28 >= _0x722dc5)
                return _0x4adccc(_0x9b3fbd, _0x137b28, _0x722dc5, 'day');
            if (_0x137b28 >= _0x4a7ea4)
                return _0x4adccc(_0x9b3fbd, _0x137b28, _0x4a7ea4, 'hour');
            if (_0x137b28 >= _0x4b8d32)
                return _0x4adccc(_0x9b3fbd, _0x137b28, _0x4b8d32, 'minute');
            if (_0x137b28 >= _0x5f5a5a)
                return _0x4adccc(_0x9b3fbd, _0x137b28, _0x5f5a5a, 'second');
            return _0x9b3fbd + '\x20ms';
        }
        function _0x4adccc(_0x4235f6, _0xaf583f, _0x540f6e, _0x4514dd) {
            var _0x350da6 = _0xaf583f >= _0x540f6e * 1.5;
            return Math['round'](_0x4235f6 / _0x540f6e) + '\x20' + _0x4514dd + (_0x350da6 ? 's' : '');
        }
    }
};
;
