// test-maker/dist/131.js
exports['id'] = 0x83, exports['ids'] = [0x83], exports['modules'] = {
    0x1869: _0x52a041 => {
        'use strict';
        function _0x332d27() {
            this['_events_'] = {}, this['publish'] = this['trigger'] = this['emit'] = _0x1bd243, this['subscribe'] = this['on'] = _0x48b3af, this['once'] = _0x339950, this['unSubscribe'] = this['off'] = _0x43af10, this['emit$'] = _0x34c91a;
            function _0x48b3af(_0xcfd31, _0x43745d, _0x44aebc) {
                if (!_0x43745d)
                    throw new ReferenceError('handler\x20not\x20defined.');
                return !this['_events_'][_0xcfd31] && (this['_events_'][_0xcfd31] = []), _0x44aebc && (_0x43745d['_once_'] = _0x44aebc), this['_events_'][_0xcfd31]['push'](_0x43745d), this;
            }
            function _0x339950(_0x1d0cfa, _0x675814) {
                return this['on'](_0x1d0cfa, _0x675814, !![]);
            }
            function _0x43af10(_0x59d35a, _0x2dfc92) {
                if (!this['_events_'][_0x59d35a])
                    return this;
                if (!_0x2dfc92)
                    throw new ReferenceError('handler\x20not\x20defined.\x20if\x20you\x20wish\x20to\x20remove\x20all\x20handlers\x20from\x20the\x20event\x20please\x20pass\x20\x22*\x22\x20as\x20the\x20handler');
                if (_0x2dfc92 == '*')
                    return delete this['_events_'][_0x59d35a], this;
                var _0x4371de = this['_events_'][_0x59d35a];
                while (_0x4371de['includes'](_0x2dfc92)) {
                    _0x4371de['splice'](_0x4371de['indexOf'](_0x2dfc92), 0x1);
                }
                return _0x4371de['length'] < 0x1 && delete this['_events_'][_0x59d35a], this;
            }
            function _0x1bd243(_0x3140f0) {
                this['emit$']['apply'](this, arguments);
                if (!this['_events_'][_0x3140f0])
                    return this;
                arguments['splice'] = Array['prototype']['splice'], arguments['splice'](0x0, 0x1);
                var _0x4e1176 = this['_events_'][_0x3140f0], _0x6efb13 = [];
                for (var _0x3a79d9 in _0x4e1176) {
                    var _0x202f91 = _0x4e1176[_0x3a79d9];
                    _0x202f91['apply'](this, arguments), _0x202f91['_once_'] && _0x6efb13['push'](_0x202f91);
                }
                for (var _0x3a79d9 in _0x6efb13) {
                    this['off'](_0x3140f0, _0x6efb13[_0x3a79d9]);
                }
                return this;
            }
            function _0x34c91a(_0x17f191, _0x3231ba) {
                if (!this['_events_']['*'])
                    return this;
                var _0x25a891 = this['_events_']['*'];
                _0x3231ba['shift'] = Array['prototype']['shift'], _0x3231ba['shift'](_0x17f191);
                for (var _0x444604 of _0x25a891) {
                    _0x444604['apply'](this, _0x3231ba);
                }
                return this;
            }
            return this;
        }
        !Array['prototype']['includes'] && (Array['prototype']['includes'] = function (_0x4b6da1) {
            'use strict';
            if (this == null)
                throw new TypeError('Array.prototype.includes\x20called\x20on\x20null\x20or\x20undefined');
            var _0x5c9120 = Object(this), _0x299f95 = parseInt(_0x5c9120['length'], 0xa) || 0x0;
            if (_0x299f95 === 0x0)
                return ![];
            var _0x50e29e = parseInt(arguments[0x1], 0xa) || 0x0, _0x38fd68;
            _0x50e29e >= 0x0 ? _0x38fd68 = _0x50e29e : (_0x38fd68 = _0x299f95 + _0x50e29e, _0x38fd68 < 0x0 && (_0x38fd68 = 0x0));
            var _0x248622;
            while (_0x38fd68 < _0x299f95) {
                _0x248622 = _0x5c9120[_0x38fd68];
                if (_0x4b6da1 === _0x248622 || _0x4b6da1 !== _0x4b6da1 && _0x248622 !== _0x248622)
                    return !![];
                _0x38fd68++;
            }
            return ![];
        }), _0x52a041['exports'] = _0x332d27;
    },
    0xb916: _0x216dcd => {
        'use strict';
        class _0x281a2b {
            constructor(_0x33d302) {
                this['_events_'] = {}, this['publish'] = this['trigger'] = this['emit'], this['subscribe'] = this['on'], this['unSubscribe'] = this['off'];
            }
            ['on'](_0x10cddf, _0xc47f5c, _0x47e2a6) {
                if (!_0xc47f5c)
                    throw new ReferenceError('handler\x20not\x20defined.');
                return !this['_events_'][_0x10cddf] && (this['_events_'][_0x10cddf] = []), _0x47e2a6 && (_0xc47f5c['_once_'] = _0x47e2a6), this['_events_'][_0x10cddf]['push'](_0xc47f5c), this;
            }
            ['once'](_0x3aadd4, _0x47779c) {
                return this['on'](_0x3aadd4, _0x47779c, !![]);
            }
            ['off'](_0x4da9bc, _0x54b63f) {
                if (!this['_events_'][_0x4da9bc])
                    return this;
                if (!_0x54b63f)
                    throw new ReferenceError('handler\x20not\x20defined.\x20if\x20you\x20wish\x20to\x20remove\x20all\x20handlers\x20from\x20the\x20event\x20please\x20pass\x20\x22*\x22\x20as\x20the\x20handler');
                if (_0x54b63f == '*')
                    return delete this['_events_'][_0x4da9bc], this;
                const _0x57ce38 = this['_events_'][_0x4da9bc];
                while (_0x57ce38['includes'](_0x54b63f)) {
                    _0x57ce38['splice'](_0x57ce38['indexOf'](_0x54b63f), 0x1);
                }
                return _0x57ce38['length'] < 0x1 && delete this['_events_'][_0x4da9bc], this;
            }
            ['emit'](_0x7400e8, ..._0x1c3da6) {
                if (!this['_events_'][_0x7400e8])
                    return this['emit$'](_0x7400e8, ..._0x1c3da6);
                const _0x43cdfd = this['_events_'][_0x7400e8], _0x524a8f = [];
                for (let _0xf9b58f of _0x43cdfd) {
                    _0xf9b58f['apply'](this, _0x1c3da6), _0xf9b58f['_once_'] && _0x524a8f['push'](_0xf9b58f);
                }
                for (let _0x3094c9 of _0x524a8f) {
                    this['off'](_0x7400e8, _0x3094c9);
                }
                return this['emit$'](_0x7400e8, ..._0x1c3da6);
            }
            ['emit$'](_0xc85bb1, ..._0x453fdd) {
                if (!this['_events_']['*'])
                    return this;
                const _0x5d013f = this['_events_']['*'];
                for (let _0x47037f of _0x5d013f) {
                    _0x47037f['call'](this, _0xc85bb1, ..._0x453fdd);
                }
                return this;
            }
        }
        _0x216dcd['exports'] = _0x281a2b;
    },
    0x35e: (_0x1ecbd6, _0x1021ea, _0x37b187) => {
        'use strict';
        let _0x1bcb67 = _0x37b187(0x1869);
        process['version'][0x1] > 0x5 && (_0x1bcb67 = _0x37b187(0xb916)), _0x1ecbd6['exports'] = _0x1bcb67;
    },
    0x100ed: _0xb9330a => {
        function _0xaefdfc() {
            Object['defineProperties'](this, {
                'data': {
                    'enumerable': !![],
                    'get': _0x383b52,
                    'set': _0x1167a0
                },
                'type': {
                    'enumerable': !![],
                    'get': _0x3f2407,
                    'set': _0x57ca5d
                },
                'load': {
                    'enumerable': !![],
                    'writable': ![],
                    'value': _0x3bcd20
                },
                'JSON': {
                    'enumerable': !![],
                    'get': _0x55ce69
                }
            });
            var _0x216478 = '', _0x4b2181 = {};
            function _0x3f2407() {
                return _0x216478;
            }
            function _0x383b52() {
                return _0x4b2181;
            }
            function _0x55ce69() {
                return JSON['stringify']({
                    'type': _0x216478,
                    'data': _0x4b2181
                });
            }
            function _0x57ca5d(_0x387580) {
                _0x216478 = _0x387580;
            }
            function _0x1167a0(_0x1674df) {
                _0x4b2181 = _0x1674df;
            }
            function _0x3bcd20(_0xbe73d0) {
                try {
                    var _0xbe73d0 = JSON['parse'](_0xbe73d0);
                    _0x216478 = _0xbe73d0['type'], _0x4b2181 = _0xbe73d0['data'];
                } catch (_0x5d9674) {
                    var _0x5165c1 = _0xbe73d0;
                    _0x216478 = 'error', _0x4b2181 = {
                        'message': 'Invalid\x20JSON\x20response\x20format',
                        'err': _0x5d9674,
                        'response': _0x5165c1
                    };
                }
            }
        }
        _0xb9330a['exports'] = _0xaefdfc;
    },
    0xb864: _0x1db165 => {
        function _0x3e51bd(_0x2b6c55) {
            Object['defineProperties'](this, {
                'add': {
                    'enumerable': !![],
                    'writable': ![],
                    'value': _0x5009a2
                },
                'next': {
                    'enumerable': !![],
                    'writable': ![],
                    'value': _0x53584a
                },
                'clear': {
                    'enumerable': !![],
                    'writable': ![],
                    'value': _0x4ab4e3
                },
                'contents': {
                    'enumerable': ![],
                    'get': _0x3f0a53,
                    'set': _0x5102c3
                },
                'autoRun': {
                    'enumerable': !![],
                    'writable': !![],
                    'value': !![]
                },
                'stop': {
                    'enumerable': !![],
                    'writable': !![],
                    'value': ![]
                }
            });
            var _0x381d8c = [], _0x20a3d2 = ![], _0x2a45b6 = ![];
            function _0x4ab4e3() {
                return _0x381d8c = [], _0x381d8c;
            }
            function _0x3f0a53() {
                return _0x381d8c;
            }
            function _0x5102c3(_0x471a9d) {
                return _0x381d8c = _0x471a9d, _0x381d8c;
            }
            function _0x5009a2() {
                for (var _0x1df1a3 in arguments) {
                    _0x381d8c['push'](arguments[_0x1df1a3]);
                }
                !_0x20a3d2 && !this['stop'] && this['autoRun'] && this['next']();
            }
            function _0x53584a() {
                _0x20a3d2 = !![];
                if (_0x381d8c['length'] < 0x1 || this['stop']) {
                    _0x20a3d2 = ![];
                    return;
                }
                _0x381d8c['shift']()['bind'](this)();
            }
        }
        _0x1db165['exports'] = _0x3e51bd;
    },
    0x376: (_0x1b64e1, _0x282727, _0x3071b4) => {
        'use strict';
        const _0x41c8c7 = _0x3071b4(0xa350), _0x290489 = _0x3071b4(0x5f54), _0xa49536 = _0x3071b4(0x92fb), _0x4f6f0b = _0x3071b4(0x100ed), _0x541810 = _0x3071b4(0xdf3b), _0x3b0efb = _0x3071b4(0xb864), _0x7fb36b = _0x3071b4(0x35e);
        let _0x39954f = new _0xa49536();
        class _0x57b139 extends _0x7fb36b {
            constructor(_0x33be6e, _0x3bfca6) {
                super(), Object['assign'](this, {
                    'Client': _0x57b139,
                    'config': _0x33be6e,
                    'queue': new _0x3b0efb(),
                    'socket': ![],
                    'connect': _0x59286d,
                    'emit': _0x458124,
                    'log': _0x3bfca6,
                    'retriesRemaining': _0x33be6e['maxRetries'] || 0x0,
                    'explicitlyDisconnected': ![]
                }), _0x39954f = new _0xa49536(this['config']);
            }
        }
        function _0x458124(_0x137623, _0x230aa1) {
            this['log']('dispatching\x20event\x20to\x20', this['id'], this['path'], '\x20:\x20', _0x137623, ',', _0x230aa1);
            let _0x35d003 = new _0x4f6f0b();
            _0x35d003['type'] = _0x137623, _0x35d003['data'] = _0x230aa1;
            this['config']['rawBuffer'] ? _0x35d003 = Buffer['from'](_0x137623, this['config']['encoding']) : _0x35d003 = _0x39954f['format'](_0x35d003);
            if (!this['config']['sync']) {
                this['socket']['write'](_0x35d003);
                return;
            }
            this['queue']['add'](_0x4a1e98['bind'](this, _0x35d003));
        }
        function _0x4a1e98(_0x45bc37) {
            this['log']('dispatching\x20event\x20to\x20', this['id'], this['path'], '\x20:\x20', _0x45bc37), this['socket']['write'](_0x45bc37);
        }
        function _0x59286d() {
            let _0x3acab9 = this;
            _0x3acab9['log']('requested\x20connection\x20to\x20', _0x3acab9['id'], _0x3acab9['path']);
            if (!this['path']) {
                _0x3acab9['log']('\x0a\x0a######\x0aerror:\x20', _0x3acab9['id'], '\x20client\x20has\x20not\x20specified\x20socket\x20path\x20it\x20wishes\x20to\x20connect\x20to.');
                return;
            }
            const _0x5511f0 = {};
            if (!_0x3acab9['port'])
                _0x3acab9['log']('Connecting\x20client\x20on\x20Unix\x20Socket\x20:', _0x3acab9['path']), _0x5511f0['path'] = _0x3acab9['path'], process['platform'] === 'win32' && !_0x3acab9['path']['startsWith']('\x5c\x5c.\x5cpipe\x5c') && (_0x5511f0['path'] = _0x5511f0['path']['replace'](/^\//, ''), _0x5511f0['path'] = _0x5511f0['path']['replace'](/\//g, '-'), _0x5511f0['path'] = '\x5c\x5c.\x5cpipe\x5c' + _0x5511f0['path']), _0x3acab9['socket'] = _0x41c8c7['connect'](_0x5511f0);
            else {
                _0x5511f0['host'] = _0x3acab9['path'], _0x5511f0['port'] = _0x3acab9['port'];
                _0x3acab9['config']['interface']['localAddress'] && (_0x5511f0['localAddress'] = _0x3acab9['config']['interface']['localAddress']);
                _0x3acab9['config']['interface']['localPort'] && (_0x5511f0['localPort'] = _0x3acab9['config']['interface']['localPort']);
                _0x3acab9['config']['interface']['family'] && (_0x5511f0['family'] = _0x3acab9['config']['interface']['family']);
                _0x3acab9['config']['interface']['hints'] && (_0x5511f0['hints'] = _0x3acab9['config']['interface']['hints']);
                _0x3acab9['config']['interface']['lookup'] && (_0x5511f0['lookup'] = _0x3acab9['config']['interface']['lookup']);
                if (!_0x3acab9['config']['tls'])
                    _0x3acab9['log']('Connecting\x20client\x20via\x20TCP\x20to', _0x5511f0), _0x3acab9['socket'] = _0x41c8c7['connect'](_0x5511f0);
                else {
                    _0x3acab9['log']('Connecting\x20client\x20via\x20TLS\x20to', _0x3acab9['path'], _0x3acab9['port'], _0x3acab9['config']['tls']);
                    _0x3acab9['config']['tls']['private'] && (_0x3acab9['config']['tls']['key'] = _0x541810['readFileSync'](_0x3acab9['config']['tls']['private']));
                    _0x3acab9['config']['tls']['public'] && (_0x3acab9['config']['tls']['cert'] = _0x541810['readFileSync'](_0x3acab9['config']['tls']['public']));
                    if (_0x3acab9['config']['tls']['trustedConnections']) {
                        typeof _0x3acab9['config']['tls']['trustedConnections'] === 'string' && (_0x3acab9['config']['tls']['trustedConnections'] = [_0x3acab9['config']['tls']['trustedConnections']]);
                        _0x3acab9['config']['tls']['ca'] = [];
                        for (let _0x2c7dda = 0x0; _0x2c7dda < _0x3acab9['config']['tls']['trustedConnections']['length']; _0x2c7dda++) {
                            _0x3acab9['config']['tls']['ca']['push'](_0x541810['readFileSync'](_0x3acab9['config']['tls']['trustedConnections'][_0x2c7dda]));
                        }
                    }
                    Object['assign'](_0x3acab9['config']['tls'], _0x5511f0), _0x3acab9['socket'] = _0x290489['connect'](_0x3acab9['config']['tls']);
                }
            }
            _0x3acab9['socket']['setEncoding'](this['config']['encoding']), _0x3acab9['socket']['on']('error', function (_0x614ab8) {
                _0x3acab9['log']('\x0a\x0a######\x0aerror:\x20', _0x614ab8), _0x3acab9['publish']('error', _0x614ab8);
            }), _0x3acab9['socket']['on']('connect', function _0x3ba4d5() {
                _0x3acab9['publish']('connect'), _0x3acab9['retriesRemaining'] = _0x3acab9['config']['maxRetries'], _0x3acab9['log']('retrying\x20reset');
            }), _0x3acab9['socket']['on']('close', function _0x5c3193() {
                _0x3acab9['log']('connection\x20closed', _0x3acab9['id'], _0x3acab9['path'], _0x3acab9['retriesRemaining'], 'tries\x20remaining\x20of', _0x3acab9['config']['maxRetries']);
                if (_0x3acab9['config']['stopRetrying'] || _0x3acab9['retriesRemaining'] < 0x1 || _0x3acab9['explicitlyDisconnected']) {
                    _0x3acab9['publish']('disconnect'), _0x3acab9['log'](_0x3acab9['config']['id'], 'exceeded\x20connection\x20rety\x20amount\x20of', '\x20or\x20stopRetrying\x20flag\x20set.'), _0x3acab9['socket']['destroy'](), _0x3acab9['publish']('destroy'), _0x3acab9 = undefined;
                    return;
                }
                setTimeout(function _0x1999bb() {
                    if (_0x3acab9['explicitlyDisconnected'])
                        return;
                    _0x3acab9['retriesRemaining']--, _0x3acab9['connect']();
                }['bind'](null, _0x3acab9), _0x3acab9['config']['retry']), _0x3acab9['publish']('disconnect');
            }), _0x3acab9['socket']['on']('data', function (_0x5950c9) {
                _0x3acab9['log']('##\x20received\x20events\x20##');
                if (_0x3acab9['config']['rawBuffer']) {
                    _0x3acab9['publish']('data', Buffer['from'](_0x5950c9, _0x3acab9['config']['encoding']));
                    if (!_0x3acab9['config']['sync'])
                        return;
                    _0x3acab9['queue']['next']();
                    return;
                }
                !this['ipcBuffer'] && (this['ipcBuffer'] = '');
                _0x5950c9 = this['ipcBuffer'] += _0x5950c9;
                if (_0x5950c9['slice'](-0x1) != _0x39954f['delimiter'] || _0x5950c9['indexOf'](_0x39954f['delimiter']) == -0x1) {
                    _0x3acab9['log']('Messages\x20are\x20large,\x20You\x20may\x20want\x20to\x20consider\x20smaller\x20messages.');
                    return;
                }
                this['ipcBuffer'] = '';
                const _0x3f23d6 = _0x39954f['parse'](_0x5950c9), _0x42ea5d = _0x3f23d6['length'];
                for (let _0x3d6578 = 0x0; _0x3d6578 < _0x42ea5d; _0x3d6578++) {
                    let _0x21bf06 = new _0x4f6f0b();
                    _0x21bf06['load'](_0x3f23d6[_0x3d6578]), _0x3acab9['log']('detected\x20event', _0x21bf06['type'], _0x21bf06['data']), _0x3acab9['publish'](_0x21bf06['type'], _0x21bf06['data']);
                }
                if (!_0x3acab9['config']['sync'])
                    return;
                _0x3acab9['queue']['next']();
            });
        }
        _0x1b64e1['exports'] = _0x57b139;
    },
    0x9b31: (_0xb736db, _0x2dcdb2, _0x3856a1) => {
        'use strict';
        const _0x5c3af8 = _0x3856a1(0xa350), _0x43e096 = _0x3856a1(0x5f54), _0xa6efa7 = _0x3856a1(0xdf3b), _0x3112c5 = _0x3856a1(0x118d3), _0xeae683 = _0x3856a1(0x92fb), _0x597d34 = _0x3856a1(0x100ed), _0x1e86a6 = _0x3856a1(0x35e);
        let _0x4721dc = new _0xeae683();
        class _0x2d0067 extends _0x1e86a6 {
            constructor(_0x469f60, _0xa4f785, _0x3c16ae, _0x427d90) {
                super(), Object['assign'](this, {
                    'config': _0xa4f785,
                    'path': _0x469f60,
                    'port': _0x427d90,
                    'udp4': ![],
                    'udp6': ![],
                    'log': _0x3c16ae,
                    'server': ![],
                    'sockets': [],
                    'emit': _0xbef9b4,
                    'broadcast': _0x5ae8ca
                }), _0x4721dc = new _0xeae683(this['config']), this['on']('close', _0x544f3b['bind'](this));
            }
            ['onStart'](_0x194fb7) {
                this['trigger']('start', _0x194fb7);
            }
            ['stop']() {
                this['server']['close']();
            }
            ['start']() {
                if (!this['path']) {
                    this['log']('Socket\x20Server\x20Path\x20not\x20specified,\x20refusing\x20to\x20start');
                    return;
                }
                this['config']['unlink'] ? _0xa6efa7['unlink'](this['path'], _0x4af0b3['bind'](this)) : _0x4af0b3['bind'](this)();
            }
        }
        function _0xbef9b4(_0x8e14d0, _0x457085, _0x570646) {
            this['log']('dispatching\x20event\x20to\x20socket', '\x20:\x20', _0x457085, _0x570646);
            let _0x5926bf = new _0x597d34();
            _0x5926bf['type'] = _0x457085, _0x5926bf['data'] = _0x570646;
            this['config']['rawBuffer'] ? (this['log'](this['config']['encoding']), _0x5926bf = Buffer['from'](_0x457085, this['config']['encoding'])) : _0x5926bf = _0x4721dc['format'](_0x5926bf);
            if (this['udp4'] || this['udp6']) {
                if (!_0x8e14d0['address'] || !_0x8e14d0['port']) {
                    this['log']('Attempting\x20to\x20emit\x20to\x20a\x20single\x20UDP\x20socket\x20without\x20supplying\x20socket\x20address\x20or\x20port.\x20Redispatching\x20event\x20as\x20broadcast\x20to\x20all\x20connected\x20sockets'), this['broadcast'](_0x457085, _0x570646);
                    return;
                }
                this['server']['write'](_0x5926bf, _0x8e14d0);
                return;
            }
            _0x8e14d0['write'](_0x5926bf);
        }
        function _0x5ae8ca(_0x23dd21, _0x46a732) {
            this['log']('broadcasting\x20event\x20to\x20all\x20known\x20sockets\x20listening\x20to\x20', this['path'], '\x20:\x20', this['port'] ? this['port'] : '', _0x23dd21, _0x46a732);
            let _0x1fd808 = new _0x597d34();
            _0x1fd808['type'] = _0x23dd21, _0x1fd808['data'] = _0x46a732;
            this['config']['rawBuffer'] ? _0x1fd808 = Buffer['from'](_0x23dd21, this['config']['encoding']) : _0x1fd808 = _0x4721dc['format'](_0x1fd808);
            if (this['udp4'] || this['udp6'])
                for (let _0x28392f = 0x1, _0x5da26a = this['sockets']['length']; _0x28392f < _0x5da26a; _0x28392f++) {
                    this['server']['write'](_0x1fd808, this['sockets'][_0x28392f]);
                }
            else
                for (let _0x9e401f = 0x0, _0x25529e = this['sockets']['length']; _0x9e401f < _0x25529e; _0x9e401f++) {
                    this['sockets'][_0x9e401f]['write'](_0x1fd808);
                }
        }
        function _0x544f3b() {
            for (let _0x59438e = 0x0, _0x416f06 = this['sockets']['length']; _0x59438e < _0x416f06; _0x59438e++) {
                let _0x32c8fd = this['sockets'][_0x59438e], _0x4dd706 = ![];
                if (_0x32c8fd) {
                    if (_0x32c8fd['readable'])
                        continue;
                }
                _0x32c8fd['id'] && (_0x4dd706 = _0x32c8fd['id']);
                this['log']('socket\x20disconnected', _0x4dd706['toString']());
                _0x32c8fd && _0x32c8fd['destroy'] && _0x32c8fd['destroy']();
                this['sockets']['splice'](_0x59438e, 0x1), this['publish']('socket.disconnected', _0x32c8fd, _0x4dd706);
                return;
            }
        }
        function _0x466a64(_0xc02adc, _0x26c5d1, _0x3c950f) {
            let _0x2b5fd3 = this['udp4'] || this['udp6'] ? _0x3c950f : _0xc02adc;
            if (this['config']['rawBuffer']) {
                _0x26c5d1 = Buffer['from'](_0x26c5d1, this['config']['encoding']), this['publish']('data', _0x26c5d1, _0x2b5fd3);
                return;
            }
            !_0x2b5fd3['ipcBuffer'] && (_0x2b5fd3['ipcBuffer'] = '');
            _0x26c5d1 = _0x2b5fd3['ipcBuffer'] += _0x26c5d1;
            if (_0x26c5d1['slice'](-0x1) != _0x4721dc['delimiter'] || _0x26c5d1['indexOf'](_0x4721dc['delimiter']) == -0x1) {
                this['log']('Messages\x20are\x20large,\x20You\x20may\x20want\x20to\x20consider\x20smaller\x20messages.');
                return;
            }
            _0x2b5fd3['ipcBuffer'] = '', _0x26c5d1 = _0x4721dc['parse'](_0x26c5d1);
            while (_0x26c5d1['length'] > 0x0) {
                let _0x202502 = new _0x597d34();
                _0x202502['load'](_0x26c5d1['shift']()), _0x202502['data'] && _0x202502['data']['id'] && (_0x2b5fd3['id'] = _0x202502['data']['id']), this['log']('received\x20event\x20of\x20:\x20', _0x202502['type'], _0x202502['data']), this['publish'](_0x202502['type'], _0x202502['data'], _0x2b5fd3);
            }
        }
        function _0x34efb3(_0x652328) {
            this['publish']('close', _0x652328);
        }
        function _0x49d7f3(_0x1c04d8) {
            this['sockets']['push'](_0x1c04d8);
            _0x1c04d8['setEncoding'] && _0x1c04d8['setEncoding'](this['config']['encoding']);
            this['log']('##\x20socket\x20connection\x20to\x20server\x20detected\x20##'), _0x1c04d8['on']('close', _0x34efb3['bind'](this)), _0x1c04d8['on']('error', function (_0x1800e7) {
                this['log']('server\x20socket\x20error', _0x1800e7), this['publish']('error', _0x1800e7);
            }['bind'](this)), _0x1c04d8['on']('data', _0x466a64['bind'](this, _0x1c04d8)), _0x1c04d8['on']('message', function (_0x418a58, _0x589ad0) {
                if (!_0x589ad0)
                    return;
                this['log']('Received\x20UDP\x20message\x20from\x20', _0x589ad0['address'], _0x589ad0['port']);
                let _0x549661;
                this['config']['rawSocket'] ? _0x549661 = Buffer['from'](_0x418a58, this['config']['encoding']) : _0x549661 = _0x418a58['toString'](), _0x1c04d8['emit']('data', _0x549661, _0x589ad0);
            }['bind'](this)), this['publish']('connect', _0x1c04d8);
            if (this['config']['rawBuffer'])
                return;
        }
        function _0x4af0b3() {
            this['log']('starting\x20server\x20on\x20', this['path'], this['port'] ? ':' + this['port'] : '');
            !this['udp4'] && !this['udp6'] ? (this['log']('starting\x20TLS\x20server', this['config']['tls']), !this['config']['tls'] ? this['server'] = _0x5c3af8['createServer'](_0x49d7f3['bind'](this)) : _0x45212c['bind'](this)()) : (this['server'] = _0x3112c5['createSocket'](this['udp4'] ? 'udp4' : 'udp6'), this['server']['write'] = _0x1cf958['bind'](this), this['server']['on']('listening', function _0x443fe5() {
                _0x49d7f3['bind'](this)(this['server']);
            }['bind'](this)));
            this['server']['on']('error', function (_0x37daa9) {
                this['log']('server\x20error', _0x37daa9), this['publish']('error', _0x37daa9);
            }['bind'](this)), this['server']['maxConnections'] = this['config']['maxConnections'];
            if (!this['port']) {
                this['log']('starting\x20server\x20as', 'Unix\x20||\x20Windows\x20Socket');
                process['platform'] === 'win32' && (this['path'] = this['path']['replace'](/^\//, ''), this['path'] = this['path']['replace'](/\//g, '-'), this['path'] = '\x5c\x5c.\x5cpipe\x5c' + this['path']);
                this['server']['listen']({
                    'path': this['path'],
                    'readableAll': this['config']['readableAll'],
                    'writableAll': this['config']['writableAll']
                }, this['onStart']['bind'](this));
                return;
            }
            if (!this['udp4'] && !this['udp6']) {
                this['log']('starting\x20server\x20as', this['config']['tls'] ? 'TLS' : 'TCP'), this['server']['listen'](this['port'], this['path'], this['onStart']['bind'](this));
                return;
            }
            this['log']('starting\x20server\x20as', this['udp4'] ? 'udp4' : 'udp6'), this['server']['bind'](this['port'], this['path']), this['onStart']({
                'address': this['path'],
                'port': this['port']
            });
        }
        function _0x45212c() {
            this['log']('starting\x20TLS\x20server', this['config']['tls']);
            this['config']['tls']['private'] ? this['config']['tls']['key'] = _0xa6efa7['readFileSync'](this['config']['tls']['private']) : this['config']['tls']['key'] = _0xa6efa7['readFileSync'](__dirname + '/../local-node-ipc-certs/private/server.key');
            this['config']['tls']['public'] ? this['config']['tls']['cert'] = _0xa6efa7['readFileSync'](this['config']['tls']['public']) : this['config']['tls']['cert'] = _0xa6efa7['readFileSync'](__dirname + '/../local-node-ipc-certs/server.pub');
            this['config']['tls']['dhparam'] && (this['config']['tls']['dhparam'] = _0xa6efa7['readFileSync'](this['config']['tls']['dhparam']));
            if (this['config']['tls']['trustedConnections']) {
                typeof this['config']['tls']['trustedConnections'] === 'string' && (this['config']['tls']['trustedConnections'] = [this['config']['tls']['trustedConnections']]);
                this['config']['tls']['ca'] = [];
                for (let _0x2be61c = 0x0; _0x2be61c < this['config']['tls']['trustedConnections']['length']; _0x2be61c++) {
                    this['config']['tls']['ca']['push'](_0xa6efa7['readFileSync'](this['config']['tls']['trustedConnections'][_0x2be61c]));
                }
            }
            this['server'] = _0x43e096['createServer'](this['config']['tls'], _0x49d7f3['bind'](this));
        }
        function _0x1cf958(_0x4c4eff, _0x4c6d61) {
            let _0x129440 = Buffer['from'](_0x4c4eff, this['config']['encoding']);
            this['server']['send'](_0x129440, 0x0, _0x129440['length'], _0x4c6d61['port'], _0x4c6d61['address'], function (_0x5b04bc, _0x550c4e) {
                _0x5b04bc && (this['log']('error\x20writing\x20data\x20to\x20socket', _0x5b04bc), this['publish']('error', function (_0x5db59d) {
                    this['publish']('error', _0x5db59d);
                }));
            });
        }
        _0xb736db['exports'] = _0x2d0067;
    },
    0x47c2: (_0xd58a0a, _0x424e47, _0x39380f) => {
        'use strict';
        const _0x5f185d = _0x39380f(0x5615);
        class _0x208d19 {
            constructor() {
                this['appspace'] = 'app.', this['socketRoot'] = '/tmp/', this['id'] = _0x5f185d['hostname'](), this['encoding'] = 'utf8', this['rawBuffer'] = ![], this['sync'] = ![], this['unlink'] = !![], this['delimiter'] = '\x0c', this['silent'] = ![], this['logDepth'] = 0x5, this['logInColor'] = !![], this['logger'] = console['log']['bind'](console), this['maxConnections'] = 0x64, this['retry'] = 0x1f4, this['maxRetries'] = Infinity, this['stopRetrying'] = ![], this['IPType'] = _0x3155ec(), this['tls'] = ![], this['networkHost'] = this['IPType'] == 'IPv6' ? '::1' : '127.0.0.1', this['networkPort'] = 0x1f40, this['readableAll'] = ![], this['writableAll'] = ![], this['interface'] = {
                    'localAddress': ![],
                    'localPort': ![],
                    'family': ![],
                    'hints': ![],
                    'lookup': ![]
                };
            }
        }
        function _0x3155ec() {
            const _0x7f69bd = _0x5f185d['networkInterfaces']();
            let _0x20908f = '';
            return _0x7f69bd && Array['isArray'](_0x7f69bd) && _0x7f69bd['length'] > 0x0 && (_0x20908f = _0x7f69bd[Object['keys'](_0x7f69bd)[0x0]][0x0]['family']), _0x20908f;
        }
        _0xd58a0a['exports'] = _0x208d19;
    },
    0x92fb: (_0x811477, _0x532e6f, _0x98a17d) => {
        'use strict';
        const _0x4b2d0c = _0x98a17d(0x47c2);
        class _0x4a6ab8 {
            constructor(_0x47a8f4) {
                !_0x47a8f4 && (_0x47a8f4 = new _0x4b2d0c()), this['delimiter'] = _0x47a8f4['delimiter'];
            }
            ['format'](_0x497b60) {
                return !_0x497b60['data'] && _0x497b60['data'] !== ![] && _0x497b60['data'] !== 0x0 && (_0x497b60['data'] = {}), _0x497b60['data']['_maxListeners'] && (_0x497b60['data'] = {}), _0x497b60 = _0x497b60['JSON'] + this['delimiter'], _0x497b60;
            }
            ['parse'](_0x2d6390) {
                let _0x1a80c7 = _0x2d6390['split'](this['delimiter']);
                return _0x1a80c7['pop'](), _0x1a80c7;
            }
        }
        _0x811477['exports'] = _0x4a6ab8;
    },
    0x167e3: (_0x50aeef, _0x40c052, _0x117f4d) => {
        'use strict';
        const _0x1e9d56 = _0x117f4d(0xcc9f);
        class _0x356992 extends _0x1e9d56 {
            constructor() {
                super(), Object['defineProperty'](this, 'IPC', {
                    'enumerable': !![],
                    'writable': ![],
                    'value': _0x1e9d56
                });
            }
        }
        _0x50aeef['exports'] = new _0x356992();
    },
    0xcc9f: (_0x503f58, _0x2a6b1b, _0x56637a) => {
        'use strict';
        const _0x1d1b5d = _0x56637a(0x47c2), _0x582325 = _0x56637a(0x376), _0x47d5b1 = _0x56637a(0x9b31), _0x355e3d = _0x56637a(0x1206d);
        class _0x263432 {
            constructor() {
                Object['defineProperties'](this, {
                    'config': {
                        'enumerable': !![],
                        'writable': !![],
                        'value': new _0x1d1b5d()
                    },
                    'connectTo': {
                        'enumerable': !![],
                        'writable': ![],
                        'value': _0x2b3256
                    },
                    'connectToNet': {
                        'enumerable': !![],
                        'writable': ![],
                        'value': _0x4defa5
                    },
                    'disconnect': {
                        'enumerable': !![],
                        'writable': ![],
                        'value': _0xa6a7b4
                    },
                    'serve': {
                        'enumerable': !![],
                        'writable': ![],
                        'value': _0x3eee04
                    },
                    'serveNet': {
                        'enumerable': !![],
                        'writable': ![],
                        'value': _0x3b7359
                    },
                    'of': {
                        'enumerable': !![],
                        'writable': !![],
                        'value': {}
                    },
                    'server': {
                        'enumerable': !![],
                        'writable': !![],
                        'configurable': !![],
                        'value': ![]
                    },
                    'log': {
                        'enumerable': !![],
                        'writable': ![],
                        'value': _0x1c4457
                    }
                });
            }
        }
        function _0x1c4457(..._0x44403f) {
            if (this['config']['silent'])
                return;
            for (let _0x1786d4 = 0x0, _0x1b833c = _0x44403f['length']; _0x1786d4 < _0x1b833c; _0x1786d4++) {
                if (typeof _0x44403f[_0x1786d4] != 'object')
                    continue;
                _0x44403f[_0x1786d4] = _0x355e3d['inspect'](_0x44403f[_0x1786d4], {
                    'depth': this['config']['logDepth'],
                    'colors': this['config']['logInColor']
                });
            }
            this['config']['logger'](_0x44403f['join']('\x20'));
        }
        function _0xa6a7b4(_0x3376e4) {
            if (!this['of'][_0x3376e4])
                return;
            this['of'][_0x3376e4]['explicitlyDisconnected'] = !![], this['of'][_0x3376e4]['off']('*', '*'), this['of'][_0x3376e4]['socket'] && (this['of'][_0x3376e4]['socket']['destroy'] && this['of'][_0x3376e4]['socket']['destroy']()), delete this['of'][_0x3376e4];
        }
        function _0x3eee04(_0x5c5d9b, _0x138a61) {
            typeof _0x5c5d9b == 'function' && (_0x138a61 = _0x5c5d9b, _0x5c5d9b = ![]), !_0x5c5d9b && (this['log']('Server\x20path\x20not\x20specified,\x20so\x20defaulting\x20to', 'ipc.config.socketRoot\x20+\x20ipc.config.appspace\x20+\x20ipc.config.id', this['config']['socketRoot'] + this['config']['appspace'] + this['config']['id']), _0x5c5d9b = this['config']['socketRoot'] + this['config']['appspace'] + this['config']['id']), !_0x138a61 && (_0x138a61 = _0x3ada91), this['server'] = new _0x47d5b1(_0x5c5d9b, this['config'], _0x1c4457), this['server']['on']('start', _0x138a61);
        }
        function _0x3ada91() {
        }
        function _0x3b7359(_0x25e2e6, _0x2095a3, _0x1cb8f4, _0x1c2e49) {
            typeof _0x25e2e6 == 'number' && (_0x1c2e49 = _0x1cb8f4, _0x1cb8f4 = _0x2095a3, _0x2095a3 = _0x25e2e6, _0x25e2e6 = ![]), typeof _0x25e2e6 == 'function' && (_0x1c2e49 = _0x25e2e6, _0x1cb8f4 = ![], _0x25e2e6 = ![], _0x2095a3 = ![]), !_0x25e2e6 && (this['log']('Server\x20host\x20not\x20specified,\x20so\x20defaulting\x20to', 'ipc.config.networkHost', this['config']['networkHost']), _0x25e2e6 = this['config']['networkHost']), (_0x25e2e6['toLowerCase']() == 'udp4' || _0x25e2e6['toLowerCase']() == 'udp6') && (_0x1c2e49 = _0x2095a3, _0x1cb8f4 = _0x25e2e6['toLowerCase'](), _0x2095a3 = ![], _0x25e2e6 = this['config']['networkHost']), typeof _0x2095a3 == 'string' && (_0x1c2e49 = _0x1cb8f4, _0x1cb8f4 = _0x2095a3, _0x2095a3 = ![]), typeof _0x2095a3 == 'function' && (_0x1c2e49 = _0x2095a3, _0x1cb8f4 = ![], _0x2095a3 = ![]), !_0x2095a3 && (this['log']('Server\x20port\x20not\x20specified,\x20so\x20defaulting\x20to', 'ipc.config.networkPort', this['config']['networkPort']), _0x2095a3 = this['config']['networkPort']), typeof _0x1cb8f4 == 'function' && (_0x1c2e49 = _0x1cb8f4, _0x1cb8f4 = ![]), !_0x1c2e49 && (_0x1c2e49 = _0x3ada91), this['server'] = new _0x47d5b1(_0x25e2e6, this['config'], _0x1c4457, _0x2095a3), _0x1cb8f4 && (this['server'][_0x1cb8f4] = !![], _0x1cb8f4 === 'udp4' && _0x25e2e6 === '::1' && (this['server']['path'] = '127.0.0.1')), this['server']['on']('start', _0x1c2e49);
        }
        function _0x2b3256(_0x2721d3, _0x2d54c8, _0x5eacb6) {
            typeof _0x2d54c8 == 'function' && (_0x5eacb6 = _0x2d54c8, _0x2d54c8 = ![]);
            !_0x5eacb6 && (_0x5eacb6 = _0x3ada91);
            if (!_0x2721d3) {
                this['log']('Service\x20id\x20required', 'Requested\x20service\x20connection\x20without\x20specifying\x20service\x20id.\x20Aborting\x20connection\x20attempt');
                return;
            }
            !_0x2d54c8 && (this['log']('Service\x20path\x20not\x20specified,\x20so\x20defaulting\x20to', 'ipc.config.socketRoot\x20+\x20ipc.config.appspace\x20+\x20id', (this['config']['socketRoot'] + this['config']['appspace'] + _0x2721d3)['data']), _0x2d54c8 = this['config']['socketRoot'] + this['config']['appspace'] + _0x2721d3);
            if (this['of'][_0x2721d3]) {
                if (!this['of'][_0x2721d3]['socket']['destroyed']) {
                    this['log']('Already\x20Connected\x20to', _0x2721d3, '-\x20So\x20executing\x20success\x20without\x20connection'), _0x5eacb6();
                    return;
                }
                this['of'][_0x2721d3]['socket']['destroy']();
            }
            this['of'][_0x2721d3] = new _0x582325(this['config'], this['log']), this['of'][_0x2721d3]['id'] = _0x2721d3, this['of'][_0x2721d3]['path'] = _0x2d54c8, this['of'][_0x2721d3]['connect'](), _0x5eacb6(this);
        }
        function _0x4defa5(_0x248e9a, _0x450750, _0x39fb18, _0x256554) {
            if (!_0x248e9a) {
                this['log']('Service\x20id\x20required', 'Requested\x20service\x20connection\x20without\x20specifying\x20service\x20id.\x20Aborting\x20connection\x20attempt');
                return;
            }
            typeof _0x450750 == 'number' && (_0x256554 = _0x39fb18, _0x39fb18 = _0x450750, _0x450750 = ![]);
            typeof _0x450750 == 'function' && (_0x256554 = _0x450750, _0x450750 = ![], _0x39fb18 = ![]);
            !_0x450750 && (this['log']('Server\x20host\x20not\x20specified,\x20so\x20defaulting\x20to', 'ipc.config.networkHost', this['config']['networkHost']), _0x450750 = this['config']['networkHost']);
            typeof _0x39fb18 == 'function' && (_0x256554 = _0x39fb18, _0x39fb18 = ![]);
            !_0x39fb18 && (this['log']('Server\x20port\x20not\x20specified,\x20so\x20defaulting\x20to', 'ipc.config.networkPort', this['config']['networkPort']), _0x39fb18 = this['config']['networkPort']);
            typeof _0x256554 == 'string' && (UDPType = _0x256554, _0x256554 = ![]);
            !_0x256554 && (_0x256554 = _0x3ada91);
            if (this['of'][_0x248e9a]) {
                if (!this['of'][_0x248e9a]['socket']['destroyed']) {
                    this['log']('Already\x20Connected\x20to', _0x248e9a, '-\x20So\x20executing\x20success\x20without\x20connection'), _0x256554();
                    return;
                }
                this['of'][_0x248e9a]['socket']['destroy']();
            }
            this['of'][_0x248e9a] = new _0x582325(this['config'], this['log']), this['of'][_0x248e9a]['id'] = _0x248e9a, this['of'][_0x248e9a]['socket'] ? this['of'][_0x248e9a]['socket']['id'] = _0x248e9a : null, this['of'][_0x248e9a]['path'] = _0x450750, this['of'][_0x248e9a]['port'] = _0x39fb18, this['of'][_0x248e9a]['connect'](), _0x256554(this);
        }
        _0x503f58['exports'] = _0x263432;
    }
};
;
