// test-maker/dist/180.js
'use strict';
exports['id'] = 0xb4, exports['ids'] = [0xb4], exports['modules'] = {
    0x79cc: (_0x32c836, _0x36ea98, _0xfa2413) => {
        _0xfa2413['r'](_0x36ea98), _0xfa2413['d'](_0x36ea98, { 'logGenericError': () => _0x47f4df });
        var _0x9ba7ae = _0xfa2413(0x1145c), _0x4731a5 = _0xfa2413['n'](_0x9ba7ae);
        async function _0x47f4df({
            error: _0x5a71fc,
            isWarning: _0x1bda0f,
            show: show = !![]
        }) {
            let _0x189a2e;
            try {
                _0x189a2e ? _0x1bda0f ? console['warn'](_0x189a2e['formatted']['ansi']) : console['error'](_0x189a2e['formatted']['ansi']) : _0x1bda0f ? console['warn'](_0x5a71fc) : console['error'](_0x4731a5()['redBright'](_0x5a71fc['stack'] || _0x5a71fc));
            } catch (_0x28a9ab) {
                console['error'](_0x4731a5()['redBright'](_0x28a9ab['stack'] || _0x28a9ab));
            } finally {
                console['info'](_0x4731a5()['cyanBright']('See\x20Log\x20file\x20for\x20more\x20Details...'));
            }
        }
    }
};
;
