// test-maker/dist/238.js
exports['id'] = 0xee, exports['ids'] = [0xee], exports['modules'] = {
    0xdcf3: (_0x260e23, _0x2c8910, _0x5c03b8) => {
        'use strict';
        const _0x13bc0b = _0x5c03b8(0x15d1a), _0x74b1f2 = _0x5c03b8(0x11569), _0x1a47ce = _0x5c03b8(0x10c1d)['mkdirsSync'], _0x3cc3b8 = _0x5c03b8(0x13743)['utimesMillisSync'], _0x114596 = _0x5c03b8(0x7c9d);
        function _0x11a6b9(_0x29d99b, _0x4eab45, _0x51b0ef) {
            typeof _0x51b0ef === 'function' && (_0x51b0ef = { 'filter': _0x51b0ef });
            _0x51b0ef = _0x51b0ef || {}, _0x51b0ef['clobber'] = 'clobber' in _0x51b0ef ? !!_0x51b0ef['clobber'] : !![], _0x51b0ef['overwrite'] = 'overwrite' in _0x51b0ef ? !!_0x51b0ef['overwrite'] : _0x51b0ef['clobber'];
            _0x51b0ef['preserveTimestamps'] && process['arch'] === 'ia32' && process['emitWarning']('Using\x20the\x20preserveTimestamps\x20option\x20in\x2032-bit\x20node\x20is\x20not\x20recommended;\x0a\x0a' + '\x09see\x20https://github.com/jprichardson/node-fs-extra/issues/269', 'Warning', 'fs-extra-WARN0002');
            const {
                srcStat: _0x449f6d,
                destStat: _0xaf4732
            } = _0x114596['checkPathsSync'](_0x29d99b, _0x4eab45, 'copy', _0x51b0ef);
            return _0x114596['checkParentPathsSync'](_0x29d99b, _0x449f6d, _0x4eab45, 'copy'), _0x57f110(_0xaf4732, _0x29d99b, _0x4eab45, _0x51b0ef);
        }
        function _0x57f110(_0x590c5d, _0x2cbe11, _0x47a26f, _0x1cd353) {
            if (_0x1cd353['filter'] && !_0x1cd353['filter'](_0x2cbe11, _0x47a26f))
                return;
            const _0x27b9e0 = _0x74b1f2['dirname'](_0x47a26f);
            if (!_0x13bc0b['existsSync'](_0x27b9e0))
                _0x1a47ce(_0x27b9e0);
            return _0x77a312(_0x590c5d, _0x2cbe11, _0x47a26f, _0x1cd353);
        }
        function _0x581b82(_0x11647a, _0x1dd6a9, _0x22051f, _0x426ff5) {
            if (_0x426ff5['filter'] && !_0x426ff5['filter'](_0x1dd6a9, _0x22051f))
                return;
            return _0x77a312(_0x11647a, _0x1dd6a9, _0x22051f, _0x426ff5);
        }
        function _0x77a312(_0x1a063f, _0x1ce00f, _0x3b6889, _0x225caa) {
            const _0x109bfa = _0x225caa['dereference'] ? _0x13bc0b['statSync'] : _0x13bc0b['lstatSync'], _0x1080f8 = _0x109bfa(_0x1ce00f);
            if (_0x1080f8['isDirectory']())
                return _0x46a4b0(_0x1080f8, _0x1a063f, _0x1ce00f, _0x3b6889, _0x225caa);
            else {
                if (_0x1080f8['isFile']() || _0x1080f8['isCharacterDevice']() || _0x1080f8['isBlockDevice']())
                    return _0x1a4c5d(_0x1080f8, _0x1a063f, _0x1ce00f, _0x3b6889, _0x225caa);
                else {
                    if (_0x1080f8['isSymbolicLink']())
                        return _0x372d67(_0x1a063f, _0x1ce00f, _0x3b6889, _0x225caa);
                    else {
                        if (_0x1080f8['isSocket']())
                            throw new Error('Cannot\x20copy\x20a\x20socket\x20file:\x20' + _0x1ce00f);
                        else {
                            if (_0x1080f8['isFIFO']())
                                throw new Error('Cannot\x20copy\x20a\x20FIFO\x20pipe:\x20' + _0x1ce00f);
                        }
                    }
                }
            }
            throw new Error('Unknown\x20file:\x20' + _0x1ce00f);
        }
        function _0x1a4c5d(_0x239d5d, _0x3f8404, _0x5e4d38, _0x1a5868, _0x34ba63) {
            if (!_0x3f8404)
                return _0x349956(_0x239d5d, _0x5e4d38, _0x1a5868, _0x34ba63);
            return _0x9a45db(_0x239d5d, _0x5e4d38, _0x1a5868, _0x34ba63);
        }
        function _0x9a45db(_0x36d47f, _0x33ba78, _0x483b0a, _0x2c8c03) {
            if (_0x2c8c03['overwrite'])
                return _0x13bc0b['unlinkSync'](_0x483b0a), _0x349956(_0x36d47f, _0x33ba78, _0x483b0a, _0x2c8c03);
            else {
                if (_0x2c8c03['errorOnExist'])
                    throw new Error('\x27' + _0x483b0a + '\x27\x20already\x20exists');
            }
        }
        function _0x349956(_0x5b14a7, _0x1c9f34, _0x49875f, _0x2117df) {
            _0x13bc0b['copyFileSync'](_0x1c9f34, _0x49875f);
            if (_0x2117df['preserveTimestamps'])
                _0x155e53(_0x5b14a7['mode'], _0x1c9f34, _0x49875f);
            return _0xaf71b1(_0x49875f, _0x5b14a7['mode']);
        }
        function _0x155e53(_0x19bb46, _0x49de8c, _0x1cfe0f) {
            if (_0x353d23(_0x19bb46))
                _0x188ba6(_0x1cfe0f, _0x19bb46);
            return _0x5d69df(_0x49de8c, _0x1cfe0f);
        }
        function _0x353d23(_0xa51232) {
            return (_0xa51232 & 0x80) === 0x0;
        }
        function _0x188ba6(_0x5c3466, _0xa6d0af) {
            return _0xaf71b1(_0x5c3466, _0xa6d0af | 0x80);
        }
        function _0xaf71b1(_0x42ee0a, _0xb811d4) {
            return _0x13bc0b['chmodSync'](_0x42ee0a, _0xb811d4);
        }
        function _0x5d69df(_0x41b6e7, _0x41f0df) {
            const _0x1b7cc3 = _0x13bc0b['statSync'](_0x41b6e7);
            return _0x3cc3b8(_0x41f0df, _0x1b7cc3['atime'], _0x1b7cc3['mtime']);
        }
        function _0x46a4b0(_0x8d9882, _0x1ffe66, _0x66a938, _0x1965a3, _0x4ea0bd) {
            if (!_0x1ffe66)
                return _0x1e6b81(_0x8d9882['mode'], _0x66a938, _0x1965a3, _0x4ea0bd);
            return _0x4ba7d0(_0x66a938, _0x1965a3, _0x4ea0bd);
        }
        function _0x1e6b81(_0x451a68, _0x1101d9, _0x1f80fd, _0xb04195) {
            return _0x13bc0b['mkdirSync'](_0x1f80fd), _0x4ba7d0(_0x1101d9, _0x1f80fd, _0xb04195), _0xaf71b1(_0x1f80fd, _0x451a68);
        }
        function _0x4ba7d0(_0x2bb550, _0x59a10a, _0x5a8b0a) {
            _0x13bc0b['readdirSync'](_0x2bb550)['forEach'](_0x516d08 => _0x194748(_0x516d08, _0x2bb550, _0x59a10a, _0x5a8b0a));
        }
        function _0x194748(_0x499c7e, _0x12b83c, _0x298c14, _0x21b7a9) {
            const _0x1c220f = _0x74b1f2['join'](_0x12b83c, _0x499c7e), _0x846769 = _0x74b1f2['join'](_0x298c14, _0x499c7e), {destStat: _0x4cf641} = _0x114596['checkPathsSync'](_0x1c220f, _0x846769, 'copy', _0x21b7a9);
            return _0x581b82(_0x4cf641, _0x1c220f, _0x846769, _0x21b7a9);
        }
        function _0x372d67(_0x1751b3, _0x12e312, _0x445f4a, _0x113a83) {
            let _0x11ffbf = _0x13bc0b['readlinkSync'](_0x12e312);
            _0x113a83['dereference'] && (_0x11ffbf = _0x74b1f2['resolve'](process['cwd'](), _0x11ffbf));
            if (!_0x1751b3)
                return _0x13bc0b['symlinkSync'](_0x11ffbf, _0x445f4a);
            else {
                let _0x1237f4;
                try {
                    _0x1237f4 = _0x13bc0b['readlinkSync'](_0x445f4a);
                } catch (_0x32e7f6) {
                    if (_0x32e7f6['code'] === 'EINVAL' || _0x32e7f6['code'] === 'UNKNOWN')
                        return _0x13bc0b['symlinkSync'](_0x11ffbf, _0x445f4a);
                    throw _0x32e7f6;
                }
                _0x113a83['dereference'] && (_0x1237f4 = _0x74b1f2['resolve'](process['cwd'](), _0x1237f4));
                if (_0x114596['isSrcSubdir'](_0x11ffbf, _0x1237f4))
                    throw new Error('Cannot\x20copy\x20\x27' + _0x11ffbf + '\x27\x20to\x20a\x20subdirectory\x20of\x20itself,\x20\x27' + _0x1237f4 + '\x27.');
                if (_0x13bc0b['statSync'](_0x445f4a)['isDirectory']() && _0x114596['isSrcSubdir'](_0x1237f4, _0x11ffbf))
                    throw new Error('Cannot\x20overwrite\x20\x27' + _0x1237f4 + '\x27\x20with\x20\x27' + _0x11ffbf + '\x27.');
                return _0x39d876(_0x11ffbf, _0x445f4a);
            }
        }
        function _0x39d876(_0x49c579, _0x2b4332) {
            return _0x13bc0b['unlinkSync'](_0x2b4332), _0x13bc0b['symlinkSync'](_0x49c579, _0x2b4332);
        }
        _0x260e23['exports'] = _0x11a6b9;
    },
    0x11885: (_0x2b7b4b, _0x2898b3, _0x16045c) => {
        'use strict';
        const _0x7652d9 = _0x16045c(0x15d1a), _0x143204 = _0x16045c(0x11569), _0x1d224a = _0x16045c(0x10c1d)['mkdirs'], _0x1a2c46 = _0x16045c(0x15330)['pathExists'], _0x4457af = _0x16045c(0x13743)['utimesMillis'], _0x1138b0 = _0x16045c(0x7c9d);
        function _0x3cfdea(_0x6732e8, _0x1e7aaa, _0x452515, _0x47c72f) {
            if (typeof _0x452515 === 'function' && !_0x47c72f)
                _0x47c72f = _0x452515, _0x452515 = {};
            else
                typeof _0x452515 === 'function' && (_0x452515 = { 'filter': _0x452515 });
            _0x47c72f = _0x47c72f || function () {
            }, _0x452515 = _0x452515 || {}, _0x452515['clobber'] = 'clobber' in _0x452515 ? !!_0x452515['clobber'] : !![], _0x452515['overwrite'] = 'overwrite' in _0x452515 ? !!_0x452515['overwrite'] : _0x452515['clobber'], _0x452515['preserveTimestamps'] && process['arch'] === 'ia32' && process['emitWarning']('Using\x20the\x20preserveTimestamps\x20option\x20in\x2032-bit\x20node\x20is\x20not\x20recommended;\x0a\x0a' + '\x09see\x20https://github.com/jprichardson/node-fs-extra/issues/269', 'Warning', 'fs-extra-WARN0001'), _0x1138b0['checkPaths'](_0x6732e8, _0x1e7aaa, 'copy', _0x452515, (_0x507973, _0x2ffb72) => {
                if (_0x507973)
                    return _0x47c72f(_0x507973);
                const {
                    srcStat: _0x368444,
                    destStat: _0x51e217
                } = _0x2ffb72;
                _0x1138b0['checkParentPaths'](_0x6732e8, _0x368444, _0x1e7aaa, 'copy', _0x2c76a9 => {
                    if (_0x2c76a9)
                        return _0x47c72f(_0x2c76a9);
                    if (_0x452515['filter'])
                        return _0x53c8ea(_0x2ec4ac, _0x51e217, _0x6732e8, _0x1e7aaa, _0x452515, _0x47c72f);
                    return _0x2ec4ac(_0x51e217, _0x6732e8, _0x1e7aaa, _0x452515, _0x47c72f);
                });
            });
        }
        function _0x2ec4ac(_0x362dc3, _0x4ff009, _0x442516, _0x511d4f, _0x57dd24) {
            const _0x4ddb04 = _0x143204['dirname'](_0x442516);
            _0x1a2c46(_0x4ddb04, (_0x71bf4f, _0x1738b2) => {
                if (_0x71bf4f)
                    return _0x57dd24(_0x71bf4f);
                if (_0x1738b2)
                    return _0x32d730(_0x362dc3, _0x4ff009, _0x442516, _0x511d4f, _0x57dd24);
                _0x1d224a(_0x4ddb04, _0x1453b6 => {
                    if (_0x1453b6)
                        return _0x57dd24(_0x1453b6);
                    return _0x32d730(_0x362dc3, _0x4ff009, _0x442516, _0x511d4f, _0x57dd24);
                });
            });
        }
        function _0x53c8ea(_0x43a9e4, _0x2db443, _0x46ba19, _0x4baf0d, _0xef27b, _0x4ff249) {
            Promise['resolve'](_0xef27b['filter'](_0x46ba19, _0x4baf0d))['then'](_0x155254 => {
                if (_0x155254)
                    return _0x43a9e4(_0x2db443, _0x46ba19, _0x4baf0d, _0xef27b, _0x4ff249);
                return _0x4ff249();
            }, _0x45d308 => _0x4ff249(_0x45d308));
        }
        function _0x2f6a5d(_0x249e49, _0x5bb2b0, _0x14470d, _0x40b2cc, _0x26729a) {
            if (_0x40b2cc['filter'])
                return _0x53c8ea(_0x32d730, _0x249e49, _0x5bb2b0, _0x14470d, _0x40b2cc, _0x26729a);
            return _0x32d730(_0x249e49, _0x5bb2b0, _0x14470d, _0x40b2cc, _0x26729a);
        }
        function _0x32d730(_0x40f1b7, _0x3b3e99, _0x31b645, _0x5b89a0, _0x2c60bc) {
            const _0x4a8d47 = _0x5b89a0['dereference'] ? _0x7652d9['stat'] : _0x7652d9['lstat'];
            _0x4a8d47(_0x3b3e99, (_0x33a043, _0x2429c7) => {
                if (_0x33a043)
                    return _0x2c60bc(_0x33a043);
                if (_0x2429c7['isDirectory']())
                    return _0xaa53b1(_0x2429c7, _0x40f1b7, _0x3b3e99, _0x31b645, _0x5b89a0, _0x2c60bc);
                else {
                    if (_0x2429c7['isFile']() || _0x2429c7['isCharacterDevice']() || _0x2429c7['isBlockDevice']())
                        return _0x246b7c(_0x2429c7, _0x40f1b7, _0x3b3e99, _0x31b645, _0x5b89a0, _0x2c60bc);
                    else {
                        if (_0x2429c7['isSymbolicLink']())
                            return _0x920d2d(_0x40f1b7, _0x3b3e99, _0x31b645, _0x5b89a0, _0x2c60bc);
                        else {
                            if (_0x2429c7['isSocket']())
                                return _0x2c60bc(new Error('Cannot\x20copy\x20a\x20socket\x20file:\x20' + _0x3b3e99));
                            else {
                                if (_0x2429c7['isFIFO']())
                                    return _0x2c60bc(new Error('Cannot\x20copy\x20a\x20FIFO\x20pipe:\x20' + _0x3b3e99));
                            }
                        }
                    }
                }
                return _0x2c60bc(new Error('Unknown\x20file:\x20' + _0x3b3e99));
            });
        }
        function _0x246b7c(_0x2b795b, _0x1d1e8a, _0x13fa08, _0x58f460, _0x778941, _0x3c5d87) {
            if (!_0x1d1e8a)
                return _0x1f49ab(_0x2b795b, _0x13fa08, _0x58f460, _0x778941, _0x3c5d87);
            return _0x417b30(_0x2b795b, _0x13fa08, _0x58f460, _0x778941, _0x3c5d87);
        }
        function _0x417b30(_0x3d797c, _0xa4c589, _0x320ba5, _0x27eabc, _0x3f1f41) {
            if (_0x27eabc['overwrite'])
                _0x7652d9['unlink'](_0x320ba5, _0x1e5282 => {
                    if (_0x1e5282)
                        return _0x3f1f41(_0x1e5282);
                    return _0x1f49ab(_0x3d797c, _0xa4c589, _0x320ba5, _0x27eabc, _0x3f1f41);
                });
            else {
                if (_0x27eabc['errorOnExist'])
                    return _0x3f1f41(new Error('\x27' + _0x320ba5 + '\x27\x20already\x20exists'));
                else
                    return _0x3f1f41();
            }
        }
        function _0x1f49ab(_0x13f589, _0x45848f, _0x3e1844, _0x29b9d6, _0xb3c5a4) {
            _0x7652d9['copyFile'](_0x45848f, _0x3e1844, _0x4967e5 => {
                if (_0x4967e5)
                    return _0xb3c5a4(_0x4967e5);
                if (_0x29b9d6['preserveTimestamps'])
                    return _0x16fb5c(_0x13f589['mode'], _0x45848f, _0x3e1844, _0xb3c5a4);
                return _0x3a946f(_0x3e1844, _0x13f589['mode'], _0xb3c5a4);
            });
        }
        function _0x16fb5c(_0x38d224, _0x5efcc0, _0x4d48ce, _0x1c55fe) {
            if (_0x1b27b7(_0x38d224))
                return _0x462968(_0x4d48ce, _0x38d224, _0x1bbef3 => {
                    if (_0x1bbef3)
                        return _0x1c55fe(_0x1bbef3);
                    return _0x12e6cf(_0x38d224, _0x5efcc0, _0x4d48ce, _0x1c55fe);
                });
            return _0x12e6cf(_0x38d224, _0x5efcc0, _0x4d48ce, _0x1c55fe);
        }
        function _0x1b27b7(_0x4ef602) {
            return (_0x4ef602 & 0x80) === 0x0;
        }
        function _0x462968(_0x5eedf1, _0x1000eb, _0x5478ff) {
            return _0x3a946f(_0x5eedf1, _0x1000eb | 0x80, _0x5478ff);
        }
        function _0x12e6cf(_0x4a71e1, _0x459c56, _0xb093ad, _0x476ac9) {
            _0x5509bc(_0x459c56, _0xb093ad, _0x30dc60 => {
                if (_0x30dc60)
                    return _0x476ac9(_0x30dc60);
                return _0x3a946f(_0xb093ad, _0x4a71e1, _0x476ac9);
            });
        }
        function _0x3a946f(_0x29af92, _0x1769d9, _0x431202) {
            return _0x7652d9['chmod'](_0x29af92, _0x1769d9, _0x431202);
        }
        function _0x5509bc(_0x330d5a, _0x264ac9, _0x387255) {
            _0x7652d9['stat'](_0x330d5a, (_0x3ec06e, _0x32b626) => {
                if (_0x3ec06e)
                    return _0x387255(_0x3ec06e);
                return _0x4457af(_0x264ac9, _0x32b626['atime'], _0x32b626['mtime'], _0x387255);
            });
        }
        function _0xaa53b1(_0x269e7c, _0x231fb6, _0x382149, _0x7ac668, _0x5d81ce, _0x27244f) {
            if (!_0x231fb6)
                return _0x1e683d(_0x269e7c['mode'], _0x382149, _0x7ac668, _0x5d81ce, _0x27244f);
            return _0x357b8a(_0x382149, _0x7ac668, _0x5d81ce, _0x27244f);
        }
        function _0x1e683d(_0x2cfe38, _0x40ef8f, _0x339f64, _0x211b8d, _0x5c369e) {
            _0x7652d9['mkdir'](_0x339f64, _0x2efd21 => {
                if (_0x2efd21)
                    return _0x5c369e(_0x2efd21);
                _0x357b8a(_0x40ef8f, _0x339f64, _0x211b8d, _0x276268 => {
                    if (_0x276268)
                        return _0x5c369e(_0x276268);
                    return _0x3a946f(_0x339f64, _0x2cfe38, _0x5c369e);
                });
            });
        }
        function _0x357b8a(_0x44c81c, _0x42e0b0, _0xa0ee10, _0x383532) {
            _0x7652d9['readdir'](_0x44c81c, (_0x355d75, _0x2cbf42) => {
                if (_0x355d75)
                    return _0x383532(_0x355d75);
                return _0x553815(_0x2cbf42, _0x44c81c, _0x42e0b0, _0xa0ee10, _0x383532);
            });
        }
        function _0x553815(_0x529008, _0x144951, _0x37e932, _0x52d51f, _0x3a9f1e) {
            const _0x2ce6e6 = _0x529008['pop']();
            if (!_0x2ce6e6)
                return _0x3a9f1e();
            return _0x27c1aa(_0x529008, _0x2ce6e6, _0x144951, _0x37e932, _0x52d51f, _0x3a9f1e);
        }
        function _0x27c1aa(_0x5918aa, _0xc3f993, _0xbf9ade, _0x3f7ed7, _0x355876, _0x31d14d) {
            const _0x4ff863 = _0x143204['join'](_0xbf9ade, _0xc3f993), _0x1178d8 = _0x143204['join'](_0x3f7ed7, _0xc3f993);
            _0x1138b0['checkPaths'](_0x4ff863, _0x1178d8, 'copy', _0x355876, (_0x4e90a6, _0x1dc0dd) => {
                if (_0x4e90a6)
                    return _0x31d14d(_0x4e90a6);
                const {destStat: _0x19dd53} = _0x1dc0dd;
                _0x2f6a5d(_0x19dd53, _0x4ff863, _0x1178d8, _0x355876, _0x501a19 => {
                    if (_0x501a19)
                        return _0x31d14d(_0x501a19);
                    return _0x553815(_0x5918aa, _0xbf9ade, _0x3f7ed7, _0x355876, _0x31d14d);
                });
            });
        }
        function _0x920d2d(_0x440992, _0x41fd98, _0x2ae1fc, _0x18ab7e, _0x4a3d30) {
            _0x7652d9['readlink'](_0x41fd98, (_0x2c2d64, _0x20f8be) => {
                if (_0x2c2d64)
                    return _0x4a3d30(_0x2c2d64);
                _0x18ab7e['dereference'] && (_0x20f8be = _0x143204['resolve'](process['cwd'](), _0x20f8be));
                if (!_0x440992)
                    return _0x7652d9['symlink'](_0x20f8be, _0x2ae1fc, _0x4a3d30);
                else
                    _0x7652d9['readlink'](_0x2ae1fc, (_0x33f869, _0x50c94c) => {
                        if (_0x33f869) {
                            if (_0x33f869['code'] === 'EINVAL' || _0x33f869['code'] === 'UNKNOWN')
                                return _0x7652d9['symlink'](_0x20f8be, _0x2ae1fc, _0x4a3d30);
                            return _0x4a3d30(_0x33f869);
                        }
                        _0x18ab7e['dereference'] && (_0x50c94c = _0x143204['resolve'](process['cwd'](), _0x50c94c));
                        if (_0x1138b0['isSrcSubdir'](_0x20f8be, _0x50c94c))
                            return _0x4a3d30(new Error('Cannot\x20copy\x20\x27' + _0x20f8be + '\x27\x20to\x20a\x20subdirectory\x20of\x20itself,\x20\x27' + _0x50c94c + '\x27.'));
                        if (_0x440992['isDirectory']() && _0x1138b0['isSrcSubdir'](_0x50c94c, _0x20f8be))
                            return _0x4a3d30(new Error('Cannot\x20overwrite\x20\x27' + _0x50c94c + '\x27\x20with\x20\x27' + _0x20f8be + '\x27.'));
                        return _0x472159(_0x20f8be, _0x2ae1fc, _0x4a3d30);
                    });
            });
        }
        function _0x472159(_0x4b6f67, _0x4883ac, _0x1a648b) {
            _0x7652d9['unlink'](_0x4883ac, _0x40ab9e => {
                if (_0x40ab9e)
                    return _0x1a648b(_0x40ab9e);
                return _0x7652d9['symlink'](_0x4b6f67, _0x4883ac, _0x1a648b);
            });
        }
        _0x2b7b4b['exports'] = _0x3cfdea;
    },
    0xb839: (_0x5e627c, _0x1fea2b, _0x530b3f) => {
        'use strict';
        const _0x276a16 = _0x530b3f(0x15f6a)['fromCallback'];
        _0x5e627c['exports'] = {
            'copy': _0x276a16(_0x530b3f(0x11885)),
            'copySync': _0x530b3f(0xdcf3)
        };
    },
    0x112ad: (_0x28ec47, _0x59a22c, _0x48de32) => {
        'use strict';
        const _0x284ab6 = _0x48de32(0x15f6a)['fromPromise'], _0x5b2a6c = _0x48de32(0x15a17), _0x4729cc = _0x48de32(0x11569), _0xaf45e7 = _0x48de32(0x10c1d), _0x425c2b = _0x48de32(0x983d), _0x4049f3 = _0x284ab6(async function _0x4c62e3(_0x1a6188) {
                let _0x571a45;
                try {
                    _0x571a45 = await _0x5b2a6c['readdir'](_0x1a6188);
                } catch {
                    return _0xaf45e7['mkdirs'](_0x1a6188);
                }
                return Promise['all'](_0x571a45['map'](_0xcdb9a2 => _0x425c2b['remove'](_0x4729cc['join'](_0x1a6188, _0xcdb9a2))));
            });
        function _0x4d7591(_0x31863f) {
            let _0x2199da;
            try {
                _0x2199da = _0x5b2a6c['readdirSync'](_0x31863f);
            } catch {
                return _0xaf45e7['mkdirsSync'](_0x31863f);
            }
            _0x2199da['forEach'](_0x89ed2f => {
                _0x89ed2f = _0x4729cc['join'](_0x31863f, _0x89ed2f), _0x425c2b['removeSync'](_0x89ed2f);
            });
        }
        _0x28ec47['exports'] = {
            'emptyDirSync': _0x4d7591,
            'emptydirSync': _0x4d7591,
            'emptyDir': _0x4049f3,
            'emptydir': _0x4049f3
        };
    },
    0x2c43: (_0x99a0e8, _0x14db7c, _0x37be33) => {
        'use strict';
        const _0x1ac6f9 = _0x37be33(0x15f6a)['fromCallback'], _0x4cfdac = _0x37be33(0x11569), _0x1a5402 = _0x37be33(0x15d1a), _0x197b11 = _0x37be33(0x10c1d);
        function _0x28bb5f(_0x53942e, _0x229f32) {
            function _0x3dd6a6() {
                _0x1a5402['writeFile'](_0x53942e, '', _0x222867 => {
                    if (_0x222867)
                        return _0x229f32(_0x222867);
                    _0x229f32();
                });
            }
            _0x1a5402['stat'](_0x53942e, (_0x3cef00, _0x4449df) => {
                if (!_0x3cef00 && _0x4449df['isFile']())
                    return _0x229f32();
                const _0x33790c = _0x4cfdac['dirname'](_0x53942e);
                _0x1a5402['stat'](_0x33790c, (_0x3e8389, _0x598ece) => {
                    if (_0x3e8389) {
                        if (_0x3e8389['code'] === 'ENOENT')
                            return _0x197b11['mkdirs'](_0x33790c, _0xfeb7dc => {
                                if (_0xfeb7dc)
                                    return _0x229f32(_0xfeb7dc);
                                _0x3dd6a6();
                            });
                        return _0x229f32(_0x3e8389);
                    }
                    if (_0x598ece['isDirectory']())
                        _0x3dd6a6();
                    else
                        _0x1a5402['readdir'](_0x33790c, _0x3689b8 => {
                            if (_0x3689b8)
                                return _0x229f32(_0x3689b8);
                        });
                });
            });
        }
        function _0xebedce(_0x530688) {
            let _0x3a209a;
            try {
                _0x3a209a = _0x1a5402['statSync'](_0x530688);
            } catch {
            }
            if (_0x3a209a && _0x3a209a['isFile']())
                return;
            const _0x4332ab = _0x4cfdac['dirname'](_0x530688);
            try {
                !_0x1a5402['statSync'](_0x4332ab)['isDirectory']() && _0x1a5402['readdirSync'](_0x4332ab);
            } catch (_0x3354ed) {
                if (_0x3354ed && _0x3354ed['code'] === 'ENOENT')
                    _0x197b11['mkdirsSync'](_0x4332ab);
                else
                    throw _0x3354ed;
            }
            _0x1a5402['writeFileSync'](_0x530688, '');
        }
        _0x99a0e8['exports'] = {
            'createFile': _0x1ac6f9(_0x28bb5f),
            'createFileSync': _0xebedce
        };
    },
    0x143a2: (_0x21f6d3, _0x1ef342, _0x54ebf5) => {
        'use strict';
        const {
                createFile: _0x13c068,
                createFileSync: _0x4d011a
            } = _0x54ebf5(0x2c43), {
                createLink: _0x314f1f,
                createLinkSync: _0x164b48
            } = _0x54ebf5(0x150ea), {
                createSymlink: _0x253630,
                createSymlinkSync: _0x33c1d6
            } = _0x54ebf5(0x16c50);
        _0x21f6d3['exports'] = {
            'createFile': _0x13c068,
            'createFileSync': _0x4d011a,
            'ensureFile': _0x13c068,
            'ensureFileSync': _0x4d011a,
            'createLink': _0x314f1f,
            'createLinkSync': _0x164b48,
            'ensureLink': _0x314f1f,
            'ensureLinkSync': _0x164b48,
            'createSymlink': _0x253630,
            'createSymlinkSync': _0x33c1d6,
            'ensureSymlink': _0x253630,
            'ensureSymlinkSync': _0x33c1d6
        };
    },
    0x150ea: (_0x5b2c18, _0x285acb, _0xbd2cd8) => {
        'use strict';
        const _0xb1022 = _0xbd2cd8(0x15f6a)['fromCallback'], _0x11e85b = _0xbd2cd8(0x11569), _0x4d3062 = _0xbd2cd8(0x15d1a), _0x3508e0 = _0xbd2cd8(0x10c1d), _0x2406d3 = _0xbd2cd8(0x15330)['pathExists'], {areIdentical: _0x348b2e} = _0xbd2cd8(0x7c9d);
        function _0x3ba402(_0x46921, _0x2651f3, _0x466d5d) {
            function _0x1f9aa6(_0x2156d3, _0x59266a) {
                _0x4d3062['link'](_0x2156d3, _0x59266a, _0x33088d => {
                    if (_0x33088d)
                        return _0x466d5d(_0x33088d);
                    _0x466d5d(null);
                });
            }
            _0x4d3062['lstat'](_0x2651f3, (_0x18922e, _0x30c599) => {
                _0x4d3062['lstat'](_0x46921, (_0x15b4fd, _0xe65454) => {
                    if (_0x15b4fd)
                        return _0x15b4fd['message'] = _0x15b4fd['message']['replace']('lstat', 'ensureLink'), _0x466d5d(_0x15b4fd);
                    if (_0x30c599 && _0x348b2e(_0xe65454, _0x30c599))
                        return _0x466d5d(null);
                    const _0x7b659a = _0x11e85b['dirname'](_0x2651f3);
                    _0x2406d3(_0x7b659a, (_0x340483, _0x39e473) => {
                        if (_0x340483)
                            return _0x466d5d(_0x340483);
                        if (_0x39e473)
                            return _0x1f9aa6(_0x46921, _0x2651f3);
                        _0x3508e0['mkdirs'](_0x7b659a, _0x778372 => {
                            if (_0x778372)
                                return _0x466d5d(_0x778372);
                            _0x1f9aa6(_0x46921, _0x2651f3);
                        });
                    });
                });
            });
        }
        function _0x289744(_0x1a7b35, _0x51b7a3) {
            let _0x3634c5;
            try {
                _0x3634c5 = _0x4d3062['lstatSync'](_0x51b7a3);
            } catch {
            }
            try {
                const _0x12e6b9 = _0x4d3062['lstatSync'](_0x1a7b35);
                if (_0x3634c5 && _0x348b2e(_0x12e6b9, _0x3634c5))
                    return;
            } catch (_0x58389c) {
                _0x58389c['message'] = _0x58389c['message']['replace']('lstat', 'ensureLink');
                throw _0x58389c;
            }
            const _0x2948d2 = _0x11e85b['dirname'](_0x51b7a3), _0x52fc13 = _0x4d3062['existsSync'](_0x2948d2);
            if (_0x52fc13)
                return _0x4d3062['linkSync'](_0x1a7b35, _0x51b7a3);
            return _0x3508e0['mkdirsSync'](_0x2948d2), _0x4d3062['linkSync'](_0x1a7b35, _0x51b7a3);
        }
        _0x5b2c18['exports'] = {
            'createLink': _0xb1022(_0x3ba402),
            'createLinkSync': _0x289744
        };
    },
    0x1496b: (_0x4cfdf4, _0x481e9a, _0x15131c) => {
        'use strict';
        const _0x13caa5 = _0x15131c(0x11569), _0xcb25cd = _0x15131c(0x15d1a), _0x42fbcf = _0x15131c(0x15330)['pathExists'];
        function _0x3687ab(_0x5b796a, _0x461197, _0x194b54) {
            if (_0x13caa5['isAbsolute'](_0x5b796a))
                return _0xcb25cd['lstat'](_0x5b796a, _0x220f27 => {
                    if (_0x220f27)
                        return _0x220f27['message'] = _0x220f27['message']['replace']('lstat', 'ensureSymlink'), _0x194b54(_0x220f27);
                    return _0x194b54(null, {
                        'toCwd': _0x5b796a,
                        'toDst': _0x5b796a
                    });
                });
            else {
                const _0x5c3f14 = _0x13caa5['dirname'](_0x461197), _0x136f1e = _0x13caa5['join'](_0x5c3f14, _0x5b796a);
                return _0x42fbcf(_0x136f1e, (_0x5411db, _0x4389e5) => {
                    if (_0x5411db)
                        return _0x194b54(_0x5411db);
                    return _0x4389e5 ? _0x194b54(null, {
                        'toCwd': _0x136f1e,
                        'toDst': _0x5b796a
                    }) : _0xcb25cd['lstat'](_0x5b796a, _0x566337 => {
                        if (_0x566337)
                            return _0x566337['message'] = _0x566337['message']['replace']('lstat', 'ensureSymlink'), _0x194b54(_0x566337);
                        return _0x194b54(null, {
                            'toCwd': _0x5b796a,
                            'toDst': _0x13caa5['relative'](_0x5c3f14, _0x5b796a)
                        });
                    });
                });
            }
        }
        function _0x4c73b3(_0x515bd7, _0x272dcf) {
            let _0x325715;
            if (_0x13caa5['isAbsolute'](_0x515bd7)) {
                _0x325715 = _0xcb25cd['existsSync'](_0x515bd7);
                if (!_0x325715)
                    throw new Error('absolute\x20srcpath\x20does\x20not\x20exist');
                return {
                    'toCwd': _0x515bd7,
                    'toDst': _0x515bd7
                };
            } else {
                const _0x4fcdf0 = _0x13caa5['dirname'](_0x272dcf), _0x3cb089 = _0x13caa5['join'](_0x4fcdf0, _0x515bd7);
                _0x325715 = _0xcb25cd['existsSync'](_0x3cb089);
                if (_0x325715)
                    return {
                        'toCwd': _0x3cb089,
                        'toDst': _0x515bd7
                    };
                else {
                    _0x325715 = _0xcb25cd['existsSync'](_0x515bd7);
                    if (!_0x325715)
                        throw new Error('relative\x20srcpath\x20does\x20not\x20exist');
                    return {
                        'toCwd': _0x515bd7,
                        'toDst': _0x13caa5['relative'](_0x4fcdf0, _0x515bd7)
                    };
                }
            }
        }
        _0x4cfdf4['exports'] = {
            'symlinkPaths': _0x3687ab,
            'symlinkPathsSync': _0x4c73b3
        };
    },
    0x1ca0: (_0x5b8db3, _0x4655e9, _0x9b1357) => {
        'use strict';
        const _0x53a396 = _0x9b1357(0x15d1a);
        function _0x871b27(_0x2b365d, _0x10f42e, _0x19a014) {
            _0x19a014 = typeof _0x10f42e === 'function' ? _0x10f42e : _0x19a014, _0x10f42e = typeof _0x10f42e === 'function' ? ![] : _0x10f42e;
            if (_0x10f42e)
                return _0x19a014(null, _0x10f42e);
            _0x53a396['lstat'](_0x2b365d, (_0x19bf91, _0x55d015) => {
                if (_0x19bf91)
                    return _0x19a014(null, 'file');
                _0x10f42e = _0x55d015 && _0x55d015['isDirectory']() ? 'dir' : 'file', _0x19a014(null, _0x10f42e);
            });
        }
        function _0x52cf29(_0x2c4520, _0x50d0f7) {
            let _0x489dbe;
            if (_0x50d0f7)
                return _0x50d0f7;
            try {
                _0x489dbe = _0x53a396['lstatSync'](_0x2c4520);
            } catch {
                return 'file';
            }
            return _0x489dbe && _0x489dbe['isDirectory']() ? 'dir' : 'file';
        }
        _0x5b8db3['exports'] = {
            'symlinkType': _0x871b27,
            'symlinkTypeSync': _0x52cf29
        };
    },
    0x16c50: (_0x3e6391, _0x29021f, _0xb69054) => {
        'use strict';
        const _0x1d4bb6 = _0xb69054(0x15f6a)['fromCallback'], _0x4c882a = _0xb69054(0x11569), _0x2a71b1 = _0xb69054(0x15a17), _0x2ee95f = _0xb69054(0x10c1d), _0x386005 = _0x2ee95f['mkdirs'], _0x29f885 = _0x2ee95f['mkdirsSync'], _0x5d3a6a = _0xb69054(0x1496b), _0x132775 = _0x5d3a6a['symlinkPaths'], _0x59e7ed = _0x5d3a6a['symlinkPathsSync'], _0x4c2d98 = _0xb69054(0x1ca0), _0x2760cc = _0x4c2d98['symlinkType'], _0x2bcf5d = _0x4c2d98['symlinkTypeSync'], _0x52429b = _0xb69054(0x15330)['pathExists'], {areIdentical: _0x5e8112} = _0xb69054(0x7c9d);
        function _0x2fbea3(_0x1baaaf, _0xa1f35, _0x11938e, _0x4cc443) {
            _0x4cc443 = typeof _0x11938e === 'function' ? _0x11938e : _0x4cc443, _0x11938e = typeof _0x11938e === 'function' ? ![] : _0x11938e, _0x2a71b1['lstat'](_0xa1f35, (_0x2ea8c1, _0xae11bf) => {
                if (!_0x2ea8c1 && _0xae11bf['isSymbolicLink']())
                    Promise['all']([
                        _0x2a71b1['stat'](_0x1baaaf),
                        _0x2a71b1['stat'](_0xa1f35)
                    ])['then'](([_0x2cf2f6, _0x34664c]) => {
                        if (_0x5e8112(_0x2cf2f6, _0x34664c))
                            return _0x4cc443(null);
                        _0x161130(_0x1baaaf, _0xa1f35, _0x11938e, _0x4cc443);
                    });
                else
                    _0x161130(_0x1baaaf, _0xa1f35, _0x11938e, _0x4cc443);
            });
        }
        function _0x161130(_0x37f9a7, _0x59d6c9, _0x472b07, _0x50b716) {
            _0x132775(_0x37f9a7, _0x59d6c9, (_0xa0268c, _0x4b725c) => {
                if (_0xa0268c)
                    return _0x50b716(_0xa0268c);
                _0x37f9a7 = _0x4b725c['toDst'], _0x2760cc(_0x4b725c['toCwd'], _0x472b07, (_0xf067d3, _0x40cf19) => {
                    if (_0xf067d3)
                        return _0x50b716(_0xf067d3);
                    const _0x45bced = _0x4c882a['dirname'](_0x59d6c9);
                    _0x52429b(_0x45bced, (_0x1bbce4, _0xe68377) => {
                        if (_0x1bbce4)
                            return _0x50b716(_0x1bbce4);
                        if (_0xe68377)
                            return _0x2a71b1['symlink'](_0x37f9a7, _0x59d6c9, _0x40cf19, _0x50b716);
                        _0x386005(_0x45bced, _0x858ef5 => {
                            if (_0x858ef5)
                                return _0x50b716(_0x858ef5);
                            _0x2a71b1['symlink'](_0x37f9a7, _0x59d6c9, _0x40cf19, _0x50b716);
                        });
                    });
                });
            });
        }
        function _0x37cb7c(_0x581445, _0x2d1225, _0xab7bc5) {
            let _0x40357c;
            try {
                _0x40357c = _0x2a71b1['lstatSync'](_0x2d1225);
            } catch {
            }
            if (_0x40357c && _0x40357c['isSymbolicLink']()) {
                const _0x3eac5e = _0x2a71b1['statSync'](_0x581445), _0x67b25b = _0x2a71b1['statSync'](_0x2d1225);
                if (_0x5e8112(_0x3eac5e, _0x67b25b))
                    return;
            }
            const _0x127c0b = _0x59e7ed(_0x581445, _0x2d1225);
            _0x581445 = _0x127c0b['toDst'], _0xab7bc5 = _0x2bcf5d(_0x127c0b['toCwd'], _0xab7bc5);
            const _0x111efb = _0x4c882a['dirname'](_0x2d1225), _0x53bf31 = _0x2a71b1['existsSync'](_0x111efb);
            if (_0x53bf31)
                return _0x2a71b1['symlinkSync'](_0x581445, _0x2d1225, _0xab7bc5);
            return _0x29f885(_0x111efb), _0x2a71b1['symlinkSync'](_0x581445, _0x2d1225, _0xab7bc5);
        }
        _0x3e6391['exports'] = {
            'createSymlink': _0x1d4bb6(_0x2fbea3),
            'createSymlinkSync': _0x37cb7c
        };
    },
    0x15a17: (_0x387d4e, _0x482d2f, _0x47afe4) => {
        'use strict';
        const _0x209fd0 = _0x47afe4(0x15f6a)['fromCallback'], _0x53c3e5 = _0x47afe4(0x15d1a), _0x2ed93a = [
                'access',
                'appendFile',
                'chmod',
                'chown',
                'close',
                'copyFile',
                'fchmod',
                'fchown',
                'fdatasync',
                'fstat',
                'fsync',
                'ftruncate',
                'futimes',
                'lchmod',
                'lchown',
                'link',
                'lstat',
                'mkdir',
                'mkdtemp',
                'open',
                'opendir',
                'readdir',
                'readFile',
                'readlink',
                'realpath',
                'rename',
                'rm',
                'rmdir',
                'stat',
                'symlink',
                'truncate',
                'unlink',
                'utimes',
                'writeFile'
            ]['filter'](_0x3e3725 => {
                return typeof _0x53c3e5[_0x3e3725] === 'function';
            });
        Object['assign'](_0x482d2f, _0x53c3e5), _0x2ed93a['forEach'](_0x3d0200 => {
            _0x482d2f[_0x3d0200] = _0x209fd0(_0x53c3e5[_0x3d0200]);
        }), _0x482d2f['exists'] = function (_0x3dbf7e, _0x3a73a6) {
            if (typeof _0x3a73a6 === 'function')
                return _0x53c3e5['exists'](_0x3dbf7e, _0x3a73a6);
            return new Promise(_0x3862f0 => {
                return _0x53c3e5['exists'](_0x3dbf7e, _0x3862f0);
            });
        }, _0x482d2f['read'] = function (_0x561cb3, _0x2bf9c8, _0x2ad0d6, _0x5d0c65, _0x1f4251, _0x35f496) {
            if (typeof _0x35f496 === 'function')
                return _0x53c3e5['read'](_0x561cb3, _0x2bf9c8, _0x2ad0d6, _0x5d0c65, _0x1f4251, _0x35f496);
            return new Promise((_0x295379, _0x3dcb4f) => {
                _0x53c3e5['read'](_0x561cb3, _0x2bf9c8, _0x2ad0d6, _0x5d0c65, _0x1f4251, (_0x2f1369, _0x1b4206, _0x5df1d2) => {
                    if (_0x2f1369)
                        return _0x3dcb4f(_0x2f1369);
                    _0x295379({
                        'bytesRead': _0x1b4206,
                        'buffer': _0x5df1d2
                    });
                });
            });
        }, _0x482d2f['write'] = function (_0x4d4df7, _0x5962d3, ..._0x31027f) {
            if (typeof _0x31027f[_0x31027f['length'] - 0x1] === 'function')
                return _0x53c3e5['write'](_0x4d4df7, _0x5962d3, ..._0x31027f);
            return new Promise((_0x2dedb1, _0x5460d2) => {
                _0x53c3e5['write'](_0x4d4df7, _0x5962d3, ..._0x31027f, (_0x41073c, _0x6010e2, _0x428943) => {
                    if (_0x41073c)
                        return _0x5460d2(_0x41073c);
                    _0x2dedb1({
                        'bytesWritten': _0x6010e2,
                        'buffer': _0x428943
                    });
                });
            });
        }, typeof _0x53c3e5['writev'] === 'function' && (_0x482d2f['writev'] = function (_0x4a7676, _0x102af6, ..._0x44a856) {
            if (typeof _0x44a856[_0x44a856['length'] - 0x1] === 'function')
                return _0x53c3e5['writev'](_0x4a7676, _0x102af6, ..._0x44a856);
            return new Promise((_0xbc8f2, _0x23b7df) => {
                _0x53c3e5['writev'](_0x4a7676, _0x102af6, ..._0x44a856, (_0x4a1b7a, _0x16833d, _0x453bc4) => {
                    if (_0x4a1b7a)
                        return _0x23b7df(_0x4a1b7a);
                    _0xbc8f2({
                        'bytesWritten': _0x16833d,
                        'buffers': _0x453bc4
                    });
                });
            });
        }), typeof _0x53c3e5['realpath']['native'] === 'function' ? _0x482d2f['realpath']['native'] = _0x209fd0(_0x53c3e5['realpath']['native']) : process['emitWarning']('fs.realpath.native\x20is\x20not\x20a\x20function.\x20Is\x20fs\x20being\x20monkey-patched?', 'Warning', 'fs-extra-WARN0003');
    },
    0x17fbe: (_0x28e7d7, _0x261bfe, _0x79c2fd) => {
        'use strict';
        _0x28e7d7['exports'] = {
            ..._0x79c2fd(0x15a17),
            ..._0x79c2fd(0xb839),
            ..._0x79c2fd(0x112ad),
            ..._0x79c2fd(0x143a2),
            ..._0x79c2fd(0x14bf2),
            ..._0x79c2fd(0x10c1d),
            ..._0x79c2fd(0x1462d),
            ..._0x79c2fd(0x7096),
            ..._0x79c2fd(0x15330),
            ..._0x79c2fd(0x983d)
        };
    },
    0x14bf2: (_0x351fa6, _0x11f2f2, _0x98aa27) => {
        'use strict';
        const _0x5d33fc = _0x98aa27(0x15f6a)['fromPromise'], _0x590e6e = _0x98aa27(0xd091);
        _0x590e6e['outputJson'] = _0x5d33fc(_0x98aa27(0x10fbe)), _0x590e6e['outputJsonSync'] = _0x98aa27(0x43a8), _0x590e6e['outputJSON'] = _0x590e6e['outputJson'], _0x590e6e['outputJSONSync'] = _0x590e6e['outputJsonSync'], _0x590e6e['writeJSON'] = _0x590e6e['writeJson'], _0x590e6e['writeJSONSync'] = _0x590e6e['writeJsonSync'], _0x590e6e['readJSON'] = _0x590e6e['readJson'], _0x590e6e['readJSONSync'] = _0x590e6e['readJsonSync'], _0x351fa6['exports'] = _0x590e6e;
    },
    0xd091: (_0x30e2ae, _0x22f926, _0x672dc4) => {
        'use strict';
        const _0x50ee5e = _0x672dc4(0x8517);
        _0x30e2ae['exports'] = {
            'readJson': _0x50ee5e['readFile'],
            'readJsonSync': _0x50ee5e['readFileSync'],
            'writeJson': _0x50ee5e['writeFile'],
            'writeJsonSync': _0x50ee5e['writeFileSync']
        };
    },
    0x43a8: (_0x500691, _0x1591d7, _0x53361c) => {
        'use strict';
        const {stringify: _0x2f2713} = _0x53361c(0x14415), {outputFileSync: _0x4c16d8} = _0x53361c(0x7096);
        function _0x3c5feb(_0x4a43fc, _0x28cb39, _0x354ee7) {
            const _0x296119 = _0x2f2713(_0x28cb39, _0x354ee7);
            _0x4c16d8(_0x4a43fc, _0x296119, _0x354ee7);
        }
        _0x500691['exports'] = _0x3c5feb;
    },
    0x10fbe: (_0x28d7dd, _0x54f391, _0x236f8e) => {
        'use strict';
        const {stringify: _0x378671} = _0x236f8e(0x14415), {outputFile: _0x4b41c7} = _0x236f8e(0x7096);
        async function _0x43d2a3(_0x193574, _0x506172, _0x2dd731 = {}) {
            const _0x183a28 = _0x378671(_0x506172, _0x2dd731);
            await _0x4b41c7(_0x193574, _0x183a28, _0x2dd731);
        }
        _0x28d7dd['exports'] = _0x43d2a3;
    },
    0x10c1d: (_0x8e65b1, _0x5efe4a, _0x108dc0) => {
        'use strict';
        const _0x5d1daa = _0x108dc0(0x15f6a)['fromPromise'], {
                makeDir: _0x4c1cc4,
                makeDirSync: _0x417c4f
            } = _0x108dc0(0x1010), _0x3cf4d5 = _0x5d1daa(_0x4c1cc4);
        _0x8e65b1['exports'] = {
            'mkdirs': _0x3cf4d5,
            'mkdirsSync': _0x417c4f,
            'mkdirp': _0x3cf4d5,
            'mkdirpSync': _0x417c4f,
            'ensureDir': _0x3cf4d5,
            'ensureDirSync': _0x417c4f
        };
    },
    0x1010: (_0x1990c5, _0x240c3f, _0x2117f2) => {
        'use strict';
        const _0x5513b3 = _0x2117f2(0x15a17), {checkPath: _0xbd1d13} = _0x2117f2(0x4517), _0x3645e6 = _0x2bd665 => {
                const _0x15b1d4 = { 'mode': 0x1ff };
                if (typeof _0x2bd665 === 'number')
                    return _0x2bd665;
                return {
                    ..._0x15b1d4,
                    ..._0x2bd665
                }['mode'];
            };
        _0x1990c5['exports']['makeDir'] = async (_0xfa2ae2, _0x838a66) => {
            return _0xbd1d13(_0xfa2ae2), _0x5513b3['mkdir'](_0xfa2ae2, {
                'mode': _0x3645e6(_0x838a66),
                'recursive': !![]
            });
        }, _0x1990c5['exports']['makeDirSync'] = (_0x332e74, _0x48d1f5) => {
            return _0xbd1d13(_0x332e74), _0x5513b3['mkdirSync'](_0x332e74, {
                'mode': _0x3645e6(_0x48d1f5),
                'recursive': !![]
            });
        };
    },
    0x4517: (_0x2bf4e7, _0x4e60b4, _0x5646f2) => {
        'use strict';
        const _0x210036 = _0x5646f2(0x11569);
        _0x2bf4e7['exports']['checkPath'] = function _0x3259b9(_0x4f49b3) {
            if (process['platform'] === 'win32') {
                const _0x17606d = /[<>:"|?*]/['test'](_0x4f49b3['replace'](_0x210036['parse'](_0x4f49b3)['root'], ''));
                if (_0x17606d) {
                    const _0x1b4486 = new Error('Path\x20contains\x20invalid\x20characters:\x20' + _0x4f49b3);
                    _0x1b4486['code'] = 'EINVAL';
                    throw _0x1b4486;
                }
            }
        };
    },
    0x1462d: (_0x7d3b5d, _0x2a104d, _0x1db2f5) => {
        'use strict';
        const _0x57fc4f = _0x1db2f5(0x15f6a)['fromCallback'];
        _0x7d3b5d['exports'] = {
            'move': _0x57fc4f(_0x1db2f5(0xda7d)),
            'moveSync': _0x1db2f5(0xea9b)
        };
    },
    0xea9b: (_0x51a943, _0xeae802, _0x43ee58) => {
        'use strict';
        const _0x43f9dc = _0x43ee58(0x15d1a), _0x22791d = _0x43ee58(0x11569), _0x43d1fc = _0x43ee58(0xb839)['copySync'], _0x16a510 = _0x43ee58(0x983d)['removeSync'], _0x3c601a = _0x43ee58(0x10c1d)['mkdirpSync'], _0x5299a8 = _0x43ee58(0x7c9d);
        function _0x2caf34(_0x135f99, _0x154b0d, _0x2a424b) {
            _0x2a424b = _0x2a424b || {};
            const _0x38593c = _0x2a424b['overwrite'] || _0x2a424b['clobber'] || ![], {
                    srcStat: _0x52f41a,
                    isChangingCase: isChangingCase = ![]
                } = _0x5299a8['checkPathsSync'](_0x135f99, _0x154b0d, 'move', _0x2a424b);
            _0x5299a8['checkParentPathsSync'](_0x135f99, _0x52f41a, _0x154b0d, 'move');
            if (!_0x4acab9(_0x154b0d))
                _0x3c601a(_0x22791d['dirname'](_0x154b0d));
            return _0x28643c(_0x135f99, _0x154b0d, _0x38593c, isChangingCase);
        }
        function _0x4acab9(_0x458ad1) {
            const _0xad1f92 = _0x22791d['dirname'](_0x458ad1), _0x4ecffa = _0x22791d['parse'](_0xad1f92);
            return _0x4ecffa['root'] === _0xad1f92;
        }
        function _0x28643c(_0x43ff9e, _0x45cf6e, _0x2ed6d2, _0x2ac11a) {
            if (_0x2ac11a)
                return _0x5a0915(_0x43ff9e, _0x45cf6e, _0x2ed6d2);
            if (_0x2ed6d2)
                return _0x16a510(_0x45cf6e), _0x5a0915(_0x43ff9e, _0x45cf6e, _0x2ed6d2);
            if (_0x43f9dc['existsSync'](_0x45cf6e))
                throw new Error('dest\x20already\x20exists.');
            return _0x5a0915(_0x43ff9e, _0x45cf6e, _0x2ed6d2);
        }
        function _0x5a0915(_0x4d4c70, _0x230c07, _0x454be8) {
            try {
                _0x43f9dc['renameSync'](_0x4d4c70, _0x230c07);
            } catch (_0x180915) {
                if (_0x180915['code'] !== 'EXDEV')
                    throw _0x180915;
                return _0x13f8bc(_0x4d4c70, _0x230c07, _0x454be8);
            }
        }
        function _0x13f8bc(_0x2a8fdd, _0x20b1b2, _0x177574) {
            const _0x16c595 = {
                'overwrite': _0x177574,
                'errorOnExist': !![]
            };
            return _0x43d1fc(_0x2a8fdd, _0x20b1b2, _0x16c595), _0x16a510(_0x2a8fdd);
        }
        _0x51a943['exports'] = _0x2caf34;
    },
    0xda7d: (_0x46e58d, _0x3fbad1, _0x5d0b0c) => {
        'use strict';
        const _0x1ec58b = _0x5d0b0c(0x15d1a), _0xff636e = _0x5d0b0c(0x11569), _0x45a367 = _0x5d0b0c(0xb839)['copy'], _0x107dc9 = _0x5d0b0c(0x983d)['remove'], _0x597cc8 = _0x5d0b0c(0x10c1d)['mkdirp'], _0x3bfe8f = _0x5d0b0c(0x15330)['pathExists'], _0x2f2013 = _0x5d0b0c(0x7c9d);
        function _0x30e6c4(_0x564284, _0x218466, _0x220d4c, _0x2174bf) {
            typeof _0x220d4c === 'function' && (_0x2174bf = _0x220d4c, _0x220d4c = {});
            _0x220d4c = _0x220d4c || {};
            const _0x224e2f = _0x220d4c['overwrite'] || _0x220d4c['clobber'] || ![];
            _0x2f2013['checkPaths'](_0x564284, _0x218466, 'move', _0x220d4c, (_0x426945, _0x437dd4) => {
                if (_0x426945)
                    return _0x2174bf(_0x426945);
                const {
                    srcStat: _0x133b3e,
                    isChangingCase: isChangingCase = ![]
                } = _0x437dd4;
                _0x2f2013['checkParentPaths'](_0x564284, _0x133b3e, _0x218466, 'move', _0x8230f8 => {
                    if (_0x8230f8)
                        return _0x2174bf(_0x8230f8);
                    if (_0x588e14(_0x218466))
                        return _0x38bd4b(_0x564284, _0x218466, _0x224e2f, isChangingCase, _0x2174bf);
                    _0x597cc8(_0xff636e['dirname'](_0x218466), _0x45582a => {
                        if (_0x45582a)
                            return _0x2174bf(_0x45582a);
                        return _0x38bd4b(_0x564284, _0x218466, _0x224e2f, isChangingCase, _0x2174bf);
                    });
                });
            });
        }
        function _0x588e14(_0x2f5efa) {
            const _0x8bb0f2 = _0xff636e['dirname'](_0x2f5efa), _0x509920 = _0xff636e['parse'](_0x8bb0f2);
            return _0x509920['root'] === _0x8bb0f2;
        }
        function _0x38bd4b(_0x9aef60, _0x535ae2, _0x52acf4, _0xbbadea, _0x591a73) {
            if (_0xbbadea)
                return _0x5a9acd(_0x9aef60, _0x535ae2, _0x52acf4, _0x591a73);
            if (_0x52acf4)
                return _0x107dc9(_0x535ae2, _0x450446 => {
                    if (_0x450446)
                        return _0x591a73(_0x450446);
                    return _0x5a9acd(_0x9aef60, _0x535ae2, _0x52acf4, _0x591a73);
                });
            _0x3bfe8f(_0x535ae2, (_0x1a81a0, _0x1dc9d3) => {
                if (_0x1a81a0)
                    return _0x591a73(_0x1a81a0);
                if (_0x1dc9d3)
                    return _0x591a73(new Error('dest\x20already\x20exists.'));
                return _0x5a9acd(_0x9aef60, _0x535ae2, _0x52acf4, _0x591a73);
            });
        }
        function _0x5a9acd(_0x14e43a, _0x2f3864, _0x28e53a, _0x263a10) {
            _0x1ec58b['rename'](_0x14e43a, _0x2f3864, _0x8d768d => {
                if (!_0x8d768d)
                    return _0x263a10();
                if (_0x8d768d['code'] !== 'EXDEV')
                    return _0x263a10(_0x8d768d);
                return _0x256c81(_0x14e43a, _0x2f3864, _0x28e53a, _0x263a10);
            });
        }
        function _0x256c81(_0xca2b95, _0x5b1df2, _0x3db32e, _0x195654) {
            const _0x3969a8 = {
                'overwrite': _0x3db32e,
                'errorOnExist': !![]
            };
            _0x45a367(_0xca2b95, _0x5b1df2, _0x3969a8, _0x50c629 => {
                if (_0x50c629)
                    return _0x195654(_0x50c629);
                return _0x107dc9(_0xca2b95, _0x195654);
            });
        }
        _0x46e58d['exports'] = _0x30e6c4;
    },
    0x7096: (_0x53eec0, _0x20ae13, _0x5d86c7) => {
        'use strict';
        const _0x2c01b7 = _0x5d86c7(0x15f6a)['fromCallback'], _0x390500 = _0x5d86c7(0x15d1a), _0x5c3360 = _0x5d86c7(0x11569), _0x4ce745 = _0x5d86c7(0x10c1d), _0xe7c73d = _0x5d86c7(0x15330)['pathExists'];
        function _0x1ebda2(_0x4956f3, _0x3e266f, _0x4b1800, _0x359efc) {
            typeof _0x4b1800 === 'function' && (_0x359efc = _0x4b1800, _0x4b1800 = 'utf8');
            const _0x1215d2 = _0x5c3360['dirname'](_0x4956f3);
            _0xe7c73d(_0x1215d2, (_0x12429f, _0x34cce4) => {
                if (_0x12429f)
                    return _0x359efc(_0x12429f);
                if (_0x34cce4)
                    return _0x390500['writeFile'](_0x4956f3, _0x3e266f, _0x4b1800, _0x359efc);
                _0x4ce745['mkdirs'](_0x1215d2, _0xa036e6 => {
                    if (_0xa036e6)
                        return _0x359efc(_0xa036e6);
                    _0x390500['writeFile'](_0x4956f3, _0x3e266f, _0x4b1800, _0x359efc);
                });
            });
        }
        function _0xd44a55(_0x527e53, ..._0x25118a) {
            const _0x421b26 = _0x5c3360['dirname'](_0x527e53);
            if (_0x390500['existsSync'](_0x421b26))
                return _0x390500['writeFileSync'](_0x527e53, ..._0x25118a);
            _0x4ce745['mkdirsSync'](_0x421b26), _0x390500['writeFileSync'](_0x527e53, ..._0x25118a);
        }
        _0x53eec0['exports'] = {
            'outputFile': _0x2c01b7(_0x1ebda2),
            'outputFileSync': _0xd44a55
        };
    },
    0x15330: (_0x394ea3, _0x346ad9, _0x2faae5) => {
        'use strict';
        const _0x4504ef = _0x2faae5(0x15f6a)['fromPromise'], _0x5a5df5 = _0x2faae5(0x15a17);
        function _0x4a0470(_0x456cee) {
            return _0x5a5df5['access'](_0x456cee)['then'](() => !![])['catch'](() => ![]);
        }
        _0x394ea3['exports'] = {
            'pathExists': _0x4504ef(_0x4a0470),
            'pathExistsSync': _0x5a5df5['existsSync']
        };
    },
    0x983d: (_0x3416ea, _0x441356, _0x2185fa) => {
        'use strict';
        const _0x11e39e = _0x2185fa(0x15d1a), _0x427d61 = _0x2185fa(0x15f6a)['fromCallback'], _0x25b047 = _0x2185fa(0x61bd);
        function _0x20cc48(_0x236e47, _0x42f90e) {
            if (_0x11e39e['rm'])
                return _0x11e39e['rm'](_0x236e47, {
                    'recursive': !![],
                    'force': !![]
                }, _0x42f90e);
            _0x25b047(_0x236e47, _0x42f90e);
        }
        function _0x5878d1(_0x59a785) {
            if (_0x11e39e['rmSync'])
                return _0x11e39e['rmSync'](_0x59a785, {
                    'recursive': !![],
                    'force': !![]
                });
            _0x25b047['sync'](_0x59a785);
        }
        _0x3416ea['exports'] = {
            'remove': _0x427d61(_0x20cc48),
            'removeSync': _0x5878d1
        };
    },
    0x61bd: (_0x420925, _0x6d5567, _0x333aad) => {
        'use strict';
        const _0x25f26b = _0x333aad(0x15d1a), _0x17f530 = _0x333aad(0x11569), _0x4c2c42 = _0x333aad(0x9a43), _0x20a05c = process['platform'] === 'win32';
        function _0x1bcc29(_0x1e8da6) {
            const _0x3258f9 = [
                'unlink',
                'chmod',
                'stat',
                'lstat',
                'rmdir',
                'readdir'
            ];
            _0x3258f9['forEach'](_0x4ce20d => {
                _0x1e8da6[_0x4ce20d] = _0x1e8da6[_0x4ce20d] || _0x25f26b[_0x4ce20d], _0x4ce20d = _0x4ce20d + 'Sync', _0x1e8da6[_0x4ce20d] = _0x1e8da6[_0x4ce20d] || _0x25f26b[_0x4ce20d];
            }), _0x1e8da6['maxBusyTries'] = _0x1e8da6['maxBusyTries'] || 0x3;
        }
        function _0x380439(_0x51003d, _0x5dc8c7, _0x3f9e2e) {
            let _0x141ddc = 0x0;
            typeof _0x5dc8c7 === 'function' && (_0x3f9e2e = _0x5dc8c7, _0x5dc8c7 = {}), _0x4c2c42(_0x51003d, 'rimraf:\x20missing\x20path'), _0x4c2c42['strictEqual'](typeof _0x51003d, 'string', 'rimraf:\x20path\x20should\x20be\x20a\x20string'), _0x4c2c42['strictEqual'](typeof _0x3f9e2e, 'function', 'rimraf:\x20callback\x20function\x20required'), _0x4c2c42(_0x5dc8c7, 'rimraf:\x20invalid\x20options\x20argument\x20provided'), _0x4c2c42['strictEqual'](typeof _0x5dc8c7, 'object', 'rimraf:\x20options\x20should\x20be\x20object'), _0x1bcc29(_0x5dc8c7), _0x1ba778(_0x51003d, _0x5dc8c7, function _0x40207f(_0x10dab2) {
                if (_0x10dab2) {
                    if ((_0x10dab2['code'] === 'EBUSY' || _0x10dab2['code'] === 'ENOTEMPTY' || _0x10dab2['code'] === 'EPERM') && _0x141ddc < _0x5dc8c7['maxBusyTries']) {
                        _0x141ddc++;
                        const _0x34362b = _0x141ddc * 0x64;
                        return setTimeout(() => _0x1ba778(_0x51003d, _0x5dc8c7, _0x40207f), _0x34362b);
                    }
                    if (_0x10dab2['code'] === 'ENOENT')
                        _0x10dab2 = null;
                }
                _0x3f9e2e(_0x10dab2);
            });
        }
        function _0x1ba778(_0x5b006c, _0xdfdc4e, _0x1576be) {
            _0x4c2c42(_0x5b006c), _0x4c2c42(_0xdfdc4e), _0x4c2c42(typeof _0x1576be === 'function'), _0xdfdc4e['lstat'](_0x5b006c, (_0x545a04, _0x4a0dee) => {
                if (_0x545a04 && _0x545a04['code'] === 'ENOENT')
                    return _0x1576be(null);
                if (_0x545a04 && _0x545a04['code'] === 'EPERM' && _0x20a05c)
                    return _0x2e896a(_0x5b006c, _0xdfdc4e, _0x545a04, _0x1576be);
                if (_0x4a0dee && _0x4a0dee['isDirectory']())
                    return _0x3501ff(_0x5b006c, _0xdfdc4e, _0x545a04, _0x1576be);
                _0xdfdc4e['unlink'](_0x5b006c, _0xa6644c => {
                    if (_0xa6644c) {
                        if (_0xa6644c['code'] === 'ENOENT')
                            return _0x1576be(null);
                        if (_0xa6644c['code'] === 'EPERM')
                            return _0x20a05c ? _0x2e896a(_0x5b006c, _0xdfdc4e, _0xa6644c, _0x1576be) : _0x3501ff(_0x5b006c, _0xdfdc4e, _0xa6644c, _0x1576be);
                        if (_0xa6644c['code'] === 'EISDIR')
                            return _0x3501ff(_0x5b006c, _0xdfdc4e, _0xa6644c, _0x1576be);
                    }
                    return _0x1576be(_0xa6644c);
                });
            });
        }
        function _0x2e896a(_0x50f1f0, _0x444d72, _0x195cb9, _0x94c199) {
            _0x4c2c42(_0x50f1f0), _0x4c2c42(_0x444d72), _0x4c2c42(typeof _0x94c199 === 'function'), _0x444d72['chmod'](_0x50f1f0, 0x1b6, _0x14f3c5 => {
                _0x14f3c5 ? _0x94c199(_0x14f3c5['code'] === 'ENOENT' ? null : _0x195cb9) : _0x444d72['stat'](_0x50f1f0, (_0x5f33c9, _0x29d030) => {
                    if (_0x5f33c9)
                        _0x94c199(_0x5f33c9['code'] === 'ENOENT' ? null : _0x195cb9);
                    else
                        _0x29d030['isDirectory']() ? _0x3501ff(_0x50f1f0, _0x444d72, _0x195cb9, _0x94c199) : _0x444d72['unlink'](_0x50f1f0, _0x94c199);
                });
            });
        }
        function _0x2516af(_0x54b361, _0x1e41b1, _0x50b51d) {
            let _0x15a952;
            _0x4c2c42(_0x54b361), _0x4c2c42(_0x1e41b1);
            try {
                _0x1e41b1['chmodSync'](_0x54b361, 0x1b6);
            } catch (_0x452e9d) {
                if (_0x452e9d['code'] === 'ENOENT')
                    return;
                else
                    throw _0x50b51d;
            }
            try {
                _0x15a952 = _0x1e41b1['statSync'](_0x54b361);
            } catch (_0x4254de) {
                if (_0x4254de['code'] === 'ENOENT')
                    return;
                else
                    throw _0x50b51d;
            }
            _0x15a952['isDirectory']() ? _0x1ef03c(_0x54b361, _0x1e41b1, _0x50b51d) : _0x1e41b1['unlinkSync'](_0x54b361);
        }
        function _0x3501ff(_0x3cef55, _0x513224, _0x195cc5, _0x125137) {
            _0x4c2c42(_0x3cef55), _0x4c2c42(_0x513224), _0x4c2c42(typeof _0x125137 === 'function'), _0x513224['rmdir'](_0x3cef55, _0x3619ff => {
                if (_0x3619ff && (_0x3619ff['code'] === 'ENOTEMPTY' || _0x3619ff['code'] === 'EEXIST' || _0x3619ff['code'] === 'EPERM'))
                    _0x5261bb(_0x3cef55, _0x513224, _0x125137);
                else
                    _0x3619ff && _0x3619ff['code'] === 'ENOTDIR' ? _0x125137(_0x195cc5) : _0x125137(_0x3619ff);
            });
        }
        function _0x5261bb(_0x564809, _0x185a84, _0x1e2e4e) {
            _0x4c2c42(_0x564809), _0x4c2c42(_0x185a84), _0x4c2c42(typeof _0x1e2e4e === 'function'), _0x185a84['readdir'](_0x564809, (_0x303478, _0xcb683d) => {
                if (_0x303478)
                    return _0x1e2e4e(_0x303478);
                let _0x4360b0 = _0xcb683d['length'], _0x1ade8d;
                if (_0x4360b0 === 0x0)
                    return _0x185a84['rmdir'](_0x564809, _0x1e2e4e);
                _0xcb683d['forEach'](_0x543aaa => {
                    _0x380439(_0x17f530['join'](_0x564809, _0x543aaa), _0x185a84, _0x363072 => {
                        if (_0x1ade8d)
                            return;
                        if (_0x363072)
                            return _0x1e2e4e(_0x1ade8d = _0x363072);
                        --_0x4360b0 === 0x0 && _0x185a84['rmdir'](_0x564809, _0x1e2e4e);
                    });
                });
            });
        }
        function _0x5c535a(_0x1fe429, _0x50e430) {
            let _0x338453;
            _0x50e430 = _0x50e430 || {}, _0x1bcc29(_0x50e430), _0x4c2c42(_0x1fe429, 'rimraf:\x20missing\x20path'), _0x4c2c42['strictEqual'](typeof _0x1fe429, 'string', 'rimraf:\x20path\x20should\x20be\x20a\x20string'), _0x4c2c42(_0x50e430, 'rimraf:\x20missing\x20options'), _0x4c2c42['strictEqual'](typeof _0x50e430, 'object', 'rimraf:\x20options\x20should\x20be\x20object');
            try {
                _0x338453 = _0x50e430['lstatSync'](_0x1fe429);
            } catch (_0x56d6a0) {
                if (_0x56d6a0['code'] === 'ENOENT')
                    return;
                _0x56d6a0['code'] === 'EPERM' && _0x20a05c && _0x2516af(_0x1fe429, _0x50e430, _0x56d6a0);
            }
            try {
                _0x338453 && _0x338453['isDirectory']() ? _0x1ef03c(_0x1fe429, _0x50e430, null) : _0x50e430['unlinkSync'](_0x1fe429);
            } catch (_0x5d0859) {
                if (_0x5d0859['code'] === 'ENOENT')
                    return;
                else {
                    if (_0x5d0859['code'] === 'EPERM')
                        return _0x20a05c ? _0x2516af(_0x1fe429, _0x50e430, _0x5d0859) : _0x1ef03c(_0x1fe429, _0x50e430, _0x5d0859);
                    else {
                        if (_0x5d0859['code'] !== 'EISDIR')
                            throw _0x5d0859;
                    }
                }
                _0x1ef03c(_0x1fe429, _0x50e430, _0x5d0859);
            }
        }
        function _0x1ef03c(_0x2816b0, _0x3989b6, _0x1464cd) {
            _0x4c2c42(_0x2816b0), _0x4c2c42(_0x3989b6);
            try {
                _0x3989b6['rmdirSync'](_0x2816b0);
            } catch (_0x561c6e) {
                if (_0x561c6e['code'] === 'ENOTDIR')
                    throw _0x1464cd;
                else {
                    if (_0x561c6e['code'] === 'ENOTEMPTY' || _0x561c6e['code'] === 'EEXIST' || _0x561c6e['code'] === 'EPERM')
                        _0x33dc20(_0x2816b0, _0x3989b6);
                    else {
                        if (_0x561c6e['code'] !== 'ENOENT')
                            throw _0x561c6e;
                    }
                }
            }
        }
        function _0x33dc20(_0x24b155, _0x3f2e66) {
            _0x4c2c42(_0x24b155), _0x4c2c42(_0x3f2e66), _0x3f2e66['readdirSync'](_0x24b155)['forEach'](_0x4ca323 => _0x5c535a(_0x17f530['join'](_0x24b155, _0x4ca323), _0x3f2e66));
            if (_0x20a05c) {
                const _0x1c0707 = Date['now']();
                do {
                    try {
                        const _0x5da599 = _0x3f2e66['rmdirSync'](_0x24b155, _0x3f2e66);
                        return _0x5da599;
                    } catch {
                    }
                } while (Date['now']() - _0x1c0707 < 0x1f4);
            } else {
                const _0x24b08b = _0x3f2e66['rmdirSync'](_0x24b155, _0x3f2e66);
                return _0x24b08b;
            }
        }
        _0x420925['exports'] = _0x380439, _0x380439['sync'] = _0x5c535a;
    },
    0x7c9d: (_0x2c356d, _0x19135f, _0x3f5380) => {
        'use strict';
        const _0x59f86b = _0x3f5380(0x15a17), _0x5346c5 = _0x3f5380(0x11569), _0x593af6 = _0x3f5380(0x1206d);
        function _0x28adb2(_0x4892a2, _0x39942b, _0x495932) {
            const _0x36d147 = _0x495932['dereference'] ? _0x3a4738 => _0x59f86b['stat'](_0x3a4738, { 'bigint': !![] }) : _0x132419 => _0x59f86b['lstat'](_0x132419, { 'bigint': !![] });
            return Promise['all']([
                _0x36d147(_0x4892a2),
                _0x36d147(_0x39942b)['catch'](_0x59b540 => {
                    if (_0x59b540['code'] === 'ENOENT')
                        return null;
                    throw _0x59b540;
                })
            ])['then'](([_0x5d05b6, _0x50ae04]) => ({
                'srcStat': _0x5d05b6,
                'destStat': _0x50ae04
            }));
        }
        function _0x463d09(_0x12d5e7, _0x229edd, _0x312047) {
            let _0xeb964b;
            const _0x3d82b8 = _0x312047['dereference'] ? _0x294db0 => _0x59f86b['statSync'](_0x294db0, { 'bigint': !![] }) : _0x1ed1ca => _0x59f86b['lstatSync'](_0x1ed1ca, { 'bigint': !![] }), _0x22d6cf = _0x3d82b8(_0x12d5e7);
            try {
                _0xeb964b = _0x3d82b8(_0x229edd);
            } catch (_0x22ccb5) {
                if (_0x22ccb5['code'] === 'ENOENT')
                    return {
                        'srcStat': _0x22d6cf,
                        'destStat': null
                    };
                throw _0x22ccb5;
            }
            return {
                'srcStat': _0x22d6cf,
                'destStat': _0xeb964b
            };
        }
        function _0x4d4be0(_0x55ec1a, _0x42324f, _0x36b3fb, _0x4b4778, _0x43b3b0) {
            _0x593af6['callbackify'](_0x28adb2)(_0x55ec1a, _0x42324f, _0x4b4778, (_0x42dfa6, _0x36a839) => {
                if (_0x42dfa6)
                    return _0x43b3b0(_0x42dfa6);
                const {
                    srcStat: _0x28bd61,
                    destStat: _0x29cd53
                } = _0x36a839;
                if (_0x29cd53) {
                    if (_0x13d9f6(_0x28bd61, _0x29cd53)) {
                        const _0x464c6a = _0x5346c5['basename'](_0x55ec1a), _0xf600b3 = _0x5346c5['basename'](_0x42324f);
                        if (_0x36b3fb === 'move' && _0x464c6a !== _0xf600b3 && _0x464c6a['toLowerCase']() === _0xf600b3['toLowerCase']())
                            return _0x43b3b0(null, {
                                'srcStat': _0x28bd61,
                                'destStat': _0x29cd53,
                                'isChangingCase': !![]
                            });
                        return _0x43b3b0(new Error('Source\x20and\x20destination\x20must\x20not\x20be\x20the\x20same.'));
                    }
                    if (_0x28bd61['isDirectory']() && !_0x29cd53['isDirectory']())
                        return _0x43b3b0(new Error('Cannot\x20overwrite\x20non-directory\x20\x27' + _0x42324f + '\x27\x20with\x20directory\x20\x27' + _0x55ec1a + '\x27.'));
                    if (!_0x28bd61['isDirectory']() && _0x29cd53['isDirectory']())
                        return _0x43b3b0(new Error('Cannot\x20overwrite\x20directory\x20\x27' + _0x42324f + '\x27\x20with\x20non-directory\x20\x27' + _0x55ec1a + '\x27.'));
                }
                if (_0x28bd61['isDirectory']() && _0x595892(_0x55ec1a, _0x42324f))
                    return _0x43b3b0(new Error(_0x574022(_0x55ec1a, _0x42324f, _0x36b3fb)));
                return _0x43b3b0(null, {
                    'srcStat': _0x28bd61,
                    'destStat': _0x29cd53
                });
            });
        }
        function _0x68b59a(_0x45ac20, _0x129693, _0xb5ab5a, _0xd30b6a) {
            const {
                srcStat: _0x29e010,
                destStat: _0x1d875e
            } = _0x463d09(_0x45ac20, _0x129693, _0xd30b6a);
            if (_0x1d875e) {
                if (_0x13d9f6(_0x29e010, _0x1d875e)) {
                    const _0x263cc5 = _0x5346c5['basename'](_0x45ac20), _0x84dd96 = _0x5346c5['basename'](_0x129693);
                    if (_0xb5ab5a === 'move' && _0x263cc5 !== _0x84dd96 && _0x263cc5['toLowerCase']() === _0x84dd96['toLowerCase']())
                        return {
                            'srcStat': _0x29e010,
                            'destStat': _0x1d875e,
                            'isChangingCase': !![]
                        };
                    throw new Error('Source\x20and\x20destination\x20must\x20not\x20be\x20the\x20same.');
                }
                if (_0x29e010['isDirectory']() && !_0x1d875e['isDirectory']())
                    throw new Error('Cannot\x20overwrite\x20non-directory\x20\x27' + _0x129693 + '\x27\x20with\x20directory\x20\x27' + _0x45ac20 + '\x27.');
                if (!_0x29e010['isDirectory']() && _0x1d875e['isDirectory']())
                    throw new Error('Cannot\x20overwrite\x20directory\x20\x27' + _0x129693 + '\x27\x20with\x20non-directory\x20\x27' + _0x45ac20 + '\x27.');
            }
            if (_0x29e010['isDirectory']() && _0x595892(_0x45ac20, _0x129693))
                throw new Error(_0x574022(_0x45ac20, _0x129693, _0xb5ab5a));
            return {
                'srcStat': _0x29e010,
                'destStat': _0x1d875e
            };
        }
        function _0x27e3da(_0x5e7f62, _0x58ea01, _0x2923c0, _0x118d74, _0x28ccef) {
            const _0x5650b2 = _0x5346c5['resolve'](_0x5346c5['dirname'](_0x5e7f62)), _0x1f8c56 = _0x5346c5['resolve'](_0x5346c5['dirname'](_0x2923c0));
            if (_0x1f8c56 === _0x5650b2 || _0x1f8c56 === _0x5346c5['parse'](_0x1f8c56)['root'])
                return _0x28ccef();
            _0x59f86b['stat'](_0x1f8c56, { 'bigint': !![] }, (_0x139850, _0x2a0a08) => {
                if (_0x139850) {
                    if (_0x139850['code'] === 'ENOENT')
                        return _0x28ccef();
                    return _0x28ccef(_0x139850);
                }
                if (_0x13d9f6(_0x58ea01, _0x2a0a08))
                    return _0x28ccef(new Error(_0x574022(_0x5e7f62, _0x2923c0, _0x118d74)));
                return _0x27e3da(_0x5e7f62, _0x58ea01, _0x1f8c56, _0x118d74, _0x28ccef);
            });
        }
        function _0x42ce0d(_0x145492, _0x529fcd, _0x34ee41, _0x44bd22) {
            const _0x4aa4f3 = _0x5346c5['resolve'](_0x5346c5['dirname'](_0x145492)), _0x445995 = _0x5346c5['resolve'](_0x5346c5['dirname'](_0x34ee41));
            if (_0x445995 === _0x4aa4f3 || _0x445995 === _0x5346c5['parse'](_0x445995)['root'])
                return;
            let _0x4e289c;
            try {
                _0x4e289c = _0x59f86b['statSync'](_0x445995, { 'bigint': !![] });
            } catch (_0x1852e2) {
                if (_0x1852e2['code'] === 'ENOENT')
                    return;
                throw _0x1852e2;
            }
            if (_0x13d9f6(_0x529fcd, _0x4e289c))
                throw new Error(_0x574022(_0x145492, _0x34ee41, _0x44bd22));
            return _0x42ce0d(_0x145492, _0x529fcd, _0x445995, _0x44bd22);
        }
        function _0x13d9f6(_0x3ee50f, _0x4b2afc) {
            return _0x4b2afc['ino'] && _0x4b2afc['dev'] && _0x4b2afc['ino'] === _0x3ee50f['ino'] && _0x4b2afc['dev'] === _0x3ee50f['dev'];
        }
        function _0x595892(_0x3fa02c, _0x5b4aa4) {
            const _0x29ffd1 = _0x5346c5['resolve'](_0x3fa02c)['split'](_0x5346c5['sep'])['filter'](_0xcfb4a9 => _0xcfb4a9), _0x447225 = _0x5346c5['resolve'](_0x5b4aa4)['split'](_0x5346c5['sep'])['filter'](_0x84c386 => _0x84c386);
            return _0x29ffd1['reduce']((_0x3af63f, _0x2787fb, _0x23278d) => _0x3af63f && _0x447225[_0x23278d] === _0x2787fb, !![]);
        }
        function _0x574022(_0x39cdef, _0x5d1c83, _0x35d87e) {
            return 'Cannot\x20' + _0x35d87e + '\x20\x27' + _0x39cdef + '\x27\x20to\x20a\x20subdirectory\x20of\x20itself,\x20\x27' + _0x5d1c83 + '\x27.';
        }
        _0x2c356d['exports'] = {
            'checkPaths': _0x4d4be0,
            'checkPathsSync': _0x68b59a,
            'checkParentPaths': _0x27e3da,
            'checkParentPathsSync': _0x42ce0d,
            'isSrcSubdir': _0x595892,
            'areIdentical': _0x13d9f6
        };
    },
    0x13743: (_0x49e8a3, _0xfa9510, _0x413aa3) => {
        'use strict';
        const _0x31c2e8 = _0x413aa3(0x15d1a);
        function _0x393fd8(_0x568c0f, _0x662637, _0x1a0a79, _0x51c737) {
            _0x31c2e8['open'](_0x568c0f, 'r+', (_0x4dae92, _0x3700f7) => {
                if (_0x4dae92)
                    return _0x51c737(_0x4dae92);
                _0x31c2e8['futimes'](_0x3700f7, _0x662637, _0x1a0a79, _0x1ead19 => {
                    _0x31c2e8['close'](_0x3700f7, _0x344f85 => {
                        if (_0x51c737)
                            _0x51c737(_0x1ead19 || _0x344f85);
                    });
                });
            });
        }
        function _0x49bdab(_0x112ebf, _0x1c8825, _0x37e6bb) {
            const _0xc924c2 = _0x31c2e8['openSync'](_0x112ebf, 'r+');
            return _0x31c2e8['futimesSync'](_0xc924c2, _0x1c8825, _0x37e6bb), _0x31c2e8['closeSync'](_0xc924c2);
        }
        _0x49e8a3['exports'] = {
            'utimesMillis': _0x393fd8,
            'utimesMillisSync': _0x49bdab
        };
    },
    0x1217f: _0x1bf691 => {
        'use strict';
        _0x1bf691['exports'] = _0x49f979;
        var _0x30e21b = Object['getPrototypeOf'] || function (_0x32903a) {
            return _0x32903a['__proto__'];
        };
        function _0x49f979(_0x5f31e4) {
            if (_0x5f31e4 === null || typeof _0x5f31e4 !== 'object')
                return _0x5f31e4;
            if (_0x5f31e4 instanceof Object)
                var _0x57bd4a = { '__proto__': _0x30e21b(_0x5f31e4) };
            else
                var _0x57bd4a = Object['create'](null);
            return Object['getOwnPropertyNames'](_0x5f31e4)['forEach'](function (_0x6faa72) {
                Object['defineProperty'](_0x57bd4a, _0x6faa72, Object['getOwnPropertyDescriptor'](_0x5f31e4, _0x6faa72));
            }), _0x57bd4a;
        }
    },
    0x15d1a: (_0x144ad6, _0x263e00, _0x5b2266) => {
        var _0x18f32f = _0x5b2266(0xdf3b), _0xbf2b61 = _0x5b2266(0x15c1b), _0x278b71 = _0x5b2266(0x1458c), _0x37e85e = _0x5b2266(0x1217f), _0x2c87ce = _0x5b2266(0x1206d), _0x467c41, _0x492cf6;
        typeof Symbol === 'function' && typeof Symbol['for'] === 'function' ? (_0x467c41 = Symbol['for']('graceful-fs.queue'), _0x492cf6 = Symbol['for']('graceful-fs.previous')) : (_0x467c41 = '___graceful-fs.queue', _0x492cf6 = '___graceful-fs.previous');
        function _0x14104a() {
        }
        function _0x302112(_0x38f3a1, _0x594aa8) {
            Object['defineProperty'](_0x38f3a1, _0x467c41, {
                'get': function () {
                    return _0x594aa8;
                }
            });
        }
        var _0x2e3939 = _0x14104a;
        if (_0x2c87ce['debuglog'])
            _0x2e3939 = _0x2c87ce['debuglog']('gfs4');
        else {
            if (/\bgfs4\b/i['test'](process['env']['NODE_DEBUG'] || ''))
                _0x2e3939 = function () {
                    var _0x241093 = _0x2c87ce['format']['apply'](_0x2c87ce, arguments);
                    _0x241093 = 'GFS4:\x20' + _0x241093['split'](/\n/)['join']('\x0aGFS4:\x20'), console['error'](_0x241093);
                };
        }
        if (!_0x18f32f[_0x467c41]) {
            var _0xd995f0 = global[_0x467c41] || [];
            _0x302112(_0x18f32f, _0xd995f0), _0x18f32f['close'] = function (_0x34bcda) {
                function _0xe41b1a(_0x4ac0e8, _0x2bb219) {
                    return _0x34bcda['call'](_0x18f32f, _0x4ac0e8, function (_0x142b5b) {
                        !_0x142b5b && _0x5ad7dc();
                        if (typeof _0x2bb219 === 'function')
                            _0x2bb219['apply'](this, arguments);
                    });
                }
                return Object['defineProperty'](_0xe41b1a, _0x492cf6, { 'value': _0x34bcda }), _0xe41b1a;
            }(_0x18f32f['close']), _0x18f32f['closeSync'] = function (_0x3e0ddb) {
                function _0x4a07ed(_0x5dddf4) {
                    _0x3e0ddb['apply'](_0x18f32f, arguments), _0x5ad7dc();
                }
                return Object['defineProperty'](_0x4a07ed, _0x492cf6, { 'value': _0x3e0ddb }), _0x4a07ed;
            }(_0x18f32f['closeSync']), /\bgfs4\b/i['test'](process['env']['NODE_DEBUG'] || '') && process['on']('exit', function () {
                _0x2e3939(_0x18f32f[_0x467c41]), _0x5b2266(0x9a43)['equal'](_0x18f32f[_0x467c41]['length'], 0x0);
            });
        }
        !global[_0x467c41] && _0x302112(global, _0x18f32f[_0x467c41]);
        _0x144ad6['exports'] = _0xf16cdf(_0x37e85e(_0x18f32f));
        process['env']['TEST_GRACEFUL_FS_GLOBAL_PATCH'] && !_0x18f32f['__patched'] && (_0x144ad6['exports'] = _0xf16cdf(_0x18f32f), _0x18f32f['__patched'] = !![]);
        function _0xf16cdf(_0x549d4e) {
            _0xbf2b61(_0x549d4e), _0x549d4e['gracefulify'] = _0xf16cdf, _0x549d4e['createReadStream'] = _0x3d4346, _0x549d4e['createWriteStream'] = _0x303ed9;
            var _0x24bd82 = _0x549d4e['readFile'];
            _0x549d4e['readFile'] = _0x22bd7f;
            function _0x22bd7f(_0x5a2319, _0x268fdf, _0x1a109d) {
                if (typeof _0x268fdf === 'function')
                    _0x1a109d = _0x268fdf, _0x268fdf = null;
                return _0xe18e7a(_0x5a2319, _0x268fdf, _0x1a109d);
                function _0xe18e7a(_0x39f1c5, _0x4004b7, _0x21eced, _0x1fce8d) {
                    return _0x24bd82(_0x39f1c5, _0x4004b7, function (_0x405b28) {
                        if (_0x405b28 && (_0x405b28['code'] === 'EMFILE' || _0x405b28['code'] === 'ENFILE'))
                            _0x483e55([
                                _0xe18e7a,
                                [
                                    _0x39f1c5,
                                    _0x4004b7,
                                    _0x21eced
                                ],
                                _0x405b28,
                                _0x1fce8d || Date['now'](),
                                Date['now']()
                            ]);
                        else {
                            if (typeof _0x21eced === 'function')
                                _0x21eced['apply'](this, arguments);
                        }
                    });
                }
            }
            var _0x1f9121 = _0x549d4e['writeFile'];
            _0x549d4e['writeFile'] = _0x52e607;
            function _0x52e607(_0x19ad5c, _0x200c4f, _0x5bd74a, _0x58097e) {
                if (typeof _0x5bd74a === 'function')
                    _0x58097e = _0x5bd74a, _0x5bd74a = null;
                return _0x8ae20(_0x19ad5c, _0x200c4f, _0x5bd74a, _0x58097e);
                function _0x8ae20(_0x337169, _0x304061, _0x2f145c, _0x5a0c99, _0x21a8c3) {
                    return _0x1f9121(_0x337169, _0x304061, _0x2f145c, function (_0x51bd7e) {
                        if (_0x51bd7e && (_0x51bd7e['code'] === 'EMFILE' || _0x51bd7e['code'] === 'ENFILE'))
                            _0x483e55([
                                _0x8ae20,
                                [
                                    _0x337169,
                                    _0x304061,
                                    _0x2f145c,
                                    _0x5a0c99
                                ],
                                _0x51bd7e,
                                _0x21a8c3 || Date['now'](),
                                Date['now']()
                            ]);
                        else {
                            if (typeof _0x5a0c99 === 'function')
                                _0x5a0c99['apply'](this, arguments);
                        }
                    });
                }
            }
            var _0x177a71 = _0x549d4e['appendFile'];
            if (_0x177a71)
                _0x549d4e['appendFile'] = _0x43938a;
            function _0x43938a(_0x254438, _0x18716e, _0x638049, _0x566c51) {
                if (typeof _0x638049 === 'function')
                    _0x566c51 = _0x638049, _0x638049 = null;
                return _0x25a63b(_0x254438, _0x18716e, _0x638049, _0x566c51);
                function _0x25a63b(_0x2264ef, _0x4121a6, _0x5f3da4, _0x44c00a, _0x3acee2) {
                    return _0x177a71(_0x2264ef, _0x4121a6, _0x5f3da4, function (_0x5f4114) {
                        if (_0x5f4114 && (_0x5f4114['code'] === 'EMFILE' || _0x5f4114['code'] === 'ENFILE'))
                            _0x483e55([
                                _0x25a63b,
                                [
                                    _0x2264ef,
                                    _0x4121a6,
                                    _0x5f3da4,
                                    _0x44c00a
                                ],
                                _0x5f4114,
                                _0x3acee2 || Date['now'](),
                                Date['now']()
                            ]);
                        else {
                            if (typeof _0x44c00a === 'function')
                                _0x44c00a['apply'](this, arguments);
                        }
                    });
                }
            }
            var _0x49a816 = _0x549d4e['copyFile'];
            if (_0x49a816)
                _0x549d4e['copyFile'] = _0x1dd735;
            function _0x1dd735(_0x3fb38a, _0x534c04, _0x64c5c8, _0x3380b5) {
                typeof _0x64c5c8 === 'function' && (_0x3380b5 = _0x64c5c8, _0x64c5c8 = 0x0);
                return _0x1d91a8(_0x3fb38a, _0x534c04, _0x64c5c8, _0x3380b5);
                function _0x1d91a8(_0x1ca096, _0x23e588, _0x185476, _0x490052, _0x2a72db) {
                    return _0x49a816(_0x1ca096, _0x23e588, _0x185476, function (_0x418ac5) {
                        if (_0x418ac5 && (_0x418ac5['code'] === 'EMFILE' || _0x418ac5['code'] === 'ENFILE'))
                            _0x483e55([
                                _0x1d91a8,
                                [
                                    _0x1ca096,
                                    _0x23e588,
                                    _0x185476,
                                    _0x490052
                                ],
                                _0x418ac5,
                                _0x2a72db || Date['now'](),
                                Date['now']()
                            ]);
                        else {
                            if (typeof _0x490052 === 'function')
                                _0x490052['apply'](this, arguments);
                        }
                    });
                }
            }
            var _0xe63d69 = _0x549d4e['readdir'];
            _0x549d4e['readdir'] = _0x3f9d9d;
            var _0x34a4a8 = /^v[0-5]\./;
            function _0x3f9d9d(_0x35c8fa, _0x43e931, _0x5709d1) {
                if (typeof _0x43e931 === 'function')
                    _0x5709d1 = _0x43e931, _0x43e931 = null;
                var _0x3abb17 = _0x34a4a8['test'](process['version']) ? function _0x49951c(_0x7dfd1e, _0x4b9a13, _0x17f624, _0xeb45fc) {
                    return _0xe63d69(_0x7dfd1e, _0x211a06(_0x7dfd1e, _0x4b9a13, _0x17f624, _0xeb45fc));
                } : function _0x56fc1b(_0x17d0a3, _0x4339e2, _0x5dc299, _0x246ee5) {
                    return _0xe63d69(_0x17d0a3, _0x4339e2, _0x211a06(_0x17d0a3, _0x4339e2, _0x5dc299, _0x246ee5));
                };
                return _0x3abb17(_0x35c8fa, _0x43e931, _0x5709d1);
                function _0x211a06(_0x201cad, _0x59c125, _0x5e36c1, _0x515e1c) {
                    return function (_0x4fd42b, _0x567b38) {
                        if (_0x4fd42b && (_0x4fd42b['code'] === 'EMFILE' || _0x4fd42b['code'] === 'ENFILE'))
                            _0x483e55([
                                _0x3abb17,
                                [
                                    _0x201cad,
                                    _0x59c125,
                                    _0x5e36c1
                                ],
                                _0x4fd42b,
                                _0x515e1c || Date['now'](),
                                Date['now']()
                            ]);
                        else {
                            if (_0x567b38 && _0x567b38['sort'])
                                _0x567b38['sort']();
                            if (typeof _0x5e36c1 === 'function')
                                _0x5e36c1['call'](this, _0x4fd42b, _0x567b38);
                        }
                    };
                }
            }
            if (process['version']['substr'](0x0, 0x4) === 'v0.8') {
                var _0x5405ad = _0x278b71(_0x549d4e);
                _0x8bcdf0 = _0x5405ad['ReadStream'], _0x44950e = _0x5405ad['WriteStream'];
            }
            var _0x33bd47 = _0x549d4e['ReadStream'];
            _0x33bd47 && (_0x8bcdf0['prototype'] = Object['create'](_0x33bd47['prototype']), _0x8bcdf0['prototype']['open'] = _0x3abc22);
            var _0x34fae8 = _0x549d4e['WriteStream'];
            _0x34fae8 && (_0x44950e['prototype'] = Object['create'](_0x34fae8['prototype']), _0x44950e['prototype']['open'] = _0x2813c8);
            Object['defineProperty'](_0x549d4e, 'ReadStream', {
                'get': function () {
                    return _0x8bcdf0;
                },
                'set': function (_0x57a434) {
                    _0x8bcdf0 = _0x57a434;
                },
                'enumerable': !![],
                'configurable': !![]
            }), Object['defineProperty'](_0x549d4e, 'WriteStream', {
                'get': function () {
                    return _0x44950e;
                },
                'set': function (_0x24a038) {
                    _0x44950e = _0x24a038;
                },
                'enumerable': !![],
                'configurable': !![]
            });
            var _0x2bb6e9 = _0x8bcdf0;
            Object['defineProperty'](_0x549d4e, 'FileReadStream', {
                'get': function () {
                    return _0x2bb6e9;
                },
                'set': function (_0xe7346a) {
                    _0x2bb6e9 = _0xe7346a;
                },
                'enumerable': !![],
                'configurable': !![]
            });
            var _0x2fed52 = _0x44950e;
            Object['defineProperty'](_0x549d4e, 'FileWriteStream', {
                'get': function () {
                    return _0x2fed52;
                },
                'set': function (_0x52ca75) {
                    _0x2fed52 = _0x52ca75;
                },
                'enumerable': !![],
                'configurable': !![]
            });
            function _0x8bcdf0(_0x1fbffb, _0x17c52b) {
                if (this instanceof _0x8bcdf0)
                    return _0x33bd47['apply'](this, arguments), this;
                else
                    return _0x8bcdf0['apply'](Object['create'](_0x8bcdf0['prototype']), arguments);
            }
            function _0x3abc22() {
                var _0x2ea4d4 = this;
                _0x296dbe(_0x2ea4d4['path'], _0x2ea4d4['flags'], _0x2ea4d4['mode'], function (_0xc70e82, _0x56a239) {
                    if (_0xc70e82) {
                        if (_0x2ea4d4['autoClose'])
                            _0x2ea4d4['destroy']();
                        _0x2ea4d4['emit']('error', _0xc70e82);
                    } else
                        _0x2ea4d4['fd'] = _0x56a239, _0x2ea4d4['emit']('open', _0x56a239), _0x2ea4d4['read']();
                });
            }
            function _0x44950e(_0xc6d6f3, _0x596586) {
                if (this instanceof _0x44950e)
                    return _0x34fae8['apply'](this, arguments), this;
                else
                    return _0x44950e['apply'](Object['create'](_0x44950e['prototype']), arguments);
            }
            function _0x2813c8() {
                var _0x18c553 = this;
                _0x296dbe(_0x18c553['path'], _0x18c553['flags'], _0x18c553['mode'], function (_0x480df3, _0x199fc4) {
                    _0x480df3 ? (_0x18c553['destroy'](), _0x18c553['emit']('error', _0x480df3)) : (_0x18c553['fd'] = _0x199fc4, _0x18c553['emit']('open', _0x199fc4));
                });
            }
            function _0x3d4346(_0x318e3b, _0x4d0982) {
                return new _0x549d4e['ReadStream'](_0x318e3b, _0x4d0982);
            }
            function _0x303ed9(_0x1f38ef, _0x4eb65c) {
                return new _0x549d4e['WriteStream'](_0x1f38ef, _0x4eb65c);
            }
            var _0x354f16 = _0x549d4e['open'];
            _0x549d4e['open'] = _0x296dbe;
            function _0x296dbe(_0x4a42c6, _0x312b4d, _0x442412, _0x5b6681) {
                if (typeof _0x442412 === 'function')
                    _0x5b6681 = _0x442412, _0x442412 = null;
                return _0x35cafd(_0x4a42c6, _0x312b4d, _0x442412, _0x5b6681);
                function _0x35cafd(_0x15054d, _0x3c672c, _0x3b3d87, _0x41d356, _0x4644f4) {
                    return _0x354f16(_0x15054d, _0x3c672c, _0x3b3d87, function (_0x576a74, _0x520e17) {
                        if (_0x576a74 && (_0x576a74['code'] === 'EMFILE' || _0x576a74['code'] === 'ENFILE'))
                            _0x483e55([
                                _0x35cafd,
                                [
                                    _0x15054d,
                                    _0x3c672c,
                                    _0x3b3d87,
                                    _0x41d356
                                ],
                                _0x576a74,
                                _0x4644f4 || Date['now'](),
                                Date['now']()
                            ]);
                        else {
                            if (typeof _0x41d356 === 'function')
                                _0x41d356['apply'](this, arguments);
                        }
                    });
                }
            }
            return _0x549d4e;
        }
        function _0x483e55(_0x209d18) {
            _0x2e3939('ENQUEUE', _0x209d18[0x0]['name'], _0x209d18[0x1]), _0x18f32f[_0x467c41]['push'](_0x209d18), _0x3e7f50();
        }
        var _0x420a04;
        function _0x5ad7dc() {
            var _0x119b25 = Date['now']();
            for (var _0xc839b = 0x0; _0xc839b < _0x18f32f[_0x467c41]['length']; ++_0xc839b) {
                _0x18f32f[_0x467c41][_0xc839b]['length'] > 0x2 && (_0x18f32f[_0x467c41][_0xc839b][0x3] = _0x119b25, _0x18f32f[_0x467c41][_0xc839b][0x4] = _0x119b25);
            }
            _0x3e7f50();
        }
        function _0x3e7f50() {
            clearTimeout(_0x420a04), _0x420a04 = undefined;
            if (_0x18f32f[_0x467c41]['length'] === 0x0)
                return;
            var _0xd31e22 = _0x18f32f[_0x467c41]['shift'](), _0x48e065 = _0xd31e22[0x0], _0x2881f6 = _0xd31e22[0x1], _0x1f72e8 = _0xd31e22[0x2], _0x41918e = _0xd31e22[0x3], _0x16854f = _0xd31e22[0x4];
            if (_0x41918e === undefined)
                _0x2e3939('RETRY', _0x48e065['name'], _0x2881f6), _0x48e065['apply'](null, _0x2881f6);
            else {
                if (Date['now']() - _0x41918e >= 0xea60) {
                    _0x2e3939('TIMEOUT', _0x48e065['name'], _0x2881f6);
                    var _0x435862 = _0x2881f6['pop']();
                    if (typeof _0x435862 === 'function')
                        _0x435862['call'](null, _0x1f72e8);
                } else {
                    var _0x131322 = Date['now']() - _0x16854f, _0x1ec09f = Math['max'](_0x16854f - _0x41918e, 0x1), _0x6e9dbf = Math['min'](_0x1ec09f * 1.2, 0x64);
                    _0x131322 >= _0x6e9dbf ? (_0x2e3939('RETRY', _0x48e065['name'], _0x2881f6), _0x48e065['apply'](null, _0x2881f6['concat']([_0x41918e]))) : _0x18f32f[_0x467c41]['push'](_0xd31e22);
                }
            }
            _0x420a04 === undefined && (_0x420a04 = setTimeout(_0x3e7f50, 0x0));
        }
    },
    0x1458c: (_0x3ad6eb, _0x5b0d21, _0x38862b) => {
        var _0x3539aa = _0x38862b(0x31ed)['Stream'];
        _0x3ad6eb['exports'] = _0x435e41;
        function _0x435e41(_0x2dd4bd) {
            return {
                'ReadStream': _0xc7b709,
                'WriteStream': _0x343e4a
            };
            function _0xc7b709(_0xef739, _0x5e7884) {
                if (!(this instanceof _0xc7b709))
                    return new _0xc7b709(_0xef739, _0x5e7884);
                _0x3539aa['call'](this);
                var _0x1eb718 = this;
                this['path'] = _0xef739, this['fd'] = null, this['readable'] = !![], this['paused'] = ![], this['flags'] = 'r', this['mode'] = 0x1b6, this['bufferSize'] = 0x40 * 0x400, _0x5e7884 = _0x5e7884 || {};
                var _0xb63ee0 = Object['keys'](_0x5e7884);
                for (var _0x1c4b44 = 0x0, _0x39aeb7 = _0xb63ee0['length']; _0x1c4b44 < _0x39aeb7; _0x1c4b44++) {
                    var _0x28c04d = _0xb63ee0[_0x1c4b44];
                    this[_0x28c04d] = _0x5e7884[_0x28c04d];
                }
                if (this['encoding'])
                    this['setEncoding'](this['encoding']);
                if (this['start'] !== undefined) {
                    if ('number' !== typeof this['start'])
                        throw TypeError('start\x20must\x20be\x20a\x20Number');
                    if (this['end'] === undefined)
                        this['end'] = Infinity;
                    else {
                        if ('number' !== typeof this['end'])
                            throw TypeError('end\x20must\x20be\x20a\x20Number');
                    }
                    if (this['start'] > this['end'])
                        throw new Error('start\x20must\x20be\x20<=\x20end');
                    this['pos'] = this['start'];
                }
                if (this['fd'] !== null) {
                    process['nextTick'](function () {
                        _0x1eb718['_read']();
                    });
                    return;
                }
                _0x2dd4bd['open'](this['path'], this['flags'], this['mode'], function (_0x5d8689, _0x18250c) {
                    if (_0x5d8689) {
                        _0x1eb718['emit']('error', _0x5d8689), _0x1eb718['readable'] = ![];
                        return;
                    }
                    _0x1eb718['fd'] = _0x18250c, _0x1eb718['emit']('open', _0x18250c), _0x1eb718['_read']();
                });
            }
            function _0x343e4a(_0x18add6, _0x27b217) {
                if (!(this instanceof _0x343e4a))
                    return new _0x343e4a(_0x18add6, _0x27b217);
                _0x3539aa['call'](this), this['path'] = _0x18add6, this['fd'] = null, this['writable'] = !![], this['flags'] = 'w', this['encoding'] = 'binary', this['mode'] = 0x1b6, this['bytesWritten'] = 0x0, _0x27b217 = _0x27b217 || {};
                var _0x3c82c4 = Object['keys'](_0x27b217);
                for (var _0x53ce08 = 0x0, _0x48b17d = _0x3c82c4['length']; _0x53ce08 < _0x48b17d; _0x53ce08++) {
                    var _0x1542b6 = _0x3c82c4[_0x53ce08];
                    this[_0x1542b6] = _0x27b217[_0x1542b6];
                }
                if (this['start'] !== undefined) {
                    if ('number' !== typeof this['start'])
                        throw TypeError('start\x20must\x20be\x20a\x20Number');
                    if (this['start'] < 0x0)
                        throw new Error('start\x20must\x20be\x20>=\x20zero');
                    this['pos'] = this['start'];
                }
                this['busy'] = ![], this['_queue'] = [], this['fd'] === null && (this['_open'] = _0x2dd4bd['open'], this['_queue']['push']([
                    this['_open'],
                    this['path'],
                    this['flags'],
                    this['mode'],
                    undefined
                ]), this['flush']());
            }
        }
    },
    0x15c1b: (_0x7a8b55, _0x23a1c9, _0x568e97) => {
        var _0x261975 = _0x568e97(0x5629), _0x595330 = process['cwd'], _0x48538d = null, _0x767efc = process['env']['GRACEFUL_FS_PLATFORM'] || process['platform'];
        process['cwd'] = function () {
            if (!_0x48538d)
                _0x48538d = _0x595330['call'](process);
            return _0x48538d;
        };
        try {
            process['cwd']();
        } catch (_0x2f3fc2) {
        }
        if (typeof process['chdir'] === 'function') {
            var _0x4deca4 = process['chdir'];
            process['chdir'] = function (_0x5def8b) {
                _0x48538d = null, _0x4deca4['call'](process, _0x5def8b);
            };
            if (Object['setPrototypeOf'])
                Object['setPrototypeOf'](process['chdir'], _0x4deca4);
        }
        _0x7a8b55['exports'] = _0x577717;
        function _0x577717(_0x57adf0) {
            _0x261975['hasOwnProperty']('O_SYMLINK') && process['version']['match'](/^v0\.6\.[0-2]|^v0\.5\./) && _0x2168dc(_0x57adf0);
            !_0x57adf0['lutimes'] && _0x396ba7(_0x57adf0);
            _0x57adf0['chown'] = _0x4ba822(_0x57adf0['chown']), _0x57adf0['fchown'] = _0x4ba822(_0x57adf0['fchown']), _0x57adf0['lchown'] = _0x4ba822(_0x57adf0['lchown']), _0x57adf0['chmod'] = _0x127c04(_0x57adf0['chmod']), _0x57adf0['fchmod'] = _0x127c04(_0x57adf0['fchmod']), _0x57adf0['lchmod'] = _0x127c04(_0x57adf0['lchmod']), _0x57adf0['chownSync'] = _0x72fdb7(_0x57adf0['chownSync']), _0x57adf0['fchownSync'] = _0x72fdb7(_0x57adf0['fchownSync']), _0x57adf0['lchownSync'] = _0x72fdb7(_0x57adf0['lchownSync']), _0x57adf0['chmodSync'] = _0x1306cc(_0x57adf0['chmodSync']), _0x57adf0['fchmodSync'] = _0x1306cc(_0x57adf0['fchmodSync']), _0x57adf0['lchmodSync'] = _0x1306cc(_0x57adf0['lchmodSync']), _0x57adf0['stat'] = _0x2da589(_0x57adf0['stat']), _0x57adf0['fstat'] = _0x2da589(_0x57adf0['fstat']), _0x57adf0['lstat'] = _0x2da589(_0x57adf0['lstat']), _0x57adf0['statSync'] = _0x5d97ac(_0x57adf0['statSync']), _0x57adf0['fstatSync'] = _0x5d97ac(_0x57adf0['fstatSync']), _0x57adf0['lstatSync'] = _0x5d97ac(_0x57adf0['lstatSync']);
            _0x57adf0['chmod'] && !_0x57adf0['lchmod'] && (_0x57adf0['lchmod'] = function (_0x3d61b8, _0x20eb2c, _0x518fc7) {
                if (_0x518fc7)
                    process['nextTick'](_0x518fc7);
            }, _0x57adf0['lchmodSync'] = function () {
            });
            _0x57adf0['chown'] && !_0x57adf0['lchown'] && (_0x57adf0['lchown'] = function (_0x32ed1c, _0x5d4362, _0x3e3910, _0xc0057a) {
                if (_0xc0057a)
                    process['nextTick'](_0xc0057a);
            }, _0x57adf0['lchownSync'] = function () {
            });
            _0x767efc === 'win32' && (_0x57adf0['rename'] = typeof _0x57adf0['rename'] !== 'function' ? _0x57adf0['rename'] : function (_0x5d54fe) {
                function _0x5db876(_0x331aee, _0x575556, _0x14ef2a) {
                    var _0x15d8b7 = Date['now'](), _0x228186 = 0x0;
                    _0x5d54fe(_0x331aee, _0x575556, function _0x22bb5a(_0x546dcd) {
                        if (_0x546dcd && (_0x546dcd['code'] === 'EACCES' || _0x546dcd['code'] === 'EPERM') && Date['now']() - _0x15d8b7 < 0xea60) {
                            setTimeout(function () {
                                _0x57adf0['stat'](_0x575556, function (_0x4ac476, _0x3c25b8) {
                                    if (_0x4ac476 && _0x4ac476['code'] === 'ENOENT')
                                        _0x5d54fe(_0x331aee, _0x575556, _0x22bb5a);
                                    else
                                        _0x14ef2a(_0x546dcd);
                                });
                            }, _0x228186);
                            if (_0x228186 < 0x64)
                                _0x228186 += 0xa;
                            return;
                        }
                        if (_0x14ef2a)
                            _0x14ef2a(_0x546dcd);
                    });
                }
                if (Object['setPrototypeOf'])
                    Object['setPrototypeOf'](_0x5db876, _0x5d54fe);
                return _0x5db876;
            }(_0x57adf0['rename']));
            _0x57adf0['read'] = typeof _0x57adf0['read'] !== 'function' ? _0x57adf0['read'] : function (_0xb79bdf) {
                function _0x14b723(_0x20a8b0, _0x5ca40d, _0x36e2c7, _0xf27d77, _0x30558c, _0xbcb655) {
                    var _0x50c676;
                    if (_0xbcb655 && typeof _0xbcb655 === 'function') {
                        var _0x307def = 0x0;
                        _0x50c676 = function (_0x38f084, _0x30ac8a, _0x510ba9) {
                            if (_0x38f084 && _0x38f084['code'] === 'EAGAIN' && _0x307def < 0xa)
                                return _0x307def++, _0xb79bdf['call'](_0x57adf0, _0x20a8b0, _0x5ca40d, _0x36e2c7, _0xf27d77, _0x30558c, _0x50c676);
                            _0xbcb655['apply'](this, arguments);
                        };
                    }
                    return _0xb79bdf['call'](_0x57adf0, _0x20a8b0, _0x5ca40d, _0x36e2c7, _0xf27d77, _0x30558c, _0x50c676);
                }
                if (Object['setPrototypeOf'])
                    Object['setPrototypeOf'](_0x14b723, _0xb79bdf);
                return _0x14b723;
            }(_0x57adf0['read']), _0x57adf0['readSync'] = typeof _0x57adf0['readSync'] !== 'function' ? _0x57adf0['readSync'] : function (_0x48e769) {
                return function (_0x296dad, _0x14a10c, _0x203ddb, _0x3aec4, _0x40bdaa) {
                    var _0xee5563 = 0x0;
                    while (!![]) {
                        try {
                            return _0x48e769['call'](_0x57adf0, _0x296dad, _0x14a10c, _0x203ddb, _0x3aec4, _0x40bdaa);
                        } catch (_0x2ac2d3) {
                            if (_0x2ac2d3['code'] === 'EAGAIN' && _0xee5563 < 0xa) {
                                _0xee5563++;
                                continue;
                            }
                            throw _0x2ac2d3;
                        }
                    }
                };
            }(_0x57adf0['readSync']);
            function _0x2168dc(_0xa32bd5) {
                _0xa32bd5['lchmod'] = function (_0x278945, _0x2fd4f9, _0x130b3f) {
                    _0xa32bd5['open'](_0x278945, _0x261975['O_WRONLY'] | _0x261975['O_SYMLINK'], _0x2fd4f9, function (_0x82a235, _0x13e58f) {
                        if (_0x82a235) {
                            if (_0x130b3f)
                                _0x130b3f(_0x82a235);
                            return;
                        }
                        _0xa32bd5['fchmod'](_0x13e58f, _0x2fd4f9, function (_0xa526ef) {
                            _0xa32bd5['close'](_0x13e58f, function (_0x41e63c) {
                                if (_0x130b3f)
                                    _0x130b3f(_0xa526ef || _0x41e63c);
                            });
                        });
                    });
                }, _0xa32bd5['lchmodSync'] = function (_0x21986e, _0x2d51ca) {
                    var _0x393848 = _0xa32bd5['openSync'](_0x21986e, _0x261975['O_WRONLY'] | _0x261975['O_SYMLINK'], _0x2d51ca), _0x27964d = !![], _0xe046c;
                    try {
                        _0xe046c = _0xa32bd5['fchmodSync'](_0x393848, _0x2d51ca), _0x27964d = ![];
                    } finally {
                        if (_0x27964d)
                            try {
                                _0xa32bd5['closeSync'](_0x393848);
                            } catch (_0x10f46d) {
                            }
                        else
                            _0xa32bd5['closeSync'](_0x393848);
                    }
                    return _0xe046c;
                };
            }
            function _0x396ba7(_0x13fd50) {
                if (_0x261975['hasOwnProperty']('O_SYMLINK') && _0x13fd50['futimes'])
                    _0x13fd50['lutimes'] = function (_0x543a44, _0x13db90, _0x3118df, _0xa2c5bf) {
                        _0x13fd50['open'](_0x543a44, _0x261975['O_SYMLINK'], function (_0x261e74, _0x1993d5) {
                            if (_0x261e74) {
                                if (_0xa2c5bf)
                                    _0xa2c5bf(_0x261e74);
                                return;
                            }
                            _0x13fd50['futimes'](_0x1993d5, _0x13db90, _0x3118df, function (_0x3b6f08) {
                                _0x13fd50['close'](_0x1993d5, function (_0x8139dd) {
                                    if (_0xa2c5bf)
                                        _0xa2c5bf(_0x3b6f08 || _0x8139dd);
                                });
                            });
                        });
                    }, _0x13fd50['lutimesSync'] = function (_0x324494, _0x207776, _0x27a349) {
                        var _0x513403 = _0x13fd50['openSync'](_0x324494, _0x261975['O_SYMLINK']), _0xd790af, _0x4349e6 = !![];
                        try {
                            _0xd790af = _0x13fd50['futimesSync'](_0x513403, _0x207776, _0x27a349), _0x4349e6 = ![];
                        } finally {
                            if (_0x4349e6)
                                try {
                                    _0x13fd50['closeSync'](_0x513403);
                                } catch (_0xefd73f) {
                                }
                            else
                                _0x13fd50['closeSync'](_0x513403);
                        }
                        return _0xd790af;
                    };
                else
                    _0x13fd50['futimes'] && (_0x13fd50['lutimes'] = function (_0x50a018, _0x224024, _0x1b36df, _0x1b40ea) {
                        if (_0x1b40ea)
                            process['nextTick'](_0x1b40ea);
                    }, _0x13fd50['lutimesSync'] = function () {
                    });
            }
            function _0x127c04(_0x43a0f0) {
                if (!_0x43a0f0)
                    return _0x43a0f0;
                return function (_0x5576f2, _0x100987, _0x2673b8) {
                    return _0x43a0f0['call'](_0x57adf0, _0x5576f2, _0x100987, function (_0xac2936) {
                        if (_0x2c07a8(_0xac2936))
                            _0xac2936 = null;
                        if (_0x2673b8)
                            _0x2673b8['apply'](this, arguments);
                    });
                };
            }
            function _0x1306cc(_0x4cdc7f) {
                if (!_0x4cdc7f)
                    return _0x4cdc7f;
                return function (_0x240b35, _0x45fb87) {
                    try {
                        return _0x4cdc7f['call'](_0x57adf0, _0x240b35, _0x45fb87);
                    } catch (_0xc3ec97) {
                        if (!_0x2c07a8(_0xc3ec97))
                            throw _0xc3ec97;
                    }
                };
            }
            function _0x4ba822(_0x41b95f) {
                if (!_0x41b95f)
                    return _0x41b95f;
                return function (_0x4ab8cf, _0x1452fe, _0x18dcf9, _0x57119a) {
                    return _0x41b95f['call'](_0x57adf0, _0x4ab8cf, _0x1452fe, _0x18dcf9, function (_0x1ffa6f) {
                        if (_0x2c07a8(_0x1ffa6f))
                            _0x1ffa6f = null;
                        if (_0x57119a)
                            _0x57119a['apply'](this, arguments);
                    });
                };
            }
            function _0x72fdb7(_0x291de4) {
                if (!_0x291de4)
                    return _0x291de4;
                return function (_0x16cee5, _0x47af7a, _0x1a2fa4) {
                    try {
                        return _0x291de4['call'](_0x57adf0, _0x16cee5, _0x47af7a, _0x1a2fa4);
                    } catch (_0x5179cf) {
                        if (!_0x2c07a8(_0x5179cf))
                            throw _0x5179cf;
                    }
                };
            }
            function _0x2da589(_0x1444d7) {
                if (!_0x1444d7)
                    return _0x1444d7;
                return function (_0x49c595, _0x8310f9, _0x399245) {
                    typeof _0x8310f9 === 'function' && (_0x399245 = _0x8310f9, _0x8310f9 = null);
                    function _0x5896d9(_0x178622, _0x56d518) {
                        if (_0x56d518) {
                            if (_0x56d518['uid'] < 0x0)
                                _0x56d518['uid'] += 0x100000000;
                            if (_0x56d518['gid'] < 0x0)
                                _0x56d518['gid'] += 0x100000000;
                        }
                        if (_0x399245)
                            _0x399245['apply'](this, arguments);
                    }
                    return _0x8310f9 ? _0x1444d7['call'](_0x57adf0, _0x49c595, _0x8310f9, _0x5896d9) : _0x1444d7['call'](_0x57adf0, _0x49c595, _0x5896d9);
                };
            }
            function _0x5d97ac(_0x1cac7e) {
                if (!_0x1cac7e)
                    return _0x1cac7e;
                return function (_0x4bf52b, _0x2ec78f) {
                    var _0x3df643 = _0x2ec78f ? _0x1cac7e['call'](_0x57adf0, _0x4bf52b, _0x2ec78f) : _0x1cac7e['call'](_0x57adf0, _0x4bf52b);
                    if (_0x3df643) {
                        if (_0x3df643['uid'] < 0x0)
                            _0x3df643['uid'] += 0x100000000;
                        if (_0x3df643['gid'] < 0x0)
                            _0x3df643['gid'] += 0x100000000;
                    }
                    return _0x3df643;
                };
            }
            function _0x2c07a8(_0x30e45e) {
                if (!_0x30e45e)
                    return !![];
                if (_0x30e45e['code'] === 'ENOSYS')
                    return !![];
                var _0x4d4c4f = !process['getuid'] || process['getuid']() !== 0x0;
                if (_0x4d4c4f) {
                    if (_0x30e45e['code'] === 'EINVAL' || _0x30e45e['code'] === 'EPERM')
                        return !![];
                }
                return ![];
            }
        }
    },
    0x8517: (_0x3827a8, _0x230783, _0x1cb16f) => {
        let _0x1d3888;
        try {
            _0x1d3888 = _0x1cb16f(0x15d1a);
        } catch (_0x3c6798) {
            _0x1d3888 = _0x1cb16f(0xdf3b);
        }
        const _0x5b2caa = _0x1cb16f(0x15f6a), {
                stringify: _0x25841a,
                stripBom: _0x58beb7
            } = _0x1cb16f(0x14415);
        async function _0x5cbbef(_0x56b2f3, _0xff516 = {}) {
            typeof _0xff516 === 'string' && (_0xff516 = { 'encoding': _0xff516 });
            const _0x1c6a71 = _0xff516['fs'] || _0x1d3888, _0x2516cc = 'throws' in _0xff516 ? _0xff516['throws'] : !![];
            let _0x22658e = await _0x5b2caa['fromCallback'](_0x1c6a71['readFile'])(_0x56b2f3, _0xff516);
            _0x22658e = _0x58beb7(_0x22658e);
            let _0x45b2e9;
            try {
                _0x45b2e9 = JSON['parse'](_0x22658e, _0xff516 ? _0xff516['reviver'] : null);
            } catch (_0x37973a) {
                if (_0x2516cc) {
                    _0x37973a['message'] = _0x56b2f3 + ':\x20' + _0x37973a['message'];
                    throw _0x37973a;
                } else
                    return null;
            }
            return _0x45b2e9;
        }
        const _0x4105ab = _0x5b2caa['fromPromise'](_0x5cbbef);
        function _0x6138c6(_0x428473, _0x430c32 = {}) {
            typeof _0x430c32 === 'string' && (_0x430c32 = { 'encoding': _0x430c32 });
            const _0x313aa2 = _0x430c32['fs'] || _0x1d3888, _0x5212ad = 'throws' in _0x430c32 ? _0x430c32['throws'] : !![];
            try {
                let _0x25b32f = _0x313aa2['readFileSync'](_0x428473, _0x430c32);
                return _0x25b32f = _0x58beb7(_0x25b32f), JSON['parse'](_0x25b32f, _0x430c32['reviver']);
            } catch (_0x3d9732) {
                if (_0x5212ad) {
                    _0x3d9732['message'] = _0x428473 + ':\x20' + _0x3d9732['message'];
                    throw _0x3d9732;
                } else
                    return null;
            }
        }
        async function _0x446305(_0x3b1de8, _0x3427f9, _0x24e4ba = {}) {
            const _0x51d726 = _0x24e4ba['fs'] || _0x1d3888, _0x575b42 = _0x25841a(_0x3427f9, _0x24e4ba);
            await _0x5b2caa['fromCallback'](_0x51d726['writeFile'])(_0x3b1de8, _0x575b42, _0x24e4ba);
        }
        const _0x50512d = _0x5b2caa['fromPromise'](_0x446305);
        function _0x264033(_0x522be9, _0x5b9aa3, _0x2148bd = {}) {
            const _0x4809a7 = _0x2148bd['fs'] || _0x1d3888, _0x3af59c = _0x25841a(_0x5b9aa3, _0x2148bd);
            return _0x4809a7['writeFileSync'](_0x522be9, _0x3af59c, _0x2148bd);
        }
        const _0x427449 = {
            'readFile': _0x4105ab,
            'readFileSync': _0x6138c6,
            'writeFile': _0x50512d,
            'writeFileSync': _0x264033
        };
        _0x3827a8['exports'] = _0x427449;
    },
    0x14415: _0x510e8a => {
        function _0x486529(_0x28a7f6, {
            EOL: EOL = '\x0a',
            finalEOL: finalEOL = !![],
            replacer: replacer = null,
            spaces: _0x3356df
        } = {}) {
            const _0x27ff35 = finalEOL ? EOL : '', _0x577579 = JSON['stringify'](_0x28a7f6, replacer, _0x3356df);
            return _0x577579['replace'](/\n/g, EOL) + _0x27ff35;
        }
        function _0x37a76c(_0x5d3b88) {
            if (Buffer['isBuffer'](_0x5d3b88))
                _0x5d3b88 = _0x5d3b88['toString']('utf8');
            return _0x5d3b88['replace'](/^\uFEFF/, '');
        }
        _0x510e8a['exports'] = {
            'stringify': _0x486529,
            'stripBom': _0x37a76c
        };
    },
    0x15f6a: (_0x3c18ef, _0x1b9c5a) => {
        'use strict';
        _0x1b9c5a['fromCallback'] = function (_0x5ab4f7) {
            return Object['defineProperty'](function (..._0x2522a5) {
                if (typeof _0x2522a5[_0x2522a5['length'] - 0x1] === 'function')
                    _0x5ab4f7['apply'](this, _0x2522a5);
                else
                    return new Promise((_0x288358, _0x365214) => {
                        _0x5ab4f7['call'](this, ..._0x2522a5, (_0x146205, _0x8273a2) => _0x146205 != null ? _0x365214(_0x146205) : _0x288358(_0x8273a2));
                    });
            }, 'name', { 'value': _0x5ab4f7['name'] });
        }, _0x1b9c5a['fromPromise'] = function (_0x25c683) {
            return Object['defineProperty'](function (..._0x37728f) {
                const _0x54aca = _0x37728f[_0x37728f['length'] - 0x1];
                if (typeof _0x54aca !== 'function')
                    return _0x25c683['apply'](this, _0x37728f);
                else
                    _0x25c683['apply'](this, _0x37728f['slice'](0x0, -0x1))['then'](_0x36551a => _0x54aca(null, _0x36551a), _0x54aca);
            }, 'name', { 'value': _0x25c683['name'] });
        };
    }
};
;
