// test-maker/dist/266.js
exports['id'] = 0x10a, exports['ids'] = [
    0x10a,
    0x3a4
], exports['modules'] = {
    0xc30c: _0x5336ed => {
        function _0x18ff4e(_0x2f5d61) {
            var _0x552dd1 = new Error('Cannot\x20find\x20module\x20\x27' + _0x2f5d61 + '\x27');
            _0x552dd1['code'] = 'MODULE_NOT_FOUND';
            throw _0x552dd1;
        }
        _0x18ff4e['keys'] = () => [], _0x18ff4e['resolve'] = _0x18ff4e, _0x18ff4e['id'] = 0xc30c, _0x5336ed['exports'] = _0x18ff4e;
    },
    0xbe3: (_0x3df191, _0x3d991c, _0x191dea) => {
        'use strict';
        _0x191dea['r'](_0x3d991c), _0x191dea['d'](_0x3d991c, { 'Validate': () => _0x4d2cb9 });
        var _0x5f2d70 = _0x191dea(0xc165), _0x5722cf = _0x191dea['n'](_0x5f2d70), _0x21bbac = _0x191dea(0xcaad);
        class _0x4d2cb9 {
            constructor() {
                _0x5f2d70['program']['command']('validate', 'Validate\x20Installation')['cast'](![])['strict'](!![])['action'](async () => {
                    await (0x0, _0x21bbac['validateDeps'])(), process['exit'](0x0);
                });
            }
        }
    },
    0xcaad: (_0x21c57a, _0x158124, _0xc8432f) => {
        'use strict';
        _0xc8432f['r'](_0x158124), _0xc8432f['d'](_0x158124, { 'validateDeps': () => _0x30a141 });
        var _0x356bc5 = _0xc8432f(0x5395), _0x3b14e9 = _0xc8432f['n'](_0x356bc5), _0x2d7fe9 = _0xc8432f(0x1145c), _0x29aeac = _0xc8432f['n'](_0x2d7fe9), _0x14574d = _0xc8432f(0x17fbe), _0x35ff30 = _0xc8432f['n'](_0x14574d), _0x4403ba = _0xc8432f(0x3b1), _0x22b6a5 = _0xc8432f['n'](_0x4403ba), _0x3dcd33 = _0xc8432f(0xbb05);
        async function _0x30a141() {
            let _0x6d40ac = ![];
            const _0x4c44b2 = _0x3dcd33['testMakerInstallPath'] + '/node_modules/test-maker-dummy/bundles-hash.txt', _0x37f918 = await _0x35ff30()['pathExists'](_0x4c44b2), _0x14a224 = './package-lock.json', _0x350ede = './pnpm-lock.yaml', _0x3d81ec = './yarn.lock';
            let _0x3fe672 = '', _0x102309 = '', _0x2dfac7 = '';
            await _0x35ff30()['pathExists'](_0x14a224) && (_0x3fe672 = (await _0x35ff30()['stat'](_0x14a224))['mtime']['toString']());
            await _0x35ff30()['pathExists'](_0x350ede) && (_0x102309 = (await _0x35ff30()['stat'](_0x350ede))['mtime']['toString']());
            await _0x35ff30()['pathExists'](_0x3d81ec) && (_0x2dfac7 = (await _0x35ff30()['stat'](_0x3d81ec))['mtime']['toString']());
            const _0x120986 = _0x3fe672 + _0x102309 + _0x2dfac7;
            if (_0x37f918) {
            } else
                _0x6d40ac = !![];
            if (_0x6d40ac) {
                console['info'](_0x29aeac()['cyanBright']('\x0aValidating\x20Test\x20Maker\x20installation\x20(This\x20is\x20only\x20required\x20on\x20first\x20time\x20install\x20or\x20when\x20extra\x20packages\x20are\x20installed/uninstalled)\x0a'));
                let _0x339416 = new (_0x3b14e9())(_0x3dcd33['testMakerDistPath'] + '/bundled_modules.zip');
                _0x339416['extractAllTo'](_0x3dcd33['testMakerInstallPath'] + '/node_modules', !![]), await _0x35ff30()['outputFile'](_0x4c44b2, _0x120986);
                const _0x178e7c = [];
                for (let _0x49fe14 of _0x178e7c) {
                    await _0x35ff30()['copy'](_0x3dcd33['testMakerInstallPath'] + '/node_modules/' + _0x49fe14, '' + _0x22b6a5()['join'](_0x3dcd33['testMakerInstallPath'], '../', _0x49fe14), { 'overwrite': !![] });
                }
            }
        }
    }
};
;
