// test-maker/dist/315.js
'use strict';
exports['id'] = 0x13b, exports['ids'] = [0x13b], exports['modules'] = {
    0x1495b: (_0x52857c, _0x5076a0, _0x3de3c1) => {
        _0x3de3c1['r'](_0x5076a0), _0x3de3c1['d'](_0x5076a0, {
            'IpcSlave': () => _0xe4bd85,
            'ipcSlave': () => _0x4489fc
        });
        var _0x5bb3ee = _0x3de3c1(0x167e3), _0x27f731 = _0x3de3c1['n'](_0x5bb3ee), _0x3b5c01 = _0x3de3c1(0xb712);
        class _0xe4bd85 {
            constructor() {
                const _0x456da2 = process['argv']['indexOf']('--terminal-eye-slave-uid') + 0x1;
                let _0x985a7 = process['argv'][_0x456da2];
                const _0x259aec = 'test-maker-terminal-eye-master-' + _0x985a7;
                _0x5bb3ee['config']['id'] = 'test-maker-terminal-eye-slave', _0x5bb3ee['config']['retry'] = 0x5dc, _0x5bb3ee['config']['silent'] = !![], _0x5bb3ee['connectTo'](_0x259aec, () => {
                    this['messenger'] = _0x5bb3ee['of'][_0x259aec], _0x5bb3ee['of'][_0x259aec]['on']('connect', async () => {
                        this['messenger']['emit']('message', { 'action': 'connected' }), setInterval(() => {
                            this['updateResourcesUsage']();
                        }, 0x3e8), this['messenger']['on']('message', _0x1c01e3 => {
                            _0x1c01e3['action'] === 'updateScreenInfo' && (process['stdout']['columns'] = _0x1c01e3['columns'] - 0x4, process['stdout']['rows'] = _0x1c01e3['rows']);
                        });
                    });
                });
            }
            async ['updateResourcesUsage']() {
                this['messenger']['emit']('message', {
                    'action': 'updateResourcesUsage',
                    'cpu': await (0x0, _0x3b5c01['getCpuAverageLoad'])(),
                    'memory': process['memoryUsage']()['heapUsed']
                });
            }
        }
        const _0x115726 = global['__TEST_MAKER__IPC_SLAVE__'], _0x4489fc = global['__TEST_MAKER__IPC_SLAVE__'] = _0x115726 ? _0x115726 : new _0xe4bd85();
    },
    0xb712: (_0x43f585, _0x20070c, _0x7b9dfc) => {
        _0x7b9dfc['r'](_0x20070c), _0x7b9dfc['d'](_0x20070c, { 'getCpuAverageLoad': () => _0xe2b7d5 });
        var _0x2459bb = _0x7b9dfc(0x5615), _0x142472 = _0x7b9dfc['n'](_0x2459bb);
        function _0x25cf84() {
            let _0x4165e9 = 0x0, _0x10652a = 0x0, _0x1b510d = _0x142472()['cpus']();
            for (let _0x118c5d = 0x0, _0x5635e2 = _0x1b510d['length']; _0x118c5d < _0x5635e2; _0x118c5d++) {
                let _0x329491 = _0x1b510d[_0x118c5d];
                for (let _0x553dcb in _0x329491['times']) {
                    _0x10652a += _0x329491['times'][_0x553dcb];
                }
                _0x4165e9 += _0x329491['times']['idle'];
            }
            return {
                'idle': _0x4165e9 / _0x1b510d['length'],
                'total': _0x10652a / _0x1b510d['length']
            };
        }
        const _0x4f62a3 = _0x175400 => {
            if (_0x175400 && _0x175400['length'] >= 0x1) {
                const _0x5d5428 = _0x175400['reduce']((_0x446cdf, _0x1fe0f7) => _0x446cdf + _0x1fe0f7, 0x0);
                return _0x5d5428 / _0x175400['length'];
            } else
                return 0x0;
        };
        function _0xe2b7d5(_0x563e81 = 0x3e8, _0x3b218d = 0x64) {
            return new Promise((_0x9557be, _0x21e689) => {
                const _0xca1d21 = ~~(_0x563e81 / _0x3b218d);
                _0xca1d21 <= 0x1 && _0x21e689(new Error('Error:\x20interval\x20to\x20small'));
                let _0x54207d = 0x0, _0x3b1d23 = [];
                const _0x200a9 = _0x25cf84();
                let _0x375925 = setInterval(() => {
                    _0x54207d >= _0xca1d21 && (clearInterval(_0x375925), _0x9557be(~~(_0x4f62a3(_0x3b1d23) * 0x64)));
                    const _0x3be82e = _0x25cf84(), _0x188642 = _0x3be82e['total'] - _0x200a9['total'], _0xb383b4 = _0x3be82e['idle'] - _0x200a9['idle'];
                    _0x3b1d23[_0x54207d] = 0x1 - _0xb383b4 / _0x188642, _0x54207d++;
                }, _0x3b218d);
            });
        }
    }
};
;
