// test-maker/dist/33.js
'use strict';
exports['id'] = 0x21, exports['ids'] = [0x21], exports['modules'] = {
    0xdae1: (_0x1e10be, _0x33efba, _0x221421) => {
        _0x221421['r'](_0x33efba), _0x221421['d'](_0x33efba, { 'exitTasks': () => _0x2323e5 });
        var _0x3a443e = _0x221421(0x1201f), _0x5a5fc2 = _0x221421(0x7fdd);
        _0x3a443e['internalLogger']['logFile']['enabled'] = !![], _0x3a443e['internalLogger']['logFile']['prefixDate'] = !![], _0x3a443e['internalLogger']['logFile']['prefixLevel'] = !![], _0x3a443e['internalLogger']['overrideConsole'](), _0x3a443e['internalLogger']['hookToStdOut']();
        const _0x2323e5 = new _0x5a5fc2['default']({ 'autoStart': ![] });
    },
    0x84d2: (_0x24cc47, _0x5300d3, _0x5cd9a7) => {
        _0x5cd9a7['r'](_0x5300d3), _0x5cd9a7['d'](_0x5300d3, { 'ansiRegex': () => _0xb21b20 });
        const _0xb21b20 = _0x55457a => {
            const _0x405b7d = [
                '[\x5cu001B\x5cu009B][[\x5c]()#;?]*(?:(?:(?:[a-zA-Z\x5cd]*(?:;[-a-zA-Z\x5cd\x5c/#&.:=?%@~_]*)*)?\x5cu0007)',
                '(?:(?:\x5cd{1,4}(?:;\x5cd{0,4})*)?[\x5cdA-PR-TZcf-ntqry=><~]))'
            ]['join']('|');
            return new RegExp(_0x405b7d, _0x55457a ? undefined : 'g');
        };
    },
    0x14ffb: (_0x55a737, _0xce50e3, _0x570c61) => {
        _0x570c61['r'](_0xce50e3), _0x570c61['d'](_0xce50e3, { 'hasAnsi': () => _0x10617a });
        var _0x3d98d3 = _0x570c61(0x84d2);
        const _0x22a049 = new RegExp((0x0, _0x3d98d3['ansiRegex'])()['source']);
        function _0x10617a(_0x46917f) {
            return _0x22a049['test'](_0x46917f);
        }
    },
    0x20db: (_0x168057, _0x1a57fb, _0x1dd53d) => {
        _0x1dd53d['r'](_0x1a57fb), _0x1dd53d['d'](_0x1a57fb, { 'hookToStdOut': () => _0x1247ce });
        function _0x1247ce({
            callback: _0x357fd1,
            takeControl: _0x374fa5,
            canOutput: canOutput = !![]
        }) {
            const _0xca6b4b = process['stdout']['write']['bind'](process['stdout']), _0x3265d0 = process['stderr']['write']['bind'](process['stderr']);
            return process['stdout']['write'] = (_0x73e403, _0x5cb9f3, _0x2a70fe) => {
                const _0x307487 = typeof canOutput === 'function' ? canOutput(_0x73e403) : canOutput;
                if (_0x374fa5)
                    _0x357fd1({
                        'chunk': _0x73e403,
                        'encoding': _0x5cb9f3,
                        'cb': _0x2a70fe,
                        'isError': ![],
                        'stdOutWriter': _0xca6b4b
                    });
                else {
                    _0x357fd1({
                        'chunk': _0x73e403,
                        'encoding': _0x5cb9f3,
                        'cb': _0x2a70fe,
                        'isError': ![],
                        'stdOutWriter': _0xca6b4b
                    });
                    if (_0x307487)
                        _0xca6b4b(_0x73e403, _0x5cb9f3, _0x2a70fe);
                }
            }, process['stderr']['write'] = (_0x45feba, _0x5b570d, _0x335028) => {
                const _0x410ba8 = typeof canOutput === 'function' ? canOutput(_0x45feba) : canOutput;
                if (_0x374fa5)
                    _0x357fd1({
                        'chunk': _0x45feba,
                        'encoding': _0x5b570d,
                        'isError': !![],
                        'stdOutWriter': _0x3265d0
                    });
                else {
                    _0x357fd1({
                        'chunk': _0x45feba,
                        'encoding': _0x5b570d,
                        'isError': !![],
                        'stdOutWriter': _0x3265d0
                    });
                    if (_0x410ba8)
                        _0x3265d0(_0x45feba, _0x5b570d, _0x335028);
                }
            }, () => {
                process['stdout']['write'] = _0xca6b4b, process['stderr']['write'] = _0x3265d0;
            };
        }
    },
    0x9205: (_0x4e3387, _0x166ef1, _0x2c0de3) => {
        _0x2c0de3['r'](_0x166ef1), _0x2c0de3['d'](_0x166ef1, { 'stripAnsi': () => _0x2d38b5 });
        var _0x5760d6 = _0x2c0de3(0x84d2);
        const _0x2d38b5 = _0x53cf71 => typeof _0x53cf71 === 'string' ? _0x53cf71['replace']((0x0, _0x5760d6['ansiRegex'])(), '') : _0x53cf71;
    },
    0x1201f: (_0x4a6126, _0x2318cc, _0x20f9f8) => {
        _0x20f9f8['r'](_0x2318cc), _0x20f9f8['d'](_0x2318cc, {
            'InternalLogger': () => _0x9787d5,
            'internalLogger': () => _0x41aa1a
        });
        var _0x379692 = _0x20f9f8(0x1145c), _0x51604a = _0x20f9f8['n'](_0x379692), _0x132c0d = _0x20f9f8(0x10ce7), _0x2fc923 = _0x20f9f8(0x17fbe), _0x4ad77d = _0x20f9f8['n'](_0x2fc923), _0x132b07 = _0x20f9f8(0x3b1), _0x433edd = _0x20f9f8['n'](_0x132b07), _0x1f9bf7 = _0x20f9f8(0x1206d), _0x27ef79 = _0x20f9f8['n'](_0x1f9bf7), _0x1b43b6 = _0x20f9f8(0xbb05), _0x55c0ce = _0x20f9f8(0x14ffb), _0xbbb075 = _0x20f9f8(0x20db), _0x4c7faf = _0x20f9f8(0x9205);
        const _0x26173f = console['log'], _0x22dd91 = console['info'], _0x2acee5 = console['warn'], _0x433217 = console['error'], _0x374aca = console['trace'], _0x36c614 = console['debug'];
        class _0x9787d5 {
            constructor() {
                this['enabled'] = !![], this['level'] = 'all', this['logFile'] = {
                    'enabled': ![],
                    'name': '',
                    'prefixDate': ![],
                    'prefixLevel': ![]
                }, this['isHooked'] = ![], this['theme'] = {
                    'trace': _0x51604a()['reset'],
                    'debug': _0x51604a()['reset'],
                    'info': _0x51604a()['cyanBright'],
                    'warn': _0x51604a()['yellowBright'],
                    'error': _0x51604a()['redBright'],
                    'critical': _0x51604a()['redBright'],
                    'fatal': _0x51604a()['redBright']
                }, this['logLevels'] = [
                    'all',
                    'trace',
                    'debug',
                    'info',
                    'log',
                    'warn',
                    'error',
                    'critical',
                    'fatal',
                    'always'
                ];
            }
            ['overrideConsole']() {
                console['log'] = (..._0x15af6e) => {
                    this['log']['apply'](this, _0x15af6e);
                }, console['info'] = (..._0x2fe6df) => {
                    this['info']['apply'](this, _0x2fe6df);
                }, console['warn'] = (..._0x33df84) => {
                    this['warn']['apply'](this, _0x33df84);
                }, console['error'] = (..._0x4acf35) => {
                    this['error']['apply'](this, _0x4acf35);
                }, console['trace'] = (..._0x1bbedc) => {
                    this['trace']['apply'](this, _0x1bbedc);
                }, console['debug'] = (..._0x457308) => {
                    this['debug']['apply'](this, _0x457308);
                };
            }
            ['restoreConsole']() {
                console['log'] = _0x26173f, console['info'] = _0x22dd91, console['warn'] = _0x2acee5, console['error'] = _0x433217, console['trace'] = _0x374aca, console['debug'] = _0x36c614;
            }
            ['trace'](..._0x41f525) {
                this['stdOut']('trace', ..._0x41f525);
            }
            ['debug'](..._0x48671a) {
                this['stdOut']('debug', ..._0x48671a);
            }
            ['info'](..._0x8a580d) {
                this['stdOut']('info', ..._0x8a580d);
            }
            ['log'](..._0x4db98b) {
                this['stdOut']('log', ..._0x4db98b);
            }
            ['warn'](..._0x169e9d) {
                this['stdError']('warn', ..._0x169e9d);
            }
            ['error'](..._0x1e67b3) {
                this['stdError']('error', ..._0x1e67b3);
            }
            ['critical'](..._0x3fa4b7) {
                this['stdError']('critical', ..._0x3fa4b7);
            }
            ['fatal'](..._0x235403) {
                this['stdError']('fatal', ..._0x235403);
            }
            ['always'](..._0x255712) {
                this['stdOut']('always', ..._0x255712);
            }
            ['hookToStdOut']() {
                !this['isHooked'] && (this['isHooked'] = !![], this['createLogFile'](), (0x0, _0xbbb075['hookToStdOut'])({
                    'callback': _0x5bfb97 => {
                        if (!this['stdOutWriter'])
                            this['stdOutWriter'] = _0x5bfb97['stdErrorWriter'];
                        if (!this['stdErrorWriter'])
                            this['stdErrorWriter'] = _0x5bfb97['stdErrorWriter'];
                        if (!this['enabled'])
                            return;
                        let _0x195bae = _0x5bfb97['chunk'];
                        if (_0x5bfb97['isError'] && _0x5bfb97['stdErrorWriter'])
                            _0x5bfb97['stdErrorWriter'](_0x195bae, _0x5bfb97['encoding'], _0x5bfb97['cb']);
                        else
                            _0x5bfb97['stdOutWriter'] && _0x5bfb97['stdOutWriter'](_0x195bae, _0x5bfb97['encoding'], _0x5bfb97['cb']);
                    },
                    'takeControl': !![],
                    'canOutput': () => !![]
                }));
            }
            ['stdOut'](_0x40afd6, ..._0xdcd79e) {
                var _0x5c4eb1;
                _0xdcd79e['unshift'](_0x40afd6);
                if (this['canLog'](_0x40afd6)) {
                    const _0x42ff87 = this['format']['apply'](this, _0xdcd79e);
                    this['stdOutWriter'] ? this['stdOutWriter'](_0x42ff87, 'utf8') : process['stdout']['write'](_0x42ff87), this['logFile']['enabled'] && ((_0x5c4eb1 = this['fileStream']) === null || _0x5c4eb1 === void 0x0 ? void 0x0 : _0x5c4eb1['write'](this['formatWriteToFile'](_0x40afd6, _0x42ff87)));
                }
            }
            ['stdError'](_0x553b20, ..._0x3c46af) {
                var _0x27c613;
                _0x3c46af['unshift'](_0x553b20);
                if (this['canLog'](_0x553b20)) {
                    const _0x1cd040 = this['format']['apply'](this, _0x3c46af);
                    this['stdErrorWriter'] ? this['stdErrorWriter'](_0x1cd040, 'utf8') : process['stderr']['write'](_0x1cd040), this['logFile']['enabled'] && ((_0x27c613 = this['fileStream']) === null || _0x27c613 === void 0x0 ? void 0x0 : _0x27c613['write'](this['formatWriteToFile'](_0x553b20, _0x1cd040)));
                }
            }
            ['canLog'](_0x2e63af) {
                if (this['level'] === 'silent')
                    return ![];
                return _0x2e63af === 'always' || _0x2e63af === 'all' ? !![] : this['logLevels']['indexOf'](_0x2e63af) >= this['logLevels']['indexOf'](this['level']);
            }
            ['format'](..._0x38f949) {
                const _0x10c3ac = _0x38f949['shift']();
                let _0x39df2e = _0x1f9bf7['format']['apply'](this, _0x38f949) + '\x0a';
                if (this['theme'][_0x10c3ac])
                    _0x39df2e = this['theme'][_0x10c3ac](_0x39df2e);
                return _0x39df2e;
            }
            ['formatWriteToFile'](_0x943f6, _0x3511c3) {
                let _0x31fd45 = '', _0x454471 = '', _0x264ed1 = _0x3511c3 ? (0x0, _0x55c0ce['hasAnsi'])(_0x3511c3) ? (0x0, _0x4c7faf['stripAnsi'])(_0x3511c3) : _0x3511c3 : '';
                if (!_0x264ed1)
                    return _0x264ed1;
                if (this['logFile']['prefixLevel']) {
                    if (_0x943f6 !== 'always' && _0x943f6 !== 'log')
                        _0x31fd45 = '[' + _0x943f6 + ']\x20';
                }
                if (this['logFile']['prefixDate'])
                    _0x454471 = '[' + new Date()['toISOString']() + ']';
                if (_0x31fd45 || _0x454471)
                    _0x264ed1 = '' + _0x31fd45 + _0x454471 + '\x0a' + _0x264ed1;
                return _0x264ed1;
            }
            ['createLogFile']() {
                if (!this['fileStream']) {
                    let _0x42a00d = this['logFile']['name'] ? this['logFile']['name'] : _0x1b43b6['testMakerArtifactsPath'] + '/logs/test-maker-' + (0x0, _0x132c0d['format'])(new Date(), 'DD-MM-YYYY\x20hh-mm-ss\x20A') + '.txt';
                    _0x42a00d = _0x433edd()['normalizeTrim'](_0x42a00d), (0x0, _0x2fc923['outputFileSync'])(_0x42a00d, ''), this['fileStream'] = (0x0, _0x2fc923['createWriteStream'])(_0x42a00d, { 'flags': 'w' });
                }
            }
        }
        const _0x434e99 = global[_0x1b43b6['globalTestMakerLogMangerName']], _0x41aa1a = global[_0x1b43b6['globalTestMakerLogMangerName']] = _0x434e99 ? _0x434e99 : new _0x9787d5();
    }
};
;
