// test-maker/dist/386.js
exports['id'] = 0x182, exports['ids'] = [0x182], exports['modules'] = {
    0x7d39: (_0x1b614c, _0x3e1dad, _0x9f99d3) => {
        'use strict';
        _0x1b614c = _0x9f99d3['nmd'](_0x1b614c);
        const _0x5cd291 = (_0x25dff7, _0x5b01cc) => (..._0x5606eb) => {
                const _0x1e7b09 = _0x25dff7(..._0x5606eb);
                return '\x1b[' + (_0x1e7b09 + _0x5b01cc) + 'm';
            }, _0x17e023 = (_0x2892ec, _0xe5f11) => (..._0x47fa68) => {
                const _0x5c1107 = _0x2892ec(..._0x47fa68);
                return '\x1b[' + (0x26 + _0xe5f11) + ';5;' + _0x5c1107 + 'm';
            }, _0x42d0d2 = (_0x224b69, _0x45fc0d) => (..._0xe20948) => {
                const _0x35c2c9 = _0x224b69(..._0xe20948);
                return '\x1b[' + (0x26 + _0x45fc0d) + ';2;' + _0x35c2c9[0x0] + ';' + _0x35c2c9[0x1] + ';' + _0x35c2c9[0x2] + 'm';
            }, _0x41f3e6 = _0x19729c => _0x19729c, _0x31970e = (_0x41405d, _0x3c17e0, _0x2c9851) => [
                _0x41405d,
                _0x3c17e0,
                _0x2c9851
            ], _0x20d5c8 = (_0x2e3273, _0x5ee9c4, _0x5f2c77) => {
                Object['defineProperty'](_0x2e3273, _0x5ee9c4, {
                    'get': () => {
                        const _0x5b7553 = _0x5f2c77();
                        return Object['defineProperty'](_0x2e3273, _0x5ee9c4, {
                            'value': _0x5b7553,
                            'enumerable': !![],
                            'configurable': !![]
                        }), _0x5b7553;
                    },
                    'enumerable': !![],
                    'configurable': !![]
                });
            };
        let _0x476f40;
        const _0xc123c9 = (_0x5e41a1, _0x17fb55, _0x183f7f, _0x2d9256) => {
            _0x476f40 === undefined && (_0x476f40 = _0x9f99d3(0xad3));
            const _0xe35dfb = _0x2d9256 ? 0xa : 0x0, _0x3f9ae2 = {};
            for (const [_0x6890c9, _0x1a01a7] of Object['entries'](_0x476f40)) {
                const _0x152a0d = _0x6890c9 === 'ansi16' ? 'ansi' : _0x6890c9;
                if (_0x6890c9 === _0x17fb55)
                    _0x3f9ae2[_0x152a0d] = _0x5e41a1(_0x183f7f, _0xe35dfb);
                else
                    typeof _0x1a01a7 === 'object' && (_0x3f9ae2[_0x152a0d] = _0x5e41a1(_0x1a01a7[_0x17fb55], _0xe35dfb));
            }
            return _0x3f9ae2;
        };
        function _0x38e6c4() {
            const _0x3c316e = new Map(), _0x2e6306 = {
                    'modifier': {
                        'reset': [
                            0x0,
                            0x0
                        ],
                        'bold': [
                            0x1,
                            0x16
                        ],
                        'dim': [
                            0x2,
                            0x16
                        ],
                        'italic': [
                            0x3,
                            0x17
                        ],
                        'underline': [
                            0x4,
                            0x18
                        ],
                        'inverse': [
                            0x7,
                            0x1b
                        ],
                        'hidden': [
                            0x8,
                            0x1c
                        ],
                        'strikethrough': [
                            0x9,
                            0x1d
                        ]
                    },
                    'color': {
                        'black': [
                            0x1e,
                            0x27
                        ],
                        'red': [
                            0x1f,
                            0x27
                        ],
                        'green': [
                            0x20,
                            0x27
                        ],
                        'yellow': [
                            0x21,
                            0x27
                        ],
                        'blue': [
                            0x22,
                            0x27
                        ],
                        'magenta': [
                            0x23,
                            0x27
                        ],
                        'cyan': [
                            0x24,
                            0x27
                        ],
                        'white': [
                            0x25,
                            0x27
                        ],
                        'blackBright': [
                            0x5a,
                            0x27
                        ],
                        'redBright': [
                            0x5b,
                            0x27
                        ],
                        'greenBright': [
                            0x5c,
                            0x27
                        ],
                        'yellowBright': [
                            0x5d,
                            0x27
                        ],
                        'blueBright': [
                            0x5e,
                            0x27
                        ],
                        'magentaBright': [
                            0x5f,
                            0x27
                        ],
                        'cyanBright': [
                            0x60,
                            0x27
                        ],
                        'whiteBright': [
                            0x61,
                            0x27
                        ]
                    },
                    'bgColor': {
                        'bgBlack': [
                            0x28,
                            0x31
                        ],
                        'bgRed': [
                            0x29,
                            0x31
                        ],
                        'bgGreen': [
                            0x2a,
                            0x31
                        ],
                        'bgYellow': [
                            0x2b,
                            0x31
                        ],
                        'bgBlue': [
                            0x2c,
                            0x31
                        ],
                        'bgMagenta': [
                            0x2d,
                            0x31
                        ],
                        'bgCyan': [
                            0x2e,
                            0x31
                        ],
                        'bgWhite': [
                            0x2f,
                            0x31
                        ],
                        'bgBlackBright': [
                            0x64,
                            0x31
                        ],
                        'bgRedBright': [
                            0x65,
                            0x31
                        ],
                        'bgGreenBright': [
                            0x66,
                            0x31
                        ],
                        'bgYellowBright': [
                            0x67,
                            0x31
                        ],
                        'bgBlueBright': [
                            0x68,
                            0x31
                        ],
                        'bgMagentaBright': [
                            0x69,
                            0x31
                        ],
                        'bgCyanBright': [
                            0x6a,
                            0x31
                        ],
                        'bgWhiteBright': [
                            0x6b,
                            0x31
                        ]
                    }
                };
            _0x2e6306['color']['gray'] = _0x2e6306['color']['blackBright'], _0x2e6306['bgColor']['bgGray'] = _0x2e6306['bgColor']['bgBlackBright'], _0x2e6306['color']['grey'] = _0x2e6306['color']['blackBright'], _0x2e6306['bgColor']['bgGrey'] = _0x2e6306['bgColor']['bgBlackBright'];
            for (const [_0x44e25f, _0x9ac080] of Object['entries'](_0x2e6306)) {
                for (const [_0x15ce04, _0x11add4] of Object['entries'](_0x9ac080)) {
                    _0x2e6306[_0x15ce04] = {
                        'open': '\x1b[' + _0x11add4[0x0] + 'm',
                        'close': '\x1b[' + _0x11add4[0x1] + 'm'
                    }, _0x9ac080[_0x15ce04] = _0x2e6306[_0x15ce04], _0x3c316e['set'](_0x11add4[0x0], _0x11add4[0x1]);
                }
                Object['defineProperty'](_0x2e6306, _0x44e25f, {
                    'value': _0x9ac080,
                    'enumerable': ![]
                });
            }
            return Object['defineProperty'](_0x2e6306, 'codes', {
                'value': _0x3c316e,
                'enumerable': ![]
            }), _0x2e6306['color']['close'] = '\x1b[39m', _0x2e6306['bgColor']['close'] = '\x1b[49m', _0x20d5c8(_0x2e6306['color'], 'ansi', () => _0xc123c9(_0x5cd291, 'ansi16', _0x41f3e6, ![])), _0x20d5c8(_0x2e6306['color'], 'ansi256', () => _0xc123c9(_0x17e023, 'ansi256', _0x41f3e6, ![])), _0x20d5c8(_0x2e6306['color'], 'ansi16m', () => _0xc123c9(_0x42d0d2, 'rgb', _0x31970e, ![])), _0x20d5c8(_0x2e6306['bgColor'], 'ansi', () => _0xc123c9(_0x5cd291, 'ansi16', _0x41f3e6, !![])), _0x20d5c8(_0x2e6306['bgColor'], 'ansi256', () => _0xc123c9(_0x17e023, 'ansi256', _0x41f3e6, !![])), _0x20d5c8(_0x2e6306['bgColor'], 'ansi16m', () => _0xc123c9(_0x42d0d2, 'rgb', _0x31970e, !![])), _0x2e6306;
        }
        Object['defineProperty'](_0x1b614c, 'exports', {
            'enumerable': !![],
            'get': _0x38e6c4
        });
    },
    0x1145c: (_0x16fc02, _0x5c7ffc, _0x5c0e49) => {
        'use strict';
        const _0x49fb03 = _0x5c0e49(0x7d39), {
                stdout: _0x27e5e5,
                stderr: _0x50d0b4
            } = _0x5c0e49(0x9583), {
                stringReplaceAll: _0x38ddbd,
                stringEncaseCRLFWithFirstIndex: _0x503371
            } = _0x5c0e49(0x2d), {isArray: _0x2d9258} = Array, _0x16ba1d = [
                'ansi',
                'ansi',
                'ansi256',
                'ansi16m'
            ], _0x29e5bb = Object['create'](null), _0x12f468 = (_0x2ff9d3, _0x5b9828 = {}) => {
                if (_0x5b9828['level'] && !(Number['isInteger'](_0x5b9828['level']) && _0x5b9828['level'] >= 0x0 && _0x5b9828['level'] <= 0x3))
                    throw new Error('The\x20`level`\x20option\x20should\x20be\x20an\x20integer\x20from\x200\x20to\x203');
                const _0x3875e1 = _0x27e5e5 ? _0x27e5e5['level'] : 0x0;
                _0x2ff9d3['level'] = _0x5b9828['level'] === undefined ? _0x3875e1 : _0x5b9828['level'];
            };
        class _0xe33b37 {
            constructor(_0x5a4e9d) {
                return _0x37d059(_0x5a4e9d);
            }
        }
        const _0x37d059 = _0x5b13f0 => {
            const _0x21602a = {};
            return _0x12f468(_0x21602a, _0x5b13f0), _0x21602a['template'] = (..._0x536985) => _0x4c2dd9(_0x21602a['template'], ..._0x536985), Object['setPrototypeOf'](_0x21602a, _0x11d9b1['prototype']), Object['setPrototypeOf'](_0x21602a['template'], _0x21602a), _0x21602a['template']['constructor'] = () => {
                throw new Error('`chalk.constructor()`\x20is\x20deprecated.\x20Use\x20`new\x20chalk.Instance()`\x20instead.');
            }, _0x21602a['template']['Instance'] = _0xe33b37, _0x21602a['template'];
        };
        function _0x11d9b1(_0x3fe641) {
            return _0x37d059(_0x3fe641);
        }
        for (const [_0xb7d8dd, _0x2f4842] of Object['entries'](_0x49fb03)) {
            _0x29e5bb[_0xb7d8dd] = {
                'get'() {
                    const _0x5688e7 = _0x2f5fc0(this, _0x233965(_0x2f4842['open'], _0x2f4842['close'], this['_styler']), this['_isEmpty']);
                    return Object['defineProperty'](this, _0xb7d8dd, { 'value': _0x5688e7 }), _0x5688e7;
                }
            };
        }
        _0x29e5bb['visible'] = {
            'get'() {
                const _0x511c2d = _0x2f5fc0(this, this['_styler'], !![]);
                return Object['defineProperty'](this, 'visible', { 'value': _0x511c2d }), _0x511c2d;
            }
        };
        const _0x3a988e = [
            'rgb',
            'hex',
            'keyword',
            'hsl',
            'hsv',
            'hwb',
            'ansi',
            'ansi256'
        ];
        for (const _0x4f0c38 of _0x3a988e) {
            _0x29e5bb[_0x4f0c38] = {
                'get'() {
                    const {level: _0xe8cf31} = this;
                    return function (..._0x5668eb) {
                        const _0x9fe1ac = _0x233965(_0x49fb03['color'][_0x16ba1d[_0xe8cf31]][_0x4f0c38](..._0x5668eb), _0x49fb03['color']['close'], this['_styler']);
                        return _0x2f5fc0(this, _0x9fe1ac, this['_isEmpty']);
                    };
                }
            };
        }
        for (const _0x166097 of _0x3a988e) {
            const _0x27bdf6 = 'bg' + _0x166097[0x0]['toUpperCase']() + _0x166097['slice'](0x1);
            _0x29e5bb[_0x27bdf6] = {
                'get'() {
                    const {level: _0xb99e0a} = this;
                    return function (..._0x4c3aa2) {
                        const _0x594a11 = _0x233965(_0x49fb03['bgColor'][_0x16ba1d[_0xb99e0a]][_0x166097](..._0x4c3aa2), _0x49fb03['bgColor']['close'], this['_styler']);
                        return _0x2f5fc0(this, _0x594a11, this['_isEmpty']);
                    };
                }
            };
        }
        const _0x37b5f3 = Object['defineProperties'](() => {
            }, {
                ..._0x29e5bb,
                'level': {
                    'enumerable': !![],
                    'get'() {
                        return this['_generator']['level'];
                    },
                    'set'(_0x853152) {
                        this['_generator']['level'] = _0x853152;
                    }
                }
            }), _0x233965 = (_0x3f6cc5, _0x1927b6, _0x572b49) => {
                let _0x521448, _0x50b72d;
                return _0x572b49 === undefined ? (_0x521448 = _0x3f6cc5, _0x50b72d = _0x1927b6) : (_0x521448 = _0x572b49['openAll'] + _0x3f6cc5, _0x50b72d = _0x1927b6 + _0x572b49['closeAll']), {
                    'open': _0x3f6cc5,
                    'close': _0x1927b6,
                    'openAll': _0x521448,
                    'closeAll': _0x50b72d,
                    'parent': _0x572b49
                };
            }, _0x2f5fc0 = (_0x578a92, _0x24e456, _0x4c9eb0) => {
                const _0x5a0eef = (..._0x1c152f) => {
                    if (_0x2d9258(_0x1c152f[0x0]) && _0x2d9258(_0x1c152f[0x0]['raw']))
                        return _0x2eec59(_0x5a0eef, _0x4c2dd9(_0x5a0eef, ..._0x1c152f));
                    return _0x2eec59(_0x5a0eef, _0x1c152f['length'] === 0x1 ? '' + _0x1c152f[0x0] : _0x1c152f['join']('\x20'));
                };
                return Object['setPrototypeOf'](_0x5a0eef, _0x37b5f3), _0x5a0eef['_generator'] = _0x578a92, _0x5a0eef['_styler'] = _0x24e456, _0x5a0eef['_isEmpty'] = _0x4c9eb0, _0x5a0eef;
            }, _0x2eec59 = (_0x157da7, _0x1a832e) => {
                if (_0x157da7['level'] <= 0x0 || !_0x1a832e)
                    return _0x157da7['_isEmpty'] ? '' : _0x1a832e;
                let _0x446ed5 = _0x157da7['_styler'];
                if (_0x446ed5 === undefined)
                    return _0x1a832e;
                const {
                    openAll: _0x296117,
                    closeAll: _0x5b87c7
                } = _0x446ed5;
                if (_0x1a832e['indexOf']('\x1b') !== -0x1)
                    while (_0x446ed5 !== undefined) {
                        _0x1a832e = _0x38ddbd(_0x1a832e, _0x446ed5['close'], _0x446ed5['open']), _0x446ed5 = _0x446ed5['parent'];
                    }
                const _0x6e1b97 = _0x1a832e['indexOf']('\x0a');
                return _0x6e1b97 !== -0x1 && (_0x1a832e = _0x503371(_0x1a832e, _0x5b87c7, _0x296117, _0x6e1b97)), _0x296117 + _0x1a832e + _0x5b87c7;
            };
        let _0x18ec14;
        const _0x4c2dd9 = (_0x30b809, ..._0x282bd6) => {
            const [_0x24aaaa] = _0x282bd6;
            if (!_0x2d9258(_0x24aaaa) || !_0x2d9258(_0x24aaaa['raw']))
                return _0x282bd6['join']('\x20');
            const _0x46d0c2 = _0x282bd6['slice'](0x1), _0x1ec736 = [_0x24aaaa['raw'][0x0]];
            for (let _0x4c07d4 = 0x1; _0x4c07d4 < _0x24aaaa['length']; _0x4c07d4++) {
                _0x1ec736['push'](String(_0x46d0c2[_0x4c07d4 - 0x1])['replace'](/[{}\\]/g, '\x5c$&'), String(_0x24aaaa['raw'][_0x4c07d4]));
            }
            return _0x18ec14 === undefined && (_0x18ec14 = _0x5c0e49(0xee76)), _0x18ec14(_0x30b809, _0x1ec736['join'](''));
        };
        Object['defineProperties'](_0x11d9b1['prototype'], _0x29e5bb);
        const _0x1c32a7 = _0x11d9b1();
        _0x1c32a7['supportsColor'] = _0x27e5e5, _0x1c32a7['stderr'] = _0x11d9b1({ 'level': _0x50d0b4 ? _0x50d0b4['level'] : 0x0 }), _0x1c32a7['stderr']['supportsColor'] = _0x50d0b4, _0x16fc02['exports'] = _0x1c32a7;
    },
    0xee76: _0x24fe79 => {
        'use strict';
        const _0x2ec3c3 = /(?:\\(u(?:[a-f\d]{4}|\{[a-f\d]{1,6}\})|x[a-f\d]{2}|.))|(?:\{(~)?(\w+(?:\([^)]*\))?(?:\.\w+(?:\([^)]*\))?)*)(?:[ \t]|(?=\r?\n)))|(\})|((?:.|[\r\n\f])+?)/gi, _0x39cf83 = /(?:^|\.)(\w+)(?:\(([^)]*)\))?/g, _0x1e83ac = /^(['"])((?:\\.|(?!\1)[^\\])*)\1$/, _0x3961a4 = /\\(u(?:[a-f\d]{4}|{[a-f\d]{1,6}})|x[a-f\d]{2}|.)|([^\\])/gi, _0x4946d2 = new Map([
                [
                    'n',
                    '\x0a'
                ],
                [
                    'r',
                    '\x0d'
                ],
                [
                    't',
                    '\x09'
                ],
                [
                    'b',
                    '\x08'
                ],
                [
                    'f',
                    '\x0c'
                ],
                [
                    'v',
                    '\x0b'
                ],
                [
                    '0',
                    '\x00'
                ],
                [
                    '\x5c',
                    '\x5c'
                ],
                [
                    'e',
                    '\x1b'
                ],
                [
                    'a',
                    '\x07'
                ]
            ]);
        function _0x54a021(_0x5ed446) {
            const _0x17e3f7 = _0x5ed446[0x0] === 'u', _0x22da0c = _0x5ed446[0x1] === '{';
            if (_0x17e3f7 && !_0x22da0c && _0x5ed446['length'] === 0x5 || _0x5ed446[0x0] === 'x' && _0x5ed446['length'] === 0x3)
                return String['fromCharCode'](parseInt(_0x5ed446['slice'](0x1), 0x10));
            if (_0x17e3f7 && _0x22da0c)
                return String['fromCodePoint'](parseInt(_0x5ed446['slice'](0x2, -0x1), 0x10));
            return _0x4946d2['get'](_0x5ed446) || _0x5ed446;
        }
        function _0x2f5353(_0xd00f34, _0xb22fd8) {
            const _0x4baa51 = [], _0x268139 = _0xb22fd8['trim']()['split'](/\s*,\s*/g);
            let _0x3b2ced;
            for (const _0x441667 of _0x268139) {
                const _0x378171 = Number(_0x441667);
                if (!Number['isNaN'](_0x378171))
                    _0x4baa51['push'](_0x378171);
                else {
                    if (_0x3b2ced = _0x441667['match'](_0x1e83ac))
                        _0x4baa51['push'](_0x3b2ced[0x2]['replace'](_0x3961a4, (_0x17303e, _0x2049cc, _0x85d99b) => _0x2049cc ? _0x54a021(_0x2049cc) : _0x85d99b));
                    else
                        throw new Error('Invalid\x20Chalk\x20template\x20style\x20argument:\x20' + _0x441667 + '\x20(in\x20style\x20\x27' + _0xd00f34 + '\x27)');
                }
            }
            return _0x4baa51;
        }
        function _0x5d1493(_0xd7ce8d) {
            _0x39cf83['lastIndex'] = 0x0;
            const _0x1dfe5c = [];
            let _0x3d51d3;
            while ((_0x3d51d3 = _0x39cf83['exec'](_0xd7ce8d)) !== null) {
                const _0x5d194d = _0x3d51d3[0x1];
                if (_0x3d51d3[0x2]) {
                    const _0x30775d = _0x2f5353(_0x5d194d, _0x3d51d3[0x2]);
                    _0x1dfe5c['push']([_0x5d194d]['concat'](_0x30775d));
                } else
                    _0x1dfe5c['push']([_0x5d194d]);
            }
            return _0x1dfe5c;
        }
        function _0x21d4ca(_0x31edc6, _0x285fb8) {
            const _0x36322d = {};
            for (const _0x600229 of _0x285fb8) {
                for (const _0x33ed34 of _0x600229['styles']) {
                    _0x36322d[_0x33ed34[0x0]] = _0x600229['inverse'] ? null : _0x33ed34['slice'](0x1);
                }
            }
            let _0x2a8c49 = _0x31edc6;
            for (const [_0x5a754b, _0x45e721] of Object['entries'](_0x36322d)) {
                if (!Array['isArray'](_0x45e721))
                    continue;
                if (!(_0x5a754b in _0x2a8c49))
                    throw new Error('Unknown\x20Chalk\x20style:\x20' + _0x5a754b);
                _0x2a8c49 = _0x45e721['length'] > 0x0 ? _0x2a8c49[_0x5a754b](..._0x45e721) : _0x2a8c49[_0x5a754b];
            }
            return _0x2a8c49;
        }
        _0x24fe79['exports'] = (_0x491c15, _0x3cb28f) => {
            const _0x2b42e3 = [], _0x3e14d4 = [];
            let _0x451f7a = [];
            _0x3cb28f['replace'](_0x2ec3c3, (_0x464d8e, _0x15e6c0, _0xcc7906, _0x5c77e1, _0x546aab, _0x242233) => {
                if (_0x15e6c0)
                    _0x451f7a['push'](_0x54a021(_0x15e6c0));
                else {
                    if (_0x5c77e1) {
                        const _0x55e5e6 = _0x451f7a['join']('');
                        _0x451f7a = [], _0x3e14d4['push'](_0x2b42e3['length'] === 0x0 ? _0x55e5e6 : _0x21d4ca(_0x491c15, _0x2b42e3)(_0x55e5e6)), _0x2b42e3['push']({
                            'inverse': _0xcc7906,
                            'styles': _0x5d1493(_0x5c77e1)
                        });
                    } else {
                        if (_0x546aab) {
                            if (_0x2b42e3['length'] === 0x0)
                                throw new Error('Found\x20extraneous\x20}\x20in\x20Chalk\x20template\x20literal');
                            _0x3e14d4['push'](_0x21d4ca(_0x491c15, _0x2b42e3)(_0x451f7a['join'](''))), _0x451f7a = [], _0x2b42e3['pop']();
                        } else
                            _0x451f7a['push'](_0x242233);
                    }
                }
            }), _0x3e14d4['push'](_0x451f7a['join'](''));
            if (_0x2b42e3['length'] > 0x0) {
                const _0x4aca8d = 'Chalk\x20template\x20literal\x20is\x20missing\x20' + _0x2b42e3['length'] + '\x20closing\x20bracket' + (_0x2b42e3['length'] === 0x1 ? '' : 's') + '\x20(`}`)';
                throw new Error(_0x4aca8d);
            }
            return _0x3e14d4['join']('');
        };
    },
    0x2d: _0x5cfd86 => {
        'use strict';
        const _0x540e0d = (_0x5888ae, _0xe2b7a0, _0x572ec1) => {
                let _0x4ff8fd = _0x5888ae['indexOf'](_0xe2b7a0);
                if (_0x4ff8fd === -0x1)
                    return _0x5888ae;
                const _0x2dd967 = _0xe2b7a0['length'];
                let _0x23b4ae = 0x0, _0x3f23de = '';
                do {
                    _0x3f23de += _0x5888ae['substr'](_0x23b4ae, _0x4ff8fd - _0x23b4ae) + _0xe2b7a0 + _0x572ec1, _0x23b4ae = _0x4ff8fd + _0x2dd967, _0x4ff8fd = _0x5888ae['indexOf'](_0xe2b7a0, _0x23b4ae);
                } while (_0x4ff8fd !== -0x1);
                return _0x3f23de += _0x5888ae['substr'](_0x23b4ae), _0x3f23de;
            }, _0x39fee5 = (_0x408607, _0x175d41, _0x4a797f, _0xa4698b) => {
                let _0xebae0 = 0x0, _0x2592e7 = '';
                do {
                    const _0xca50ef = _0x408607[_0xa4698b - 0x1] === '\x0d';
                    _0x2592e7 += _0x408607['substr'](_0xebae0, (_0xca50ef ? _0xa4698b - 0x1 : _0xa4698b) - _0xebae0) + _0x175d41 + (_0xca50ef ? '\x0d\x0a' : '\x0a') + _0x4a797f, _0xebae0 = _0xa4698b + 0x1, _0xa4698b = _0x408607['indexOf']('\x0a', _0xebae0);
                } while (_0xa4698b !== -0x1);
                return _0x2592e7 += _0x408607['substr'](_0xebae0), _0x2592e7;
            };
        _0x5cfd86['exports'] = {
            'stringReplaceAll': _0x540e0d,
            'stringEncaseCRLFWithFirstIndex': _0x39fee5
        };
    },
    0x13ed: (_0x2e5119, _0x245a25, _0x3ca43c) => {
        const _0x2ddfb5 = _0x3ca43c(0x1845b), _0x4f0f11 = {};
        for (const _0xed891f of Object['keys'](_0x2ddfb5)) {
            _0x4f0f11[_0x2ddfb5[_0xed891f]] = _0xed891f;
        }
        const _0x2678cf = {
            'rgb': {
                'channels': 0x3,
                'labels': 'rgb'
            },
            'hsl': {
                'channels': 0x3,
                'labels': 'hsl'
            },
            'hsv': {
                'channels': 0x3,
                'labels': 'hsv'
            },
            'hwb': {
                'channels': 0x3,
                'labels': 'hwb'
            },
            'cmyk': {
                'channels': 0x4,
                'labels': 'cmyk'
            },
            'xyz': {
                'channels': 0x3,
                'labels': 'xyz'
            },
            'lab': {
                'channels': 0x3,
                'labels': 'lab'
            },
            'lch': {
                'channels': 0x3,
                'labels': 'lch'
            },
            'hex': {
                'channels': 0x1,
                'labels': ['hex']
            },
            'keyword': {
                'channels': 0x1,
                'labels': ['keyword']
            },
            'ansi16': {
                'channels': 0x1,
                'labels': ['ansi16']
            },
            'ansi256': {
                'channels': 0x1,
                'labels': ['ansi256']
            },
            'hcg': {
                'channels': 0x3,
                'labels': [
                    'h',
                    'c',
                    'g'
                ]
            },
            'apple': {
                'channels': 0x3,
                'labels': [
                    'r16',
                    'g16',
                    'b16'
                ]
            },
            'gray': {
                'channels': 0x1,
                'labels': ['gray']
            }
        };
        _0x2e5119['exports'] = _0x2678cf;
        for (const _0x1af274 of Object['keys'](_0x2678cf)) {
            if (!('channels' in _0x2678cf[_0x1af274]))
                throw new Error('missing\x20channels\x20property:\x20' + _0x1af274);
            if (!('labels' in _0x2678cf[_0x1af274]))
                throw new Error('missing\x20channel\x20labels\x20property:\x20' + _0x1af274);
            if (_0x2678cf[_0x1af274]['labels']['length'] !== _0x2678cf[_0x1af274]['channels'])
                throw new Error('channel\x20and\x20label\x20counts\x20mismatch:\x20' + _0x1af274);
            const {
                channels: _0x9223ed,
                labels: _0x3e3f91
            } = _0x2678cf[_0x1af274];
            delete _0x2678cf[_0x1af274]['channels'], delete _0x2678cf[_0x1af274]['labels'], Object['defineProperty'](_0x2678cf[_0x1af274], 'channels', { 'value': _0x9223ed }), Object['defineProperty'](_0x2678cf[_0x1af274], 'labels', { 'value': _0x3e3f91 });
        }
        _0x2678cf['rgb']['hsl'] = function (_0x44ac24) {
            const _0x5d3d24 = _0x44ac24[0x0] / 0xff, _0x4d9b5a = _0x44ac24[0x1] / 0xff, _0x4656b5 = _0x44ac24[0x2] / 0xff, _0x2e5cde = Math['min'](_0x5d3d24, _0x4d9b5a, _0x4656b5), _0x5a93b4 = Math['max'](_0x5d3d24, _0x4d9b5a, _0x4656b5), _0x2fcc80 = _0x5a93b4 - _0x2e5cde;
            let _0x498a94, _0x22bf66;
            if (_0x5a93b4 === _0x2e5cde)
                _0x498a94 = 0x0;
            else {
                if (_0x5d3d24 === _0x5a93b4)
                    _0x498a94 = (_0x4d9b5a - _0x4656b5) / _0x2fcc80;
                else {
                    if (_0x4d9b5a === _0x5a93b4)
                        _0x498a94 = 0x2 + (_0x4656b5 - _0x5d3d24) / _0x2fcc80;
                    else
                        _0x4656b5 === _0x5a93b4 && (_0x498a94 = 0x4 + (_0x5d3d24 - _0x4d9b5a) / _0x2fcc80);
                }
            }
            _0x498a94 = Math['min'](_0x498a94 * 0x3c, 0x168);
            _0x498a94 < 0x0 && (_0x498a94 += 0x168);
            const _0x1876bc = (_0x2e5cde + _0x5a93b4) / 0x2;
            if (_0x5a93b4 === _0x2e5cde)
                _0x22bf66 = 0x0;
            else
                _0x1876bc <= 0.5 ? _0x22bf66 = _0x2fcc80 / (_0x5a93b4 + _0x2e5cde) : _0x22bf66 = _0x2fcc80 / (0x2 - _0x5a93b4 - _0x2e5cde);
            return [
                _0x498a94,
                _0x22bf66 * 0x64,
                _0x1876bc * 0x64
            ];
        }, _0x2678cf['rgb']['hsv'] = function (_0x42b5c8) {
            let _0x50c69f, _0x3f0896, _0x5423d7, _0x1f3b9b, _0x1d47ff;
            const _0x1cc62a = _0x42b5c8[0x0] / 0xff, _0x1aab17 = _0x42b5c8[0x1] / 0xff, _0x4b4ef0 = _0x42b5c8[0x2] / 0xff, _0x29b76a = Math['max'](_0x1cc62a, _0x1aab17, _0x4b4ef0), _0xbe38d4 = _0x29b76a - Math['min'](_0x1cc62a, _0x1aab17, _0x4b4ef0), _0x77814b = function (_0x226d67) {
                    return (_0x29b76a - _0x226d67) / 0x6 / _0xbe38d4 + 0x1 / 0x2;
                };
            if (_0xbe38d4 === 0x0)
                _0x1f3b9b = 0x0, _0x1d47ff = 0x0;
            else {
                _0x1d47ff = _0xbe38d4 / _0x29b76a, _0x50c69f = _0x77814b(_0x1cc62a), _0x3f0896 = _0x77814b(_0x1aab17), _0x5423d7 = _0x77814b(_0x4b4ef0);
                if (_0x1cc62a === _0x29b76a)
                    _0x1f3b9b = _0x5423d7 - _0x3f0896;
                else {
                    if (_0x1aab17 === _0x29b76a)
                        _0x1f3b9b = 0x1 / 0x3 + _0x50c69f - _0x5423d7;
                    else
                        _0x4b4ef0 === _0x29b76a && (_0x1f3b9b = 0x2 / 0x3 + _0x3f0896 - _0x50c69f);
                }
                if (_0x1f3b9b < 0x0)
                    _0x1f3b9b += 0x1;
                else
                    _0x1f3b9b > 0x1 && (_0x1f3b9b -= 0x1);
            }
            return [
                _0x1f3b9b * 0x168,
                _0x1d47ff * 0x64,
                _0x29b76a * 0x64
            ];
        }, _0x2678cf['rgb']['hwb'] = function (_0x17fe3a) {
            const _0x892693 = _0x17fe3a[0x0], _0x149b1e = _0x17fe3a[0x1];
            let _0x2c7ea1 = _0x17fe3a[0x2];
            const _0x202a26 = _0x2678cf['rgb']['hsl'](_0x17fe3a)[0x0], _0x19db3a = 0x1 / 0xff * Math['min'](_0x892693, Math['min'](_0x149b1e, _0x2c7ea1));
            return _0x2c7ea1 = 0x1 - 0x1 / 0xff * Math['max'](_0x892693, Math['max'](_0x149b1e, _0x2c7ea1)), [
                _0x202a26,
                _0x19db3a * 0x64,
                _0x2c7ea1 * 0x64
            ];
        }, _0x2678cf['rgb']['cmyk'] = function (_0x736b84) {
            const _0x200201 = _0x736b84[0x0] / 0xff, _0x376ff5 = _0x736b84[0x1] / 0xff, _0x3cc5cf = _0x736b84[0x2] / 0xff, _0xddef1c = Math['min'](0x1 - _0x200201, 0x1 - _0x376ff5, 0x1 - _0x3cc5cf), _0x25bfe0 = (0x1 - _0x200201 - _0xddef1c) / (0x1 - _0xddef1c) || 0x0, _0x1a9998 = (0x1 - _0x376ff5 - _0xddef1c) / (0x1 - _0xddef1c) || 0x0, _0x302f75 = (0x1 - _0x3cc5cf - _0xddef1c) / (0x1 - _0xddef1c) || 0x0;
            return [
                _0x25bfe0 * 0x64,
                _0x1a9998 * 0x64,
                _0x302f75 * 0x64,
                _0xddef1c * 0x64
            ];
        };
        function _0x2402d8(_0x51cabc, _0x5b8dea) {
            return (_0x51cabc[0x0] - _0x5b8dea[0x0]) ** 0x2 + (_0x51cabc[0x1] - _0x5b8dea[0x1]) ** 0x2 + (_0x51cabc[0x2] - _0x5b8dea[0x2]) ** 0x2;
        }
        _0x2678cf['rgb']['keyword'] = function (_0x3f67cf) {
            const _0x59961f = _0x4f0f11[_0x3f67cf];
            if (_0x59961f)
                return _0x59961f;
            let _0x7d3d13 = Infinity, _0x1cd5aa;
            for (const _0x531377 of Object['keys'](_0x2ddfb5)) {
                const _0xe8d844 = _0x2ddfb5[_0x531377], _0x1e05f2 = _0x2402d8(_0x3f67cf, _0xe8d844);
                _0x1e05f2 < _0x7d3d13 && (_0x7d3d13 = _0x1e05f2, _0x1cd5aa = _0x531377);
            }
            return _0x1cd5aa;
        }, _0x2678cf['keyword']['rgb'] = function (_0x47c9e4) {
            return _0x2ddfb5[_0x47c9e4];
        }, _0x2678cf['rgb']['xyz'] = function (_0x14fc84) {
            let _0x493dc9 = _0x14fc84[0x0] / 0xff, _0xc162fe = _0x14fc84[0x1] / 0xff, _0x1e4ae1 = _0x14fc84[0x2] / 0xff;
            _0x493dc9 = _0x493dc9 > 0.04045 ? ((_0x493dc9 + 0.055) / 1.055) ** 2.4 : _0x493dc9 / 12.92, _0xc162fe = _0xc162fe > 0.04045 ? ((_0xc162fe + 0.055) / 1.055) ** 2.4 : _0xc162fe / 12.92, _0x1e4ae1 = _0x1e4ae1 > 0.04045 ? ((_0x1e4ae1 + 0.055) / 1.055) ** 2.4 : _0x1e4ae1 / 12.92;
            const _0x577b4c = _0x493dc9 * 0.4124 + _0xc162fe * 0.3576 + _0x1e4ae1 * 0.1805, _0x4630ad = _0x493dc9 * 0.2126 + _0xc162fe * 0.7152 + _0x1e4ae1 * 0.0722, _0x541450 = _0x493dc9 * 0.0193 + _0xc162fe * 0.1192 + _0x1e4ae1 * 0.9505;
            return [
                _0x577b4c * 0x64,
                _0x4630ad * 0x64,
                _0x541450 * 0x64
            ];
        }, _0x2678cf['rgb']['lab'] = function (_0x35c891) {
            const _0x14c47b = _0x2678cf['rgb']['xyz'](_0x35c891);
            let _0x4fb478 = _0x14c47b[0x0], _0x57471f = _0x14c47b[0x1], _0x4986c4 = _0x14c47b[0x2];
            _0x4fb478 /= 95.047, _0x57471f /= 0x64, _0x4986c4 /= 108.883, _0x4fb478 = _0x4fb478 > 0.008856 ? _0x4fb478 ** (0x1 / 0x3) : 7.787 * _0x4fb478 + 0x10 / 0x74, _0x57471f = _0x57471f > 0.008856 ? _0x57471f ** (0x1 / 0x3) : 7.787 * _0x57471f + 0x10 / 0x74, _0x4986c4 = _0x4986c4 > 0.008856 ? _0x4986c4 ** (0x1 / 0x3) : 7.787 * _0x4986c4 + 0x10 / 0x74;
            const _0x3c9fa5 = 0x74 * _0x57471f - 0x10, _0x4e7558 = 0x1f4 * (_0x4fb478 - _0x57471f), _0x108808 = 0xc8 * (_0x57471f - _0x4986c4);
            return [
                _0x3c9fa5,
                _0x4e7558,
                _0x108808
            ];
        }, _0x2678cf['hsl']['rgb'] = function (_0x45f113) {
            const _0xa88d39 = _0x45f113[0x0] / 0x168, _0x2b1fab = _0x45f113[0x1] / 0x64, _0x43389e = _0x45f113[0x2] / 0x64;
            let _0x5d68cd, _0x12dba6, _0x53a35f;
            if (_0x2b1fab === 0x0)
                return _0x53a35f = _0x43389e * 0xff, [
                    _0x53a35f,
                    _0x53a35f,
                    _0x53a35f
                ];
            _0x43389e < 0.5 ? _0x5d68cd = _0x43389e * (0x1 + _0x2b1fab) : _0x5d68cd = _0x43389e + _0x2b1fab - _0x43389e * _0x2b1fab;
            const _0x1f7aba = 0x2 * _0x43389e - _0x5d68cd, _0x6b18b8 = [
                    0x0,
                    0x0,
                    0x0
                ];
            for (let _0x5ef77e = 0x0; _0x5ef77e < 0x3; _0x5ef77e++) {
                _0x12dba6 = _0xa88d39 + 0x1 / 0x3 * -(_0x5ef77e - 0x1);
                _0x12dba6 < 0x0 && _0x12dba6++;
                _0x12dba6 > 0x1 && _0x12dba6--;
                if (0x6 * _0x12dba6 < 0x1)
                    _0x53a35f = _0x1f7aba + (_0x5d68cd - _0x1f7aba) * 0x6 * _0x12dba6;
                else {
                    if (0x2 * _0x12dba6 < 0x1)
                        _0x53a35f = _0x5d68cd;
                    else
                        0x3 * _0x12dba6 < 0x2 ? _0x53a35f = _0x1f7aba + (_0x5d68cd - _0x1f7aba) * (0x2 / 0x3 - _0x12dba6) * 0x6 : _0x53a35f = _0x1f7aba;
                }
                _0x6b18b8[_0x5ef77e] = _0x53a35f * 0xff;
            }
            return _0x6b18b8;
        }, _0x2678cf['hsl']['hsv'] = function (_0x1e6854) {
            const _0x5b0b8f = _0x1e6854[0x0];
            let _0x51341f = _0x1e6854[0x1] / 0x64, _0x585ab5 = _0x1e6854[0x2] / 0x64, _0x3a99a4 = _0x51341f;
            const _0x44ae87 = Math['max'](_0x585ab5, 0.01);
            _0x585ab5 *= 0x2, _0x51341f *= _0x585ab5 <= 0x1 ? _0x585ab5 : 0x2 - _0x585ab5, _0x3a99a4 *= _0x44ae87 <= 0x1 ? _0x44ae87 : 0x2 - _0x44ae87;
            const _0x124474 = (_0x585ab5 + _0x51341f) / 0x2, _0x1cdbe = _0x585ab5 === 0x0 ? 0x2 * _0x3a99a4 / (_0x44ae87 + _0x3a99a4) : 0x2 * _0x51341f / (_0x585ab5 + _0x51341f);
            return [
                _0x5b0b8f,
                _0x1cdbe * 0x64,
                _0x124474 * 0x64
            ];
        }, _0x2678cf['hsv']['rgb'] = function (_0x3ef158) {
            const _0x1b847b = _0x3ef158[0x0] / 0x3c, _0x473580 = _0x3ef158[0x1] / 0x64;
            let _0x49a621 = _0x3ef158[0x2] / 0x64;
            const _0x1c4121 = Math['floor'](_0x1b847b) % 0x6, _0x558ddf = _0x1b847b - Math['floor'](_0x1b847b), _0xa4fe7c = 0xff * _0x49a621 * (0x1 - _0x473580), _0x2c9d70 = 0xff * _0x49a621 * (0x1 - _0x473580 * _0x558ddf), _0x2d91f5 = 0xff * _0x49a621 * (0x1 - _0x473580 * (0x1 - _0x558ddf));
            _0x49a621 *= 0xff;
            switch (_0x1c4121) {
            case 0x0:
                return [
                    _0x49a621,
                    _0x2d91f5,
                    _0xa4fe7c
                ];
            case 0x1:
                return [
                    _0x2c9d70,
                    _0x49a621,
                    _0xa4fe7c
                ];
            case 0x2:
                return [
                    _0xa4fe7c,
                    _0x49a621,
                    _0x2d91f5
                ];
            case 0x3:
                return [
                    _0xa4fe7c,
                    _0x2c9d70,
                    _0x49a621
                ];
            case 0x4:
                return [
                    _0x2d91f5,
                    _0xa4fe7c,
                    _0x49a621
                ];
            case 0x5:
                return [
                    _0x49a621,
                    _0xa4fe7c,
                    _0x2c9d70
                ];
            }
        }, _0x2678cf['hsv']['hsl'] = function (_0x65b8c) {
            const _0x31c190 = _0x65b8c[0x0], _0x46204c = _0x65b8c[0x1] / 0x64, _0x2ca14a = _0x65b8c[0x2] / 0x64, _0xd37bc2 = Math['max'](_0x2ca14a, 0.01);
            let _0x157a72, _0x368e8a;
            _0x368e8a = (0x2 - _0x46204c) * _0x2ca14a;
            const _0x5c3a39 = (0x2 - _0x46204c) * _0xd37bc2;
            return _0x157a72 = _0x46204c * _0xd37bc2, _0x157a72 /= _0x5c3a39 <= 0x1 ? _0x5c3a39 : 0x2 - _0x5c3a39, _0x157a72 = _0x157a72 || 0x0, _0x368e8a /= 0x2, [
                _0x31c190,
                _0x157a72 * 0x64,
                _0x368e8a * 0x64
            ];
        }, _0x2678cf['hwb']['rgb'] = function (_0x2272ed) {
            const _0x182f7e = _0x2272ed[0x0] / 0x168;
            let _0x139f7d = _0x2272ed[0x1] / 0x64, _0x2e6e53 = _0x2272ed[0x2] / 0x64;
            const _0x394285 = _0x139f7d + _0x2e6e53;
            let _0x27d51d;
            _0x394285 > 0x1 && (_0x139f7d /= _0x394285, _0x2e6e53 /= _0x394285);
            const _0x122d60 = Math['floor'](0x6 * _0x182f7e), _0x37a63c = 0x1 - _0x2e6e53;
            _0x27d51d = 0x6 * _0x182f7e - _0x122d60;
            (_0x122d60 & 0x1) !== 0x0 && (_0x27d51d = 0x1 - _0x27d51d);
            const _0x561fcc = _0x139f7d + _0x27d51d * (_0x37a63c - _0x139f7d);
            let _0x512c1f, _0xf76233, _0x19768c;
            switch (_0x122d60) {
            default:
            case 0x6:
            case 0x0:
                _0x512c1f = _0x37a63c, _0xf76233 = _0x561fcc, _0x19768c = _0x139f7d;
                break;
            case 0x1:
                _0x512c1f = _0x561fcc, _0xf76233 = _0x37a63c, _0x19768c = _0x139f7d;
                break;
            case 0x2:
                _0x512c1f = _0x139f7d, _0xf76233 = _0x37a63c, _0x19768c = _0x561fcc;
                break;
            case 0x3:
                _0x512c1f = _0x139f7d, _0xf76233 = _0x561fcc, _0x19768c = _0x37a63c;
                break;
            case 0x4:
                _0x512c1f = _0x561fcc, _0xf76233 = _0x139f7d, _0x19768c = _0x37a63c;
                break;
            case 0x5:
                _0x512c1f = _0x37a63c, _0xf76233 = _0x139f7d, _0x19768c = _0x561fcc;
                break;
            }
            return [
                _0x512c1f * 0xff,
                _0xf76233 * 0xff,
                _0x19768c * 0xff
            ];
        }, _0x2678cf['cmyk']['rgb'] = function (_0x584d19) {
            const _0x402714 = _0x584d19[0x0] / 0x64, _0x598651 = _0x584d19[0x1] / 0x64, _0x3db186 = _0x584d19[0x2] / 0x64, _0x4887a9 = _0x584d19[0x3] / 0x64, _0x342523 = 0x1 - Math['min'](0x1, _0x402714 * (0x1 - _0x4887a9) + _0x4887a9), _0x2b08bf = 0x1 - Math['min'](0x1, _0x598651 * (0x1 - _0x4887a9) + _0x4887a9), _0x291b95 = 0x1 - Math['min'](0x1, _0x3db186 * (0x1 - _0x4887a9) + _0x4887a9);
            return [
                _0x342523 * 0xff,
                _0x2b08bf * 0xff,
                _0x291b95 * 0xff
            ];
        }, _0x2678cf['xyz']['rgb'] = function (_0x2857ed) {
            const _0x1cbe0a = _0x2857ed[0x0] / 0x64, _0x36b631 = _0x2857ed[0x1] / 0x64, _0x4f8105 = _0x2857ed[0x2] / 0x64;
            let _0x402c60, _0x47f422, _0x110474;
            return _0x402c60 = _0x1cbe0a * 3.2406 + _0x36b631 * -1.5372 + _0x4f8105 * -0.4986, _0x47f422 = _0x1cbe0a * -0.9689 + _0x36b631 * 1.8758 + _0x4f8105 * 0.0415, _0x110474 = _0x1cbe0a * 0.0557 + _0x36b631 * -0.204 + _0x4f8105 * 1.057, _0x402c60 = _0x402c60 > 0.0031308 ? 1.055 * _0x402c60 ** (0x1 / 2.4) - 0.055 : _0x402c60 * 12.92, _0x47f422 = _0x47f422 > 0.0031308 ? 1.055 * _0x47f422 ** (0x1 / 2.4) - 0.055 : _0x47f422 * 12.92, _0x110474 = _0x110474 > 0.0031308 ? 1.055 * _0x110474 ** (0x1 / 2.4) - 0.055 : _0x110474 * 12.92, _0x402c60 = Math['min'](Math['max'](0x0, _0x402c60), 0x1), _0x47f422 = Math['min'](Math['max'](0x0, _0x47f422), 0x1), _0x110474 = Math['min'](Math['max'](0x0, _0x110474), 0x1), [
                _0x402c60 * 0xff,
                _0x47f422 * 0xff,
                _0x110474 * 0xff
            ];
        }, _0x2678cf['xyz']['lab'] = function (_0x291bef) {
            let _0x3b5e8e = _0x291bef[0x0], _0x1233c5 = _0x291bef[0x1], _0x1761fe = _0x291bef[0x2];
            _0x3b5e8e /= 95.047, _0x1233c5 /= 0x64, _0x1761fe /= 108.883, _0x3b5e8e = _0x3b5e8e > 0.008856 ? _0x3b5e8e ** (0x1 / 0x3) : 7.787 * _0x3b5e8e + 0x10 / 0x74, _0x1233c5 = _0x1233c5 > 0.008856 ? _0x1233c5 ** (0x1 / 0x3) : 7.787 * _0x1233c5 + 0x10 / 0x74, _0x1761fe = _0x1761fe > 0.008856 ? _0x1761fe ** (0x1 / 0x3) : 7.787 * _0x1761fe + 0x10 / 0x74;
            const _0x9cc8e8 = 0x74 * _0x1233c5 - 0x10, _0x365d81 = 0x1f4 * (_0x3b5e8e - _0x1233c5), _0x4575fa = 0xc8 * (_0x1233c5 - _0x1761fe);
            return [
                _0x9cc8e8,
                _0x365d81,
                _0x4575fa
            ];
        }, _0x2678cf['lab']['xyz'] = function (_0x208e28) {
            const _0x5f0701 = _0x208e28[0x0], _0x1cd4af = _0x208e28[0x1], _0x6f5809 = _0x208e28[0x2];
            let _0x2041a9, _0x1574cb, _0x382936;
            _0x1574cb = (_0x5f0701 + 0x10) / 0x74, _0x2041a9 = _0x1cd4af / 0x1f4 + _0x1574cb, _0x382936 = _0x1574cb - _0x6f5809 / 0xc8;
            const _0x547fd4 = _0x1574cb ** 0x3, _0x48132b = _0x2041a9 ** 0x3, _0x3efec3 = _0x382936 ** 0x3;
            return _0x1574cb = _0x547fd4 > 0.008856 ? _0x547fd4 : (_0x1574cb - 0x10 / 0x74) / 7.787, _0x2041a9 = _0x48132b > 0.008856 ? _0x48132b : (_0x2041a9 - 0x10 / 0x74) / 7.787, _0x382936 = _0x3efec3 > 0.008856 ? _0x3efec3 : (_0x382936 - 0x10 / 0x74) / 7.787, _0x2041a9 *= 95.047, _0x1574cb *= 0x64, _0x382936 *= 108.883, [
                _0x2041a9,
                _0x1574cb,
                _0x382936
            ];
        }, _0x2678cf['lab']['lch'] = function (_0x5a2047) {
            const _0x420a1c = _0x5a2047[0x0], _0x124089 = _0x5a2047[0x1], _0x654b1b = _0x5a2047[0x2];
            let _0x1e258b;
            const _0x3a85c8 = Math['atan2'](_0x654b1b, _0x124089);
            _0x1e258b = _0x3a85c8 * 0x168 / 0x2 / Math['PI'];
            _0x1e258b < 0x0 && (_0x1e258b += 0x168);
            const _0x43818e = Math['sqrt'](_0x124089 * _0x124089 + _0x654b1b * _0x654b1b);
            return [
                _0x420a1c,
                _0x43818e,
                _0x1e258b
            ];
        }, _0x2678cf['lch']['lab'] = function (_0x121738) {
            const _0xa33fe8 = _0x121738[0x0], _0x500768 = _0x121738[0x1], _0x1771f8 = _0x121738[0x2], _0x3afbe0 = _0x1771f8 / 0x168 * 0x2 * Math['PI'], _0x2f1880 = _0x500768 * Math['cos'](_0x3afbe0), _0x59d1ab = _0x500768 * Math['sin'](_0x3afbe0);
            return [
                _0xa33fe8,
                _0x2f1880,
                _0x59d1ab
            ];
        }, _0x2678cf['rgb']['ansi16'] = function (_0x35265b, _0x26b854 = null) {
            const [_0x238916, _0x8aab20, _0x27a069] = _0x35265b;
            let _0x38cd30 = _0x26b854 === null ? _0x2678cf['rgb']['hsv'](_0x35265b)[0x2] : _0x26b854;
            _0x38cd30 = Math['round'](_0x38cd30 / 0x32);
            if (_0x38cd30 === 0x0)
                return 0x1e;
            let _0x774783 = 0x1e + (Math['round'](_0x27a069 / 0xff) << 0x2 | Math['round'](_0x8aab20 / 0xff) << 0x1 | Math['round'](_0x238916 / 0xff));
            return _0x38cd30 === 0x2 && (_0x774783 += 0x3c), _0x774783;
        }, _0x2678cf['hsv']['ansi16'] = function (_0x182e75) {
            return _0x2678cf['rgb']['ansi16'](_0x2678cf['hsv']['rgb'](_0x182e75), _0x182e75[0x2]);
        }, _0x2678cf['rgb']['ansi256'] = function (_0x23c822) {
            const _0x49151e = _0x23c822[0x0], _0x2e9f89 = _0x23c822[0x1], _0x207b78 = _0x23c822[0x2];
            if (_0x49151e === _0x2e9f89 && _0x2e9f89 === _0x207b78) {
                if (_0x49151e < 0x8)
                    return 0x10;
                if (_0x49151e > 0xf8)
                    return 0xe7;
                return Math['round']((_0x49151e - 0x8) / 0xf7 * 0x18) + 0xe8;
            }
            const _0x57117c = 0x10 + 0x24 * Math['round'](_0x49151e / 0xff * 0x5) + 0x6 * Math['round'](_0x2e9f89 / 0xff * 0x5) + Math['round'](_0x207b78 / 0xff * 0x5);
            return _0x57117c;
        }, _0x2678cf['ansi16']['rgb'] = function (_0x35f5d6) {
            let _0xa3e8ff = _0x35f5d6 % 0xa;
            if (_0xa3e8ff === 0x0 || _0xa3e8ff === 0x7)
                return _0x35f5d6 > 0x32 && (_0xa3e8ff += 3.5), _0xa3e8ff = _0xa3e8ff / 10.5 * 0xff, [
                    _0xa3e8ff,
                    _0xa3e8ff,
                    _0xa3e8ff
                ];
            const _0x12763c = (~~(_0x35f5d6 > 0x32) + 0x1) * 0.5, _0x30718d = (_0xa3e8ff & 0x1) * _0x12763c * 0xff, _0x1565ba = (_0xa3e8ff >> 0x1 & 0x1) * _0x12763c * 0xff, _0x2afb61 = (_0xa3e8ff >> 0x2 & 0x1) * _0x12763c * 0xff;
            return [
                _0x30718d,
                _0x1565ba,
                _0x2afb61
            ];
        }, _0x2678cf['ansi256']['rgb'] = function (_0x1805d6) {
            if (_0x1805d6 >= 0xe8) {
                const _0x13acaf = (_0x1805d6 - 0xe8) * 0xa + 0x8;
                return [
                    _0x13acaf,
                    _0x13acaf,
                    _0x13acaf
                ];
            }
            _0x1805d6 -= 0x10;
            let _0x5e6508;
            const _0x2d9f5f = Math['floor'](_0x1805d6 / 0x24) / 0x5 * 0xff, _0x73e63b = Math['floor']((_0x5e6508 = _0x1805d6 % 0x24) / 0x6) / 0x5 * 0xff, _0x38e9df = _0x5e6508 % 0x6 / 0x5 * 0xff;
            return [
                _0x2d9f5f,
                _0x73e63b,
                _0x38e9df
            ];
        }, _0x2678cf['rgb']['hex'] = function (_0x2ebff9) {
            const _0x58a309 = ((Math['round'](_0x2ebff9[0x0]) & 0xff) << 0x10) + ((Math['round'](_0x2ebff9[0x1]) & 0xff) << 0x8) + (Math['round'](_0x2ebff9[0x2]) & 0xff), _0x305ac1 = _0x58a309['toString'](0x10)['toUpperCase']();
            return '000000'['substring'](_0x305ac1['length']) + _0x305ac1;
        }, _0x2678cf['hex']['rgb'] = function (_0x3fca3f) {
            const _0x39223e = _0x3fca3f['toString'](0x10)['match'](/[a-f0-9]{6}|[a-f0-9]{3}/i);
            if (!_0x39223e)
                return [
                    0x0,
                    0x0,
                    0x0
                ];
            let _0x3b3b3f = _0x39223e[0x0];
            _0x39223e[0x0]['length'] === 0x3 && (_0x3b3b3f = _0x3b3b3f['split']('')['map'](_0x260052 => {
                return _0x260052 + _0x260052;
            })['join'](''));
            const _0x472d3c = parseInt(_0x3b3b3f, 0x10), _0x5ca4a1 = _0x472d3c >> 0x10 & 0xff, _0x4f1a1b = _0x472d3c >> 0x8 & 0xff, _0x2fb7a4 = _0x472d3c & 0xff;
            return [
                _0x5ca4a1,
                _0x4f1a1b,
                _0x2fb7a4
            ];
        }, _0x2678cf['rgb']['hcg'] = function (_0x1cfb3c) {
            const _0x237b10 = _0x1cfb3c[0x0] / 0xff, _0x41c0bc = _0x1cfb3c[0x1] / 0xff, _0x285335 = _0x1cfb3c[0x2] / 0xff, _0x1cc314 = Math['max'](Math['max'](_0x237b10, _0x41c0bc), _0x285335), _0x5285b3 = Math['min'](Math['min'](_0x237b10, _0x41c0bc), _0x285335), _0x52bd5d = _0x1cc314 - _0x5285b3;
            let _0x41c642, _0xbeb89f;
            _0x52bd5d < 0x1 ? _0x41c642 = _0x5285b3 / (0x1 - _0x52bd5d) : _0x41c642 = 0x0;
            if (_0x52bd5d <= 0x0)
                _0xbeb89f = 0x0;
            else {
                if (_0x1cc314 === _0x237b10)
                    _0xbeb89f = (_0x41c0bc - _0x285335) / _0x52bd5d % 0x6;
                else
                    _0x1cc314 === _0x41c0bc ? _0xbeb89f = 0x2 + (_0x285335 - _0x237b10) / _0x52bd5d : _0xbeb89f = 0x4 + (_0x237b10 - _0x41c0bc) / _0x52bd5d;
            }
            return _0xbeb89f /= 0x6, _0xbeb89f %= 0x1, [
                _0xbeb89f * 0x168,
                _0x52bd5d * 0x64,
                _0x41c642 * 0x64
            ];
        }, _0x2678cf['hsl']['hcg'] = function (_0x392129) {
            const _0x1d11b6 = _0x392129[0x1] / 0x64, _0x5a9bbe = _0x392129[0x2] / 0x64, _0x1393c6 = _0x5a9bbe < 0.5 ? 0x2 * _0x1d11b6 * _0x5a9bbe : 0x2 * _0x1d11b6 * (0x1 - _0x5a9bbe);
            let _0x2f94a7 = 0x0;
            return _0x1393c6 < 0x1 && (_0x2f94a7 = (_0x5a9bbe - 0.5 * _0x1393c6) / (0x1 - _0x1393c6)), [
                _0x392129[0x0],
                _0x1393c6 * 0x64,
                _0x2f94a7 * 0x64
            ];
        }, _0x2678cf['hsv']['hcg'] = function (_0x13b0c3) {
            const _0x4a51db = _0x13b0c3[0x1] / 0x64, _0x199850 = _0x13b0c3[0x2] / 0x64, _0x42cae4 = _0x4a51db * _0x199850;
            let _0x3388bf = 0x0;
            return _0x42cae4 < 0x1 && (_0x3388bf = (_0x199850 - _0x42cae4) / (0x1 - _0x42cae4)), [
                _0x13b0c3[0x0],
                _0x42cae4 * 0x64,
                _0x3388bf * 0x64
            ];
        }, _0x2678cf['hcg']['rgb'] = function (_0x167ee3) {
            const _0x26a5c1 = _0x167ee3[0x0] / 0x168, _0x147358 = _0x167ee3[0x1] / 0x64, _0x3b8c69 = _0x167ee3[0x2] / 0x64;
            if (_0x147358 === 0x0)
                return [
                    _0x3b8c69 * 0xff,
                    _0x3b8c69 * 0xff,
                    _0x3b8c69 * 0xff
                ];
            const _0x2fec50 = [
                    0x0,
                    0x0,
                    0x0
                ], _0x280fa6 = _0x26a5c1 % 0x1 * 0x6, _0x722e0a = _0x280fa6 % 0x1, _0x277f08 = 0x1 - _0x722e0a;
            let _0x13fbd3 = 0x0;
            switch (Math['floor'](_0x280fa6)) {
            case 0x0:
                _0x2fec50[0x0] = 0x1, _0x2fec50[0x1] = _0x722e0a, _0x2fec50[0x2] = 0x0;
                break;
            case 0x1:
                _0x2fec50[0x0] = _0x277f08, _0x2fec50[0x1] = 0x1, _0x2fec50[0x2] = 0x0;
                break;
            case 0x2:
                _0x2fec50[0x0] = 0x0, _0x2fec50[0x1] = 0x1, _0x2fec50[0x2] = _0x722e0a;
                break;
            case 0x3:
                _0x2fec50[0x0] = 0x0, _0x2fec50[0x1] = _0x277f08, _0x2fec50[0x2] = 0x1;
                break;
            case 0x4:
                _0x2fec50[0x0] = _0x722e0a, _0x2fec50[0x1] = 0x0, _0x2fec50[0x2] = 0x1;
                break;
            default:
                _0x2fec50[0x0] = 0x1, _0x2fec50[0x1] = 0x0, _0x2fec50[0x2] = _0x277f08;
            }
            return _0x13fbd3 = (0x1 - _0x147358) * _0x3b8c69, [
                (_0x147358 * _0x2fec50[0x0] + _0x13fbd3) * 0xff,
                (_0x147358 * _0x2fec50[0x1] + _0x13fbd3) * 0xff,
                (_0x147358 * _0x2fec50[0x2] + _0x13fbd3) * 0xff
            ];
        }, _0x2678cf['hcg']['hsv'] = function (_0x89164b) {
            const _0x19086e = _0x89164b[0x1] / 0x64, _0x4d5a4a = _0x89164b[0x2] / 0x64, _0x614497 = _0x19086e + _0x4d5a4a * (0x1 - _0x19086e);
            let _0x5d6c29 = 0x0;
            return _0x614497 > 0x0 && (_0x5d6c29 = _0x19086e / _0x614497), [
                _0x89164b[0x0],
                _0x5d6c29 * 0x64,
                _0x614497 * 0x64
            ];
        }, _0x2678cf['hcg']['hsl'] = function (_0x5b93d6) {
            const _0x3b5f41 = _0x5b93d6[0x1] / 0x64, _0x27ed28 = _0x5b93d6[0x2] / 0x64, _0x4ec32a = _0x27ed28 * (0x1 - _0x3b5f41) + 0.5 * _0x3b5f41;
            let _0x1706b7 = 0x0;
            if (_0x4ec32a > 0x0 && _0x4ec32a < 0.5)
                _0x1706b7 = _0x3b5f41 / (0x2 * _0x4ec32a);
            else
                _0x4ec32a >= 0.5 && _0x4ec32a < 0x1 && (_0x1706b7 = _0x3b5f41 / (0x2 * (0x1 - _0x4ec32a)));
            return [
                _0x5b93d6[0x0],
                _0x1706b7 * 0x64,
                _0x4ec32a * 0x64
            ];
        }, _0x2678cf['hcg']['hwb'] = function (_0x1c0684) {
            const _0x26f904 = _0x1c0684[0x1] / 0x64, _0x43e9e6 = _0x1c0684[0x2] / 0x64, _0x2f1205 = _0x26f904 + _0x43e9e6 * (0x1 - _0x26f904);
            return [
                _0x1c0684[0x0],
                (_0x2f1205 - _0x26f904) * 0x64,
                (0x1 - _0x2f1205) * 0x64
            ];
        }, _0x2678cf['hwb']['hcg'] = function (_0x5178d8) {
            const _0x12fd70 = _0x5178d8[0x1] / 0x64, _0xbb3baa = _0x5178d8[0x2] / 0x64, _0x23eb3d = 0x1 - _0xbb3baa, _0x520a80 = _0x23eb3d - _0x12fd70;
            let _0x4486af = 0x0;
            return _0x520a80 < 0x1 && (_0x4486af = (_0x23eb3d - _0x520a80) / (0x1 - _0x520a80)), [
                _0x5178d8[0x0],
                _0x520a80 * 0x64,
                _0x4486af * 0x64
            ];
        }, _0x2678cf['apple']['rgb'] = function (_0x51cf9c) {
            return [
                _0x51cf9c[0x0] / 0xffff * 0xff,
                _0x51cf9c[0x1] / 0xffff * 0xff,
                _0x51cf9c[0x2] / 0xffff * 0xff
            ];
        }, _0x2678cf['rgb']['apple'] = function (_0x158c1c) {
            return [
                _0x158c1c[0x0] / 0xff * 0xffff,
                _0x158c1c[0x1] / 0xff * 0xffff,
                _0x158c1c[0x2] / 0xff * 0xffff
            ];
        }, _0x2678cf['gray']['rgb'] = function (_0x151079) {
            return [
                _0x151079[0x0] / 0x64 * 0xff,
                _0x151079[0x0] / 0x64 * 0xff,
                _0x151079[0x0] / 0x64 * 0xff
            ];
        }, _0x2678cf['gray']['hsl'] = function (_0x629e66) {
            return [
                0x0,
                0x0,
                _0x629e66[0x0]
            ];
        }, _0x2678cf['gray']['hsv'] = _0x2678cf['gray']['hsl'], _0x2678cf['gray']['hwb'] = function (_0xa66a2f) {
            return [
                0x0,
                0x64,
                _0xa66a2f[0x0]
            ];
        }, _0x2678cf['gray']['cmyk'] = function (_0x55a0a3) {
            return [
                0x0,
                0x0,
                0x0,
                _0x55a0a3[0x0]
            ];
        }, _0x2678cf['gray']['lab'] = function (_0x14b103) {
            return [
                _0x14b103[0x0],
                0x0,
                0x0
            ];
        }, _0x2678cf['gray']['hex'] = function (_0x562b73) {
            const _0x12fb57 = Math['round'](_0x562b73[0x0] / 0x64 * 0xff) & 0xff, _0xe8ba37 = (_0x12fb57 << 0x10) + (_0x12fb57 << 0x8) + _0x12fb57, _0xc025a3 = _0xe8ba37['toString'](0x10)['toUpperCase']();
            return '000000'['substring'](_0xc025a3['length']) + _0xc025a3;
        }, _0x2678cf['rgb']['gray'] = function (_0x3e8a33) {
            const _0x242de6 = (_0x3e8a33[0x0] + _0x3e8a33[0x1] + _0x3e8a33[0x2]) / 0x3;
            return [_0x242de6 / 0xff * 0x64];
        };
    },
    0xad3: (_0x431dcc, _0x3ce339, _0x17353b) => {
        const _0x167fb = _0x17353b(0x13ed), _0x3ac46b = _0x17353b(0xc51), _0x346617 = {}, _0x37fceb = Object['keys'](_0x167fb);
        function _0x3c9c18(_0x3d5910) {
            const _0x4168f6 = function (..._0xf23cf7) {
                const _0x6eb27c = _0xf23cf7[0x0];
                if (_0x6eb27c === undefined || _0x6eb27c === null)
                    return _0x6eb27c;
                return _0x6eb27c['length'] > 0x1 && (_0xf23cf7 = _0x6eb27c), _0x3d5910(_0xf23cf7);
            };
            return 'conversion' in _0x3d5910 && (_0x4168f6['conversion'] = _0x3d5910['conversion']), _0x4168f6;
        }
        function _0x1c1c7b(_0x11c23a) {
            const _0x4eb19e = function (..._0x1879e3) {
                const _0x1d3e97 = _0x1879e3[0x0];
                if (_0x1d3e97 === undefined || _0x1d3e97 === null)
                    return _0x1d3e97;
                _0x1d3e97['length'] > 0x1 && (_0x1879e3 = _0x1d3e97);
                const _0x4cd291 = _0x11c23a(_0x1879e3);
                if (typeof _0x4cd291 === 'object')
                    for (let _0x495374 = _0x4cd291['length'], _0x3799ac = 0x0; _0x3799ac < _0x495374; _0x3799ac++) {
                        _0x4cd291[_0x3799ac] = Math['round'](_0x4cd291[_0x3799ac]);
                    }
                return _0x4cd291;
            };
            return 'conversion' in _0x11c23a && (_0x4eb19e['conversion'] = _0x11c23a['conversion']), _0x4eb19e;
        }
        _0x37fceb['forEach'](_0x4cf8cd => {
            _0x346617[_0x4cf8cd] = {}, Object['defineProperty'](_0x346617[_0x4cf8cd], 'channels', { 'value': _0x167fb[_0x4cf8cd]['channels'] }), Object['defineProperty'](_0x346617[_0x4cf8cd], 'labels', { 'value': _0x167fb[_0x4cf8cd]['labels'] });
            const _0x1a53ae = _0x3ac46b(_0x4cf8cd), _0x1042f3 = Object['keys'](_0x1a53ae);
            _0x1042f3['forEach'](_0x269a3e => {
                const _0x5403f5 = _0x1a53ae[_0x269a3e];
                _0x346617[_0x4cf8cd][_0x269a3e] = _0x1c1c7b(_0x5403f5), _0x346617[_0x4cf8cd][_0x269a3e]['raw'] = _0x3c9c18(_0x5403f5);
            });
        }), _0x431dcc['exports'] = _0x346617;
    },
    0xc51: (_0x38c13c, _0xaca51, _0x212b76) => {
        const _0xf3b89f = _0x212b76(0x13ed);
        function _0x84f9dd() {
            const _0x46076a = {}, _0xb7c020 = Object['keys'](_0xf3b89f);
            for (let _0x2eeb5a = _0xb7c020['length'], _0x527716 = 0x0; _0x527716 < _0x2eeb5a; _0x527716++) {
                _0x46076a[_0xb7c020[_0x527716]] = {
                    'distance': -0x1,
                    'parent': null
                };
            }
            return _0x46076a;
        }
        function _0xf9fd4c(_0x1a89ad) {
            const _0x303a11 = _0x84f9dd(), _0x2fb4cb = [_0x1a89ad];
            _0x303a11[_0x1a89ad]['distance'] = 0x0;
            while (_0x2fb4cb['length']) {
                const _0x204ce9 = _0x2fb4cb['pop'](), _0x175267 = Object['keys'](_0xf3b89f[_0x204ce9]);
                for (let _0x5a00a9 = _0x175267['length'], _0x588c79 = 0x0; _0x588c79 < _0x5a00a9; _0x588c79++) {
                    const _0x370958 = _0x175267[_0x588c79], _0x4c5e7 = _0x303a11[_0x370958];
                    _0x4c5e7['distance'] === -0x1 && (_0x4c5e7['distance'] = _0x303a11[_0x204ce9]['distance'] + 0x1, _0x4c5e7['parent'] = _0x204ce9, _0x2fb4cb['unshift'](_0x370958));
                }
            }
            return _0x303a11;
        }
        function _0x1967ea(_0x3b973e, _0x4d7771) {
            return function (_0xf57936) {
                return _0x4d7771(_0x3b973e(_0xf57936));
            };
        }
        function _0x16d442(_0x572ee8, _0x2137f2) {
            const _0xd453bf = [
                _0x2137f2[_0x572ee8]['parent'],
                _0x572ee8
            ];
            let _0x33b49d = _0xf3b89f[_0x2137f2[_0x572ee8]['parent']][_0x572ee8], _0x3c2dee = _0x2137f2[_0x572ee8]['parent'];
            while (_0x2137f2[_0x3c2dee]['parent']) {
                _0xd453bf['unshift'](_0x2137f2[_0x3c2dee]['parent']), _0x33b49d = _0x1967ea(_0xf3b89f[_0x2137f2[_0x3c2dee]['parent']][_0x3c2dee], _0x33b49d), _0x3c2dee = _0x2137f2[_0x3c2dee]['parent'];
            }
            return _0x33b49d['conversion'] = _0xd453bf, _0x33b49d;
        }
        _0x38c13c['exports'] = function (_0x5a512b) {
            const _0x71ca17 = _0xf9fd4c(_0x5a512b), _0x13eb4d = {}, _0x3bedd7 = Object['keys'](_0x71ca17);
            for (let _0x2cfd40 = _0x3bedd7['length'], _0x3829a1 = 0x0; _0x3829a1 < _0x2cfd40; _0x3829a1++) {
                const _0x43f72d = _0x3bedd7[_0x3829a1], _0x274be8 = _0x71ca17[_0x43f72d];
                if (_0x274be8['parent'] === null)
                    continue;
                _0x13eb4d[_0x43f72d] = _0x16d442(_0x43f72d, _0x71ca17);
            }
            return _0x13eb4d;
        };
    },
    0x1845b: _0x2314fe => {
        'use strict';
        _0x2314fe['exports'] = {
            'aliceblue': [
                0xf0,
                0xf8,
                0xff
            ],
            'antiquewhite': [
                0xfa,
                0xeb,
                0xd7
            ],
            'aqua': [
                0x0,
                0xff,
                0xff
            ],
            'aquamarine': [
                0x7f,
                0xff,
                0xd4
            ],
            'azure': [
                0xf0,
                0xff,
                0xff
            ],
            'beige': [
                0xf5,
                0xf5,
                0xdc
            ],
            'bisque': [
                0xff,
                0xe4,
                0xc4
            ],
            'black': [
                0x0,
                0x0,
                0x0
            ],
            'blanchedalmond': [
                0xff,
                0xeb,
                0xcd
            ],
            'blue': [
                0x0,
                0x0,
                0xff
            ],
            'blueviolet': [
                0x8a,
                0x2b,
                0xe2
            ],
            'brown': [
                0xa5,
                0x2a,
                0x2a
            ],
            'burlywood': [
                0xde,
                0xb8,
                0x87
            ],
            'cadetblue': [
                0x5f,
                0x9e,
                0xa0
            ],
            'chartreuse': [
                0x7f,
                0xff,
                0x0
            ],
            'chocolate': [
                0xd2,
                0x69,
                0x1e
            ],
            'coral': [
                0xff,
                0x7f,
                0x50
            ],
            'cornflowerblue': [
                0x64,
                0x95,
                0xed
            ],
            'cornsilk': [
                0xff,
                0xf8,
                0xdc
            ],
            'crimson': [
                0xdc,
                0x14,
                0x3c
            ],
            'cyan': [
                0x0,
                0xff,
                0xff
            ],
            'darkblue': [
                0x0,
                0x0,
                0x8b
            ],
            'darkcyan': [
                0x0,
                0x8b,
                0x8b
            ],
            'darkgoldenrod': [
                0xb8,
                0x86,
                0xb
            ],
            'darkgray': [
                0xa9,
                0xa9,
                0xa9
            ],
            'darkgreen': [
                0x0,
                0x64,
                0x0
            ],
            'darkgrey': [
                0xa9,
                0xa9,
                0xa9
            ],
            'darkkhaki': [
                0xbd,
                0xb7,
                0x6b
            ],
            'darkmagenta': [
                0x8b,
                0x0,
                0x8b
            ],
            'darkolivegreen': [
                0x55,
                0x6b,
                0x2f
            ],
            'darkorange': [
                0xff,
                0x8c,
                0x0
            ],
            'darkorchid': [
                0x99,
                0x32,
                0xcc
            ],
            'darkred': [
                0x8b,
                0x0,
                0x0
            ],
            'darksalmon': [
                0xe9,
                0x96,
                0x7a
            ],
            'darkseagreen': [
                0x8f,
                0xbc,
                0x8f
            ],
            'darkslateblue': [
                0x48,
                0x3d,
                0x8b
            ],
            'darkslategray': [
                0x2f,
                0x4f,
                0x4f
            ],
            'darkslategrey': [
                0x2f,
                0x4f,
                0x4f
            ],
            'darkturquoise': [
                0x0,
                0xce,
                0xd1
            ],
            'darkviolet': [
                0x94,
                0x0,
                0xd3
            ],
            'deeppink': [
                0xff,
                0x14,
                0x93
            ],
            'deepskyblue': [
                0x0,
                0xbf,
                0xff
            ],
            'dimgray': [
                0x69,
                0x69,
                0x69
            ],
            'dimgrey': [
                0x69,
                0x69,
                0x69
            ],
            'dodgerblue': [
                0x1e,
                0x90,
                0xff
            ],
            'firebrick': [
                0xb2,
                0x22,
                0x22
            ],
            'floralwhite': [
                0xff,
                0xfa,
                0xf0
            ],
            'forestgreen': [
                0x22,
                0x8b,
                0x22
            ],
            'fuchsia': [
                0xff,
                0x0,
                0xff
            ],
            'gainsboro': [
                0xdc,
                0xdc,
                0xdc
            ],
            'ghostwhite': [
                0xf8,
                0xf8,
                0xff
            ],
            'gold': [
                0xff,
                0xd7,
                0x0
            ],
            'goldenrod': [
                0xda,
                0xa5,
                0x20
            ],
            'gray': [
                0x80,
                0x80,
                0x80
            ],
            'green': [
                0x0,
                0x80,
                0x0
            ],
            'greenyellow': [
                0xad,
                0xff,
                0x2f
            ],
            'grey': [
                0x80,
                0x80,
                0x80
            ],
            'honeydew': [
                0xf0,
                0xff,
                0xf0
            ],
            'hotpink': [
                0xff,
                0x69,
                0xb4
            ],
            'indianred': [
                0xcd,
                0x5c,
                0x5c
            ],
            'indigo': [
                0x4b,
                0x0,
                0x82
            ],
            'ivory': [
                0xff,
                0xff,
                0xf0
            ],
            'khaki': [
                0xf0,
                0xe6,
                0x8c
            ],
            'lavender': [
                0xe6,
                0xe6,
                0xfa
            ],
            'lavenderblush': [
                0xff,
                0xf0,
                0xf5
            ],
            'lawngreen': [
                0x7c,
                0xfc,
                0x0
            ],
            'lemonchiffon': [
                0xff,
                0xfa,
                0xcd
            ],
            'lightblue': [
                0xad,
                0xd8,
                0xe6
            ],
            'lightcoral': [
                0xf0,
                0x80,
                0x80
            ],
            'lightcyan': [
                0xe0,
                0xff,
                0xff
            ],
            'lightgoldenrodyellow': [
                0xfa,
                0xfa,
                0xd2
            ],
            'lightgray': [
                0xd3,
                0xd3,
                0xd3
            ],
            'lightgreen': [
                0x90,
                0xee,
                0x90
            ],
            'lightgrey': [
                0xd3,
                0xd3,
                0xd3
            ],
            'lightpink': [
                0xff,
                0xb6,
                0xc1
            ],
            'lightsalmon': [
                0xff,
                0xa0,
                0x7a
            ],
            'lightseagreen': [
                0x20,
                0xb2,
                0xaa
            ],
            'lightskyblue': [
                0x87,
                0xce,
                0xfa
            ],
            'lightslategray': [
                0x77,
                0x88,
                0x99
            ],
            'lightslategrey': [
                0x77,
                0x88,
                0x99
            ],
            'lightsteelblue': [
                0xb0,
                0xc4,
                0xde
            ],
            'lightyellow': [
                0xff,
                0xff,
                0xe0
            ],
            'lime': [
                0x0,
                0xff,
                0x0
            ],
            'limegreen': [
                0x32,
                0xcd,
                0x32
            ],
            'linen': [
                0xfa,
                0xf0,
                0xe6
            ],
            'magenta': [
                0xff,
                0x0,
                0xff
            ],
            'maroon': [
                0x80,
                0x0,
                0x0
            ],
            'mediumaquamarine': [
                0x66,
                0xcd,
                0xaa
            ],
            'mediumblue': [
                0x0,
                0x0,
                0xcd
            ],
            'mediumorchid': [
                0xba,
                0x55,
                0xd3
            ],
            'mediumpurple': [
                0x93,
                0x70,
                0xdb
            ],
            'mediumseagreen': [
                0x3c,
                0xb3,
                0x71
            ],
            'mediumslateblue': [
                0x7b,
                0x68,
                0xee
            ],
            'mediumspringgreen': [
                0x0,
                0xfa,
                0x9a
            ],
            'mediumturquoise': [
                0x48,
                0xd1,
                0xcc
            ],
            'mediumvioletred': [
                0xc7,
                0x15,
                0x85
            ],
            'midnightblue': [
                0x19,
                0x19,
                0x70
            ],
            'mintcream': [
                0xf5,
                0xff,
                0xfa
            ],
            'mistyrose': [
                0xff,
                0xe4,
                0xe1
            ],
            'moccasin': [
                0xff,
                0xe4,
                0xb5
            ],
            'navajowhite': [
                0xff,
                0xde,
                0xad
            ],
            'navy': [
                0x0,
                0x0,
                0x80
            ],
            'oldlace': [
                0xfd,
                0xf5,
                0xe6
            ],
            'olive': [
                0x80,
                0x80,
                0x0
            ],
            'olivedrab': [
                0x6b,
                0x8e,
                0x23
            ],
            'orange': [
                0xff,
                0xa5,
                0x0
            ],
            'orangered': [
                0xff,
                0x45,
                0x0
            ],
            'orchid': [
                0xda,
                0x70,
                0xd6
            ],
            'palegoldenrod': [
                0xee,
                0xe8,
                0xaa
            ],
            'palegreen': [
                0x98,
                0xfb,
                0x98
            ],
            'paleturquoise': [
                0xaf,
                0xee,
                0xee
            ],
            'palevioletred': [
                0xdb,
                0x70,
                0x93
            ],
            'papayawhip': [
                0xff,
                0xef,
                0xd5
            ],
            'peachpuff': [
                0xff,
                0xda,
                0xb9
            ],
            'peru': [
                0xcd,
                0x85,
                0x3f
            ],
            'pink': [
                0xff,
                0xc0,
                0xcb
            ],
            'plum': [
                0xdd,
                0xa0,
                0xdd
            ],
            'powderblue': [
                0xb0,
                0xe0,
                0xe6
            ],
            'purple': [
                0x80,
                0x0,
                0x80
            ],
            'rebeccapurple': [
                0x66,
                0x33,
                0x99
            ],
            'red': [
                0xff,
                0x0,
                0x0
            ],
            'rosybrown': [
                0xbc,
                0x8f,
                0x8f
            ],
            'royalblue': [
                0x41,
                0x69,
                0xe1
            ],
            'saddlebrown': [
                0x8b,
                0x45,
                0x13
            ],
            'salmon': [
                0xfa,
                0x80,
                0x72
            ],
            'sandybrown': [
                0xf4,
                0xa4,
                0x60
            ],
            'seagreen': [
                0x2e,
                0x8b,
                0x57
            ],
            'seashell': [
                0xff,
                0xf5,
                0xee
            ],
            'sienna': [
                0xa0,
                0x52,
                0x2d
            ],
            'silver': [
                0xc0,
                0xc0,
                0xc0
            ],
            'skyblue': [
                0x87,
                0xce,
                0xeb
            ],
            'slateblue': [
                0x6a,
                0x5a,
                0xcd
            ],
            'slategray': [
                0x70,
                0x80,
                0x90
            ],
            'slategrey': [
                0x70,
                0x80,
                0x90
            ],
            'snow': [
                0xff,
                0xfa,
                0xfa
            ],
            'springgreen': [
                0x0,
                0xff,
                0x7f
            ],
            'steelblue': [
                0x46,
                0x82,
                0xb4
            ],
            'tan': [
                0xd2,
                0xb4,
                0x8c
            ],
            'teal': [
                0x0,
                0x80,
                0x80
            ],
            'thistle': [
                0xd8,
                0xbf,
                0xd8
            ],
            'tomato': [
                0xff,
                0x63,
                0x47
            ],
            'turquoise': [
                0x40,
                0xe0,
                0xd0
            ],
            'violet': [
                0xee,
                0x82,
                0xee
            ],
            'wheat': [
                0xf5,
                0xde,
                0xb3
            ],
            'white': [
                0xff,
                0xff,
                0xff
            ],
            'whitesmoke': [
                0xf5,
                0xf5,
                0xf5
            ],
            'yellow': [
                0xff,
                0xff,
                0x0
            ],
            'yellowgreen': [
                0x9a,
                0xcd,
                0x32
            ]
        };
    },
    0x908f: _0x3ed907 => {
        'use strict';
        _0x3ed907['exports'] = (_0x19af0a, _0x44ec7e = process['argv']) => {
            const _0x353266 = _0x19af0a['startsWith']('-') ? '' : _0x19af0a['length'] === 0x1 ? '-' : '--', _0x36ea5e = _0x44ec7e['indexOf'](_0x353266 + _0x19af0a), _0x2070c6 = _0x44ec7e['indexOf']('--');
            return _0x36ea5e !== -0x1 && (_0x2070c6 === -0x1 || _0x36ea5e < _0x2070c6);
        };
    },
    0x9583: (_0x3df1db, _0x11fc02, _0x443015) => {
        'use strict';
        const _0x54bb5e = _0x443015(0x5615), _0x3fd03b = _0x443015(0x129c0), _0x34a452 = _0x443015(0x908f), {env: _0x278b44} = process;
        let _0x1046e8;
        if (_0x34a452('no-color') || _0x34a452('no-colors') || _0x34a452('color=false') || _0x34a452('color=never'))
            _0x1046e8 = 0x0;
        else
            (_0x34a452('color') || _0x34a452('colors') || _0x34a452('color=true') || _0x34a452('color=always')) && (_0x1046e8 = 0x1);
        if ('FORCE_COLOR' in _0x278b44) {
            if (_0x278b44['FORCE_COLOR'] === 'true')
                _0x1046e8 = 0x1;
            else
                _0x278b44['FORCE_COLOR'] === 'false' ? _0x1046e8 = 0x0 : _0x1046e8 = _0x278b44['FORCE_COLOR']['length'] === 0x0 ? 0x1 : Math['min'](parseInt(_0x278b44['FORCE_COLOR'], 0xa), 0x3);
        }
        function _0x342e1f(_0x29e128) {
            if (_0x29e128 === 0x0)
                return ![];
            return {
                'level': _0x29e128,
                'hasBasic': !![],
                'has256': _0x29e128 >= 0x2,
                'has16m': _0x29e128 >= 0x3
            };
        }
        function _0x3dc1a0(_0x5cc6e2, _0x3c90e7) {
            if (_0x1046e8 === 0x0)
                return 0x0;
            if (_0x34a452('color=16m') || _0x34a452('color=full') || _0x34a452('color=truecolor'))
                return 0x3;
            if (_0x34a452('color=256'))
                return 0x2;
            if (_0x5cc6e2 && !_0x3c90e7 && _0x1046e8 === undefined)
                return 0x0;
            const _0x146681 = _0x1046e8 || 0x0;
            if (_0x278b44['TERM'] === 'dumb')
                return _0x146681;
            if (process['platform'] === 'win32') {
                const _0x4ff79a = _0x54bb5e['release']()['split']('.');
                if (Number(_0x4ff79a[0x0]) >= 0xa && Number(_0x4ff79a[0x2]) >= 0x295a)
                    return Number(_0x4ff79a[0x2]) >= 0x3a53 ? 0x3 : 0x2;
                return 0x1;
            }
            if ('CI' in _0x278b44) {
                if ([
                        'TRAVIS',
                        'CIRCLECI',
                        'APPVEYOR',
                        'GITLAB_CI',
                        'GITHUB_ACTIONS',
                        'BUILDKITE'
                    ]['some'](_0x4ce9f4 => _0x4ce9f4 in _0x278b44) || _0x278b44['CI_NAME'] === 'codeship')
                    return 0x1;
                return _0x146681;
            }
            if ('TEAMCITY_VERSION' in _0x278b44)
                return /^(9\.(0*[1-9]\d*)\.|\d{2,}\.)/['test'](_0x278b44['TEAMCITY_VERSION']) ? 0x1 : 0x0;
            if (_0x278b44['COLORTERM'] === 'truecolor')
                return 0x3;
            if ('TERM_PROGRAM' in _0x278b44) {
                const _0x23f424 = parseInt((_0x278b44['TERM_PROGRAM_VERSION'] || '')['split']('.')[0x0], 0xa);
                switch (_0x278b44['TERM_PROGRAM']) {
                case 'iTerm.app':
                    return _0x23f424 >= 0x3 ? 0x3 : 0x2;
                case 'Apple_Terminal':
                    return 0x2;
                }
            }
            if (/-256(color)?$/i['test'](_0x278b44['TERM']))
                return 0x2;
            if (/^screen|^xterm|^vt100|^vt220|^rxvt|color|ansi|cygwin|linux/i['test'](_0x278b44['TERM']))
                return 0x1;
            if ('COLORTERM' in _0x278b44)
                return 0x1;
            return _0x146681;
        }
        function _0xaa30dd(_0x2bf0a8) {
            const _0x114f6f = _0x3dc1a0(_0x2bf0a8, _0x2bf0a8 && _0x2bf0a8['isTTY']);
            return _0x342e1f(_0x114f6f);
        }
        _0x3df1db['exports'] = {
            'supportsColor': _0xaa30dd,
            'stdout': _0x342e1f(_0x3dc1a0(!![], _0x3fd03b['isatty'](0x1))),
            'stderr': _0x342e1f(_0x3dc1a0(!![], _0x3fd03b['isatty'](0x2)))
        };
    },
    0x3b1: (_0x4003be, _0x295967, _0x1aaf35) => {
        var _0x152443 = '2.0.1', _0xc93ec8, _0x466507, _0x329e65, _0x446937, _0xd281e2, _0x11404c, _0x1adc66, _0x20fde9, _0x2e26a8, _0x141eb7, _0x478c7f, _0x173523 = []['slice'], _0x3d5762 = []['indexOf'] || function (_0x4a43cf) {
                for (var _0x231c9d = 0x0, _0x360b91 = this['length']; _0x231c9d < _0x360b91; _0x231c9d++) {
                    if (_0x231c9d in this && this[_0x231c9d] === _0x4a43cf)
                        return _0x231c9d;
                }
                return -0x1;
            }, _0x39929c = {}['hasOwnProperty'];
        _0x1adc66 = _0x1aaf35(0x11569), _0x329e65 = function (_0x2e9dc8) {
            return typeof _0x2e9dc8 === 'function';
        }, _0x446937 = function (_0x1a8e64) {
            return typeof _0x1a8e64 === 'string' || !!_0x1a8e64 && typeof _0x1a8e64 === 'object' && Object['prototype']['toString']['call'](_0x1a8e64) === '[object\x20String]';
        }, _0x478c7f = _0x295967, _0x478c7f['VERSION'] = typeof _0x152443 !== 'undefined' && _0x152443 !== null ? _0x152443 : 'NO-VERSION', _0x141eb7 = function (_0x231244) {
            return _0x231244 = _0x231244['replace'](/\\/g, '/'), _0x231244 = _0x231244['replace'](/(?<!^)\/+/g, '/'), _0x231244;
        };
        for (_0x20fde9 in _0x1adc66) {
            _0x2e26a8 = _0x1adc66[_0x20fde9], _0x329e65(_0x2e26a8) ? _0x478c7f[_0x20fde9] = function (_0x74d91b) {
                return function () {
                    var _0x3b9ea0, _0x53d7c9;
                    return _0x3b9ea0 = 0x1 <= arguments['length'] ? _0x173523['call'](arguments, 0x0) : [], _0x3b9ea0 = _0x3b9ea0['map'](function (_0x315df8) {
                        return _0x446937(_0x315df8) ? _0x141eb7(_0x315df8) : _0x315df8;
                    }), _0x53d7c9 = _0x1adc66[_0x74d91b]['apply'](_0x1adc66, _0x3b9ea0), _0x446937(_0x53d7c9) ? _0x141eb7(_0x53d7c9) : _0x53d7c9;
                };
            }(_0x20fde9) : _0x478c7f[_0x20fde9] = _0x2e26a8;
        }
        _0x478c7f['sep'] = '/', _0x466507 = {
            'toUnix': _0x141eb7,
            'normalizeSafe': function (_0x505c0e) {
                var _0x47f142;
                _0x505c0e = _0x141eb7(_0x505c0e), _0x47f142 = _0x478c7f['normalize'](_0x505c0e);
                if (_0x505c0e['startsWith']('./') && !_0x47f142['startsWith']('./') && !_0x47f142['startsWith']('..'))
                    _0x47f142 = './' + _0x47f142;
                else
                    _0x505c0e['startsWith']('//') && !_0x47f142['startsWith']('//') && (_0x505c0e['startsWith']('//./') ? _0x47f142 = '//.' + _0x47f142 : _0x47f142 = '/' + _0x47f142);
                return _0x47f142;
            },
            'normalizeTrim': function (_0x282684) {
                return _0x282684 = _0x478c7f['normalizeSafe'](_0x282684), _0x282684['endsWith']('/') ? _0x282684['slice'](0x0, +(_0x282684['length'] - 0x2) + 0x1 || 0x218711a00) : _0x282684;
            },
            'joinSafe': function () {
                var _0x17bed7, _0x46f2c3, _0x4b3d11;
                _0x17bed7 = 0x1 <= arguments['length'] ? _0x173523['call'](arguments, 0x0) : [], _0x4b3d11 = _0x478c7f['join']['apply'](null, _0x17bed7);
                if (_0x17bed7['length'] > 0x0) {
                    _0x46f2c3 = _0x141eb7(_0x17bed7[0x0]);
                    if (_0x46f2c3['startsWith']('./') && !_0x4b3d11['startsWith']('./') && !_0x4b3d11['startsWith']('..'))
                        _0x4b3d11 = './' + _0x4b3d11;
                    else
                        _0x46f2c3['startsWith']('//') && !_0x4b3d11['startsWith']('//') && (_0x46f2c3['startsWith']('//./') ? _0x4b3d11 = '//.' + _0x4b3d11 : _0x4b3d11 = '/' + _0x4b3d11);
                }
                return _0x4b3d11;
            },
            'addExt': function (_0x59eb9c, _0x5aff3d) {
                return !_0x5aff3d ? _0x59eb9c : (_0x5aff3d[0x0] !== '.' && (_0x5aff3d = '.' + _0x5aff3d), _0x59eb9c + (_0x59eb9c['endsWith'](_0x5aff3d) ? '' : _0x5aff3d));
            },
            'trimExt': function (_0x4c3d37, _0x3cb219, _0xc764bd) {
                var _0x346660;
                return _0xc764bd == null && (_0xc764bd = 0x7), _0x346660 = _0x478c7f['extname'](_0x4c3d37), _0xd281e2(_0x346660, _0x3cb219, _0xc764bd) ? _0x4c3d37['slice'](0x0, +(_0x4c3d37['length'] - _0x346660['length'] - 0x1) + 0x1 || 0x218711a00) : _0x4c3d37;
            },
            'removeExt': function (_0x2f37d7, _0x26c093) {
                return !_0x26c093 ? _0x2f37d7 : (_0x26c093 = _0x26c093[0x0] === '.' ? _0x26c093 : '.' + _0x26c093, _0x478c7f['extname'](_0x2f37d7) === _0x26c093 ? _0x478c7f['trimExt'](_0x2f37d7, [], _0x26c093['length']) : _0x2f37d7);
            },
            'changeExt': function (_0x1db4a8, _0x37a505, _0x579953, _0x386c33) {
                return _0x386c33 == null && (_0x386c33 = 0x7), _0x478c7f['trimExt'](_0x1db4a8, _0x579953, _0x386c33) + (!_0x37a505 ? '' : _0x37a505[0x0] === '.' ? _0x37a505 : '.' + _0x37a505);
            },
            'defaultExt': function (_0x51fcd1, _0x584df7, _0x480105, _0x19616f) {
                var _0x472e87;
                return _0x19616f == null && (_0x19616f = 0x7), _0x472e87 = _0x478c7f['extname'](_0x51fcd1), _0xd281e2(_0x472e87, _0x480105, _0x19616f) ? _0x51fcd1 : _0x478c7f['addExt'](_0x51fcd1, _0x584df7);
            }
        }, _0xd281e2 = function (_0x3fca21, _0x57cd8e, _0x5c7692) {
            return _0x57cd8e == null && (_0x57cd8e = []), _0x3fca21 && _0x3fca21['length'] <= _0x5c7692 && _0x3d5762['call'](_0x57cd8e['map'](function (_0x466e5f) {
                return (_0x466e5f && _0x466e5f[0x0] !== '.' ? '.' : '') + _0x466e5f;
            }), _0x3fca21) < 0x0;
        };
        for (_0x11404c in _0x466507) {
            if (!_0x39929c['call'](_0x466507, _0x11404c))
                continue;
            _0xc93ec8 = _0x466507[_0x11404c];
            if (_0x478c7f[_0x11404c] !== void 0x0)
                throw new Error('path.' + _0x11404c + '\x20already\x20exists.');
            else
                _0x478c7f[_0x11404c] = _0xc93ec8;
        }
        ;
    }
};
;
