// test-maker/dist/397.js
exports['id'] = 0x18d, exports['ids'] = [0x18d], exports['modules'] = {
    0x5395: (_0x2eea46, _0x49eb09, _0x528bb6) => {
        const _0xdcf34e = _0x528bb6(0xbfe2), _0x2a03e4 = _0x528bb6(0x11569), _0x1f470e = _0x528bb6(0x4d5b), _0x13c138 = _0x528bb6(0x432b), _0x1b888f = (_0x63d4ba, _0x10b8c6) => typeof _0x63d4ba === 'boolean' ? _0x63d4ba : _0x10b8c6, _0xc20667 = (_0x3a86b7, _0x201a22) => typeof _0x3a86b7 === 'string' ? _0x3a86b7 : _0x201a22, _0x5422a9 = {
                'noSort': ![],
                'readEntries': ![],
                'method': _0xdcf34e['Constants']['NONE'],
                'fs': null
            };
        _0x2eea46['exports'] = function (_0x2de10f, _0x940134) {
            let _0x2fdd00 = null;
            const _0x34bce3 = Object['assign'](Object['create'](null), _0x5422a9);
            if (_0x2de10f && 'object' === typeof _0x2de10f) {
                if (!(_0x2de10f instanceof Uint8Array)) {
                    Object['assign'](_0x34bce3, _0x2de10f), _0x2de10f = _0x34bce3['input'] ? _0x34bce3['input'] : undefined;
                    if (_0x34bce3['input'])
                        delete _0x34bce3['input'];
                }
                Buffer['isBuffer'](_0x2de10f) && (_0x2fdd00 = _0x2de10f, _0x34bce3['method'] = _0xdcf34e['Constants']['BUFFER'], _0x2de10f = undefined);
            }
            Object['assign'](_0x34bce3, _0x940134);
            const _0x18ed9c = new _0xdcf34e(_0x34bce3);
            if (_0x2de10f && 'string' === typeof _0x2de10f) {
                if (_0x18ed9c['fs']['existsSync'](_0x2de10f))
                    _0x34bce3['method'] = _0xdcf34e['Constants']['FILE'], _0x34bce3['filename'] = _0x2de10f, _0x2fdd00 = _0x18ed9c['fs']['readFileSync'](_0x2de10f);
                else
                    throw new Error(_0xdcf34e['Errors']['INVALID_FILENAME']);
            }
            const _0x5cd37d = new _0x13c138(_0x2fdd00, _0x34bce3), {
                    canonical: _0x59ed86,
                    sanitize: _0x9bb76e
                } = _0xdcf34e;
            function _0xa676fa(_0x514052) {
                if (_0x514052 && _0x5cd37d) {
                    var _0x4fb1ac;
                    if (typeof _0x514052 === 'string')
                        _0x4fb1ac = _0x5cd37d['getEntry'](_0x514052);
                    if (typeof _0x514052 === 'object' && typeof _0x514052['entryName'] !== 'undefined' && typeof _0x514052['header'] !== 'undefined')
                        _0x4fb1ac = _0x5cd37d['getEntry'](_0x514052['entryName']);
                    if (_0x4fb1ac)
                        return _0x4fb1ac;
                }
                return null;
            }
            function _0x36ffd4(_0x2966db) {
                const {
                    join: _0x4e4e4b,
                    normalize: _0xfe252,
                    sep: _0x185777
                } = _0x2a03e4['posix'];
                return _0x4e4e4b('.', _0xfe252(_0x185777 + _0x2966db['split']('\x5c')['join'](_0x185777) + _0x185777));
            }
            return {
                'readFile': function (_0x5a1e4c, _0x17e100) {
                    var _0x3b3be5 = _0xa676fa(_0x5a1e4c);
                    return _0x3b3be5 && _0x3b3be5['getData'](_0x17e100) || null;
                },
                'readFileAsync': function (_0x4f6eaf, _0x46d728) {
                    var _0x4eac47 = _0xa676fa(_0x4f6eaf);
                    _0x4eac47 ? _0x4eac47['getDataAsync'](_0x46d728) : _0x46d728(null, 'getEntry\x20failed\x20for:' + _0x4f6eaf);
                },
                'readAsText': function (_0x5a2307, _0x41970e) {
                    var _0x2dfdfb = _0xa676fa(_0x5a2307);
                    if (_0x2dfdfb) {
                        var _0x3402cd = _0x2dfdfb['getData']();
                        if (_0x3402cd && _0x3402cd['length'])
                            return _0x3402cd['toString'](_0x41970e || 'utf8');
                    }
                    return '';
                },
                'readAsTextAsync': function (_0x282c34, _0x48a483, _0xb6ed50) {
                    var _0x393e77 = _0xa676fa(_0x282c34);
                    _0x393e77 ? _0x393e77['getDataAsync'](function (_0x5135c7, _0x37f9a1) {
                        if (_0x37f9a1) {
                            _0x48a483(_0x5135c7, _0x37f9a1);
                            return;
                        }
                        _0x5135c7 && _0x5135c7['length'] ? _0x48a483(_0x5135c7['toString'](_0xb6ed50 || 'utf8')) : _0x48a483('');
                    }) : _0x48a483('');
                },
                'deleteFile': function (_0x28ccc1) {
                    var _0x522c3a = _0xa676fa(_0x28ccc1);
                    _0x522c3a && _0x5cd37d['deleteEntry'](_0x522c3a['entryName']);
                },
                'addZipComment': function (_0x3879fa) {
                    _0x5cd37d['comment'] = _0x3879fa;
                },
                'getZipComment': function () {
                    return _0x5cd37d['comment'] || '';
                },
                'addZipEntryComment': function (_0x1e6857, _0x17dfe5) {
                    var _0x35a993 = _0xa676fa(_0x1e6857);
                    _0x35a993 && (_0x35a993['comment'] = _0x17dfe5);
                },
                'getZipEntryComment': function (_0x28d542) {
                    var _0x3ef8fd = _0xa676fa(_0x28d542);
                    if (_0x3ef8fd)
                        return _0x3ef8fd['comment'] || '';
                    return '';
                },
                'updateFile': function (_0x22a389, _0x30b44d) {
                    var _0x339f4a = _0xa676fa(_0x22a389);
                    _0x339f4a && _0x339f4a['setData'](_0x30b44d);
                },
                'addLocalFile': function (_0x14aeb0, _0x46cac2, _0x355d95, _0x4d5b55) {
                    if (_0x18ed9c['fs']['existsSync'](_0x14aeb0)) {
                        _0x46cac2 = _0x46cac2 ? _0x36ffd4(_0x46cac2) : '';
                        var _0x42efe3 = _0x14aeb0['split']('\x5c')['join']('/')['split']('/')['pop']();
                        _0x46cac2 += _0x355d95 ? _0x355d95 : _0x42efe3;
                        const _0x36383a = _0x18ed9c['fs']['statSync'](_0x14aeb0);
                        this['addFile'](_0x46cac2, _0x18ed9c['fs']['readFileSync'](_0x14aeb0), _0x4d5b55, _0x36383a);
                    } else
                        throw new Error(_0xdcf34e['Errors']['FILE_NOT_FOUND']['replace']('%s', _0x14aeb0));
                },
                'addLocalFolder': function (_0x67269f, _0x54fa92, _0x20d3e4) {
                    if (_0x20d3e4 instanceof RegExp)
                        _0x20d3e4 = function (_0x4f81dc) {
                            return function (_0x3f6af0) {
                                return _0x4f81dc['test'](_0x3f6af0);
                            };
                        }(_0x20d3e4);
                    else
                        'function' !== typeof _0x20d3e4 && (_0x20d3e4 = function () {
                            return !![];
                        });
                    _0x54fa92 = _0x54fa92 ? _0x36ffd4(_0x54fa92) : '', _0x67269f = _0x2a03e4['normalize'](_0x67269f);
                    if (_0x18ed9c['fs']['existsSync'](_0x67269f)) {
                        const _0x227261 = _0x18ed9c['findFiles'](_0x67269f), _0x217433 = this;
                        _0x227261['length'] && _0x227261['forEach'](function (_0x1830de) {
                            var _0x262e24 = _0x2a03e4['relative'](_0x67269f, _0x1830de)['split']('\x5c')['join']('/');
                            if (_0x20d3e4(_0x262e24)) {
                                var _0x2d0421 = _0x18ed9c['fs']['statSync'](_0x1830de);
                                _0x2d0421['isFile']() ? _0x217433['addFile'](_0x54fa92 + _0x262e24, _0x18ed9c['fs']['readFileSync'](_0x1830de), '', _0x2d0421) : _0x217433['addFile'](_0x54fa92 + _0x262e24 + '/', Buffer['alloc'](0x0), '', _0x2d0421);
                            }
                        });
                    } else
                        throw new Error(_0xdcf34e['Errors']['FILE_NOT_FOUND']['replace']('%s', _0x67269f));
                },
                'addLocalFolderAsync': function (_0x579bcb, _0x140e46, _0x1e627d, _0x5bfbb2) {
                    if (_0x5bfbb2 instanceof RegExp)
                        _0x5bfbb2 = function (_0x1a527a) {
                            return function (_0x500c36) {
                                return _0x1a527a['test'](_0x500c36);
                            };
                        }(_0x5bfbb2);
                    else
                        'function' !== typeof _0x5bfbb2 && (_0x5bfbb2 = function () {
                            return !![];
                        });
                    _0x1e627d = _0x1e627d ? _0x36ffd4(_0x1e627d) : '', _0x579bcb = _0x2a03e4['normalize'](_0x579bcb);
                    var _0x1412dd = this;
                    _0x18ed9c['fs']['open'](_0x579bcb, 'r', function (_0x192979) {
                        if (_0x192979 && _0x192979['code'] === 'ENOENT')
                            _0x140e46(undefined, _0xdcf34e['Errors']['FILE_NOT_FOUND']['replace']('%s', _0x579bcb));
                        else {
                            if (_0x192979)
                                _0x140e46(undefined, _0x192979);
                            else {
                                var _0x458faa = _0x18ed9c['findFiles'](_0x579bcb), _0x8f6612 = -0x1, _0x4082f5 = function () {
                                        _0x8f6612 += 0x1;
                                        if (_0x8f6612 < _0x458faa['length']) {
                                            var _0x470246 = _0x458faa[_0x8f6612], _0x12579a = _0x2a03e4['relative'](_0x579bcb, _0x470246)['split']('\x5c')['join']('/');
                                            _0x12579a = _0x12579a['normalize']('NFD')['replace'](/[\u0300-\u036f]/g, '')['replace'](/[^\x20-\x7E]/g, ''), _0x5bfbb2(_0x12579a) ? _0x18ed9c['fs']['stat'](_0x470246, function (_0x2cbcc3, _0x35a164) {
                                                if (_0x2cbcc3)
                                                    _0x140e46(undefined, _0x2cbcc3);
                                                _0x35a164['isFile']() ? _0x18ed9c['fs']['readFile'](_0x470246, function (_0x3ab5e0, _0x3bf332) {
                                                    _0x3ab5e0 ? _0x140e46(undefined, _0x3ab5e0) : (_0x1412dd['addFile'](_0x1e627d + _0x12579a, _0x3bf332, '', _0x35a164), _0x4082f5());
                                                }) : (_0x1412dd['addFile'](_0x1e627d + _0x12579a + '/', Buffer['alloc'](0x0), '', _0x35a164), _0x4082f5());
                                            }) : _0x4082f5();
                                        } else
                                            _0x140e46(!![], undefined);
                                    };
                                _0x4082f5();
                            }
                        }
                    });
                },
                'addLocalFolderPromise': function (_0x2b282a, _0x5220f4) {
                    return new Promise((_0x2b0673, _0xf79f4b) => {
                        const {
                            filter: _0x447afc,
                            zipPath: _0x2ddea9
                        } = Object['assign']({}, _0x5220f4);
                        this['addLocalFolderAsync'](_0x2b282a, (_0x4a0c37, _0x5b6a53) => {
                            if (_0x5b6a53)
                                _0xf79f4b(_0x5b6a53);
                            if (_0x4a0c37)
                                _0x2b0673(this);
                        }, _0x2ddea9, _0x447afc);
                    });
                },
                'addFile': function (_0xb684c4, _0x2c3c85, _0x2e249d, _0x2cc8ca) {
                    let _0x4df25a = _0xa676fa(_0xb684c4);
                    const _0x3abee4 = _0x4df25a != null;
                    !_0x3abee4 && (_0x4df25a = new _0x1f470e(), _0x4df25a['entryName'] = _0xb684c4);
                    _0x4df25a['comment'] = _0x2e249d || '';
                    const _0x30c733 = 'object' === typeof _0x2cc8ca && _0x2cc8ca instanceof _0x18ed9c['fs']['Stats'];
                    _0x30c733 && (_0x4df25a['header']['time'] = _0x2cc8ca['mtime']);
                    var _0x51f1ed = _0x4df25a['isDirectory'] ? 0x10 : 0x0;
                    if (!_0xdcf34e['isWin']) {
                        let _0x51bc4c = _0x4df25a['isDirectory'] ? 0x4000 : 0x8000;
                        if (_0x30c733)
                            _0x51bc4c |= 0xfff & _0x2cc8ca['mode'];
                        else
                            'number' === typeof _0x2cc8ca ? _0x51bc4c |= 0xfff & _0x2cc8ca : _0x51bc4c |= _0x4df25a['isDirectory'] ? 0x1ed : 0x1a4;
                        _0x51f1ed = (_0x51f1ed | _0x51bc4c << 0x10) >>> 0x0;
                    }
                    _0x4df25a['attr'] = _0x51f1ed, _0x4df25a['setData'](_0x2c3c85);
                    if (!_0x3abee4)
                        _0x5cd37d['setEntry'](_0x4df25a);
                },
                'getEntries': function () {
                    return _0x5cd37d ? _0x5cd37d['entries'] : [];
                },
                'getEntry': function (_0x193c70) {
                    return _0xa676fa(_0x193c70);
                },
                'getEntryCount': function () {
                    return _0x5cd37d['getEntryCount']();
                },
                'forEach': function (_0x23d2a7) {
                    return _0x5cd37d['forEach'](_0x23d2a7);
                },
                'extractEntryTo': function (_0x86a005, _0x4318e1, _0x26646a, _0x284fa7, _0x5158cd, _0x56c95e) {
                    _0x284fa7 = _0x1b888f(_0x284fa7, ![]), _0x5158cd = _0x1b888f(_0x5158cd, ![]), _0x26646a = _0x1b888f(_0x26646a, !![]), _0x56c95e = _0xc20667(_0x56c95e, _0xc20667(_0x5158cd, undefined));
                    var _0x1a6909 = _0xa676fa(_0x86a005);
                    if (!_0x1a6909)
                        throw new Error(_0xdcf34e['Errors']['NO_ENTRY']);
                    var _0x52a90c = _0x59ed86(_0x1a6909['entryName']), _0x36f12f = _0x9bb76e(_0x4318e1, _0x56c95e && !_0x1a6909['isDirectory'] ? _0x56c95e : _0x26646a ? _0x52a90c : _0x2a03e4['basename'](_0x52a90c));
                    if (_0x1a6909['isDirectory']) {
                        var _0x3d78a9 = _0x5cd37d['getEntryChildren'](_0x1a6909);
                        return _0x3d78a9['forEach'](function (_0x439457) {
                            if (_0x439457['isDirectory'])
                                return;
                            var _0x4b54c3 = _0x439457['getData']();
                            if (!_0x4b54c3)
                                throw new Error(_0xdcf34e['Errors']['CANT_EXTRACT_FILE']);
                            var _0x48653f = _0x59ed86(_0x439457['entryName']), _0x1bcf8c = _0x9bb76e(_0x4318e1, _0x26646a ? _0x48653f : _0x2a03e4['basename'](_0x48653f));
                            const _0xcc17f6 = _0x5158cd ? _0x439457['header']['fileAttr'] : undefined;
                            _0x18ed9c['writeFileTo'](_0x1bcf8c, _0x4b54c3, _0x284fa7, _0xcc17f6);
                        }), !![];
                    }
                    var _0x2a1db4 = _0x1a6909['getData']();
                    if (!_0x2a1db4)
                        throw new Error(_0xdcf34e['Errors']['CANT_EXTRACT_FILE']);
                    if (_0x18ed9c['fs']['existsSync'](_0x36f12f) && !_0x284fa7)
                        throw new Error(_0xdcf34e['Errors']['CANT_OVERRIDE']);
                    const _0xa5423d = _0x5158cd ? _0x86a005['header']['fileAttr'] : undefined;
                    return _0x18ed9c['writeFileTo'](_0x36f12f, _0x2a1db4, _0x284fa7, _0xa5423d), !![];
                },
                'test': function (_0x3f8e39) {
                    if (!_0x5cd37d)
                        return ![];
                    for (var _0x57119b in _0x5cd37d['entries']) {
                        try {
                            if (_0x57119b['isDirectory'])
                                continue;
                            var _0x38d189 = _0x5cd37d['entries'][_0x57119b]['getData'](_0x3f8e39);
                            if (!_0x38d189)
                                return ![];
                        } catch (_0x13b97a) {
                            return ![];
                        }
                    }
                    return !![];
                },
                'extractAllTo': function (_0x61ce06, _0x2fe1d9, _0x2d3a22, _0x1f2acf) {
                    _0x2fe1d9 = _0x1b888f(_0x2fe1d9, ![]), _0x1f2acf = _0xc20667(_0x2d3a22, _0x1f2acf), _0x2d3a22 = _0x1b888f(_0x2d3a22, ![]);
                    if (!_0x5cd37d)
                        throw new Error(_0xdcf34e['Errors']['NO_ZIP']);
                    _0x5cd37d['entries']['forEach'](function (_0x47e1f9) {
                        var _0x54fe2d = _0x9bb76e(_0x61ce06, _0x59ed86(_0x47e1f9['entryName']['toString']()));
                        if (_0x47e1f9['isDirectory']) {
                            _0x18ed9c['makeDir'](_0x54fe2d);
                            return;
                        }
                        var _0x5ae4fd = _0x47e1f9['getData'](_0x1f2acf);
                        if (!_0x5ae4fd)
                            throw new Error(_0xdcf34e['Errors']['CANT_EXTRACT_FILE']);
                        const _0x277aab = _0x2d3a22 ? _0x47e1f9['header']['fileAttr'] : undefined;
                        _0x18ed9c['writeFileTo'](_0x54fe2d, _0x5ae4fd, _0x2fe1d9, _0x277aab);
                        try {
                            _0x18ed9c['fs']['utimesSync'](_0x54fe2d, _0x47e1f9['header']['time'], _0x47e1f9['header']['time']);
                        } catch (_0x16e701) {
                            throw new Error(_0xdcf34e['Errors']['CANT_EXTRACT_FILE']);
                        }
                    });
                },
                'extractAllToAsync': function (_0x5a4581, _0x4c0950, _0x4330cc, _0x5daa7e) {
                    !_0x5daa7e && (_0x5daa7e = function () {
                    });
                    _0x4c0950 = _0x1b888f(_0x4c0950, ![]);
                    if (typeof _0x4330cc === 'function' && !_0x5daa7e)
                        _0x5daa7e = _0x4330cc;
                    _0x4330cc = _0x1b888f(_0x4330cc, ![]);
                    if (!_0x5cd37d) {
                        _0x5daa7e(new Error(_0xdcf34e['Errors']['NO_ZIP']));
                        return;
                    }
                    _0x5a4581 = _0x2a03e4['resolve'](_0x5a4581);
                    const _0xc71aeb = _0x4b1005 => _0x9bb76e(_0x5a4581, _0x2a03e4['normalize'](_0x59ed86(_0x4b1005['entryName']['toString']()))), _0x5a2690 = (_0x40a807, _0x25a16c) => new Error(_0x40a807 + ':\x20\x22' + _0x25a16c + '\x22'), _0x6da07e = [], _0xe4a33d = new Set();
                    _0x5cd37d['entries']['forEach'](_0x287a7f => {
                        _0x287a7f['isDirectory'] ? _0x6da07e['push'](_0x287a7f) : _0xe4a33d['add'](_0x287a7f);
                    });
                    for (const _0x2c3be4 of _0x6da07e) {
                        const _0x36bdf8 = _0xc71aeb(_0x2c3be4), _0x55a26b = _0x4330cc ? _0x2c3be4['header']['fileAttr'] : undefined;
                        try {
                            _0x18ed9c['makeDir'](_0x36bdf8);
                            if (_0x55a26b)
                                _0x18ed9c['fs']['chmodSync'](_0x36bdf8, _0x55a26b);
                            _0x18ed9c['fs']['utimesSync'](_0x36bdf8, _0x2c3be4['header']['time'], _0x2c3be4['header']['time']);
                        } catch (_0x2c6007) {
                            _0x5daa7e(_0x5a2690('Unable\x20to\x20create\x20folder', _0x36bdf8));
                        }
                    }
                    const _0x230f9b = () => {
                        _0xe4a33d['size'] === 0x0 && _0x5daa7e();
                    };
                    for (const _0x18325d of _0xe4a33d['values']()) {
                        const _0x28deac = _0x2a03e4['normalize'](_0x59ed86(_0x18325d['entryName']['toString']())), _0x227343 = _0x9bb76e(_0x5a4581, _0x28deac);
                        _0x18325d['getDataAsync'](function (_0x3fbfe4, _0x3e5cff) {
                            if (_0x3e5cff) {
                                _0x5daa7e(new Error(_0x3e5cff));
                                return;
                            }
                            if (!_0x3fbfe4)
                                _0x5daa7e(new Error(_0xdcf34e['Errors']['CANT_EXTRACT_FILE']));
                            else {
                                const _0xf5f044 = _0x4330cc ? _0x18325d['header']['fileAttr'] : undefined;
                                _0x18ed9c['writeFileToAsync'](_0x227343, _0x3fbfe4, _0x4c0950, _0xf5f044, function (_0xbcadc3) {
                                    if (!_0xbcadc3) {
                                        _0x5daa7e(_0x5a2690('Unable\x20to\x20write\x20file', _0x227343));
                                        return;
                                    }
                                    _0x18ed9c['fs']['utimes'](_0x227343, _0x18325d['header']['time'], _0x18325d['header']['time'], function (_0x2471c0) {
                                        if (_0x2471c0) {
                                            _0x5daa7e(_0x5a2690('Unable\x20to\x20set\x20times', _0x227343));
                                            return;
                                        }
                                        _0xe4a33d['delete'](_0x18325d), _0x230f9b();
                                    });
                                });
                            }
                        });
                    }
                    _0x230f9b();
                },
                'writeZip': function (_0x1d3ef7, _0x5ebd85) {
                    arguments['length'] === 0x1 && (typeof _0x1d3ef7 === 'function' && (_0x5ebd85 = _0x1d3ef7, _0x1d3ef7 = ''));
                    !_0x1d3ef7 && _0x34bce3['filename'] && (_0x1d3ef7 = _0x34bce3['filename']);
                    if (!_0x1d3ef7)
                        return;
                    var _0x2f3e51 = _0x5cd37d['compressToBuffer']();
                    if (_0x2f3e51) {
                        var _0x302db5 = _0x18ed9c['writeFileTo'](_0x1d3ef7, _0x2f3e51, !![]);
                        if (typeof _0x5ebd85 === 'function')
                            _0x5ebd85(!_0x302db5 ? new Error('failed') : null, '');
                    }
                },
                'writeZipPromise': function (_0x4c005c, _0x1f57f6) {
                    const {
                        overwrite: _0x365b4c,
                        perm: _0x2561ef
                    } = Object['assign']({ 'overwrite': !![] }, _0x1f57f6);
                    return new Promise((_0x2ee8dc, _0x5d5971) => {
                        if (!_0x4c005c && _0x34bce3['filename'])
                            _0x4c005c = _0x34bce3['filename'];
                        if (!_0x4c005c)
                            _0x5d5971('ADM-ZIP:\x20ZIP\x20File\x20Name\x20Missing');
                        this['toBufferPromise']()['then'](_0x3d009c => {
                            const _0x986356 = _0xd9f803 => _0xd9f803 ? _0x2ee8dc(_0xd9f803) : _0x5d5971('ADM-ZIP:\x20Wasn\x27t\x20able\x20to\x20write\x20zip\x20file');
                            _0x18ed9c['writeFileToAsync'](_0x4c005c, _0x3d009c, _0x365b4c, _0x2561ef, _0x986356);
                        }, _0x5d5971);
                    });
                },
                'toBufferPromise': function () {
                    return new Promise((_0x15ca23, _0x23f403) => {
                        _0x5cd37d['toAsyncBuffer'](_0x15ca23, _0x23f403);
                    });
                },
                'toBuffer': function (_0x49a39d, _0x3983dc, _0x33d9e8, _0x11e5ef) {
                    this['valueOf'] = 0x2;
                    if (typeof _0x49a39d === 'function')
                        return _0x5cd37d['toAsyncBuffer'](_0x49a39d, _0x3983dc, _0x33d9e8, _0x11e5ef), null;
                    return _0x5cd37d['compressToBuffer']();
                }
            };
        };
    },
    0x13877: (_0x279c4a, _0x2f7cd3, _0x4bd5eb) => {
        var _0x16d63b = _0x4bd5eb(0xbfe2), _0x8644cc = _0x16d63b['Constants'];
        _0x279c4a['exports'] = function () {
            var _0x19525c = 0x14, _0xfaff3c = 0xa, _0x4645ce = 0x0, _0x1c949a = 0x0, _0x452d82 = 0x0, _0x49df69 = 0x0, _0x38b191 = 0x0, _0x5efbfd = 0x0, _0x38a390 = 0x0, _0x291950 = 0x0, _0x4948b0 = 0x0, _0x1a4732 = 0x0, _0x5b5c25 = 0x0, _0x5e3ee8 = 0x0, _0x253fef = 0x0;
            _0x19525c |= _0x16d63b['isWin'] ? 0xa00 : 0x300, _0x4645ce |= _0x8644cc['FLG_EFS'];
            var _0x2c6da4 = {};
            function _0x394409(_0x46f0e5) {
                _0x46f0e5 = new Date(_0x46f0e5), _0x452d82 = (_0x46f0e5['getFullYear']() - 0x7bc & 0x7f) << 0x19 | _0x46f0e5['getMonth']() + 0x1 << 0x15 | _0x46f0e5['getDate']() << 0x10 | _0x46f0e5['getHours']() << 0xb | _0x46f0e5['getMinutes']() << 0x5 | _0x46f0e5['getSeconds']() >> 0x1;
            }
            return _0x394409(+new Date()), {
                get 'made'() {
                    return _0x19525c;
                },
                set 'made'(_0x4da097) {
                    _0x19525c = _0x4da097;
                },
                get 'version'() {
                    return _0xfaff3c;
                },
                set 'version'(_0x99f158) {
                    _0xfaff3c = _0x99f158;
                },
                get 'flags'() {
                    return _0x4645ce;
                },
                set 'flags'(_0x2dadbc) {
                    _0x4645ce = _0x2dadbc;
                },
                get 'method'() {
                    return _0x1c949a;
                },
                set 'method'(_0x2e7aa5) {
                    switch (_0x2e7aa5) {
                    case _0x8644cc['STORED']:
                        this['version'] = 0xa;
                    case _0x8644cc['DEFLATED']:
                    default:
                        this['version'] = 0x14;
                    }
                    _0x1c949a = _0x2e7aa5;
                },
                get 'time'() {
                    return new Date((_0x452d82 >> 0x19 & 0x7f) + 0x7bc, (_0x452d82 >> 0x15 & 0xf) - 0x1, _0x452d82 >> 0x10 & 0x1f, _0x452d82 >> 0xb & 0x1f, _0x452d82 >> 0x5 & 0x3f, (_0x452d82 & 0x1f) << 0x1);
                },
                set 'time'(_0x42cb2e) {
                    _0x394409(_0x42cb2e);
                },
                get 'crc'() {
                    return _0x49df69;
                },
                set 'crc'(_0x2ecf42) {
                    _0x49df69 = Math['max'](0x0, _0x2ecf42) >>> 0x0;
                },
                get 'compressedSize'() {
                    return _0x38b191;
                },
                set 'compressedSize'(_0x5b66eb) {
                    _0x38b191 = Math['max'](0x0, _0x5b66eb) >>> 0x0;
                },
                get 'size'() {
                    return _0x5efbfd;
                },
                set 'size'(_0x21b44d) {
                    _0x5efbfd = Math['max'](0x0, _0x21b44d) >>> 0x0;
                },
                get 'fileNameLength'() {
                    return _0x38a390;
                },
                set 'fileNameLength'(_0xf26291) {
                    _0x38a390 = _0xf26291;
                },
                get 'extraLength'() {
                    return _0x291950;
                },
                set 'extraLength'(_0x258121) {
                    _0x291950 = _0x258121;
                },
                get 'commentLength'() {
                    return _0x4948b0;
                },
                set 'commentLength'(_0x59b661) {
                    _0x4948b0 = _0x59b661;
                },
                get 'diskNumStart'() {
                    return _0x1a4732;
                },
                set 'diskNumStart'(_0x56eab6) {
                    _0x1a4732 = Math['max'](0x0, _0x56eab6) >>> 0x0;
                },
                get 'inAttr'() {
                    return _0x5b5c25;
                },
                set 'inAttr'(_0x490192) {
                    _0x5b5c25 = Math['max'](0x0, _0x490192) >>> 0x0;
                },
                get 'attr'() {
                    return _0x5e3ee8;
                },
                set 'attr'(_0x2f4f6f) {
                    _0x5e3ee8 = Math['max'](0x0, _0x2f4f6f) >>> 0x0;
                },
                get 'fileAttr'() {
                    return _0x5e3ee8 ? (_0x5e3ee8 >>> 0x0 | 0x0) >> 0x10 & 0xfff : 0x0;
                },
                get 'offset'() {
                    return _0x253fef;
                },
                set 'offset'(_0x5576c9) {
                    _0x253fef = Math['max'](0x0, _0x5576c9) >>> 0x0;
                },
                get 'encripted'() {
                    return (_0x4645ce & 0x1) === 0x1;
                },
                get 'entryHeaderSize'() {
                    return _0x8644cc['CENHDR'] + _0x38a390 + _0x291950 + _0x4948b0;
                },
                get 'realDataOffset'() {
                    return _0x253fef + _0x8644cc['LOCHDR'] + _0x2c6da4['fnameLen'] + _0x2c6da4['extraLen'];
                },
                get 'dataHeader'() {
                    return _0x2c6da4;
                },
                'loadDataHeaderFromBinary': function (_0x1de682) {
                    var _0x44051e = _0x1de682['slice'](_0x253fef, _0x253fef + _0x8644cc['LOCHDR']);
                    if (_0x44051e['readUInt32LE'](0x0) !== _0x8644cc['LOCSIG'])
                        throw new Error(_0x16d63b['Errors']['INVALID_LOC']);
                    _0x2c6da4 = {
                        'version': _0x44051e['readUInt16LE'](_0x8644cc['LOCVER']),
                        'flags': _0x44051e['readUInt16LE'](_0x8644cc['LOCFLG']),
                        'method': _0x44051e['readUInt16LE'](_0x8644cc['LOCHOW']),
                        'time': _0x44051e['readUInt32LE'](_0x8644cc['LOCTIM']),
                        'crc': _0x44051e['readUInt32LE'](_0x8644cc['LOCCRC']),
                        'compressedSize': _0x44051e['readUInt32LE'](_0x8644cc['LOCSIZ']),
                        'size': _0x44051e['readUInt32LE'](_0x8644cc['LOCLEN']),
                        'fnameLen': _0x44051e['readUInt16LE'](_0x8644cc['LOCNAM']),
                        'extraLen': _0x44051e['readUInt16LE'](_0x8644cc['LOCEXT'])
                    };
                },
                'loadFromBinary': function (_0x34efa4) {
                    if (_0x34efa4['length'] !== _0x8644cc['CENHDR'] || _0x34efa4['readUInt32LE'](0x0) !== _0x8644cc['CENSIG'])
                        throw new Error(_0x16d63b['Errors']['INVALID_CEN']);
                    _0x19525c = _0x34efa4['readUInt16LE'](_0x8644cc['CENVEM']), _0xfaff3c = _0x34efa4['readUInt16LE'](_0x8644cc['CENVER']), _0x4645ce = _0x34efa4['readUInt16LE'](_0x8644cc['CENFLG']), _0x1c949a = _0x34efa4['readUInt16LE'](_0x8644cc['CENHOW']), _0x452d82 = _0x34efa4['readUInt32LE'](_0x8644cc['CENTIM']), _0x49df69 = _0x34efa4['readUInt32LE'](_0x8644cc['CENCRC']), _0x38b191 = _0x34efa4['readUInt32LE'](_0x8644cc['CENSIZ']), _0x5efbfd = _0x34efa4['readUInt32LE'](_0x8644cc['CENLEN']), _0x38a390 = _0x34efa4['readUInt16LE'](_0x8644cc['CENNAM']), _0x291950 = _0x34efa4['readUInt16LE'](_0x8644cc['CENEXT']), _0x4948b0 = _0x34efa4['readUInt16LE'](_0x8644cc['CENCOM']), _0x1a4732 = _0x34efa4['readUInt16LE'](_0x8644cc['CENDSK']), _0x5b5c25 = _0x34efa4['readUInt16LE'](_0x8644cc['CENATT']), _0x5e3ee8 = _0x34efa4['readUInt32LE'](_0x8644cc['CENATX']), _0x253fef = _0x34efa4['readUInt32LE'](_0x8644cc['CENOFF']);
                },
                'dataHeaderToBinary': function () {
                    var _0x53a340 = Buffer['alloc'](_0x8644cc['LOCHDR']);
                    return _0x53a340['writeUInt32LE'](_0x8644cc['LOCSIG'], 0x0), _0x53a340['writeUInt16LE'](_0xfaff3c, _0x8644cc['LOCVER']), _0x53a340['writeUInt16LE'](_0x4645ce, _0x8644cc['LOCFLG']), _0x53a340['writeUInt16LE'](_0x1c949a, _0x8644cc['LOCHOW']), _0x53a340['writeUInt32LE'](_0x452d82, _0x8644cc['LOCTIM']), _0x53a340['writeUInt32LE'](_0x49df69, _0x8644cc['LOCCRC']), _0x53a340['writeUInt32LE'](_0x38b191, _0x8644cc['LOCSIZ']), _0x53a340['writeUInt32LE'](_0x5efbfd, _0x8644cc['LOCLEN']), _0x53a340['writeUInt16LE'](_0x38a390, _0x8644cc['LOCNAM']), _0x53a340['writeUInt16LE'](_0x291950, _0x8644cc['LOCEXT']), _0x53a340;
                },
                'entryHeaderToBinary': function () {
                    var _0x1f74d8 = Buffer['alloc'](_0x8644cc['CENHDR'] + _0x38a390 + _0x291950 + _0x4948b0);
                    return _0x1f74d8['writeUInt32LE'](_0x8644cc['CENSIG'], 0x0), _0x1f74d8['writeUInt16LE'](_0x19525c, _0x8644cc['CENVEM']), _0x1f74d8['writeUInt16LE'](_0xfaff3c, _0x8644cc['CENVER']), _0x1f74d8['writeUInt16LE'](_0x4645ce, _0x8644cc['CENFLG']), _0x1f74d8['writeUInt16LE'](_0x1c949a, _0x8644cc['CENHOW']), _0x1f74d8['writeUInt32LE'](_0x452d82, _0x8644cc['CENTIM']), _0x1f74d8['writeUInt32LE'](_0x49df69, _0x8644cc['CENCRC']), _0x1f74d8['writeUInt32LE'](_0x38b191, _0x8644cc['CENSIZ']), _0x1f74d8['writeUInt32LE'](_0x5efbfd, _0x8644cc['CENLEN']), _0x1f74d8['writeUInt16LE'](_0x38a390, _0x8644cc['CENNAM']), _0x1f74d8['writeUInt16LE'](_0x291950, _0x8644cc['CENEXT']), _0x1f74d8['writeUInt16LE'](_0x4948b0, _0x8644cc['CENCOM']), _0x1f74d8['writeUInt16LE'](_0x1a4732, _0x8644cc['CENDSK']), _0x1f74d8['writeUInt16LE'](_0x5b5c25, _0x8644cc['CENATT']), _0x1f74d8['writeUInt32LE'](_0x5e3ee8, _0x8644cc['CENATX']), _0x1f74d8['writeUInt32LE'](_0x253fef, _0x8644cc['CENOFF']), _0x1f74d8['fill'](0x0, _0x8644cc['CENHDR']), _0x1f74d8;
                },
                'toJSON': function () {
                    const _0x583397 = function (_0x422ca9) {
                        return _0x422ca9 + '\x20bytes';
                    };
                    return {
                        'made': _0x19525c,
                        'version': _0xfaff3c,
                        'flags': _0x4645ce,
                        'method': _0x16d63b['methodToString'](_0x1c949a),
                        'time': this['time'],
                        'crc': '0x' + _0x49df69['toString'](0x10)['toUpperCase'](),
                        'compressedSize': _0x583397(_0x38b191),
                        'size': _0x583397(_0x5efbfd),
                        'fileNameLength': _0x583397(_0x38a390),
                        'extraLength': _0x583397(_0x291950),
                        'commentLength': _0x583397(_0x4948b0),
                        'diskNumStart': _0x1a4732,
                        'inAttr': _0x5b5c25,
                        'attr': _0x5e3ee8,
                        'offset': _0x253fef,
                        'entryHeaderSize': _0x583397(_0x8644cc['CENHDR'] + _0x38a390 + _0x291950 + _0x4948b0)
                    };
                },
                'toString': function () {
                    return JSON['stringify'](this['toJSON'](), null, '\x09');
                }
            };
        };
    },
    0x5ac9: (_0x51c562, _0x5bede2, _0x19c2b7) => {
        _0x5bede2['EntryHeader'] = _0x19c2b7(0x13877), _0x5bede2['MainHeader'] = _0x19c2b7(0x9ea0);
    },
    0x9ea0: (_0x16688d, _0x42d3fc, _0x2f5e15) => {
        var _0x23409a = _0x2f5e15(0xbfe2), _0x3d638b = _0x23409a['Constants'];
        _0x16688d['exports'] = function () {
            var _0x4d5782 = 0x0, _0x228502 = 0x0, _0x1a4987 = 0x0, _0x2a3f35 = 0x0, _0x5d6877 = 0x0;
            return {
                get 'diskEntries'() {
                    return _0x4d5782;
                },
                set 'diskEntries'(_0x3cea16) {
                    _0x4d5782 = _0x228502 = _0x3cea16;
                },
                get 'totalEntries'() {
                    return _0x228502;
                },
                set 'totalEntries'(_0x3a8df4) {
                    _0x228502 = _0x4d5782 = _0x3a8df4;
                },
                get 'size'() {
                    return _0x1a4987;
                },
                set 'size'(_0x50bf08) {
                    _0x1a4987 = _0x50bf08;
                },
                get 'offset'() {
                    return _0x2a3f35;
                },
                set 'offset'(_0x184599) {
                    _0x2a3f35 = _0x184599;
                },
                get 'commentLength'() {
                    return _0x5d6877;
                },
                set 'commentLength'(_0x5f2833) {
                    _0x5d6877 = _0x5f2833;
                },
                get 'mainHeaderSize'() {
                    return _0x3d638b['ENDHDR'] + _0x5d6877;
                },
                'loadFromBinary': function (_0x26fea8) {
                    if ((_0x26fea8['length'] !== _0x3d638b['ENDHDR'] || _0x26fea8['readUInt32LE'](0x0) !== _0x3d638b['ENDSIG']) && (_0x26fea8['length'] < _0x3d638b['ZIP64HDR'] || _0x26fea8['readUInt32LE'](0x0) !== _0x3d638b['ZIP64SIG']))
                        throw new Error(_0x23409a['Errors']['INVALID_END']);
                    _0x26fea8['readUInt32LE'](0x0) === _0x3d638b['ENDSIG'] ? (_0x4d5782 = _0x26fea8['readUInt16LE'](_0x3d638b['ENDSUB']), _0x228502 = _0x26fea8['readUInt16LE'](_0x3d638b['ENDTOT']), _0x1a4987 = _0x26fea8['readUInt32LE'](_0x3d638b['ENDSIZ']), _0x2a3f35 = _0x26fea8['readUInt32LE'](_0x3d638b['ENDOFF']), _0x5d6877 = _0x26fea8['readUInt16LE'](_0x3d638b['ENDCOM'])) : (_0x4d5782 = _0x23409a['readBigUInt64LE'](_0x26fea8, _0x3d638b['ZIP64SUB']), _0x228502 = _0x23409a['readBigUInt64LE'](_0x26fea8, _0x3d638b['ZIP64TOT']), _0x1a4987 = _0x23409a['readBigUInt64LE'](_0x26fea8, _0x3d638b['ZIP64SIZ']), _0x2a3f35 = _0x23409a['readBigUInt64LE'](_0x26fea8, _0x3d638b['ZIP64OFF']), _0x5d6877 = 0x0);
                },
                'toBinary': function () {
                    var _0xe0bca5 = Buffer['alloc'](_0x3d638b['ENDHDR'] + _0x5d6877);
                    return _0xe0bca5['writeUInt32LE'](_0x3d638b['ENDSIG'], 0x0), _0xe0bca5['writeUInt32LE'](0x0, 0x4), _0xe0bca5['writeUInt16LE'](_0x4d5782, _0x3d638b['ENDSUB']), _0xe0bca5['writeUInt16LE'](_0x228502, _0x3d638b['ENDTOT']), _0xe0bca5['writeUInt32LE'](_0x1a4987, _0x3d638b['ENDSIZ']), _0xe0bca5['writeUInt32LE'](_0x2a3f35, _0x3d638b['ENDOFF']), _0xe0bca5['writeUInt16LE'](_0x5d6877, _0x3d638b['ENDCOM']), _0xe0bca5['fill']('\x20', _0x3d638b['ENDHDR']), _0xe0bca5;
                },
                'toJSON': function () {
                    const _0x171e69 = function (_0x2bb4e3, _0x1f0c72) {
                        let _0xc13b63 = _0x2bb4e3['toString'](0x10)['toUpperCase']();
                        while (_0xc13b63['length'] < _0x1f0c72)
                            _0xc13b63 = '0' + _0xc13b63;
                        return '0x' + _0xc13b63;
                    };
                    return {
                        'diskEntries': _0x4d5782,
                        'totalEntries': _0x228502,
                        'size': _0x1a4987 + '\x20bytes',
                        'offset': _0x171e69(_0x2a3f35, 0x4),
                        'commentLength': _0x5d6877
                    };
                },
                'toString': function () {
                    return JSON['stringify'](this['toJSON'](), null, '\x09');
                }
            };
        };
    },
    0xc05: (_0xff982c, _0x410691, _0x537c43) => {
        _0xff982c['exports'] = function (_0x14bd68) {
            var _0x56fb8d = _0x537c43(0xe994), _0x1a1ec7 = { 'chunkSize': (parseInt(_0x14bd68['length'] / 0x400) + 0x1) * 0x400 };
            return {
                'deflate': function () {
                    return _0x56fb8d['deflateRawSync'](_0x14bd68, _0x1a1ec7);
                },
                'deflateAsync': function (_0x59ba00) {
                    var _0x5d16a0 = _0x56fb8d['createDeflateRaw'](_0x1a1ec7), _0x14bd0f = [], _0x52fd4b = 0x0;
                    _0x5d16a0['on']('data', function (_0x2c39fe) {
                        _0x14bd0f['push'](_0x2c39fe), _0x52fd4b += _0x2c39fe['length'];
                    }), _0x5d16a0['on']('end', function () {
                        var _0xad2cd = Buffer['alloc'](_0x52fd4b), _0x74be16 = 0x0;
                        _0xad2cd['fill'](0x0);
                        for (var _0x508141 = 0x0; _0x508141 < _0x14bd0f['length']; _0x508141++) {
                            var _0x5ee2df = _0x14bd0f[_0x508141];
                            _0x5ee2df['copy'](_0xad2cd, _0x74be16), _0x74be16 += _0x5ee2df['length'];
                        }
                        _0x59ba00 && _0x59ba00(_0xad2cd);
                    }), _0x5d16a0['end'](_0x14bd68);
                }
            };
        };
    },
    0x53b: (_0x310b62, _0xbbfdad, _0x310d72) => {
        _0xbbfdad['Deflater'] = _0x310d72(0xc05), _0xbbfdad['Inflater'] = _0x310d72(0xb6f5), _0xbbfdad['ZipCrypto'] = _0x310d72(0x8420);
    },
    0xb6f5: (_0x30c6d8, _0x54eb7d, _0x1ba44c) => {
        _0x30c6d8['exports'] = function (_0x34d9cb) {
            var _0x4f2572 = _0x1ba44c(0xe994);
            return {
                'inflate': function () {
                    return _0x4f2572['inflateRawSync'](_0x34d9cb);
                },
                'inflateAsync': function (_0x117b1a) {
                    var _0x250833 = _0x4f2572['createInflateRaw'](), _0x3e70f2 = [], _0x30bbc2 = 0x0;
                    _0x250833['on']('data', function (_0x16705e) {
                        _0x3e70f2['push'](_0x16705e), _0x30bbc2 += _0x16705e['length'];
                    }), _0x250833['on']('end', function () {
                        var _0x228547 = Buffer['alloc'](_0x30bbc2), _0x5dbcaa = 0x0;
                        _0x228547['fill'](0x0);
                        for (var _0x2e1ca9 = 0x0; _0x2e1ca9 < _0x3e70f2['length']; _0x2e1ca9++) {
                            var _0x1779c6 = _0x3e70f2[_0x2e1ca9];
                            _0x1779c6['copy'](_0x228547, _0x5dbcaa), _0x5dbcaa += _0x1779c6['length'];
                        }
                        _0x117b1a && _0x117b1a(_0x228547);
                    }), _0x250833['end'](_0x34d9cb);
                }
            };
        };
    },
    0x8420: (_0x48bf99, _0x38f1d6, _0x291769) => {
        'use strict';
        const {randomFillSync: _0x4cbedd} = _0x291769(0x17e1), _0x5cc4d5 = new Uint32Array(0x100)['map']((_0x492c98, _0x2b5348) => {
                for (let _0x2385e4 = 0x0; _0x2385e4 < 0x8; _0x2385e4++) {
                    0x0 !== (_0x2b5348 & 0x1) ? _0x2b5348 = _0x2b5348 >>> 0x1 ^ 0xedb88320 : _0x2b5348 >>>= 0x1;
                }
                return _0x2b5348 >>> 0x0;
            }), _0x4d54e6 = (_0x235f08, _0x4fc498) => Math['imul'](_0x235f08, _0x4fc498) >>> 0x0, _0x2ca491 = (_0x27cd04, _0x3454ec) => {
                return _0x5cc4d5[(_0x27cd04 ^ _0x3454ec) & 0xff] ^ _0x27cd04 >>> 0x8;
            }, _0x30463d = () => {
                return 'function' === typeof _0x4cbedd ? _0x4cbedd(Buffer['alloc'](0xc)) : _0x30463d['node']();
            };
        _0x30463d['node'] = () => {
            const _0x2ab748 = Buffer['alloc'](0xc), _0x5ed3f9 = _0x2ab748['length'];
            for (let _0x355f69 = 0x0; _0x355f69 < _0x5ed3f9; _0x355f69++)
                _0x2ab748[_0x355f69] = Math['random']() * 0x100 & 0xff;
            return _0x2ab748;
        };
        const _0x4113b0 = { 'genSalt': _0x30463d };
        function _0xe42a81(_0x3ec876) {
            const _0x3bf3f4 = Buffer['isBuffer'](_0x3ec876) ? _0x3ec876 : Buffer['from'](_0x3ec876);
            this['keys'] = new Uint32Array([
                0x12345678,
                0x23456789,
                0x34567890
            ]);
            for (let _0x578e9f = 0x0; _0x578e9f < _0x3bf3f4['length']; _0x578e9f++) {
                this['updateKeys'](_0x3bf3f4[_0x578e9f]);
            }
        }
        _0xe42a81['prototype']['updateKeys'] = function (_0x464ebe) {
            const _0x525207 = this['keys'];
            return _0x525207[0x0] = _0x2ca491(_0x525207[0x0], _0x464ebe), _0x525207[0x1] += _0x525207[0x0] & 0xff, _0x525207[0x1] = _0x4d54e6(_0x525207[0x1], 0x8088405) + 0x1, _0x525207[0x2] = _0x2ca491(_0x525207[0x2], _0x525207[0x1] >>> 0x18), _0x464ebe;
        }, _0xe42a81['prototype']['next'] = function () {
            const _0x19e35c = (this['keys'][0x2] | 0x2) >>> 0x0;
            return _0x4d54e6(_0x19e35c, _0x19e35c ^ 0x1) >> 0x8 & 0xff;
        };
        function _0x5a7ad5(_0x332a6c) {
            const _0x157876 = new _0xe42a81(_0x332a6c);
            return function (_0x28818f) {
                const _0x136085 = Buffer['alloc'](_0x28818f['length']);
                let _0x4579a8 = 0x0;
                for (let _0x27b997 of _0x28818f) {
                    _0x136085[_0x4579a8++] = _0x157876['updateKeys'](_0x27b997 ^ _0x157876['next']());
                }
                return _0x136085;
            };
        }
        function _0x3a4868(_0x456222) {
            const _0x52908d = new _0xe42a81(_0x456222);
            return function (_0x32bf9c, _0xb2e7d9, _0x1d7fd7 = 0x0) {
                if (!_0xb2e7d9)
                    _0xb2e7d9 = Buffer['alloc'](_0x32bf9c['length']);
                for (let _0x2fb983 of _0x32bf9c) {
                    const _0x3bfe2f = _0x52908d['next']();
                    _0xb2e7d9[_0x1d7fd7++] = _0x2fb983 ^ _0x3bfe2f, _0x52908d['updateKeys'](_0x2fb983);
                }
                return _0xb2e7d9;
            };
        }
        function _0x41b91b(_0x28ddb3, _0x26f0ff, _0x55bd26) {
            if (!_0x28ddb3 || !Buffer['isBuffer'](_0x28ddb3) || _0x28ddb3['length'] < 0xc)
                return Buffer['alloc'](0x0);
            const _0x21b762 = _0x5a7ad5(_0x55bd26), _0x496676 = _0x21b762(_0x28ddb3['slice'](0x0, 0xc));
            if (_0x496676[0xb] !== _0x26f0ff['crc'] >>> 0x18)
                throw 'ADM-ZIP:\x20Wrong\x20Password';
            return _0x21b762(_0x28ddb3['slice'](0xc));
        }
        function _0x1cc0e9(_0x372d84) {
            if (Buffer['isBuffer'](_0x372d84) && _0x372d84['length'] >= 0xc)
                _0x4113b0['genSalt'] = function () {
                    return _0x372d84['slice'](0x0, 0xc);
                };
            else
                _0x372d84 === 'node' ? _0x4113b0['genSalt'] = _0x30463d['node'] : _0x4113b0['genSalt'] = _0x30463d;
        }
        function _0x468554(_0x4de307, _0x3344c1, _0x5b9be9, _0x1278ae = ![]) {
            if (_0x4de307 == null)
                _0x4de307 = Buffer['alloc'](0x0);
            if (!Buffer['isBuffer'](_0x4de307))
                _0x4de307 = Buffer['from'](_0x4de307['toString']());
            const _0x4a1335 = _0x3a4868(_0x5b9be9), _0x4dddb4 = _0x4113b0['genSalt']();
            _0x4dddb4[0xb] = _0x3344c1['crc'] >>> 0x18 & 0xff;
            if (_0x1278ae)
                _0x4dddb4[0xa] = _0x3344c1['crc'] >>> 0x10 & 0xff;
            const _0x61945a = Buffer['alloc'](_0x4de307['length'] + 0xc);
            return _0x4a1335(_0x4dddb4, _0x61945a), _0x4a1335(_0x4de307, _0x61945a, 0xc);
        }
        _0x48bf99['exports'] = {
            'decrypt': _0x41b91b,
            'encrypt': _0x468554,
            '_salter': _0x1cc0e9
        };
    },
    0x2c9a: _0x406dba => {
        _0x406dba['exports'] = {
            'LOCHDR': 0x1e,
            'LOCSIG': 0x4034b50,
            'LOCVER': 0x4,
            'LOCFLG': 0x6,
            'LOCHOW': 0x8,
            'LOCTIM': 0xa,
            'LOCCRC': 0xe,
            'LOCSIZ': 0x12,
            'LOCLEN': 0x16,
            'LOCNAM': 0x1a,
            'LOCEXT': 0x1c,
            'EXTSIG': 0x8074b50,
            'EXTHDR': 0x10,
            'EXTCRC': 0x4,
            'EXTSIZ': 0x8,
            'EXTLEN': 0xc,
            'CENHDR': 0x2e,
            'CENSIG': 0x2014b50,
            'CENVEM': 0x4,
            'CENVER': 0x6,
            'CENFLG': 0x8,
            'CENHOW': 0xa,
            'CENTIM': 0xc,
            'CENCRC': 0x10,
            'CENSIZ': 0x14,
            'CENLEN': 0x18,
            'CENNAM': 0x1c,
            'CENEXT': 0x1e,
            'CENCOM': 0x20,
            'CENDSK': 0x22,
            'CENATT': 0x24,
            'CENATX': 0x26,
            'CENOFF': 0x2a,
            'ENDHDR': 0x16,
            'ENDSIG': 0x6054b50,
            'ENDSUB': 0x8,
            'ENDTOT': 0xa,
            'ENDSIZ': 0xc,
            'ENDOFF': 0x10,
            'ENDCOM': 0x14,
            'END64HDR': 0x14,
            'END64SIG': 0x7064b50,
            'END64START': 0x4,
            'END64OFF': 0x8,
            'END64NUMDISKS': 0x10,
            'ZIP64SIG': 0x6064b50,
            'ZIP64HDR': 0x38,
            'ZIP64LEAD': 0xc,
            'ZIP64SIZE': 0x4,
            'ZIP64VEM': 0xc,
            'ZIP64VER': 0xe,
            'ZIP64DSK': 0x10,
            'ZIP64DSKDIR': 0x14,
            'ZIP64SUB': 0x18,
            'ZIP64TOT': 0x20,
            'ZIP64SIZB': 0x28,
            'ZIP64OFF': 0x30,
            'ZIP64EXTRA': 0x38,
            'STORED': 0x0,
            'SHRUNK': 0x1,
            'REDUCED1': 0x2,
            'REDUCED2': 0x3,
            'REDUCED3': 0x4,
            'REDUCED4': 0x5,
            'IMPLODED': 0x6,
            'DEFLATED': 0x8,
            'ENHANCED_DEFLATED': 0x9,
            'PKWARE': 0xa,
            'BZIP2': 0xc,
            'LZMA': 0xe,
            'IBM_TERSE': 0x12,
            'IBM_LZ77': 0x13,
            'AES_ENCRYPT': 0x63,
            'FLG_ENC': 0x1,
            'FLG_COMP1': 0x2,
            'FLG_COMP2': 0x4,
            'FLG_DESC': 0x8,
            'FLG_ENH': 0x10,
            'FLG_PATCH': 0x20,
            'FLG_STR': 0x40,
            'FLG_EFS': 0x800,
            'FLG_MSK': 0x1000,
            'FILE': 0x2,
            'BUFFER': 0x1,
            'NONE': 0x0,
            'EF_ID': 0x0,
            'EF_SIZE': 0x2,
            'ID_ZIP64': 0x1,
            'ID_AVINFO': 0x7,
            'ID_PFS': 0x8,
            'ID_OS2': 0x9,
            'ID_NTFS': 0xa,
            'ID_OPENVMS': 0xc,
            'ID_UNIX': 0xd,
            'ID_FORK': 0xe,
            'ID_PATCH': 0xf,
            'ID_X509_PKCS7': 0x14,
            'ID_X509_CERTID_F': 0x15,
            'ID_X509_CERTID_C': 0x16,
            'ID_STRONGENC': 0x17,
            'ID_RECORD_MGT': 0x18,
            'ID_X509_PKCS7_RL': 0x19,
            'ID_IBM1': 0x65,
            'ID_IBM2': 0x66,
            'ID_POSZIP': 0x4690,
            'EF_ZIP64_OR_32': 0xffffffff,
            'EF_ZIP64_OR_16': 0xffff,
            'EF_ZIP64_SUNCOMP': 0x0,
            'EF_ZIP64_SCOMP': 0x8,
            'EF_ZIP64_RHO': 0x10,
            'EF_ZIP64_DSN': 0x18
        };
    },
    0x56ac: _0x347226 => {
        _0x347226['exports'] = {
            'INVALID_LOC': 'Invalid\x20LOC\x20header\x20(bad\x20signature)',
            'INVALID_CEN': 'Invalid\x20CEN\x20header\x20(bad\x20signature)',
            'INVALID_END': 'Invalid\x20END\x20header\x20(bad\x20signature)',
            'NO_DATA': 'Nothing\x20to\x20decompress',
            'BAD_CRC': 'CRC32\x20checksum\x20failed',
            'FILE_IN_THE_WAY': 'There\x20is\x20a\x20file\x20in\x20the\x20way:\x20%s',
            'UNKNOWN_METHOD': 'Invalid/unsupported\x20compression\x20method',
            'AVAIL_DATA': 'inflate::Available\x20inflate\x20data\x20did\x20not\x20terminate',
            'INVALID_DISTANCE': 'inflate::Invalid\x20literal/length\x20or\x20distance\x20code\x20in\x20fixed\x20or\x20dynamic\x20block',
            'TO_MANY_CODES': 'inflate::Dynamic\x20block\x20code\x20description:\x20too\x20many\x20length\x20or\x20distance\x20codes',
            'INVALID_REPEAT_LEN': 'inflate::Dynamic\x20block\x20code\x20description:\x20repeat\x20more\x20than\x20specified\x20lengths',
            'INVALID_REPEAT_FIRST': 'inflate::Dynamic\x20block\x20code\x20description:\x20repeat\x20lengths\x20with\x20no\x20first\x20length',
            'INCOMPLETE_CODES': 'inflate::Dynamic\x20block\x20code\x20description:\x20code\x20lengths\x20codes\x20incomplete',
            'INVALID_DYN_DISTANCE': 'inflate::Dynamic\x20block\x20code\x20description:\x20invalid\x20distance\x20code\x20lengths',
            'INVALID_CODES_LEN': 'inflate::Dynamic\x20block\x20code\x20description:\x20invalid\x20literal/length\x20code\x20lengths',
            'INVALID_STORE_BLOCK': 'inflate::Stored\x20block\x20length\x20did\x20not\x20match\x20one\x27s\x20complement',
            'INVALID_BLOCK_TYPE': 'inflate::Invalid\x20block\x20type\x20(type\x20==\x203)',
            'CANT_EXTRACT_FILE': 'Could\x20not\x20extract\x20the\x20file',
            'CANT_OVERRIDE': 'Target\x20file\x20already\x20exists',
            'NO_ZIP': 'No\x20zip\x20file\x20was\x20loaded',
            'NO_ENTRY': 'Entry\x20doesn\x27t\x20exist',
            'DIRECTORY_CONTENT_ERROR': 'A\x20directory\x20cannot\x20have\x20content',
            'FILE_NOT_FOUND': 'File\x20not\x20found:\x20%s',
            'NOT_IMPLEMENTED': 'Not\x20implemented',
            'INVALID_FILENAME': 'Invalid\x20filename',
            'INVALID_FORMAT': 'Invalid\x20or\x20unsupported\x20zip\x20format.\x20No\x20END\x20header\x20found'
        };
    },
    0x6edc: (_0xbe956d, _0x1871c3, _0x1c2f89) => {
        const _0x3fd800 = _0x1c2f89(0x99d6)['require'](), _0x54d1b1 = _0x1c2f89(0x11569);
        _0x3fd800['existsSync'] = _0x3fd800['existsSync'] || _0x54d1b1['existsSync'], _0xbe956d['exports'] = function (_0x53a480) {
            var _0x2d60e3 = _0x53a480 || '', _0x52e59a = _0x45eaa7(), _0x192efc = null;
            function _0x45eaa7() {
                return {
                    'directory': ![],
                    'readonly': ![],
                    'hidden': ![],
                    'executable': ![],
                    'mtime': 0x0,
                    'atime': 0x0
                };
            }
            return _0x2d60e3 && _0x3fd800['existsSync'](_0x2d60e3) ? (_0x192efc = _0x3fd800['statSync'](_0x2d60e3), _0x52e59a['directory'] = _0x192efc['isDirectory'](), _0x52e59a['mtime'] = _0x192efc['mtime'], _0x52e59a['atime'] = _0x192efc['atime'], _0x52e59a['executable'] = (0x49 & _0x192efc['mode']) !== 0x0, _0x52e59a['readonly'] = (0x80 & _0x192efc['mode']) === 0x0, _0x52e59a['hidden'] = _0x54d1b1['basename'](_0x2d60e3)[0x0] === '.') : console['warn']('Invalid\x20path:\x20' + _0x2d60e3), {
                get 'directory'() {
                    return _0x52e59a['directory'];
                },
                get 'readOnly'() {
                    return _0x52e59a['readonly'];
                },
                get 'hidden'() {
                    return _0x52e59a['hidden'];
                },
                get 'mtime'() {
                    return _0x52e59a['mtime'];
                },
                get 'atime'() {
                    return _0x52e59a['atime'];
                },
                get 'executable'() {
                    return _0x52e59a['executable'];
                },
                'decodeAttributes': function () {
                },
                'encodeAttributes': function () {
                },
                'toJSON': function () {
                    return {
                        'path': _0x2d60e3,
                        'isDirectory': _0x52e59a['directory'],
                        'isReadOnly': _0x52e59a['readonly'],
                        'isHidden': _0x52e59a['hidden'],
                        'isExecutable': _0x52e59a['executable'],
                        'mTime': _0x52e59a['mtime'],
                        'aTime': _0x52e59a['atime']
                    };
                },
                'toString': function () {
                    return JSON['stringify'](this['toJSON'](), null, '\x09');
                }
            };
        };
    },
    0x99d6: (_0x485cb7, _0x5b07d8, _0x413612) => {
        _0x5b07d8['require'] = function () {
            if (typeof process === 'object' && process['versions'] && process['versions']['electron'])
                try {
                    const _0x22a369 = _0x413612(Object(function _0x2a28e0() {
                        var _0xd2ef92 = new Error('Cannot\x20find\x20module\x20\x27original-fs\x27');
                        _0xd2ef92['code'] = 'MODULE_NOT_FOUND';
                        throw _0xd2ef92;
                    }()));
                    if (Object['keys'](_0x22a369)['length'] > 0x0)
                        return _0x22a369;
                } catch (_0x52f258) {
                }
            return _0x413612(0xdf3b);
        };
    },
    0xbfe2: (_0x387586, _0x344747, _0x439667) => {
        _0x387586['exports'] = _0x439667(0x10718), _0x387586['exports']['Constants'] = _0x439667(0x2c9a), _0x387586['exports']['Errors'] = _0x439667(0x56ac), _0x387586['exports']['FileAttr'] = _0x439667(0x6edc);
    },
    0x10718: (_0x217e20, _0x534185, _0x219f3a) => {
        const _0x4a1585 = _0x219f3a(0x99d6)['require'](), _0x37d12e = _0x219f3a(0x11569), _0x141103 = _0x219f3a(0x2c9a), _0x4189b1 = typeof process === 'object' && 'win32' === process['platform'], _0x1c2e06 = _0x22ca4d => _0x22ca4d && typeof _0x22ca4d === 'object', _0x5fe72a = new Uint32Array(0x100)['map']((_0x53e494, _0xee0c3b) => {
                for (let _0x334928 = 0x0; _0x334928 < 0x8; _0x334928++) {
                    (_0xee0c3b & 0x1) !== 0x0 ? _0xee0c3b = 0xedb88320 ^ _0xee0c3b >>> 0x1 : _0xee0c3b >>>= 0x1;
                }
                return _0xee0c3b >>> 0x0;
            });
        function _0x1ecf75(_0x19e867) {
            this['sep'] = _0x37d12e['sep'], this['fs'] = _0x4a1585, _0x1c2e06(_0x19e867) && (_0x1c2e06(_0x19e867['fs']) && typeof _0x19e867['fs']['statSync'] === 'function' && (this['fs'] = _0x19e867['fs']));
        }
        _0x217e20['exports'] = _0x1ecf75, _0x1ecf75['prototype']['makeDir'] = function (_0x1382df) {
            const _0x40a6e7 = this;
            function _0x3a2594(_0x4b3765) {
                let _0x319e9f = _0x4b3765['split'](_0x40a6e7['sep'])[0x0];
                _0x4b3765['split'](_0x40a6e7['sep'])['forEach'](function (_0x1984d6) {
                    if (!_0x1984d6 || _0x1984d6['substr'](-0x1, 0x1) === ':')
                        return;
                    _0x319e9f += _0x40a6e7['sep'] + _0x1984d6;
                    var _0x519a4c;
                    try {
                        _0x519a4c = _0x40a6e7['fs']['statSync'](_0x319e9f);
                    } catch (_0x2bc4e3) {
                        _0x40a6e7['fs']['mkdirSync'](_0x319e9f);
                    }
                    if (_0x519a4c && _0x519a4c['isFile']())
                        throw Errors['FILE_IN_THE_WAY']['replace']('%s', _0x319e9f);
                });
            }
            _0x3a2594(_0x1382df);
        }, _0x1ecf75['prototype']['writeFileTo'] = function (_0x25fe68, _0x18e659, _0x248a1c, _0x2b0929) {
            const _0x144822 = this;
            if (_0x144822['fs']['existsSync'](_0x25fe68)) {
                if (!_0x248a1c)
                    return ![];
                var _0x2ef99f = _0x144822['fs']['statSync'](_0x25fe68);
                if (_0x2ef99f['isDirectory']())
                    return ![];
            }
            var _0xe6f1d5 = _0x37d12e['dirname'](_0x25fe68);
            !_0x144822['fs']['existsSync'](_0xe6f1d5) && _0x144822['makeDir'](_0xe6f1d5);
            var _0x5324ee;
            try {
                _0x5324ee = _0x144822['fs']['openSync'](_0x25fe68, 'w', 0x1b6);
            } catch (_0x12ffb4) {
                _0x144822['fs']['chmodSync'](_0x25fe68, 0x1b6), _0x5324ee = _0x144822['fs']['openSync'](_0x25fe68, 'w', 0x1b6);
            }
            if (_0x5324ee)
                try {
                    _0x144822['fs']['writeSync'](_0x5324ee, _0x18e659, 0x0, _0x18e659['length'], 0x0);
                } finally {
                    _0x144822['fs']['closeSync'](_0x5324ee);
                }
            return _0x144822['fs']['chmodSync'](_0x25fe68, _0x2b0929 || 0x1b6), !![];
        }, _0x1ecf75['prototype']['writeFileToAsync'] = function (_0x49cdb7, _0x4e10eb, _0x505d49, _0x359ae8, _0xc89aad) {
            typeof _0x359ae8 === 'function' && (_0xc89aad = _0x359ae8, _0x359ae8 = undefined);
            const _0x1aca9b = this;
            _0x1aca9b['fs']['exists'](_0x49cdb7, function (_0x3e75fb) {
                if (_0x3e75fb && !_0x505d49)
                    return _0xc89aad(![]);
                _0x1aca9b['fs']['stat'](_0x49cdb7, function (_0x162b99, _0xab7c6a) {
                    if (_0x3e75fb && _0xab7c6a['isDirectory']())
                        return _0xc89aad(![]);
                    var _0x31cbee = _0x37d12e['dirname'](_0x49cdb7);
                    _0x1aca9b['fs']['exists'](_0x31cbee, function (_0x1658f0) {
                        if (!_0x1658f0)
                            _0x1aca9b['makeDir'](_0x31cbee);
                        _0x1aca9b['fs']['open'](_0x49cdb7, 'w', 0x1b6, function (_0x482fb3, _0x4bc030) {
                            if (_0x482fb3)
                                _0x1aca9b['fs']['chmod'](_0x49cdb7, 0x1b6, function () {
                                    _0x1aca9b['fs']['open'](_0x49cdb7, 'w', 0x1b6, function (_0x11d356, _0x117679) {
                                        _0x1aca9b['fs']['write'](_0x117679, _0x4e10eb, 0x0, _0x4e10eb['length'], 0x0, function () {
                                            _0x1aca9b['fs']['close'](_0x117679, function () {
                                                _0x1aca9b['fs']['chmod'](_0x49cdb7, _0x359ae8 || 0x1b6, function () {
                                                    _0xc89aad(!![]);
                                                });
                                            });
                                        });
                                    });
                                });
                            else
                                _0x4bc030 ? _0x1aca9b['fs']['write'](_0x4bc030, _0x4e10eb, 0x0, _0x4e10eb['length'], 0x0, function () {
                                    _0x1aca9b['fs']['close'](_0x4bc030, function () {
                                        _0x1aca9b['fs']['chmod'](_0x49cdb7, _0x359ae8 || 0x1b6, function () {
                                            _0xc89aad(!![]);
                                        });
                                    });
                                }) : _0x1aca9b['fs']['chmod'](_0x49cdb7, _0x359ae8 || 0x1b6, function () {
                                    _0xc89aad(!![]);
                                });
                        });
                    });
                });
            });
        }, _0x1ecf75['prototype']['findFiles'] = function (_0x57bdd1) {
            const _0x41b142 = this;
            function _0x3e74d2(_0x2d1671, _0x49f2f4, _0x3d3e49) {
                typeof _0x49f2f4 === 'boolean' && (_0x3d3e49 = _0x49f2f4, _0x49f2f4 = undefined);
                let _0x2fbfdf = [];
                return _0x41b142['fs']['readdirSync'](_0x2d1671)['forEach'](function (_0x1aa939) {
                    var _0x7feae4 = _0x37d12e['join'](_0x2d1671, _0x1aa939);
                    if (_0x41b142['fs']['statSync'](_0x7feae4)['isDirectory']() && _0x3d3e49)
                        _0x2fbfdf = _0x2fbfdf['concat'](_0x3e74d2(_0x7feae4, _0x49f2f4, _0x3d3e49));
                    (!_0x49f2f4 || _0x49f2f4['test'](_0x7feae4)) && _0x2fbfdf['push'](_0x37d12e['normalize'](_0x7feae4) + (_0x41b142['fs']['statSync'](_0x7feae4)['isDirectory']() ? _0x41b142['sep'] : ''));
                }), _0x2fbfdf;
            }
            return _0x3e74d2(_0x57bdd1, undefined, !![]);
        }, _0x1ecf75['prototype']['getAttributes'] = function () {
        }, _0x1ecf75['prototype']['setAttributes'] = function () {
        }, _0x1ecf75['crc32update'] = function (_0x572985, _0x286013) {
            return _0x5fe72a[(_0x572985 ^ _0x286013) & 0xff] ^ _0x572985 >>> 0x8;
        }, _0x1ecf75['crc32'] = function (_0x4e5608) {
            typeof _0x4e5608 === 'string' && (_0x4e5608 = Buffer['from'](_0x4e5608, 'utf8'));
            if (!_0x5fe72a['length'])
                genCRCTable();
            let _0x46fc76 = _0x4e5608['length'], _0x4f1ce1 = ~0x0;
            for (let _0x56dbee = 0x0; _0x56dbee < _0x46fc76;)
                _0x4f1ce1 = _0x1ecf75['crc32update'](_0x4f1ce1, _0x4e5608[_0x56dbee++]);
            return ~_0x4f1ce1 >>> 0x0;
        }, _0x1ecf75['methodToString'] = function (_0xa9da25) {
            switch (_0xa9da25) {
            case _0x141103['STORED']:
                return 'STORED\x20(' + _0xa9da25 + ')';
            case _0x141103['DEFLATED']:
                return 'DEFLATED\x20(' + _0xa9da25 + ')';
            default:
                return 'UNSUPPORTED\x20(' + _0xa9da25 + ')';
            }
        }, _0x1ecf75['canonical'] = function (_0x4a18fb) {
            if (!_0x4a18fb)
                return '';
            var _0x24beaf = _0x37d12e['posix']['normalize']('/' + _0x4a18fb['split']('\x5c')['join']('/'));
            return _0x37d12e['join']('.', _0x24beaf);
        }, _0x1ecf75['sanitize'] = function (_0x58649b, _0x3814f6) {
            _0x58649b = _0x37d12e['resolve'](_0x37d12e['normalize'](_0x58649b));
            var _0x24ae17 = _0x3814f6['split']('/');
            for (var _0xdd9119 = 0x0, _0x2cb3f3 = _0x24ae17['length']; _0xdd9119 < _0x2cb3f3; _0xdd9119++) {
                var _0x54df47 = _0x37d12e['normalize'](_0x37d12e['join'](_0x58649b, _0x24ae17['slice'](_0xdd9119, _0x2cb3f3)['join'](_0x37d12e['sep'])));
                if (_0x54df47['indexOf'](_0x58649b) === 0x0)
                    return _0x54df47;
            }
            return _0x37d12e['normalize'](_0x37d12e['join'](_0x58649b, _0x37d12e['basename'](_0x3814f6)));
        }, _0x1ecf75['toBuffer'] = function _0x3fe9a6(_0x2775fd) {
            if (Buffer['isBuffer'](_0x2775fd))
                return _0x2775fd;
            else
                return _0x2775fd instanceof Uint8Array ? Buffer['from'](_0x2775fd) : typeof _0x2775fd === 'string' ? Buffer['from'](_0x2775fd, 'utf8') : Buffer['alloc'](0x0);
        }, _0x1ecf75['readBigUInt64LE'] = function (_0x3ef449, _0x326b68) {
            var _0x48501c = Buffer['from'](_0x3ef449['slice'](_0x326b68, _0x326b68 + 0x8));
            return _0x48501c['swap64'](), parseInt('0x' + _0x48501c['toString']('hex'));
        }, _0x1ecf75['isWin'] = _0x4189b1, _0x1ecf75['crcTable'] = _0x5fe72a;
    },
    0x4d5b: (_0x4e1338, _0x279f04, _0x3128fa) => {
        var _0xaaf205 = _0x3128fa(0xbfe2), _0x144058 = _0x3128fa(0x5ac9), _0x4d13cc = _0xaaf205['Constants'], _0x3dc544 = _0x3128fa(0x53b);
        _0x4e1338['exports'] = function (_0x55b7dc) {
            var _0x91df07 = new _0x144058['EntryHeader'](), _0x1d7866 = Buffer['alloc'](0x0), _0x1cc7b1 = Buffer['alloc'](0x0), _0x4a1dd6 = ![], _0x3ec788 = null, _0x4705e9 = Buffer['alloc'](0x0);
            function _0x353742() {
                if (!_0x55b7dc || !Buffer['isBuffer'](_0x55b7dc))
                    return Buffer['alloc'](0x0);
                return _0x91df07['loadDataHeaderFromBinary'](_0x55b7dc), _0x55b7dc['slice'](_0x91df07['realDataOffset'], _0x91df07['realDataOffset'] + _0x91df07['compressedSize']);
            }
            function _0x1d5d5f(_0x4ee44a) {
                if ((_0x91df07['flags'] & 0x8) !== 0x8) {
                    if (_0xaaf205['crc32'](_0x4ee44a) !== _0x91df07['dataHeader']['crc'])
                        return ![];
                } else {
                }
                return !![];
            }
            function _0x3c91a6(_0x2b40a9, _0x445882, _0x258ae2) {
                typeof _0x445882 === 'undefined' && typeof _0x2b40a9 === 'string' && (_0x258ae2 = _0x2b40a9, _0x2b40a9 = void 0x0);
                if (_0x4a1dd6)
                    return _0x2b40a9 && _0x445882 && _0x445882(Buffer['alloc'](0x0), _0xaaf205['Errors']['DIRECTORY_CONTENT_ERROR']), Buffer['alloc'](0x0);
                var _0x5aeb80 = _0x353742();
                if (_0x5aeb80['length'] === 0x0) {
                    if (_0x2b40a9 && _0x445882)
                        _0x445882(_0x5aeb80);
                    return _0x5aeb80;
                }
                if (_0x91df07['encripted']) {
                    if ('string' !== typeof _0x258ae2 && !Buffer['isBuffer'](_0x258ae2))
                        throw new Error('ADM-ZIP:\x20Incompatible\x20password\x20parameter');
                    _0x5aeb80 = _0x3dc544['ZipCrypto']['decrypt'](_0x5aeb80, _0x91df07, _0x258ae2);
                }
                var _0x45d6ca = Buffer['alloc'](_0x91df07['size']);
                switch (_0x91df07['method']) {
                case _0xaaf205['Constants']['STORED']:
                    _0x5aeb80['copy'](_0x45d6ca);
                    if (!_0x1d5d5f(_0x45d6ca)) {
                        if (_0x2b40a9 && _0x445882)
                            _0x445882(_0x45d6ca, _0xaaf205['Errors']['BAD_CRC']);
                        throw new Error(_0xaaf205['Errors']['BAD_CRC']);
                    } else {
                        if (_0x2b40a9 && _0x445882)
                            _0x445882(_0x45d6ca);
                        return _0x45d6ca;
                    }
                case _0xaaf205['Constants']['DEFLATED']:
                    var _0x36bf61 = new _0x3dc544['Inflater'](_0x5aeb80);
                    if (!_0x2b40a9) {
                        const _0x5db7e4 = _0x36bf61['inflate'](_0x45d6ca);
                        _0x5db7e4['copy'](_0x45d6ca, 0x0);
                        if (!_0x1d5d5f(_0x45d6ca))
                            throw new Error(_0xaaf205['Errors']['BAD_CRC'] + '\x20' + _0x1d7866['toString']());
                        return _0x45d6ca;
                    } else
                        _0x36bf61['inflateAsync'](function (_0x554623) {
                            _0x554623['copy'](_0x554623, 0x0), _0x445882 && (!_0x1d5d5f(_0x554623) ? _0x445882(_0x554623, _0xaaf205['Errors']['BAD_CRC']) : _0x445882(_0x554623));
                        });
                    break;
                default:
                    if (_0x2b40a9 && _0x445882)
                        _0x445882(Buffer['alloc'](0x0), _0xaaf205['Errors']['UNKNOWN_METHOD']);
                    throw new Error(_0xaaf205['Errors']['UNKNOWN_METHOD']);
                }
            }
            function _0xb9e9f5(_0x4b6da0, _0x42774a) {
                if ((!_0x3ec788 || !_0x3ec788['length']) && Buffer['isBuffer'](_0x55b7dc)) {
                    if (_0x4b6da0 && _0x42774a)
                        _0x42774a(_0x353742());
                    return _0x353742();
                }
                if (_0x3ec788['length'] && !_0x4a1dd6) {
                    var _0x3e015d;
                    switch (_0x91df07['method']) {
                    case _0xaaf205['Constants']['STORED']:
                        _0x91df07['compressedSize'] = _0x91df07['size'], _0x3e015d = Buffer['alloc'](_0x3ec788['length']), _0x3ec788['copy'](_0x3e015d);
                        if (_0x4b6da0 && _0x42774a)
                            _0x42774a(_0x3e015d);
                        return _0x3e015d;
                    default:
                    case _0xaaf205['Constants']['DEFLATED']:
                        var _0x36db63 = new _0x3dc544['Deflater'](_0x3ec788);
                        if (!_0x4b6da0) {
                            var _0x1b073a = _0x36db63['deflate']();
                            return _0x91df07['compressedSize'] = _0x1b073a['length'], _0x1b073a;
                        } else
                            _0x36db63['deflateAsync'](function (_0x246411) {
                                _0x3e015d = Buffer['alloc'](_0x246411['length']), _0x91df07['compressedSize'] = _0x246411['length'], _0x246411['copy'](_0x3e015d), _0x42774a && _0x42774a(_0x3e015d);
                            });
                        _0x36db63 = null;
                        break;
                    }
                } else {
                    if (_0x4b6da0 && _0x42774a)
                        _0x42774a(Buffer['alloc'](0x0));
                    else
                        return Buffer['alloc'](0x0);
                }
            }
            function _0xab2d65(_0x7298d, _0x3296c3) {
                return (_0x7298d['readUInt32LE'](_0x3296c3 + 0x4) << 0x4) + _0x7298d['readUInt32LE'](_0x3296c3);
            }
            function _0x141c03(_0x3d0b69) {
                var _0x570855 = 0x0, _0x5e0023, _0x2460b8, _0x5dff02;
                while (_0x570855 < _0x3d0b69['length']) {
                    _0x5e0023 = _0x3d0b69['readUInt16LE'](_0x570855), _0x570855 += 0x2, _0x2460b8 = _0x3d0b69['readUInt16LE'](_0x570855), _0x570855 += 0x2, _0x5dff02 = _0x3d0b69['slice'](_0x570855, _0x570855 + _0x2460b8), _0x570855 += _0x2460b8, _0x4d13cc['ID_ZIP64'] === _0x5e0023 && _0x5eadad(_0x5dff02);
                }
            }
            function _0x5eadad(_0x5e7f2d) {
                var _0x1e0288, _0x8216ef, _0x4c7cc5, _0x35dd4a;
                _0x5e7f2d['length'] >= _0x4d13cc['EF_ZIP64_SCOMP'] && (_0x1e0288 = _0xab2d65(_0x5e7f2d, _0x4d13cc['EF_ZIP64_SUNCOMP']), _0x91df07['size'] === _0x4d13cc['EF_ZIP64_OR_32'] && (_0x91df07['size'] = _0x1e0288)), _0x5e7f2d['length'] >= _0x4d13cc['EF_ZIP64_RHO'] && (_0x8216ef = _0xab2d65(_0x5e7f2d, _0x4d13cc['EF_ZIP64_SCOMP']), _0x91df07['compressedSize'] === _0x4d13cc['EF_ZIP64_OR_32'] && (_0x91df07['compressedSize'] = _0x8216ef)), _0x5e7f2d['length'] >= _0x4d13cc['EF_ZIP64_DSN'] && (_0x4c7cc5 = _0xab2d65(_0x5e7f2d, _0x4d13cc['EF_ZIP64_RHO']), _0x91df07['offset'] === _0x4d13cc['EF_ZIP64_OR_32'] && (_0x91df07['offset'] = _0x4c7cc5)), _0x5e7f2d['length'] >= _0x4d13cc['EF_ZIP64_DSN'] + 0x4 && (_0x35dd4a = _0x5e7f2d['readUInt32LE'](_0x4d13cc['EF_ZIP64_DSN']), _0x91df07['diskNumStart'] === _0x4d13cc['EF_ZIP64_OR_16'] && (_0x91df07['diskNumStart'] = _0x35dd4a));
            }
            return {
                get 'entryName'() {
                    return _0x1d7866['toString']();
                },
                get 'rawEntryName'() {
                    return _0x1d7866;
                },
                set 'entryName'(_0x391394) {
                    _0x1d7866 = _0xaaf205['toBuffer'](_0x391394);
                    var _0x3c10c4 = _0x1d7866[_0x1d7866['length'] - 0x1];
                    _0x4a1dd6 = _0x3c10c4 === 0x2f || _0x3c10c4 === 0x5c, _0x91df07['fileNameLength'] = _0x1d7866['length'];
                },
                get 'extra'() {
                    return _0x4705e9;
                },
                set 'extra'(_0x167fa7) {
                    _0x4705e9 = _0x167fa7, _0x91df07['extraLength'] = _0x167fa7['length'], _0x141c03(_0x167fa7);
                },
                get 'comment'() {
                    return _0x1cc7b1['toString']();
                },
                set 'comment'(_0x10b623) {
                    _0x1cc7b1 = _0xaaf205['toBuffer'](_0x10b623), _0x91df07['commentLength'] = _0x1cc7b1['length'];
                },
                get 'name'() {
                    var _0xff8b23 = _0x1d7866['toString']();
                    return _0x4a1dd6 ? _0xff8b23['substr'](_0xff8b23['length'] - 0x1)['split']('/')['pop']() : _0xff8b23['split']('/')['pop']();
                },
                get 'isDirectory'() {
                    return _0x4a1dd6;
                },
                'getCompressedData': function () {
                    return _0xb9e9f5(![], null);
                },
                'getCompressedDataAsync': function (_0x41b2d6) {
                    _0xb9e9f5(!![], _0x41b2d6);
                },
                'setData': function (_0x48488d) {
                    _0x3ec788 = _0xaaf205['toBuffer'](_0x48488d), !_0x4a1dd6 && _0x3ec788['length'] ? (_0x91df07['size'] = _0x3ec788['length'], _0x91df07['method'] = _0xaaf205['Constants']['DEFLATED'], _0x91df07['crc'] = _0xaaf205['crc32'](_0x48488d), _0x91df07['changed'] = !![]) : _0x91df07['method'] = _0xaaf205['Constants']['STORED'];
                },
                'getData': function (_0x27f6eb) {
                    return _0x91df07['changed'] ? _0x3ec788 : _0x3c91a6(![], null, _0x27f6eb);
                },
                'getDataAsync': function (_0x7cb4cf, _0x128c87) {
                    _0x91df07['changed'] ? _0x7cb4cf(_0x3ec788) : _0x3c91a6(!![], _0x7cb4cf, _0x128c87);
                },
                set 'attr'(_0x3bff1b) {
                    _0x91df07['attr'] = _0x3bff1b;
                },
                get 'attr'() {
                    return _0x91df07['attr'];
                },
                set 'header'(_0x94014b) {
                    _0x91df07['loadFromBinary'](_0x94014b);
                },
                get 'header'() {
                    return _0x91df07;
                },
                'packHeader': function () {
                    var _0x25364f = _0x91df07['entryHeaderToBinary'](), _0x1485e9 = _0xaaf205['Constants']['CENHDR'];
                    return _0x1d7866['copy'](_0x25364f, _0x1485e9), _0x1485e9 += _0x1d7866['length'], _0x91df07['extraLength'] && (_0x4705e9['copy'](_0x25364f, _0x1485e9), _0x1485e9 += _0x91df07['extraLength']), _0x91df07['commentLength'] && _0x1cc7b1['copy'](_0x25364f, _0x1485e9), _0x25364f;
                },
                'toJSON': function () {
                    const _0x1c28ca = function (_0x54f090) {
                        return '<' + (_0x54f090 && _0x54f090['length'] + '\x20bytes\x20buffer' || 'null') + '>';
                    };
                    return {
                        'entryName': this['entryName'],
                        'name': this['name'],
                        'comment': this['comment'],
                        'isDirectory': this['isDirectory'],
                        'header': _0x91df07['toJSON'](),
                        'compressedData': _0x1c28ca(_0x55b7dc),
                        'data': _0x1c28ca(_0x3ec788)
                    };
                },
                'toString': function () {
                    return JSON['stringify'](this['toJSON'](), null, '\x09');
                }
            };
        };
    },
    0x432b: (_0x209201, _0x37946e, _0x252a6e) => {
        const _0x3171bc = _0x252a6e(0x4d5b), _0x58e8d6 = _0x252a6e(0x5ac9), _0x4c17ba = _0x252a6e(0xbfe2);
        _0x209201['exports'] = function (_0x2397f6, _0x33218a) {
            var _0x4b535d = [], _0x17f2b5 = {}, _0x1dfea7 = Buffer['alloc'](0x0), _0x2340e8 = new _0x58e8d6['MainHeader'](), _0x2a3e1f = ![];
            const _0x2b351 = Object['assign'](Object['create'](null), _0x33218a), {noSort: _0x15da44} = _0x2b351;
            _0x2397f6 ? _0x48fb33(_0x2b351['readEntries']) : _0x2a3e1f = !![];
            function _0x30a48a(_0x1fcc0b) {
                const _0x494cc8 = _0x2340e8['diskEntries'];
                let _0x4d5572 = _0x2340e8['offset'];
                for (let _0x4dfa0f = 0x0; _0x4dfa0f < _0x494cc8; _0x4dfa0f++) {
                    let _0x17937b = _0x4d5572;
                    const _0x402abf = new _0x3171bc(_0x2397f6);
                    _0x402abf['header'] = _0x2397f6['slice'](_0x17937b, _0x17937b += _0x4c17ba['Constants']['CENHDR']), _0x402abf['entryName'] = _0x2397f6['slice'](_0x17937b, _0x17937b += _0x402abf['header']['fileNameLength']), _0x4d5572 += _0x402abf['header']['entryHeaderSize'], _0x1fcc0b(_0x402abf);
                }
            }
            function _0x2cda90() {
                _0x2a3e1f = !![], _0x17f2b5 = {}, _0x4b535d = new Array(_0x2340e8['diskEntries']);
                var _0x3f1ee7 = _0x2340e8['offset'];
                for (var _0x2ce691 = 0x0; _0x2ce691 < _0x4b535d['length']; _0x2ce691++) {
                    var _0x5e32ac = _0x3f1ee7, _0x574500 = new _0x3171bc(_0x2397f6);
                    _0x574500['header'] = _0x2397f6['slice'](_0x5e32ac, _0x5e32ac += _0x4c17ba['Constants']['CENHDR']), _0x574500['entryName'] = _0x2397f6['slice'](_0x5e32ac, _0x5e32ac += _0x574500['header']['fileNameLength']);
                    _0x574500['header']['extraLength'] && (_0x574500['extra'] = _0x2397f6['slice'](_0x5e32ac, _0x5e32ac += _0x574500['header']['extraLength']));
                    if (_0x574500['header']['commentLength'])
                        _0x574500['comment'] = _0x2397f6['slice'](_0x5e32ac, _0x5e32ac + _0x574500['header']['commentLength']);
                    _0x3f1ee7 += _0x574500['header']['entryHeaderSize'], _0x4b535d[_0x2ce691] = _0x574500, _0x17f2b5[_0x574500['entryName']] = _0x574500;
                }
            }
            function _0x48fb33(_0x83a6d2) {
                var _0x285949 = _0x2397f6['length'] - _0x4c17ba['Constants']['ENDHDR'], _0x572ed7 = Math['max'](0x0, _0x285949 - 0xffff), _0x298893 = _0x572ed7, _0x32b6dc = _0x2397f6['length'], _0xa241fa = -0x1, _0x51f48d = 0x0;
                for (_0x285949; _0x285949 >= _0x298893; _0x285949--) {
                    if (_0x2397f6[_0x285949] !== 0x50)
                        continue;
                    if (_0x2397f6['readUInt32LE'](_0x285949) === _0x4c17ba['Constants']['ENDSIG']) {
                        _0xa241fa = _0x285949, _0x51f48d = _0x285949, _0x32b6dc = _0x285949 + _0x4c17ba['Constants']['ENDHDR'], _0x298893 = _0x285949 - _0x4c17ba['Constants']['END64HDR'];
                        continue;
                    }
                    if (_0x2397f6['readUInt32LE'](_0x285949) === _0x4c17ba['Constants']['END64SIG']) {
                        _0x298893 = _0x572ed7;
                        continue;
                    }
                    if (_0x2397f6['readUInt32LE'](_0x285949) === _0x4c17ba['Constants']['ZIP64SIG']) {
                        _0xa241fa = _0x285949, _0x32b6dc = _0x285949 + _0x4c17ba['readBigUInt64LE'](_0x2397f6, _0x285949 + _0x4c17ba['Constants']['ZIP64SIZE']) + _0x4c17ba['Constants']['ZIP64LEAD'];
                        break;
                    }
                }
                if (!~_0xa241fa)
                    throw new Error(_0x4c17ba['Errors']['INVALID_FORMAT']);
                _0x2340e8['loadFromBinary'](_0x2397f6['slice'](_0xa241fa, _0x32b6dc));
                _0x2340e8['commentLength'] && (_0x1dfea7 = _0x2397f6['slice'](_0x51f48d + _0x4c17ba['Constants']['ENDHDR']));
                if (_0x83a6d2)
                    _0x2cda90();
            }
            function _0x235e41() {
                _0x4b535d['length'] > 0x1 && !_0x15da44 && _0x4b535d['sort']((_0x351652, _0x4a6f00) => _0x351652['entryName']['toLowerCase']()['localeCompare'](_0x4a6f00['entryName']['toLowerCase']()));
            }
            return {
                get 'entries'() {
                    return !_0x2a3e1f && _0x2cda90(), _0x4b535d;
                },
                get 'comment'() {
                    return _0x1dfea7['toString']();
                },
                set 'comment'(_0x39a59b) {
                    _0x1dfea7 = _0x4c17ba['toBuffer'](_0x39a59b), _0x2340e8['commentLength'] = _0x1dfea7['length'];
                },
                'getEntryCount': function () {
                    if (!_0x2a3e1f)
                        return _0x2340e8['diskEntries'];
                    return _0x4b535d['length'];
                },
                'forEach': function (_0x4b8cdd) {
                    if (!_0x2a3e1f) {
                        _0x30a48a(_0x4b8cdd);
                        return;
                    }
                    _0x4b535d['forEach'](_0x4b8cdd);
                },
                'getEntry': function (_0x19fae1) {
                    return !_0x2a3e1f && _0x2cda90(), _0x17f2b5[_0x19fae1] || null;
                },
                'setEntry': function (_0xeb361f) {
                    !_0x2a3e1f && _0x2cda90(), _0x4b535d['push'](_0xeb361f), _0x17f2b5[_0xeb361f['entryName']] = _0xeb361f, _0x2340e8['totalEntries'] = _0x4b535d['length'];
                },
                'deleteEntry': function (_0x2da65a) {
                    !_0x2a3e1f && _0x2cda90();
                    var _0x417b3f = _0x17f2b5[_0x2da65a];
                    if (_0x417b3f && _0x417b3f['isDirectory']) {
                        var _0xc3fbcf = this;
                        this['getEntryChildren'](_0x417b3f)['forEach'](function (_0x92a298) {
                            _0x92a298['entryName'] !== _0x2da65a && _0xc3fbcf['deleteEntry'](_0x92a298['entryName']);
                        });
                    }
                    _0x4b535d['splice'](_0x4b535d['indexOf'](_0x417b3f), 0x1), delete _0x17f2b5[_0x2da65a], _0x2340e8['totalEntries'] = _0x4b535d['length'];
                },
                'getEntryChildren': function (_0x86206e) {
                    !_0x2a3e1f && _0x2cda90();
                    if (_0x86206e && _0x86206e['isDirectory']) {
                        const _0x1f2adf = [], _0x52dba7 = _0x86206e['entryName'], _0x5f2e67 = _0x52dba7['length'];
                        return _0x4b535d['forEach'](function (_0x5a3891) {
                            _0x5a3891['entryName']['substr'](0x0, _0x5f2e67) === _0x52dba7 && _0x1f2adf['push'](_0x5a3891);
                        }), _0x1f2adf;
                    }
                    return [];
                },
                'compressToBuffer': function () {
                    !_0x2a3e1f && _0x2cda90();
                    _0x235e41();
                    const _0xb2cbf2 = [], _0x49c5d0 = [];
                    let _0x3f6d6e = 0x0, _0x3ed945 = 0x0;
                    _0x2340e8['size'] = 0x0, _0x2340e8['offset'] = 0x0;
                    for (const _0x4c7dff of _0x4b535d) {
                        const _0x708526 = _0x4c7dff['getCompressedData']();
                        _0x4c7dff['header']['offset'] = _0x3ed945;
                        const _0x4b9b82 = _0x4c7dff['header']['dataHeaderToBinary'](), _0x2d1e92 = _0x4c7dff['rawEntryName']['length'], _0x35b2e5 = Buffer['alloc'](_0x2d1e92 + _0x4c7dff['extra']['length']);
                        _0x4c7dff['rawEntryName']['copy'](_0x35b2e5, 0x0), _0x35b2e5['copy'](_0x4c7dff['extra'], _0x2d1e92);
                        const _0x1a8a63 = _0x4b9b82['length'] + _0x35b2e5['length'] + _0x708526['length'];
                        _0x3ed945 += _0x1a8a63, _0xb2cbf2['push'](_0x4b9b82), _0xb2cbf2['push'](_0x35b2e5), _0xb2cbf2['push'](_0x708526);
                        const _0x4aafcf = _0x4c7dff['packHeader']();
                        _0x49c5d0['push'](_0x4aafcf), _0x2340e8['size'] += _0x4aafcf['length'], _0x3f6d6e += _0x1a8a63 + _0x4aafcf['length'];
                    }
                    _0x3f6d6e += _0x2340e8['mainHeaderSize'], _0x2340e8['offset'] = _0x3ed945, _0x3ed945 = 0x0;
                    const _0x946150 = Buffer['alloc'](_0x3f6d6e);
                    for (const _0x40b9b9 of _0xb2cbf2) {
                        _0x40b9b9['copy'](_0x946150, _0x3ed945), _0x3ed945 += _0x40b9b9['length'];
                    }
                    for (const _0x452ece of _0x49c5d0) {
                        _0x452ece['copy'](_0x946150, _0x3ed945), _0x3ed945 += _0x452ece['length'];
                    }
                    const _0x59018d = _0x2340e8['toBinary']();
                    return _0x1dfea7 && _0x1dfea7['copy'](_0x59018d, _0x4c17ba['Constants']['ENDHDR']), _0x59018d['copy'](_0x946150, _0x3ed945), _0x946150;
                },
                'toAsyncBuffer': function (_0x16aa7f, _0x7ca0e, _0x12dde4, _0x4827de) {
                    try {
                        !_0x2a3e1f && _0x2cda90();
                        _0x235e41();
                        const _0x469d04 = [], _0x29f702 = [];
                        let _0x153994 = 0x0, _0x31e0c1 = 0x0;
                        _0x2340e8['size'] = 0x0, _0x2340e8['offset'] = 0x0;
                        const _0x4cdfe2 = function (_0xf86461) {
                            if (_0xf86461['length']) {
                                const _0x4908c8 = _0xf86461['pop'](), _0x47096e = _0x4908c8['entryName'] + _0x4908c8['extra']['toString']();
                                if (_0x12dde4)
                                    _0x12dde4(_0x47096e);
                                _0x4908c8['getCompressedDataAsync'](function (_0x4f10bc) {
                                    if (_0x4827de)
                                        _0x4827de(_0x47096e);
                                    _0x4908c8['header']['offset'] = _0x31e0c1;
                                    const _0x3a6222 = _0x4908c8['header']['dataHeaderToBinary'](), _0xf06341 = Buffer['alloc'](_0x47096e['length'], _0x47096e), _0x2d2733 = _0x3a6222['length'] + _0xf06341['length'] + _0x4f10bc['length'];
                                    _0x31e0c1 += _0x2d2733, _0x469d04['push'](_0x3a6222), _0x469d04['push'](_0xf06341), _0x469d04['push'](_0x4f10bc);
                                    const _0x17c07f = _0x4908c8['packHeader']();
                                    _0x29f702['push'](_0x17c07f), _0x2340e8['size'] += _0x17c07f['length'], _0x153994 += _0x2d2733 + _0x17c07f['length'], _0x4cdfe2(_0xf86461);
                                });
                            } else {
                                _0x153994 += _0x2340e8['mainHeaderSize'], _0x2340e8['offset'] = _0x31e0c1, _0x31e0c1 = 0x0;
                                const _0x524262 = Buffer['alloc'](_0x153994);
                                _0x469d04['forEach'](function (_0x16157a) {
                                    _0x16157a['copy'](_0x524262, _0x31e0c1), _0x31e0c1 += _0x16157a['length'];
                                }), _0x29f702['forEach'](function (_0x160acd) {
                                    _0x160acd['copy'](_0x524262, _0x31e0c1), _0x31e0c1 += _0x160acd['length'];
                                });
                                const _0x554111 = _0x2340e8['toBinary']();
                                _0x1dfea7 && _0x1dfea7['copy'](_0x554111, _0x4c17ba['Constants']['ENDHDR']), _0x554111['copy'](_0x524262, _0x31e0c1), _0x16aa7f(_0x524262);
                            }
                        };
                        _0x4cdfe2(_0x4b535d);
                    } catch (_0x2b721d) {
                        _0x7ca0e(_0x2b721d);
                    }
                }
            };
        };
    }
};
;
