// test-maker/dist/481.js
'use strict';
exports['id'] = 0x1e1, exports['ids'] = [
    0x1e1,
    0xb4,
    0x28c,
    0x3b6
], exports['modules'] = {
    0x686d: _0x556e69 => {
        var _0x2a57dc = /[|\\{}()[\]^$+*?.]/g;
        _0x556e69['exports'] = function (_0x46c432) {
            if (typeof _0x46c432 !== 'string')
                throw new TypeError('Expected\x20a\x20string');
            return _0x46c432['replace'](_0x2a57dc, '\x5c$&');
        };
    },
    0xa2b4: (_0x597638, _0x4beafc, _0x419a3e) => {
        _0x419a3e['r'](_0x4beafc), _0x419a3e['d'](_0x4beafc, {
            'generateId': () => _0x4f665c,
            'generateIdAsync': () => _0x294181,
            'generateIdCustom': () => _0x25f82f,
            'generateIdCustomAsync': () => _0x5989b5
        });
        var _0x5bc6e6 = _0x419a3e(0x16925), _0x1dbf3 = _0x419a3e(0xccb6);
        function _0x4f665c(_0xcd6181) {
            return (0x0, _0x5bc6e6['nanoid'])(_0xcd6181);
        }
        async function _0x294181(_0x514fc7) {
            return await (0x0, _0x1dbf3['nanoid'])(_0x514fc7);
        }
        function _0x25f82f(_0x3529f4, _0x38fecb) {
            return (0x0, _0x5bc6e6['customAlphabet'])(_0x3529f4, _0x38fecb)();
        }
        async function _0x5989b5(_0x4855f3, _0x202155) {
            return await (0x0, _0x1dbf3['customAlphabet'])(_0x4855f3, _0x202155)();
        }
    },
    0x79cc: (_0x5807d4, _0x47ebbc, _0x3b73ef) => {
        _0x3b73ef['r'](_0x47ebbc), _0x3b73ef['d'](_0x47ebbc, { 'logGenericError': () => _0x1e33f7 });
        var _0x2f989b = _0x3b73ef(0x1145c), _0x9b276f = _0x3b73ef['n'](_0x2f989b);
        async function _0x1e33f7({
            error: _0x471b1a,
            isWarning: _0x1da9b0,
            show: show = !![]
        }) {
            let _0x2d0f25;
            try {
                _0x2d0f25 ? _0x1da9b0 ? console['warn'](_0x2d0f25['formatted']['ansi']) : console['error'](_0x2d0f25['formatted']['ansi']) : _0x1da9b0 ? console['warn'](_0x471b1a) : console['error'](_0x9b276f()['redBright'](_0x471b1a['stack'] || _0x471b1a));
            } catch (_0x1606e6) {
                console['error'](_0x9b276f()['redBright'](_0x1606e6['stack'] || _0x1606e6));
            } finally {
                console['info'](_0x9b276f()['cyanBright']('See\x20Log\x20file\x20for\x20more\x20Details...'));
            }
        }
    },
    0xccb6: (_0x54445e, _0x2a1e01, _0xcad95c) => {
        _0xcad95c['r'](_0x2a1e01), _0xcad95c['d'](_0x2a1e01, {
            'customAlphabet': () => _0x19d1c,
            'nanoid': () => _0x480d7f,
            'random': () => _0x3882ef
        });
        var _0x49b298 = _0xcad95c(0x17e1), _0x5acec3 = _0xcad95c(0x12bca);
        let _0x3882ef = _0xe543fa => new Promise((_0x46f86a, _0x3fc32a) => {
                _0x49b298['randomFill'](Buffer['allocUnsafe'](_0xe543fa), (_0x48203c, _0x5d4b77) => {
                    _0x48203c ? _0x3fc32a(_0x48203c) : _0x46f86a(_0x5d4b77);
                });
            }), _0x19d1c = (_0x33472c, _0xcf5308 = 0x15) => {
                let _0x2e2da3 = (0x2 << 0x1f - Math['clz32'](_0x33472c['length'] - 0x1 | 0x1)) - 0x1, _0x59afa5 = Math['ceil'](1.6 * _0x2e2da3 * _0xcf5308 / _0x33472c['length']), _0x3f877c = (_0x4d4be0, _0x2a126a = _0xcf5308) => _0x3882ef(_0x59afa5)['then'](_0x137971 => {
                        let _0x3047b6 = _0x59afa5;
                        while (_0x3047b6--) {
                            _0x4d4be0 += _0x33472c[_0x137971[_0x3047b6] & _0x2e2da3] || '';
                            if (_0x4d4be0['length'] === _0x2a126a)
                                return _0x4d4be0;
                        }
                        return _0x3f877c(_0x4d4be0, _0x2a126a);
                    });
                return _0x2a0e6c => _0x3f877c('', _0x2a0e6c);
            }, _0x480d7f = (_0x288a99 = 0x15) => _0x3882ef(_0x288a99)['then'](_0x5a998c => {
                let _0x2f493c = '';
                while (_0x288a99--) {
                    _0x2f493c += _0x5acec3['urlAlphabet'][_0x5a998c[_0x288a99] & 0x3f];
                }
                return _0x2f493c;
            });
    },
    0x16925: (_0x2617ff, _0x47c1b4, _0x49f363) => {
        _0x49f363['r'](_0x47c1b4), _0x49f363['d'](_0x47c1b4, {
            'customAlphabet': () => _0x16fb43,
            'customRandom': () => _0x451e35,
            'nanoid': () => _0x5b93eb,
            'random': () => _0x4c26f8,
            'urlAlphabet': () => _0xbc56a9['urlAlphabet']
        });
        var _0x2668e8 = _0x49f363(0x17e1), _0xbc56a9 = _0x49f363(0x12bca);
        const _0x43d377 = 0x80;
        let _0x2a7b47, _0x1098e8, _0x4d9a11 = _0x2dd41d => {
                if (!_0x2a7b47 || _0x2a7b47['length'] < _0x2dd41d)
                    _0x2a7b47 = Buffer['allocUnsafe'](_0x2dd41d * _0x43d377), _0x2668e8['randomFillSync'](_0x2a7b47), _0x1098e8 = 0x0;
                else
                    _0x1098e8 + _0x2dd41d > _0x2a7b47['length'] && (_0x2668e8['randomFillSync'](_0x2a7b47), _0x1098e8 = 0x0);
                _0x1098e8 += _0x2dd41d;
            }, _0x4c26f8 = _0x257bdf => {
                return _0x4d9a11(_0x257bdf -= 0x0), _0x2a7b47['subarray'](_0x1098e8 - _0x257bdf, _0x1098e8);
            }, _0x451e35 = (_0x299ace, _0x4c7413, _0x38a469) => {
                let _0x17d750 = (0x2 << 0x1f - Math['clz32'](_0x299ace['length'] - 0x1 | 0x1)) - 0x1, _0x248488 = Math['ceil'](1.6 * _0x17d750 * _0x4c7413 / _0x299ace['length']);
                return (_0x491034 = _0x4c7413) => {
                    let _0xc1b1da = '';
                    while (!![]) {
                        let _0x4ef815 = _0x38a469(_0x248488), _0x3c1d60 = _0x248488;
                        while (_0x3c1d60--) {
                            _0xc1b1da += _0x299ace[_0x4ef815[_0x3c1d60] & _0x17d750] || '';
                            if (_0xc1b1da['length'] === _0x491034)
                                return _0xc1b1da;
                        }
                    }
                };
            }, _0x16fb43 = (_0x9d8488, _0x3675af = 0x15) => _0x451e35(_0x9d8488, _0x3675af, _0x4c26f8), _0x5b93eb = (_0x57db7e = 0x15) => {
                _0x4d9a11(_0x57db7e -= 0x0);
                let _0x127c9c = '';
                for (let _0xe40e92 = _0x1098e8 - _0x57db7e; _0xe40e92 < _0x1098e8; _0xe40e92++) {
                    _0x127c9c += _0xbc56a9['urlAlphabet'][_0x2a7b47[_0xe40e92] & 0x3f];
                }
                return _0x127c9c;
            };
    },
    0x12bca: (_0x253c91, _0x509eeb, _0x2695fc) => {
        _0x2695fc['r'](_0x509eeb), _0x2695fc['d'](_0x509eeb, { 'urlAlphabet': () => _0x4a8467 });
        let _0x4a8467 = 'useandom-26T198340PX75pxJACKVERYMINDBUSHWOLF_GQZbfghjklqvwyzrict';
    }
};
;
