// test-maker/dist/680.js
'use strict';
exports['id'] = 0x2a8, exports['ids'] = [0x2a8], exports['modules'] = {
    0x9ee8: (_0xd1c6d2, _0x23b06c, _0x1a836c) => {
        _0x1a836c['r'](_0x23b06c);
        var _0x136d48 = _0x1a836c(0x177ce), _0x2c0f08 = _0x1a836c['n'](_0x136d48), _0x7902b2 = _0x1a836c(0xed2f), _0x3d8c2d = _0x1a836c['n'](_0x7902b2), _0x2392d1 = _0x1a836c(0x10ce7), _0x356ae2 = _0x1a836c(0xbb05), _0xeb8c94 = _0x1a836c(0x17f4f);
        ;
        async function _0x2cef36({
            error: _0x434c58,
            type: _0x3ebd5a,
            exit: exit = !![]
        }) {
            let _0x4ce93c;
            try {
                _0x4ce93c ? console['error'](_0x4ce93c['formatted']['ansi']) : console['error'](_0x434c58['stack'] || _0x434c58);
            } catch (_0x37a77b) {
                console['error'](_0x434c58['stack'] || _0x434c58);
            } finally {
                console['info']('See\x20Log\x20file\x20for\x20more\x20Details...');
                if (exit)
                    (0x0, _0xeb8c94['exitProcess'])(0x1);
            }
        }
        var _0x2bc808 = _0x1a836c(0x1145c), _0x51cfe5 = _0x1a836c['n'](_0x2bc808);
        ;
        const _0x2f52fe = '\x0a████████╗███████╗███████╗████████╗\x20\x20\x20\x20███╗\x20\x20\x20███╗\x20█████╗\x20██╗\x20\x20██╗███████╗██████╗\x20\x0a╚══██╔══╝██╔════╝██╔════╝╚══██╔══╝\x20\x20\x20\x20████╗\x20████║██╔══██╗██║\x20██╔╝██╔════╝██╔══██╗\x0a\x20\x20\x20██║\x20\x20\x20█████╗\x20\x20███████╗\x20\x20\x20██║\x20\x20\x20\x20\x20\x20\x20██╔████╔██║███████║█████╔╝\x20█████╗\x20\x20██████╔╝\x0a\x20\x20\x20██║\x20\x20\x20██╔══╝\x20\x20╚════██║\x20\x20\x20██║\x20\x20\x20\x20\x20\x20\x20██║╚██╔╝██║██╔══██║██╔═██╗\x20██╔══╝\x20\x20██╔══██╗\x0a\x20\x20\x20██║\x20\x20\x20███████╗███████║\x20\x20\x20██║\x20\x20\x20\x20\x20\x20\x20██║\x20╚═╝\x20██║██║\x20\x20██║██║\x20\x20██╗███████╗██║\x20\x20██║\x0a\x20\x20\x20╚═╝\x20\x20\x20╚══════╝╚══════╝\x20\x20\x20╚═╝\x20\x20\x20\x20\x20\x20\x20╚═╝\x20\x20\x20\x20\x20╚═╝╚═╝\x20\x20╚═╝╚═╝\x20\x20╚═╝╚══════╝╚═╝\x20\x20╚═╝\x20\x20\x20\x0a', _0xfc28af = () => {
                console['log'](_0x51cfe5()['hex']('#ff3838')['bold'](_0x2f52fe));
            };
        ;
        const _0x22ceb7 = process['argv']['indexOf']('--terminal-eye'), _0x5281b3 = process['argv']['indexOf']('--terminal-eye-slave') > -0x1;
        function _0x3cfd23(_0x433e1f) {
            process['on']('unhandledRejection', async _0x3c0995 => await _0x2cef36({
                'error': _0x3c0995,
                'type': 'unhandledRejection',
                'exit': _0x433e1f['configuration']['runner']['process']['exitOnUnhandledErrors']
            }))['on']('uncaughtException', async _0x533ddb => await _0x2cef36({
                'error': _0x533ddb,
                'type': 'uncaughtException',
                'exit': _0x433e1f['configuration']['runner']['process']['exitOnUnhandledErrors']
            }));
        }
        async function _0x14afda() {
            const _0x1b8768 = [
                Promise['all']([
                    _0x1a836c['e'](0xee),
                    _0x1a836c['e'](0x2dd),
                    _0x1a836c['e'](0x21)
                ])['then'](_0x1a836c['bind'](_0x1a836c, 0xdae1)),
                Promise['all']([
                    _0x1a836c['e'](0xee),
                    _0x1a836c['e'](0x83),
                    _0x1a836c['e'](0xfc),
                    _0x1a836c['e'](0x2dd),
                    _0x1a836c['e'](0x7b),
                    _0x1a836c['e'](0x2e6),
                    _0x1a836c['e'](0x34d),
                    _0x1a836c['e'](0x1e1)
                ])['then'](_0x1a836c['bind'](_0x1a836c, 0x17067)),
                Promise['all']([
                    _0x1a836c['e'](0xee),
                    _0x1a836c['e'](0x83),
                    _0x1a836c['e'](0xfc),
                    _0x1a836c['e'](0x2dd),
                    _0x1a836c['e'](0x7b),
                    _0x1a836c['e'](0x2e6),
                    _0x1a836c['e'](0x34d),
                    _0x1a836c['e'](0x1e1)
                ])['then'](_0x1a836c['bind'](_0x1a836c, 0x14e4b))
            ];
            if (_0x5281b3)
                _0x1b8768['push'](Promise['all']([
                    _0x1a836c['e'](0x83),
                    _0x1a836c['e'](0x13b)
                ])['then'](_0x1a836c['bind'](_0x1a836c, 0x1495b)));
            const _0x2aa685 = await Promise['all'](_0x1b8768);
            if (_0x5281b3)
                _0x2aa685[0x1]['stateManager']['state']['terminalDashBoard'] = !![];
            _0x3cfd23(_0x2aa685[0x2]['configManager']);
        }
        async function _0x5cd526() {
            const _0x2c4f23 = [
                    'run',
                    'init',
                    'generate',
                    'validate'
                ], _0x28ad65 = process['argv'][0x2];
            process['title'] = 'Test\x20Maker', process['env']['FORCE_COLOR'] = '3', _0xfc28af();
            if (_0x2c4f23['includes'](_0x28ad65))
                await _0x14afda();
            let _0x158eae = Promise['all']([
                    _0x1a836c['e'](0x39d),
                    _0x1a836c['e'](0x7b),
                    _0x1a836c['e'](0x67),
                    _0x1a836c['e'](0x1fd),
                    _0x1a836c['e'](0x3a4)
                ])['then'](_0x1a836c['t']['bind'](_0x1a836c, 0xc165, 0x17)), _0x1a3046 = await Promise['all']([
                    _0x158eae,
                    Promise['all']([
                        _0x1a836c['e'](0x39d),
                        _0x1a836c['e'](0x7b),
                        _0x1a836c['e'](0x67),
                        _0x1a836c['e'](0x1fd),
                        _0x1a836c['e'](0x200)
                    ])['then'](_0x1a836c['bind'](_0x1a836c, 0x466f)),
                    Promise['all']([
                        _0x1a836c['e'](0xee),
                        _0x1a836c['e'](0x39d),
                        _0x1a836c['e'](0x7b),
                        _0x1a836c['e'](0x67),
                        _0x1a836c['e'](0x1fd),
                        _0x1a836c['e'](0x7f),
                        _0x1a836c['e'](0x14e)
                    ])['then'](_0x1a836c['bind'](_0x1a836c, 0xc055)),
                    Promise['all']([
                        _0x1a836c['e'](0xee),
                        _0x1a836c['e'](0x39d),
                        _0x1a836c['e'](0x7b),
                        _0x1a836c['e'](0x67),
                        _0x1a836c['e'](0x18d),
                        _0x1a836c['e'](0x1fd),
                        _0x1a836c['e'](0x10a)
                    ])['then'](_0x1a836c['bind'](_0x1a836c, 0xbe3))
                ]);
            const _0x48d972 = _0x1a3046[0x0]['program'];
            try {
                await _0x48d972['version'](_0x356ae2['testMakerPackageJson']['version'])['name']('ketm')['option']('--no-bundle', '', {
                    'global': !![],
                    'hidden': !![]
                })['option']('--pnpm', '', {
                    'global': !![],
                    'hidden': !![]
                })['option']('--devtools', '', {
                    'global': !![],
                    'hidden': !![]
                })['option']('--devtools-template', '', {
                    'global': !![],
                    'hidden': !![]
                })['help']('Test\x20Maker\x20CLI')['run']();
            } catch (_0x376437) {
                _0x2c0f08()['error'](_0x376437['message']['indexOf']('Synopsis') > -0x1 ? _0x376437['message']['substring'](0x0, _0x376437['message']['indexOf']('Synopsis'))['trim']() : _0x376437), (0x0, _0xeb8c94['exitProcess'])(0x1);
            }
        }
        async function _0x1a5f4e(_0x3df6f9, _0x432f08) {
            const _0x5ba91a = process['argv']['splice'](0x1), _0x567f95 = _0x356ae2['testMakerArtifactsPath'] + '/profiling/' + (0x0, _0x2392d1['format'])(new Date(), 'DD-MM-YYYY\x20hh-mm-ss\x20A');
            _0x3df6f9 > -0x1 && (_0x5ba91a['splice'](_0x5ba91a['indexOf']('--profile-cpu'), 0x1), _0x5ba91a['splice'](0x0, 0x0, '--cpu-prof', '--cpu-prof-dir', _0x567f95)), _0x432f08 > -0x1 && (_0x5ba91a['splice'](_0x5ba91a['indexOf']('--profile-memory'), 0x1), _0x5ba91a['splice'](0x1, 0x0, '--heap-prof', '--heap-prof-dir', _0x567f95)), await _0x3d8c2d()('node', _0x5ba91a, { 'stdio': 'inherit' });
        }
        ((async () => {
            try {
                const _0x3386f5 = process['argv']['indexOf']('--profile-cpu'), _0xd86d5e = process['argv']['indexOf']('--profile-memory');
                if (_0x3386f5 > -0x1 || _0xd86d5e > -0x1) {
                    await _0x1a5f4e(_0x3386f5, _0xd86d5e);
                    return;
                }
                _0x22ceb7 > -0x1 ? await (await Promise['all']([
                    _0x1a836c['e'](0x83),
                    _0x1a836c['e'](0xe8)
                ])['then'](_0x1a836c['bind'](_0x1a836c, 0xa4f8)))['terminalDashboard']['startDashboard'](_0x22ceb7) : await _0x5cd526();
            } catch (_0xf6a2f7) {
                _0x2c0f08()['error'](_0xf6a2f7);
            }
        })());
    },
    0xbb05: (_0x107b27, _0x1c2973, _0xd0daf1) => {
        _0xd0daf1['r'](_0x1c2973), _0xd0daf1['d'](_0x1c2973, {
            'allRetriesFailedMessage': () => _0x33307a,
            'globalTestMakerConfigMangerName': () => _0x48b876,
            'globalTestMakerEventsMangerName': () => _0x54af37,
            'globalTestMakerInstanceMangerName': () => _0x5b90f7,
            'globalTestMakerLogMangerName': () => _0x34fb0b,
            'globalTestMakerLoggerName': () => _0x19b3ac,
            'globalTestMakerName': () => _0x4d71d4,
            'globalTestMakerPluginsName': () => _0x28dd93,
            'globalTestMakerStateMangerName': () => _0x2c4318,
            'isStudio': () => _0xb47758,
            'keywordTimeoutMessage': () => _0x4937cb,
            'minimumSelectorTimeOut': () => _0x175375,
            'modulesPaths': () => _0x51d85c,
            'projectPath': () => _0xb31393,
            'selectorNotVisibleMessage': () => _0x2aa8c6,
            'selectorNotfoundMessage': () => _0x28077b,
            'selectorSessionErrorMessage': () => _0x1c1007,
            'testMakerAdapterSelectorName': () => _0x3f54d9,
            'testMakerArtifactsPath': () => _0x4fc799,
            'testMakerBundledDepsPath': () => _0x17ed6e,
            'testMakerDistPath': () => _0x1ee4ed,
            'testMakerElementUidName': () => _0x541868,
            'testMakerFrameElementUidName': () => _0x15f5c7,
            'testMakerInstallPath': () => _0x56a9df,
            'testMakerPackageJson': () => _0x38d5cd,
            'testMakerPackageName': () => _0x49fdea,
            'tmLoggerContentStrip': () => _0x28f558,
            'tmLoggerFileStrip': () => _0x5c8aaf
        });
        var _0x4ee773 = _0xd0daf1(0x10e48), _0x57827f = _0xd0daf1(0x3b1), _0x4f3366 = _0xd0daf1['n'](_0x57827f), _0x1f45c8;
        const _0x38d5cd = _0x4ee773, _0x49fdea = _0x38d5cd['name'], _0x34fb0b = '__TEST_MAKER__LOG__MANAGER__', _0x48b876 = '__TEST_MA\x20\x20\x20KER__CONFIG__MANGER__', _0x5b90f7 = '__TEST_MAKER__Instance__MANGER__', _0x2c4318 = '__TEST_MAKER__STATE__MANGER__', _0x54af37 = '__TEST_MAKER__EVENTS__MANGER__', _0x4d71d4 = '__TEST__MAKER__', _0x28dd93 = '__TEST__MAKER__PLUGINS__', _0x19b3ac = '__TEST_MAKER__logger__', _0x3f54d9 = '__TM_ADAPTER_SELECTOR__', _0x541868 = 'data-test-maker-id', _0x15f5c7 = 'data-test-maker-frame-id', _0x28077b = 'Could\x20not\x20obtain\x20information\x20about\x20the\x20Selector\x20because\x20the\x20specified\x20selector\x20does\x20not\x20match\x20any\x20node\x20in\x20the\x20DOM\x20tree.\x20Make\x20sure\x20the\x20Selector\x20exists\x20or\x20the\x20operation\x20is\x20performed\x20in\x20the\x20correct\x20Frame', _0x2aa8c6 = 'Selector\x20exists\x20but\x20not\x20visible,\x20if\x20visibilityCheck\x20is\x20not\x20required\x20disable\x20it\x20in\x20Selector\x27s\x20options.', _0x1c1007 = 'Could\x20not\x20resolve\x20Selector\x20Session.\x20make\x20sure\x20you\x20are\x20in\x20Client\x20context\x20when\x20executing\x20any\x20methods\x20or\x20properties', _0x33307a = 'All\x20retries\x20failed\x20after', _0x4937cb = 'Keyword\x20Timeout\x20Error', _0x175375 = 0x64, _0x28f558 = '<tm-logger>content</tm-logger>', _0x5c8aaf = '<tm-logger>file-content</tm-logger>', _0xb31393 = typeof process !== 'undefined' ? _0x4f3366()['normalizeTrim'](process['cwd']()) : '', _0x56a9df = typeof process !== 'undefined' ? _0x4f3366()['normalizeTrim'](_0x4f3366()['dirname'](_0x4f3366()['dirname'](((_0x1f45c8 = process['mainModule']) === null || _0x1f45c8 === void 0x0 ? void 0x0 : _0x1f45c8['filename']) || ''))) : '', _0x4fc799 = typeof process !== 'undefined' ? _0x4f3366()['joinSafe'](_0xb31393, '.test-maker') : '', _0x1ee4ed = _0x56a9df + '/dist', _0x17ed6e = _0x56a9df + '/node_modules', _0x51d85c = [
                'node_modules',
                _0x17ed6e
            ], _0xb47758 = typeof process !== 'undefined' ? process['env']['isStudio'] : ![];
    },
    0x17f4f: (_0x5c9c3e, _0x268992, _0x16163e) => {
        _0x16163e['r'](_0x268992), _0x16163e['d'](_0x268992, { 'exitProcess': () => _0x3e8d86 });
        function _0x3e8d86(_0x20eab5, _0x473989 = !![]) {
            if (_0x473989)
                console['log']('\x0aExiting\x20Test\x20Maker\x20Process.\x0a');
            process['exit'](_0x20eab5);
        }
    }
};
;
