// test-maker/dist/8.js
'use strict';
exports['id'] = 0x8, exports['ids'] = [0x8], exports['modules'] = {
    0x14440: (_0x1e191c, _0x220ccf, _0x560b0c) => {
        _0x560b0c['r'](_0x220ccf), _0x560b0c['d'](_0x220ccf, { 'ReportersManager': () => _0x223f0b });
        var _0x5ee549 = _0x560b0c(0x1145c), _0xc8890a = _0x560b0c['n'](_0x5ee549), _0x30b3fa = _0x560b0c(0x17067), _0x22128d = _0x560b0c(0x2fa5), _0x58c3d0 = _0x560b0c(0xc491), _0x1a2f2a = _0x560b0c(0xbe8a), _0x3c72bd = _0x560b0c(0x79cc);
        class _0x223f0b {
            constructor() {
                this['reporters'] = new Map(), this['registerEvents']();
            }
            async ['install'](_0x5a9825, _0x22a411) {
                var _0x273ae5, _0x222dcf;
                this['engine'] = _0x5a9825, this['pluginsManager'] = _0x22a411, this['reporters'] = this['engine']['reporters'];
                for (let _0x13792c of this['reporters']['values']()) {
                    const _0x653e1 = await ((_0x222dcf = (_0x273ae5 = _0x13792c['reporter'])['install']) === null || _0x222dcf === void 0x0 ? void 0x0 : _0x222dcf['call'](_0x273ae5, _0x5a9825, _0x13792c['options']));
                    if (_0x653e1)
                        _0x30b3fa['stateManager']['state']['installQueue']['push'](..._0x653e1);
                }
            }
            async ['init'](_0x2bd20f, _0x227260) {
                this['engine'] = _0x2bd20f, this['pluginsManager'] = _0x227260, this['reporters'] = this['engine']['reporters'];
                for (let _0x2e73a8 of this['reporters']['values']())
                    await _0x2e73a8['reporter']['init'](_0x2bd20f, _0x2e73a8['options']);
            }
            async ['taskStart'](_0x4421e5) {
                var _0xa43b35, _0x3dea00;
                try {
                    for (let _0x22ff4c of this['reporters']['values']())
                        await ((_0x3dea00 = (_0xa43b35 = _0x22ff4c['reporter'])['taskStart']) === null || _0x3dea00 === void 0x0 ? void 0x0 : _0x3dea00['call'](_0xa43b35, _0x4421e5));
                    await this['pluginsManager']['runTaskStart']();
                } catch (_0x556676) {
                    await (0x0, _0x3c72bd['logGenericError'])({ 'error': _0x556676 });
                }
            }
            async ['featureStart'](_0x566527, _0x2318d8) {
                var _0x570a95, _0x436de5;
                try {
                    for (let _0x15c83e of this['reporters']['values']())
                        await ((_0x436de5 = (_0x570a95 = _0x15c83e['reporter'])['featureStart']) === null || _0x436de5 === void 0x0 ? void 0x0 : _0x436de5['call'](_0x570a95, _0x566527, _0x2318d8));
                } catch (_0x54d499) {
                    await (0x0, _0x3c72bd['logGenericError'])({ 'error': _0x54d499 });
                }
            }
            async ['featureDone'](_0x2a1e42, _0x197c46) {
                var _0x57cc73, _0x449e98;
                try {
                    for (let _0x2b010a of this['reporters']['values']())
                        await ((_0x449e98 = (_0x57cc73 = _0x2b010a['reporter'])['featureDone']) === null || _0x449e98 === void 0x0 ? void 0x0 : _0x449e98['call'](_0x57cc73, _0x2a1e42, _0x197c46));
                } catch (_0x1d82e7) {
                    await (0x0, _0x3c72bd['logGenericError'])({ 'error': _0x1d82e7 });
                }
            }
            async ['scenarioStart'](_0x2cf976, _0x3c991f) {
                var _0x2ce624, _0x299d01;
                try {
                    for (let _0xa7b17a of this['reporters']['values']())
                        await ((_0x299d01 = (_0x2ce624 = _0xa7b17a['reporter'])['scenarioStart']) === null || _0x299d01 === void 0x0 ? void 0x0 : _0x299d01['call'](_0x2ce624, _0x2cf976, _0x3c991f));
                } catch (_0x3007fc) {
                    await (0x0, _0x3c72bd['logGenericError'])({ 'error': _0x3007fc });
                }
            }
            async ['scenarioDone'](_0x27e7bd, _0x5d6b86) {
                var _0x28cf35, _0x174cb9;
                try {
                    for (let _0xe7c4d5 of this['reporters']['values']())
                        await ((_0x174cb9 = (_0x28cf35 = _0xe7c4d5['reporter'])['scenarioDone']) === null || _0x174cb9 === void 0x0 ? void 0x0 : _0x174cb9['call'](_0x28cf35, _0x27e7bd, _0x5d6b86));
                } catch (_0x16e782) {
                    await (0x0, _0x3c72bd['logGenericError'])({ 'error': _0x16e782 });
                }
            }
            async ['stepStart'](_0x3e1408, _0x1bfd37) {
                var _0x11e4c2, _0x2c23c3;
                try {
                    for (let _0x41a771 of this['reporters']['values']())
                        await ((_0x2c23c3 = (_0x11e4c2 = _0x41a771['reporter'])['stepStart']) === null || _0x2c23c3 === void 0x0 ? void 0x0 : _0x2c23c3['call'](_0x11e4c2, _0x3e1408, _0x1bfd37));
                } catch (_0x3ccb4a) {
                    await (0x0, _0x3c72bd['logGenericError'])({ 'error': _0x3ccb4a });
                }
            }
            async ['stepDone'](_0x330695, _0x5c6a43) {
                var _0x66649c, _0x49d6af;
                try {
                    for (let _0x1bfbed of this['reporters']['values']())
                        await ((_0x49d6af = (_0x66649c = _0x1bfbed['reporter'])['stepDone']) === null || _0x49d6af === void 0x0 ? void 0x0 : _0x49d6af['call'](_0x66649c, _0x330695, _0x5c6a43));
                } catch (_0x29daf3) {
                    await (0x0, _0x3c72bd['logGenericError'])({ 'error': _0x29daf3 });
                }
            }
            async ['subStepStart'](_0x499143, _0x1928a8) {
                var _0x16deb8, _0x2f2945;
                try {
                    for (let _0x4ac30c of this['reporters']['values']())
                        await ((_0x2f2945 = (_0x16deb8 = _0x4ac30c['reporter'])['subStepStart']) === null || _0x2f2945 === void 0x0 ? void 0x0 : _0x2f2945['call'](_0x16deb8, _0x499143, _0x1928a8));
                } catch (_0x3c1665) {
                    await (0x0, _0x3c72bd['logGenericError'])({ 'error': _0x3c1665 });
                }
            }
            async ['subStepDone'](_0x337e82, _0x4c4d9e) {
                var _0x10c45c, _0x5e4fb7;
                try {
                    for (let _0x53f900 of this['reporters']['values']())
                        await ((_0x5e4fb7 = (_0x10c45c = _0x53f900['reporter'])['subStepDone']) === null || _0x5e4fb7 === void 0x0 ? void 0x0 : _0x5e4fb7['call'](_0x10c45c, _0x337e82, _0x4c4d9e));
                } catch (_0x5a7b66) {
                    await (0x0, _0x3c72bd['logGenericError'])({ 'error': _0x5a7b66 });
                }
            }
            async ['hookStart'](_0x162c3a, _0x42f3d0, _0x278138) {
                var _0x43a053, _0x3717d3;
                try {
                    for (let _0x492f60 of this['reporters']['values']())
                        await ((_0x3717d3 = (_0x43a053 = _0x492f60['reporter'])['hookStart']) === null || _0x3717d3 === void 0x0 ? void 0x0 : _0x3717d3['call'](_0x43a053, _0x162c3a, _0x42f3d0, _0x278138));
                } catch (_0x2ddea3) {
                    await (0x0, _0x3c72bd['logGenericError'])({ 'error': _0x2ddea3 });
                }
            }
            async ['hookDone'](_0x1fb43f, _0x2655e3, _0x15e30e) {
                var _0x4b22db, _0xb3fce0;
                try {
                    for (let _0x2c99c0 of this['reporters']['values']())
                        await ((_0xb3fce0 = (_0x4b22db = _0x2c99c0['reporter'])['hookDone']) === null || _0xb3fce0 === void 0x0 ? void 0x0 : _0xb3fce0['call'](_0x4b22db, _0x1fb43f, _0x2655e3, _0x15e30e));
                } catch (_0x351e04) {
                    await (0x0, _0x3c72bd['logGenericError'])({ 'error': _0x351e04 });
                }
            }
            async ['selectorActionStart'](_0x5b3186) {
                var _0x132240, _0x4009db;
                try {
                    for (let _0x1ac559 of this['reporters']['values']())
                        await ((_0x4009db = (_0x132240 = _0x1ac559['reporter'])['selectorActionStart']) === null || _0x4009db === void 0x0 ? void 0x0 : _0x4009db['call'](_0x132240, _0x5b3186));
                } catch (_0x997d6e) {
                    await (0x0, _0x3c72bd['logGenericError'])({ 'error': _0x997d6e });
                }
            }
            async ['selectorActionDone'](_0x335a14) {
                var _0xb6d6d2, _0xced5b1;
                try {
                    for (let _0x1b72d2 of this['reporters']['values']())
                        await ((_0xced5b1 = (_0xb6d6d2 = _0x1b72d2['reporter'])['selectorActionDone']) === null || _0xced5b1 === void 0x0 ? void 0x0 : _0xced5b1['call'](_0xb6d6d2, _0x335a14));
                } catch (_0x29d873) {
                    await (0x0, _0x3c72bd['logGenericError'])({ 'error': _0x29d873 });
                }
            }
            async ['controllerActionStart'](_0x1cef41) {
                var _0x48083a, _0x1a8107;
                try {
                    for (let _0x24943 of this['reporters']['values']())
                        await ((_0x1a8107 = (_0x48083a = _0x24943['reporter'])['controllerActionStart']) === null || _0x1a8107 === void 0x0 ? void 0x0 : _0x1a8107['call'](_0x48083a, _0x1cef41));
                } catch (_0x51b189) {
                    await (0x0, _0x3c72bd['logGenericError'])({ 'error': _0x51b189 });
                }
            }
            async ['controllerActionDone'](_0x2f9c26) {
                var _0xe64b7, _0x43ff9d;
                try {
                    for (let _0x2a0147 of this['reporters']['values']())
                        await ((_0x43ff9d = (_0xe64b7 = _0x2a0147['reporter'])['controllerActionDone']) === null || _0x43ff9d === void 0x0 ? void 0x0 : _0x43ff9d['call'](_0xe64b7, _0x2f9c26));
                } catch (_0x2d1685) {
                    await (0x0, _0x3c72bd['logGenericError'])({ 'error': _0x2d1685 });
                }
            }
            async ['assertionActionStart'](_0x4bb7e4) {
                var _0x37f0e6, _0x19bc58;
                try {
                    for (let _0x39dc6f of this['reporters']['values']())
                        await ((_0x19bc58 = (_0x37f0e6 = _0x39dc6f['reporter'])['assertionActionStart']) === null || _0x19bc58 === void 0x0 ? void 0x0 : _0x19bc58['call'](_0x37f0e6, _0x4bb7e4));
                } catch (_0x1feeed) {
                    await (0x0, _0x3c72bd['logGenericError'])({ 'error': _0x1feeed });
                }
            }
            async ['assertionActionDone'](_0x24a97b) {
                var _0x4b17f9, _0xf504aa;
                try {
                    for (let _0x14b113 of this['reporters']['values']())
                        await ((_0xf504aa = (_0x4b17f9 = _0x14b113['reporter'])['assertionActionDone']) === null || _0xf504aa === void 0x0 ? void 0x0 : _0xf504aa['call'](_0x4b17f9, _0x24a97b));
                } catch (_0x150d6a) {
                    await (0x0, _0x3c72bd['logGenericError'])({ 'error': _0x150d6a });
                }
            }
            async ['taskDone'](_0x155cc8) {
                var _0x58c56e, _0x4b1016, _0x92f26c, _0x3b17c5;
                try {
                    for (let _0x3d66d4 of this['reporters']['values']())
                        await ((_0x4b1016 = (_0x58c56e = _0x3d66d4['reporter'])['taskDone']) === null || _0x4b1016 === void 0x0 ? void 0x0 : _0x4b1016['call'](_0x58c56e, _0x155cc8));
                    await this['pluginsManager']['runTaskDone'](), ((_0x3b17c5 = (_0x92f26c = this['engine']['configuration']['gherkin']) === null || _0x92f26c === void 0x0 ? void 0x0 : _0x92f26c['parser']) === null || _0x3b17c5 === void 0x0 ? void 0x0 : _0x3b17c5['enabled']) && (console['log'](_0xc8890a()['yellow']('Generating\x20Acceptance\x20Criteria\x20analysis\x20Report')), _0x1a2f2a['specsAnalyzer']['analyze'](this['engine']['specs']));
                } catch (_0x2d087d) {
                    await (0x0, _0x3c72bd['logGenericError'])({ 'error': _0x2d087d });
                }
            }
            ['registerEvents']() {
                _0x58c3d0['tmEventsManger']['on'](_0x22128d['TMEvents']['ReporterTaskStart'], async _0x25b6a6 => await this['taskStart'](_0x25b6a6)), _0x58c3d0['tmEventsManger']['on'](_0x22128d['TMEvents']['ReporterTaskDone'], async _0x4dd00d => await this['taskDone'](_0x4dd00d)), _0x58c3d0['tmEventsManger']['on'](_0x22128d['TMEvents']['ReporterFeatureStart'], async _0x3d328d => await this['featureStart'](_0x3d328d['feature'], _0x3d328d['info'])), _0x58c3d0['tmEventsManger']['on'](_0x22128d['TMEvents']['ReporterFeatureDone'], async _0x5a49bb => await this['featureDone'](_0x5a49bb['feature'], _0x5a49bb['info'])), _0x58c3d0['tmEventsManger']['on'](_0x22128d['TMEvents']['ReporterScenarioStart'], async _0x213074 => await this['scenarioStart'](_0x213074['scenario'], _0x213074['info'])), _0x58c3d0['tmEventsManger']['on'](_0x22128d['TMEvents']['ReporterScenarioDone'], async _0x53141b => await this['scenarioDone'](_0x53141b['scenario'], _0x53141b['info'])), _0x58c3d0['tmEventsManger']['on'](_0x22128d['TMEvents']['ReporterStepStart'], async _0x44bf05 => await this['stepStart'](_0x44bf05['step'], _0x44bf05['info'])), _0x58c3d0['tmEventsManger']['on'](_0x22128d['TMEvents']['ReporterStepDone'], async _0xd66809 => await this['stepDone'](_0xd66809['step'], _0xd66809['info'])), _0x58c3d0['tmEventsManger']['on'](_0x22128d['TMEvents']['ReporterSubStepStart'], async _0x3b28d => await this['subStepStart'](_0x3b28d['subStep'], _0x3b28d['info'])), _0x58c3d0['tmEventsManger']['on'](_0x22128d['TMEvents']['ReporterSubStepDone'], async _0x1ce506 => await this['subStepDone'](_0x1ce506['subStep'], _0x1ce506['info'])), _0x58c3d0['tmEventsManger']['on'](_0x22128d['TMEvents']['ReporterHookStart'], async _0x48dd26 => await this['hookStart'](_0x48dd26['name'], _0x48dd26['spec'], _0x48dd26['info'])), _0x58c3d0['tmEventsManger']['on'](_0x22128d['TMEvents']['ReporterHookDone'], async _0x50bd1b => await this['hookDone'](_0x50bd1b['name'], _0x50bd1b['spec'], _0x50bd1b['info'])), _0x58c3d0['tmEventsManger']['on'](_0x22128d['TMEvents']['ReporterSelectorActionStart'], async _0x59f816 => await this['selectorActionStart'](_0x59f816)), _0x58c3d0['tmEventsManger']['on'](_0x22128d['TMEvents']['ReporterSelectorActionDone'], async _0xb1a261 => await this['selectorActionDone'](_0xb1a261)), _0x58c3d0['tmEventsManger']['on'](_0x22128d['TMEvents']['ReporterControllerActionStart'], async _0x16a265 => await this['controllerActionStart'](_0x16a265)), _0x58c3d0['tmEventsManger']['on'](_0x22128d['TMEvents']['ReporterControllerActionDone'], async _0x2a1a73 => await this['controllerActionDone'](_0x2a1a73)), _0x58c3d0['tmEventsManger']['on'](_0x22128d['TMEvents']['ReporterAssertionActionStart'], async _0x28c0b4 => await this['assertionActionStart'](_0x28c0b4)), _0x58c3d0['tmEventsManger']['on'](_0x22128d['TMEvents']['ReporterAssertionActionDone'], async _0x5d5638 => await this['assertionActionDone'](_0x5d5638));
            }
        }
    }
};
;
