// test-maker/dist/feature-loader.js
(function webpackUniversalModuleDefinition(_0x2d3fb8, _0x107408) {
    if (typeof exports === 'object' && typeof module === 'object')
        module['exports'] = _0x107408();
    else {
        if (typeof define === 'function' && define['amd'])
            define([], _0x107408);
        else {
            var _0x1921c1 = _0x107408();
            for (var _0x21d0f6 in _0x1921c1)
                (typeof exports === 'object' ? exports : _0x2d3fb8)[_0x21d0f6] = _0x1921c1[_0x21d0f6];
        }
    }
}(this, () => {
    return ((() => {
        'use strict';
        var _0x3ff4ab = {
                0x14052: (_0x4ee056, _0xa8c407, _0x5d2290) => {
                    _0x5d2290['r'](_0xa8c407), _0x5d2290['d'](_0xa8c407, {
                        'FeatureLoader': () => _0x1760ed,
                        'default': () => _0x4b3bf8
                    });
                    var _0x3f0a6f = _0x5d2290(0x1145c), _0x2c0373 = _0x5d2290['n'](_0x3f0a6f), _0x515819 = _0x5d2290(0x3942), _0x26120e = _0x5d2290(0x17fbe), _0x5680d5 = _0x5d2290['n'](_0x26120e), _0x36fd7f = _0x5d2290(0x10fb0);
                    ;
                    const _0x1b39dd = require('typescript');
                    var _0x2f73c9 = _0x5d2290(0x3b1);
                    ;
                    var _0x3cb780;
                    (function (_0x5b2974) {
                        _0x5b2974['Feature'] = 'Feature', _0x5b2974['Background'] = 'Background', _0x5b2974['Example'] = 'Example', _0x5b2974['Scenario'] = 'Scenario', _0x5b2974['ScenarioOutline'] = 'ScenarioOutline', _0x5b2974['Given'] = 'Given', _0x5b2974['When'] = 'When', _0x5b2974['And'] = 'And', _0x5b2974['Then'] = 'Then', _0x5b2974['But'] = 'But';
                    }(_0x3cb780 || (_0x3cb780 = {})));
                    ;
                    const _0x32dda0 = [_0x3cb780['Feature']], _0x5a4638 = [
                            _0x3cb780['Scenario'],
                            _0x3cb780['ScenarioOutline'],
                            _0x3cb780['Example'],
                            _0x3cb780['Background']
                        ], _0x2ac922 = [
                            _0x3cb780['Given'],
                            _0x3cb780['And'],
                            _0x3cb780['When'],
                            _0x3cb780['Then'],
                            _0x3cb780['But']
                        ], _0x34b39c = [
                            ..._0x32dda0,
                            ..._0x5a4638,
                            ..._0x2ac922
                        ];
                    ;
                    var _0x3c46ef;
                    (function (_0x23c868) {
                        _0x23c868['TestMaker'] = 'Test\x20Maker', _0x23c868['BeforeAll'] = 'Before\x20All', _0x23c868['AfterAll'] = 'After\x20All', _0x23c868['Feature'] = 'Feature', _0x23c868['BeforeEachFeature'] = 'Before\x20Each\x20Feature', _0x23c868['BeforeFeature'] = 'Before\x20Feature', _0x23c868['AfterFeature'] = 'After\x20Feature', _0x23c868['AfterEachFeature'] = 'After\x20Each\x20Feature', _0x23c868['Scenario'] = 'Scenario', _0x23c868['BeforeEachScenario'] = 'Before\x20Each\x20Scenario', _0x23c868['BeforeScenario'] = 'Before\x20Scenario', _0x23c868['AfterScenario'] = 'After\x20Scenario', _0x23c868['AfterEachScenario'] = 'After\x20Each\x20Scenario', _0x23c868['Step'] = 'Step', _0x23c868['BeforeEachStep'] = 'Before\x20Each\x20Step\x20In\x20Scenario', _0x23c868['AfterEachStep'] = 'After\x20Each\x20Step\x20In\x20Scenario', _0x23c868['SupStep'] = 'Sup\x20Step';
                    }(_0x3c46ef || (_0x3c46ef = {})));
                    ;
                    function _0x3c88b8(_0x246bde, _0xeb4d3a = !![]) {
                        if (_0xeb4d3a)
                            console['log']('\x0aExiting\x20Test\x20Maker\x20Process.\x0a');
                        process['exit'](_0x246bde);
                    }
                    var _0x48082d = _0x5d2290(0x16925), _0x1705c7 = _0x5d2290(0xccb6);
                    ;
                    function _0x42082a(_0x1ff11a) {
                        return (0x0, _0x48082d['nanoid'])(_0x1ff11a);
                    }
                    async function _0x5a4bbf(_0x20010d) {
                        return await (0x0, _0x1705c7['nanoid'])(_0x20010d);
                    }
                    function _0x1026dd(_0x3ed9da, _0x5d7eb4) {
                        return (0x0, _0x48082d['customAlphabet'])(_0x3ed9da, _0x5d7eb4)();
                    }
                    async function _0x27a5ed(_0x5aef22, _0x189a58) {
                        return await (0x0, _0x1705c7['customAlphabet'])(_0x5aef22, _0x189a58)();
                    }
                    var _0x249af2 = _0x5d2290(0x9e6);
                    ;
                    class _0x1760ed {
                        constructor(_0x5adb20, _0x365c0c, _0x12392c) {
                            this['source'] = '', this['tsSource'] = '', this['traverse'] = _0x515819['traverse'], this['feature'] = {
                                'id': '',
                                'name': '',
                                'didRun': ![],
                                'keyword': _0x3cb780['Feature'],
                                'only': ![],
                                'skip': ![],
                                'isLast': !![],
                                'sourcePath': '',
                                'distPath': '',
                                'OriginalSource': '',
                                'location': {
                                    'start': {
                                        'line': 0x1,
                                        'column': 0x1
                                    },
                                    'end': {
                                        'line': 0x1,
                                        'column': 0x1
                                    }
                                },
                                'adapters': [],
                                'clients': [],
                                'scenarios': []
                            }, this['features'] = [], this['scenarios'] = [], this['steps'] = [], this['sourcePath'] = '', this['distPath'] = '', this['source'] = _0x5adb20, this['map'] = _0x12392c, this['compiler'] = _0x365c0c, this['renderCodeFrame'] = _0x365c0c['getOptions']()['renderCodeFrame'];
                        }
                        async ['load']() {
                            const _0x3c08bf = this['compiler']['async']();
                            try {
                                this['stateManager'] = this['compiler']['getOptions']()['stateManager'], this['configManger'] = this['compiler']['getOptions']()['configManager'];
                                const _0x32cdaf = Object['values'](this['compiler']['getOptions']()['whiteList'])['map'](_0x3fe205 => _0x2f73c9['resolve'](_0x3fe205)), _0x5a98ff = this['compiler']['_compiler']['outputPath'];
                                this['sourcePath'] = _0x2f73c9['normalizeTrim'](this['compiler']['resourcePath']), this['distPath'] = _0x2f73c9['joinSafe'](_0x2f73c9['changeExt'](_0x2f73c9['joinSafe'](_0x5a98ff, this['compiler']['_module']['rawRequest']), 'js'));
                                for (let _0x2ed290 of this['stateManager']['state']['runner']['featuresAst']['values']()) {
                                    if (_0x2ed290 && _0x2ed290['sourcePath'] === this['sourcePath']) {
                                        this['stateManager']['state']['runner']['featuresAst']['delete'](_0x2ed290['name']);
                                        break;
                                    }
                                }
                                this['stateManager']['state']['onlyFeatures']['delete'](this['sourcePath']), this['stateManager']['state']['onlyScenarios']['delete'](this['sourcePath']), this['stateManager']['state']['onlySteps']['delete'](this['sourcePath']);
                                if (!_0x32cdaf['includes'](this['sourcePath']))
                                    this['stateManager']['state']['runner']['featuresToRun']['delete'](this['sourcePath']);
                                else {
                                    this['configManger']['configuration']['build']['cleanUp']['tests'] && this['stateManager']['state']['runner']['runs'] > 0x1 && (console['info']('-\x20Deleting\x20Old\x20Spec\x20File:\x20' + this['distPath'] + '\x0a'), await _0x5680d5()['remove'](_0x2f73c9['resolve'](this['distPath'])));
                                    const _0x4911ca = (await _0x5680d5()['readFile'](this['sourcePath']))['toString'](), _0x596b58 = _0x1b39dd['transpileModule'](_0x4911ca, {
                                            'compilerOptions': {
                                                'module': _0x1b39dd['ModuleKind']['ESNext'],
                                                'target': _0x1b39dd['ScriptTarget']['ESNext'],
                                                'importsNotUsedAsValue': _0x1b39dd['ImportsNotUsedAsValues']['Preserve'],
                                                'preserveValueImports': !![],
                                                'preserveConstEnums': !![],
                                                'removeComments': ![],
                                                'pretty': ![]
                                            }
                                        })['outputText'], _0x3847cc = _0x249af2['parsePatch'](_0x249af2['createPatch']('file', _0x4911ca, _0x596b58, '', '')), _0x111e68 = _0x3847cc[0x0]['hunks'];
                                    for (let _0xb03d6e = 0x0; _0xb03d6e < _0x111e68['length']; ++_0xb03d6e) {
                                        let _0x4aa181 = 0x0;
                                        const _0xa1a1d = _0x111e68[_0xb03d6e];
                                        _0xa1a1d['lines'] = _0xa1a1d['lines']['map'](_0x1fa968 => {
                                            if (_0x1fa968 === '-')
                                                return _0x4aa181++, '\x20';
                                            return _0x1fa968;
                                        }), _0xa1a1d['newLines'] += _0x4aa181;
                                        for (let _0x26afe0 = _0xb03d6e + 0x1; _0x26afe0 < _0x111e68['length']; ++_0x26afe0) {
                                            _0x111e68[_0x26afe0]['newStart'] += _0x4aa181;
                                        }
                                    }
                                    this['tsSource'] = _0x249af2['applyPatch'](_0x4911ca, _0x3847cc), await this['parseCode'](), this['feature']['OriginalSource'] = this['tsSource'];
                                    this['feature']['name'] && (await this['checkDuplicateSameFileFeatures'](), await this['checkDuplicateMultipleFeatures'](), await this['checkDuplicateScenarios'](), this['setIsLast']());
                                    if (this['feature']['name'])
                                        this['stateManager']['state']['runner']['featuresAst']['set'](this['feature']['name'], this['feature']), this['stateManager']['state']['runner']['featuresToRun']['set'](this['sourcePath'], this['feature']);
                                    else {
                                        for (let _0x540572 of this['stateManager']['state']['runner']['featuresAst']['values']()) {
                                            if (_0x540572['sourcePath'] === this['sourcePath'])
                                                this['stateManager']['state']['runner']['featuresAst']['delete'](_0x540572['name']);
                                        }
                                        this['stateManager']['state']['runner']['featuresToRun']['delete'](this['sourcePath']);
                                    }
                                }
                                return this['stateManager']['state']['filesCache']['set'](this['sourcePath'], (await _0x5680d5()['readFile'](this['sourcePath']))['toString']()), _0x3c08bf(null, this['source'], this['map']);
                            } catch (_0x1380cf) {
                                return console['error']('Spec\x20Compilation\x20Error\x20for\x20File:\x20' + this['sourcePath'] + '\x0a'), _0x3c08bf(_0x1380cf);
                            }
                        }
                        async ['parseCode']() {
                            let _0x319f6c = ![];
                            this['ast'] = (0x0, _0x36fd7f['parseScript'])(this['tsSource']['toString'](), {
                                'module': !![],
                                'loc': !![],
                                'raw': !![],
                                'ranges': !![]
                            }), this['traverse'](this['ast'], {
                                'enter': _0x4f9649 => {
                                    var _0x1d6056, _0x504d18, _0x45255b, _0x425fad, _0x2343d0;
                                    if (this['isFeature'](_0x4f9649)) {
                                        const _0x170e97 = _0x4f9649['arguments'][0x0]['quasis'] ? (_0x1d6056 = _0x4f9649['arguments'][0x0]['quasis'][0x0]) === null || _0x1d6056 === void 0x0 ? void 0x0 : _0x1d6056['value']['cooked'] : _0x4f9649['arguments'][0x0]['value'], _0x46d3ee = {
                                                'id': _0x42082a(),
                                                'didRun': ![],
                                                'name': this['removeQuotes'](_0x170e97),
                                                'only': ![],
                                                'skip': ![],
                                                'isLast': !![],
                                                'keyword': _0x3cb780['Feature'],
                                                'sourcePath': this['sourcePath'],
                                                'distPath': this['distPath'],
                                                'location': _0x4f9649['loc'],
                                                'scenarios': this['scenarios']['reverse'](),
                                                'adapters': [],
                                                'clients': []
                                            };
                                        this['scenarios'] = [], this['features']['push'](JSON['parse'](JSON['stringify'](_0x46d3ee))), this['feature'] = _0x46d3ee, _0x319f6c && (this['stateManager']['state']['onlyFeatures']['set'](this['sourcePath'], this['feature']['name']), _0x46d3ee['only'] = !![], _0x319f6c = ![]);
                                    }
                                    if (this['isScenario'](_0x4f9649)) {
                                        const _0x29095d = _0x4f9649['arguments'][0x0]['quasis'] ? (_0x504d18 = _0x4f9649['arguments'][0x0]['quasis'][0x0]) === null || _0x504d18 === void 0x0 ? void 0x0 : _0x504d18['value']['cooked'] : _0x4f9649['arguments'][0x0]['value'], _0x2f6a9b = {
                                                'id': _0x42082a(),
                                                'name': this['removeQuotes'](_0x29095d),
                                                'keyword': _0x3cb780['Scenario'],
                                                'only': ![],
                                                'skip': ![],
                                                'isLast': ![],
                                                'sourcePath': this['sourcePath'],
                                                'distPath': this['distPath'],
                                                'adapters': [],
                                                'clients': [],
                                                'location': (_0x45255b = _0x4f9649['callee']) === null || _0x45255b === void 0x0 ? void 0x0 : _0x45255b['property']['loc'],
                                                'steps': JSON['parse'](JSON['stringify'](this['steps']['reverse']()))
                                            };
                                        this['scenarios']['push'](_0x2f6a9b), this['steps'] = [], _0x319f6c && (this['stateManager']['state']['onlyScenarios']['add'](this['sourcePath']), _0x2f6a9b['only'] = !![], _0x319f6c = ![]);
                                    }
                                    if (this['isStep'](_0x4f9649)) {
                                        const _0x30a6b4 = _0x4f9649['arguments'][0x0]['quasis'] ? (_0x425fad = _0x4f9649['arguments'][0x0]['quasis'][0x0]) === null || _0x425fad === void 0x0 ? void 0x0 : _0x425fad['value']['cooked'] : _0x4f9649['arguments'][0x0]['value'], _0x1d97c7 = {
                                                'id': _0x42082a(),
                                                'name': this['removeQuotes'](_0x30a6b4),
                                                'keyword': _0x3cb780['Given'],
                                                'only': ![],
                                                'skip': ![],
                                                'isLast': ![],
                                                'sourcePath': this['sourcePath'],
                                                'distPath': this['distPath'],
                                                'adapters': [],
                                                'clients': [],
                                                'location': (_0x2343d0 = _0x4f9649['callee']) === null || _0x2343d0 === void 0x0 ? void 0x0 : _0x2343d0['property']['loc']
                                            };
                                        this['steps']['push'](_0x1d97c7), _0x319f6c && (this['stateManager']['state']['onlySteps']['add'](this['sourcePath']), _0x1d97c7['only'] = !![], _0x319f6c = ![]);
                                    }
                                    if (this['isOnly'](_0x4f9649))
                                        _0x319f6c = !![];
                                }
                            });
                        }
                        ['isFeature'](_0x85da3d) {
                            var _0x4807e0;
                            return _0x85da3d['type'] === 'CallExpression' && _0x32dda0['includes']((_0x4807e0 = _0x85da3d['callee']) === null || _0x4807e0 === void 0x0 ? void 0x0 : _0x4807e0['name']);
                        }
                        ['isScenario'](_0x41c7b1) {
                            var _0x4f2aa7, _0x969551;
                            return _0x41c7b1['type'] === 'CallExpression' && _0x5a4638['includes']((_0x969551 = (_0x4f2aa7 = _0x41c7b1['callee']) === null || _0x4f2aa7 === void 0x0 ? void 0x0 : _0x4f2aa7['property']) === null || _0x969551 === void 0x0 ? void 0x0 : _0x969551['name']);
                        }
                        ['isStep'](_0x36b129) {
                            var _0x3e2de8, _0x4a8245;
                            return _0x36b129['type'] === 'CallExpression' && _0x2ac922['includes']((_0x4a8245 = (_0x3e2de8 = _0x36b129['callee']) === null || _0x3e2de8 === void 0x0 ? void 0x0 : _0x3e2de8['property']) === null || _0x4a8245 === void 0x0 ? void 0x0 : _0x4a8245['name']);
                        }
                        ['isHook'](_0x463f4c) {
                            var _0x2592e8, _0x5d48d2;
                            const _0xfe3f9e = [
                                _0x3c46ef['BeforeAll'],
                                _0x3c46ef['AfterAll'],
                                _0x3c46ef['BeforeEachFeature'],
                                _0x3c46ef['BeforeFeature'],
                                _0x3c46ef['AfterFeature'],
                                _0x3c46ef['AfterEachFeature'],
                                _0x3c46ef['BeforeEachScenario'],
                                _0x3c46ef['BeforeScenario'],
                                _0x3c46ef['AfterScenario'],
                                _0x3c46ef['AfterEachScenario'],
                                _0x3c46ef['BeforeEachStep'],
                                _0x3c46ef['AfterEachStep']
                            ];
                            return _0x463f4c['type'] === 'CallExpression' && _0xfe3f9e['includes']((_0x5d48d2 = (_0x2592e8 = _0x463f4c['callee']) === null || _0x2592e8 === void 0x0 ? void 0x0 : _0x2592e8['property']) === null || _0x5d48d2 === void 0x0 ? void 0x0 : _0x5d48d2['name']);
                        }
                        ['isOnly'](_0x16453f) {
                            var _0x173320, _0x22db14;
                            return _0x16453f['type'] === 'CallExpression' && ((_0x22db14 = (_0x173320 = _0x16453f['callee']) === null || _0x173320 === void 0x0 ? void 0x0 : _0x173320['property']) === null || _0x22db14 === void 0x0 ? void 0x0 : _0x22db14['name']) === 'only';
                        }
                        ['removeQuotes'](_0x580599) {
                            return _0x580599['replace'](/(\n|\r\n)/g, '\x0a');
                            let _0x21cc65 = _0x580599;
                            const _0x502d21 = [
                                '\x22',
                                '\x27',
                                '`'
                            ];
                            if (_0x502d21['includes'](_0x21cc65['charAt'](0x0)) && _0x502d21['includes'](_0x21cc65['charAt'](_0x21cc65['length'] - 0x1)))
                                _0x21cc65 = _0x21cc65['substr'](0x1, _0x21cc65['length'] - 0x2);
                            return _0x21cc65;
                        }
                        async ['checkDuplicateMultipleFeatures']() {
                            const _0x5d7e6c = this['stateManager']['state']['runner']['featuresAst']['get'](this['feature']['name']), _0x1b63b8 = _0x5d7e6c === null || _0x5d7e6c === void 0x0 ? void 0x0 : _0x5d7e6c['OriginalSource'];
                            if (!_0x1b63b8)
                                return;
                            _0x5d7e6c && (console['error'](_0x2c0373()['redBright']('Duplicate\x20Features\x20Files\x20found\x0a')), console['log']('\x20\x20' + _0x2c0373()['cyanBright']('Feature') + ':\x20' + this['feature']['name'] + '\x0a\x20\x20' + _0x2c0373()['cyanBright']('File') + ':\x20' + this['sourcePath'] + '\x0a'), console['log']((await this['renderCodeFrame'](this['tsSource'], this['feature']['location']['start']['line'], this['feature']['location']['start']['column']))['ansi'] + '\x0a'), console['log']('\x20\x20' + _0x2c0373()['cyanBright']('Feature') + ':\x20' + _0x5d7e6c['name'] + '\x0a\x20\x20' + _0x2c0373()['cyanBright']('File') + ':\x20' + _0x5d7e6c['sourcePath'] + '\x0a'), console['log']((await this['renderCodeFrame'](_0x1b63b8 || '', _0x5d7e6c['location']['start']['line'], _0x5d7e6c['location']['start']['column']))['ansi'] + '\x0a'), _0x3c88b8(0x1));
                        }
                        async ['checkDuplicateSameFileFeatures']() {
                            if (this['features']['length'] > 0x1) {
                                console['error'](_0x2c0373()['redBright']('A\x20file\x20can\x20have\x20only\x20one\x20Feature\x0a'));
                                for (let _0x48d17b of this['features']) {
                                    console['log']('\x20\x20' + _0x2c0373()['cyanBright']('Feature') + ':\x20' + this['feature']['name'] + '\x0a\x20\x20' + _0x2c0373()['cyanBright']('File') + ':\x20' + this['sourcePath'] + '\x0a'), console['log']((await this['renderCodeFrame'](this['tsSource'], _0x48d17b['location']['start']['line'], _0x48d17b['location']['start']['column']))['ansi'] + '\x0a');
                                }
                                _0x3c88b8(0x1);
                            }
                        }
                        async ['checkDuplicateScenarios']() {
                            if (!this['feature']['scenarios']['length'])
                                console['error'](_0x2c0373()['redBright']('A\x20Feature\x20must\x20have\x20at\x20least\x20one\x20Scenario\x0a')), console['log']('\x20\x20' + _0x2c0373()['cyanBright']('Feature') + ':\x20' + this['feature']['name'] + '\x0a\x20\x20' + _0x2c0373()['cyanBright']('File') + ':\x20' + this['sourcePath'] + '\x0a'), console['log']((await this['renderCodeFrame'](this['tsSource'], this['feature']['location']['start']['line'], this['feature']['location']['start']['column']))['ansi'] + '\x0a'), _0x3c88b8(0x1);
                            else {
                                const _0x4f10ef = new Map();
                                for (let _0x5d27d0 of this['feature']['scenarios']) {
                                    const _0x35a4c5 = _0x4f10ef['get'](_0x5d27d0['name']);
                                    _0x35a4c5 && (console['error'](_0x2c0373()['redBright']('A\x20Feature\x20can\x20not\x20have\x20duplicate\x20Scenarios\x20names\x0a')), console['log']('\x20\x20' + _0x2c0373()['cyanBright']('Scenario') + ':\x20' + _0x35a4c5['name'] + '\x0a\x20\x20' + _0x2c0373()['cyanBright']('File') + ':\x20' + this['sourcePath'] + '\x0a'), console['log']((await this['renderCodeFrame'](this['tsSource'], _0x35a4c5['location']['start']['line'], _0x35a4c5['location']['start']['column']))['ansi'] + '\x0a'), console['log']('\x20\x20' + _0x2c0373()['cyanBright']('Scenario') + ':\x20' + _0x5d27d0['name'] + '\x0a\x20\x20' + _0x2c0373()['cyanBright']('File') + ':\x20' + this['sourcePath'] + '\x0a'), console['log']((await this['renderCodeFrame'](this['tsSource'], _0x5d27d0['location']['start']['line'], _0x5d27d0['location']['start']['column']))['ansi'] + '\x0a'), _0x3c88b8(0x1)), _0x4f10ef['set'](_0x5d27d0['name'], _0x5d27d0), await this['checkDuplicateSteps'](_0x5d27d0);
                                }
                            }
                        }
                        async ['checkDuplicateSteps'](_0x39b956) {
                            if (!_0x39b956['steps']['length'])
                                console['error'](_0x2c0373()['redBright']('A\x20Scenario\x20must\x20have\x20at\x20least\x20one\x20Step\x0a')), console['log']('\x20\x20' + _0x2c0373()['cyanBright']('Scenario') + ':\x20' + _0x39b956['name'] + '\x0a\x20\x20' + _0x2c0373()['cyanBright']('File') + ':\x20' + this['sourcePath'] + '\x0a'), console['log']((await this['renderCodeFrame'](this['tsSource'], _0x39b956['location']['start']['line'], _0x39b956['location']['start']['column']))['ansi'] + '\x0a'), _0x3c88b8(0x1);
                            else {
                                const _0x4fb68d = new Map();
                                for (let _0x150b9e of _0x39b956['steps']) {
                                    const _0x5a5094 = _0x4fb68d['get'](_0x150b9e['name']);
                                    _0x5a5094 && (console['error'](_0x2c0373()['redBright']('A\x20Scenario\x20can\x20not\x20have\x20duplicate\x20Steps\x20names\x0a')), console['log']('\x20\x20' + _0x2c0373()['cyanBright']('Step') + ':\x20' + _0x5a5094['name'] + '\x0a\x20\x20' + _0x2c0373()['cyanBright']('File') + ':\x20' + this['sourcePath'] + '\x0a'), console['log']((await this['renderCodeFrame'](this['tsSource'], _0x5a5094['location']['start']['line'], _0x5a5094['location']['start']['column']))['ansi'] + '\x0a'), console['log']('\x20\x20' + _0x2c0373()['cyanBright']('Step') + ':\x20' + _0x150b9e['name'] + '\x0a\x20\x20' + _0x2c0373()['cyanBright']('File') + ':\x20' + this['sourcePath'] + '\x0a'), console['log']((await this['renderCodeFrame'](this['tsSource'], _0x150b9e['location']['start']['line'], _0x150b9e['location']['start']['column']))['ansi'] + '\x0a'), _0x3c88b8(0x1)), _0x4fb68d['set'](_0x150b9e['name'], _0x150b9e);
                                }
                            }
                        }
                        async ['checkNames'](_0x40c409) {
                            await this['logNoName'](_0x40c409['name'], _0x40c409['keyword'], _0x40c409['location']);
                            for (let _0x55620b of _0x40c409['scenarios']) {
                                await this['logNoName'](_0x55620b['name'], _0x55620b['keyword'], _0x55620b['location']);
                                for (let _0x29945b of _0x55620b['steps']) {
                                    await this['logNoName'](_0x29945b['name'], _0x29945b['keyword'], _0x29945b['location']);
                                }
                            }
                        }
                        async ['logNoName'](_0x4a05a5, _0x570ad4, _0x7f66e0) {
                            !_0x4a05a5['trim']() && (console['error'](_0x2c0373()['redBright'](_0x570ad4 + '\x20name\x20can\x20not\x20be\x20empty\x0a')), console['log']((await this['renderCodeFrame'](this['tsSource'], _0x7f66e0['start']['line'], _0x7f66e0['start']['column']))['ansi'] + '\x0a')), _0x3c88b8(0x1);
                        }
                        ['setIsLast']() {
                            const _0x37efc8 = this['feature']['scenarios'][this['feature']['scenarios']['length'] - 0x1];
                            _0x37efc8['isLast'] = !![];
                            const _0x2ae6cf = _0x37efc8['steps'][_0x37efc8['steps']['length'] - 0x1];
                            _0x2ae6cf['isLast'] = !![];
                        }
                    }
                    function _0x4b3bf8(_0x9bbeb3, _0x357a92) {
                        const _0x562fdd = new _0x1760ed(_0x9bbeb3, this, _0x357a92);
                        return _0x562fdd['load']();
                    }
                },
                0x9a43: _0x52d9a1 => {
                    _0x52d9a1['exports'] = require('assert');
                },
                0x5629: _0x16ce5a => {
                    _0x16ce5a['exports'] = require('constants');
                },
                0x17e1: _0xf0de0c => {
                    _0xf0de0c['exports'] = require('crypto');
                },
                0xdf3b: _0x4e5495 => {
                    _0x4e5495['exports'] = require('fs');
                },
                0x5615: _0x27daa1 => {
                    _0x27daa1['exports'] = require('os');
                },
                0x11569: _0x4711d6 => {
                    _0x4711d6['exports'] = require('path');
                },
                0x31ed: _0x294c07 => {
                    _0x294c07['exports'] = require('stream');
                },
                0x129c0: _0xa39e0d => {
                    _0xa39e0d['exports'] = require('tty');
                },
                0x1206d: _0xd9c1ee => {
                    _0xd9c1ee['exports'] = require('util');
                }
            }, _0x14a71b = {};
        function _0xf2a64b(_0x13ff44) {
            var _0xb6e457 = _0x14a71b[_0x13ff44];
            if (_0xb6e457 !== undefined)
                return _0xb6e457['exports'];
            var _0x447cc9 = _0x14a71b[_0x13ff44] = {
                'id': _0x13ff44,
                'loaded': ![],
                'exports': {}
            };
            return _0x3ff4ab[_0x13ff44](_0x447cc9, _0x447cc9['exports'], _0xf2a64b), _0x447cc9['loaded'] = !![], _0x447cc9['exports'];
        }
        _0xf2a64b['m'] = _0x3ff4ab, _0xf2a64b['x'] = () => {
            var _0x1f21fa = _0xf2a64b['O'](undefined, [
                0xee,
                0x182,
                0x1e7
            ], () => _0xf2a64b(0x14052));
            return _0x1f21fa = _0xf2a64b['O'](_0x1f21fa), _0x1f21fa;
        }, ((() => {
            var _0x956ccd = [];
            _0xf2a64b['O'] = (_0x65eabf, _0x1ac6e3, _0x210a0d, _0xb45692) => {
                if (_0x1ac6e3) {
                    _0xb45692 = _0xb45692 || 0x0;
                    for (var _0x1bd9da = _0x956ccd['length']; _0x1bd9da > 0x0 && _0x956ccd[_0x1bd9da - 0x1][0x2] > _0xb45692; _0x1bd9da--)
                        _0x956ccd[_0x1bd9da] = _0x956ccd[_0x1bd9da - 0x1];
                    _0x956ccd[_0x1bd9da] = [
                        _0x1ac6e3,
                        _0x210a0d,
                        _0xb45692
                    ];
                    return;
                }
                var _0x5988f7 = Infinity;
                for (var _0x1bd9da = 0x0; _0x1bd9da < _0x956ccd['length']; _0x1bd9da++) {
                    var [_0x1ac6e3, _0x210a0d, _0xb45692] = _0x956ccd[_0x1bd9da], _0x342175 = !![];
                    for (var _0x490600 = 0x0; _0x490600 < _0x1ac6e3['length']; _0x490600++) {
                        if ((_0xb45692 & 0x1 === 0x0 || _0x5988f7 >= _0xb45692) && Object['keys'](_0xf2a64b['O'])['every'](_0x5d4f71 => _0xf2a64b['O'][_0x5d4f71](_0x1ac6e3[_0x490600])))
                            _0x1ac6e3['splice'](_0x490600--, 0x1);
                        else {
                            _0x342175 = ![];
                            if (_0xb45692 < _0x5988f7)
                                _0x5988f7 = _0xb45692;
                        }
                    }
                    if (_0x342175) {
                        _0x956ccd['splice'](_0x1bd9da--, 0x1);
                        var _0x5a54bd = _0x210a0d();
                        if (_0x5a54bd !== undefined)
                            _0x65eabf = _0x5a54bd;
                    }
                }
                return _0x65eabf;
            };
        })()), ((() => {
            _0xf2a64b['n'] = _0x2f1acd => {
                var _0x368b1e = _0x2f1acd && _0x2f1acd['__esModule'] ? () => _0x2f1acd['default'] : () => _0x2f1acd;
                return _0xf2a64b['d'](_0x368b1e, { 'a': _0x368b1e }), _0x368b1e;
            };
        })()), ((() => {
            _0xf2a64b['d'] = (_0x365e9a, _0x2060c8) => {
                for (var _0x47d412 in _0x2060c8) {
                    _0xf2a64b['o'](_0x2060c8, _0x47d412) && !_0xf2a64b['o'](_0x365e9a, _0x47d412) && Object['defineProperty'](_0x365e9a, _0x47d412, {
                        'enumerable': !![],
                        'get': _0x2060c8[_0x47d412]
                    });
                }
            };
        })()), ((() => {
            _0xf2a64b['f'] = {}, _0xf2a64b['e'] = _0x38015f => {
                return Promise['all'](Object['keys'](_0xf2a64b['f'])['reduce']((_0x834cab, _0x25281d) => {
                    return _0xf2a64b['f'][_0x25281d](_0x38015f, _0x834cab), _0x834cab;
                }, []));
            };
        })()), ((() => {
            _0xf2a64b['u'] = _0x43ade1 => {
                return '' + _0x43ade1 + '.js';
            };
        })()), ((() => {
            _0xf2a64b['o'] = (_0x245462, _0x3933b5) => Object['prototype']['hasOwnProperty']['call'](_0x245462, _0x3933b5);
        })()), ((() => {
            _0xf2a64b['r'] = _0x11f5ab => {
                typeof Symbol !== 'undefined' && Symbol['toStringTag'] && Object['defineProperty'](_0x11f5ab, Symbol['toStringTag'], { 'value': 'Module' }), Object['defineProperty'](_0x11f5ab, '__esModule', { 'value': !![] });
            };
        })()), ((() => {
            _0xf2a64b['nmd'] = _0x5aa74e => {
                _0x5aa74e['paths'] = [];
                if (!_0x5aa74e['children'])
                    _0x5aa74e['children'] = [];
                return _0x5aa74e;
            };
        })()), ((() => {
            var _0x679f4 = { 0x144: 0x1 };
            _0xf2a64b['O']['require'] = _0x125bf4 => _0x679f4[_0x125bf4];
            var _0x37833c = _0x1cabed => {
                var _0x459058 = _0x1cabed['modules'], _0xc72c49 = _0x1cabed['ids'], _0x25d99f = _0x1cabed['runtime'];
                for (var _0x4acba9 in _0x459058) {
                    _0xf2a64b['o'](_0x459058, _0x4acba9) && (_0xf2a64b['m'][_0x4acba9] = _0x459058[_0x4acba9]);
                }
                if (_0x25d99f)
                    _0x25d99f(_0xf2a64b);
                for (var _0x31d443 = 0x0; _0x31d443 < _0xc72c49['length']; _0x31d443++)
                    _0x679f4[_0xc72c49[_0x31d443]] = 0x1;
                _0xf2a64b['O']();
            };
            _0xf2a64b['f']['require'] = (_0x5a0ae9, _0x4c10ff) => {
                if (!_0x679f4[_0x5a0ae9]) {
                    if (!![])
                        _0x37833c(require('./' + _0xf2a64b['u'](_0x5a0ae9)));
                    else
                        _0x679f4[_0x5a0ae9] = 0x1;
                }
            };
        })()), ((() => {
            var _0x48330e = _0xf2a64b['x'];
            _0xf2a64b['x'] = () => {
                return _0xf2a64b['e'](0xee), _0xf2a64b['e'](0x182), _0xf2a64b['e'](0x1e7), _0x48330e();
            };
        })());
        var _0x1df6f7 = _0xf2a64b['x']();
        return _0x1df6f7;
    })());
}));
