// selenidejs/built/locators/BrowserWebElementByJs.js
"use strict";
// Copyright 2018 Knowledge Expert SA
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
Object.defineProperty(exports, "__esModule", { value: true });
exports.BrowserWebElementByJs = void 0;
class BrowserWebElementByJs {
    constructor(browser, script, args) {
        this.browser = browser;
        this.script = script;
        this.args = args;
        this.browser = browser;
        this.script = script;
        this.args = args || [];
    }
    async find() {
        return this.browser.executeScript(this.script, ...this.args);
    }
    toString() {
        return `browser.element(${this.script.toString()})`;
    }
}
exports.BrowserWebElementByJs = BrowserWebElementByJs;
